<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>visualization methods in ChIPseeker</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            visualization methods in ChIPseeker 
          </h1>
          <p class="meta">Apr 30, 2014 - 2 minute read -  <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          

<p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes:6a8a65de687a162a83f53f5c5a317279">ChIP peaks over Chromosomes</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/chrCoverage.png" alt="" /></p>

<h3 id="heatmap-of-chip-binding-to-tss-regions:6a8a65de687a162a83f53f5c5a317279">Heatmap of ChIP binding to TSS regions</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">require(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
peakHeatmap(files, weightCol=&quot;V5&quot;, TxDb=txdb, upstream=3000, downstream=3000, color=rainbow(length(files)))
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/peakHeat2.png" alt="" /></p>

<h3 id="average-profile-of-chip-peaks-binding-to-tss-region:6a8a65de687a162a83f53f5c5a317279">Average Profile of ChIP peaks binding to TSS region</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">plotAvgProf(files, TranscriptDb=txdb, weightCol=&quot;V5&quot;, upstream=3000, downstream=3000)
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/avgProf2.png" alt="" /></p>

<h3 id="genomic-annotation:6a8a65de687a162a83f53f5c5a317279">Genomic Annotation</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">peakAnnoList=lapply(files, annotatePeak)
plotAnnoPie(peakAnnoList[[1]])
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/pieAnno.png" alt="" /></p>

<pre><code class="language-lang=&quot;rsplus&quot;">plotAnnoBar(peakAnnoList)
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/barAnno.png" alt="" /></p>

<h3 id="distance-to-tss:6a8a65de687a162a83f53f5c5a317279">Distance to TSS</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">plotDistToTSS(peakAnnoList)
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/distTSS.png" alt="" /></p>

<h3 id="overlap-of-peak-sets:6a8a65de687a162a83f53f5c5a317279">Overlap of peak sets</h3>

<pre><code class="language-lang=&quot;rsplus&quot;">genes= lapply(peakAnnoList, function(i) as.data.frame(i)$geneId)
vennplot(genes)
</code></pre>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/2014/04/venn.png" alt="" /></p>

<p>In the future version, ChIPseeker will support statistical comparison
among ChIP peak sets, and incorporate open access database GEO for users
to compare their own dataset to those deposited in database. Significant
overlap among peak sets can be used to infer cooperative regulation.
This feature will soon be available.</p>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Apr 30, 2014</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>  <a class="category" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/" title="ChIPseeker for ChIP peak annotation">ChIPseeker for ChIP peak annotation</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2014/05/%E6%9B%B4%E6%94%B9%E7%99%BE%E5%BA%A6%E4%BA%91%E5%90%8C%E6%AD%A5%E7%9B%98%E8%B7%AF%E5%BE%84/" title="更改百度云同步盘路径">更改百度云同步盘路径</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
     
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </li>
      
    </ul>
  </section>

  <img src="http://guangchuangyu.github.io/img/wechat.jpg" />
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


