<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="http://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="http://guangchuangyu.github.io/post/">

    <title>Posts | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#contact">Contact</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2014_visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-04-30 06:33:41 &#43;0800 CST" itemprop="datePublished">Wed, Apr 30, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>
        
    </span>
    
    



</div>

        <p><p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes:6a8a65de687a162a83f53f5c5a317279">ChIP peaks over Chromosomes</h3>

<pre><code>&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2014_chipseeker-for-chip-peak-annotation/">ChIPseeker for ChIP peak annotation</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-04-13 13:02:15 &#43;0800 CST" itemprop="datePublished">Sun, Apr 13, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>
        
    </span>
    
    



</div>

        <p><p><a href="http://www.bioconductor.org/packages/2.14/bioc/html/ChIPpeakAnno.html">ChIPpeakAnno</a>
WAS the only R package for ChIP peak annotation. I used it for
annotating peak in my recent study.</p>

<p>I found it does not consider the strand information of genes. I
<a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">reported the bug</a> to the
authors, but they are reluctant to change.</p>

<p>So I decided to develop my own package,
<a href="http://www.bioconductor.org/packages/2.14/bioc/html/ChIPseeker.html">ChIPseeker</a>,
and it&rsquo;s now available in Bioconductor.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/2014_boxplot/">boxplot</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-03-04 12:23:16 &#43;0800 CST" itemprop="datePublished">Tue, Mar 4, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/boxplot">boxplot</a>
        
    </span>
    
    



</div>

        <p><p>生物坑很多人画图只会直方图，统计只会T检验，在暨大见过太多的学生连T检验都不会，分不清SEM和SD的差别，也不清楚T检验那几个简单参数的含义。我写统计笔记也是因为不想重复性地跟学生讲解T检验。</p>

<p>Barplot和T test一样普遍而流行，barplot适合于表示计数数据和比例，显示比例也可以用pie
plot，但直方图比饼图要好，因为人类的眼睛适合于比较高度，而不是弧度。</p>

<p>多半时候生物学数据并非简单的计数数据，对于测量数据，在展示数据分布时，很多人会使用他们熟悉的barplot，用高度来表示mean，然后再加上errorbar，这样展示数据，信息量是非常低的，使用boxplot能够提供更多的数据分布信息，能更好地展现数据，但可能很多人只会在excel里画barplot，Nature
Methods
2013年的文章中有100个barplot图，而只有20个boxplot图，从这里就可以看出来，用boxplot的人远远没有barplot多，于是NPG怒了，写了两篇专栏文章<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2807.html">Points of View: Bar charts and box plots</a>和<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2813.html">Points of Significance: Visualizing samples with box plots</a>并且发表了一篇<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2811.html">BoxPlotR: a web tool for generation of box plots</a>方便大家画boxplot，如此简单的web tool能够发Nature Methods，实在是让人羡慕妒忌恨啊。</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/biology/2014_old-habits-die-hard/">old habits die hard</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-01-22 13:27:22 &#43;0800 CST" itemprop="datePublished">Wed, Jan 22, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ncbi">NCBI</a>
        
    </span>
    
    



</div>

        <p>从2011年1月我就在实验室的QQ群里发群邮件说IPI关门，时至今日，已经关门3年了，主页上一直停留在关门大吉的那一刻。 我不断在邮件里, lab meeting上强调要换成uniprot来搜库，然而时至今日，依然还是有很多的人在使用IPI，想想真可怕，实验室真是100年不更新一下数据啊。 另外一个我非常讨厌的就是GI号，它压根就不是正儿八经的ID号，但他们从来就不愿意尝试改变。 比如上面这个蛋白质序列的FASTA文件，注释行有很多信息，比如： &gt;gi|16128001|ref|NP_414548.1| putative transporter [Escherichia coli str. K-12 substr. MG1655] 显然搜库时可以使用NP_414548.1做为ID，这个问题我说过N多遍，但他们一定会用他们惯用的gi|16128001来做ID。 问题是很明显的： GI number (sometimes written in lower case, &ldquo;gi&rdquo;) is simply a series of digits that are assigned consecutively to each sequence record processed by NCBI. The GI number bears no resemblance to the Accession number of the sequence record. The GI number has been used for many years by NCBI to track sequence histories in GenBank and the other sequence databases it maintains.</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/2014_bug-of-r-package-chippeakanno/">Bug of R package ChIPpeakAnno</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-01-14 17:09:07 &#43;0800 CST" itemprop="datePublished">Tue, Jan 14, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <p><p>I used R package <strong>ChIPpeakAnno</strong> for annotating peaks, and found that
it handle the DNA strand in the wrong way. Maybe the developers were
from the computer science but not biology background.</p>

<pre><code>&gt; require(ChIPpeakAnno)
&gt; packageVersion(&quot;ChIPpeakAnno&quot;)
[1] '2.10.0'
&gt; peak &lt;- RangedData(space=&quot;chr1&quot;, IRanges(24736757, 24737528))
&gt; data(TSS.human.GRCh37)
&gt; ap &lt;- annotatePeakInBatch(peak, Annotation=TSS.human.GRCh37)
&gt; ap
RangedData with 1 row and 9 value columns across 1 space
                     space               ranges |        peak      strand
                               |  
1 ENSG00000001461        1 [24736757, 24737528] |           1           +
                          feature start_position end_position insideFeature
                                   
1 ENSG00000001461 ENSG00000001461       24742284     24799466      upstream
                  distancetoFeature shortestDistance fromOverlappingOrNearest
                                                
1 ENSG00000001461             -5527             4756             NearestStart
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/software/2013_install-emacs-on-mac/">Install Emacs on Mac</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2013-09-10 18:04:22 &#43;0800 CST" itemprop="datePublished">Tue, Sep 10, 2013</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/emacs">emacs</a>
        
    </span>
    
    



</div>

        <p><h2 id="package-manager:a0d2dabbee19e6fccf2cd8eeaa0e67d0">package manager</h2>

<p>之前一直用<a href="http://guangchuangyu.github.io/categories/bsd/">NetBSD</a>的<a href="http://guangchuangyu.github.io/2007/05/%E6%88%91%E5%88%B6%E4%BD%9C%E7%9A%84pkgsrc/">pkgsrc</a>做为各种Unix-like系统的包管理系统，pkgsrc在自己内部解决依赖关系，这点对于生产环境来说，是很好的一个优点，我通常会在服务器里装一个pkgsrc，在用户目录下，维护一套UNIX软件。但是用在自己的机器上，内部解决软件依赖，就是个缺点，浪费编译时间和硬盘空间。</p>

<p>Mac上可用的包管理系统很多，Fink移植于<a href="http://guangchuangyu.github.io/categories/linux/">Debian</a>的apt，<a href="http://guangchuangyu.github.io/categories/linux/">Gentoo</a>的portage也支持OS
X。OS X是基于BSD的系统，BSD有自己的一套包管理系统，如FreeBSD的ports和<a href="http://guangchuangyu.github.io/categories/bsd/">NetBSD</a>的<a href="http://guangchuangyu.github.io/2007/05/%E6%88%91%E5%88%B6%E4%BD%9C%E7%9A%84pkgsrc/">pkgsrc</a>，MacPorts可以看作是这一系列中的一个，不过它并不是OS
X原生的，它只是OpenDarwin的子项目，所以和pkgsrc一个样，也是内部解决依赖，真正为OS
X设计的原生包管理系统非Homebrew莫属，它会尽量依赖系统里已有的软件。</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/mypaper/2012_clusterprofiler-package-comparing-biological-themes-gene-clusters/">clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2012-05-07 03:04:46 &#43;0800 CST" itemprop="datePublished">Mon, May 7, 2012</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>
        
    </span>
    
    



</div>

        <p><p>Increasing quantitative data generated from transcriptomics and
proteomics require integrative strategies for analysis. Here, we present
an R package, clusterProfiler that automates the process of
biological-term classification and the enrichment analysis of gene
clusters. The analysis module and visualization module were combined
into a reusable workflow. Currently, clusterProfiler supports three
species, including humans, mice, and yeast. Methods provided in this
package can be easily extended to other species and ontologies. The
clusterProfiler package is released under Artistic-2.0 License within
Bioconductor project. The source code and vignette are freely available
at
<a href="http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html</a></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/stats/2012_enrichment-analysis/">enrichment analysis</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2012-04-28 08:06:05 &#43;0800 CST" itemprop="datePublished">Sat, Apr 28, 2012</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/hypergeometric">hypergeometric</a>
        
    </span>
    
    



</div>

        <p><p>经常看到一些饼图，描述某些事物的组成，比如说有钱人的学历分布，然后我们可以看到高学历所占比例并不高，根据这个比例下结论通常是错的，这些比例说明不了问题，如果把各种学历在总体人口中的分布做为背景进行考虑的话，你就会发现学历还是有点用的。
当我们用组学测定了一大堆分子之后，我们希望站在更高的角度去看这些分子和那些生物学过程相关。那么通常各种注释，对这些基因/蛋白进行分类，那么从分类的比例上，是不能草率下结论，正如上面有钱人学历分布的例子一样。我们需要把总体的分布考虑进去。
和某个注释/分类是否有相关性，把基因分成属于这一类，和不属于这一类两种，这就好比经典统计学中的白球和黑球的抽样问题。也可以列一个２x２的表，进行独立性分析。
以文章<a href="http://cancerres.aacrjournals.org/content/62/16/4722">Gene Expression in Ovarian Cancer Reflects Both Morphology and Biological Behavior, Distinguishing Clear Cell from Other Poor-Prognosis Ovarian Carcinomas</a>所鉴定的差异基因为例。</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/math/2012_support-vector-machine/">Support Vector Machine</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2012-04-12 05:37:55 &#43;0800 CST" itemprop="datePublished">Thu, Apr 12, 2012</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/machine-learning">Machine Learning</a>
        
    </span>
    
    



</div>

        <p><p>支持向量机(Support Vector Machines, SVM)最初由Vladimir
Vapnik于1997年提出，SVM最简单的形式就是找出一下区分界限（descision
boundary），也称之为超平面(hyperplane)，使得离它最近的点（称之为support
vectors）与之间隔最大。
<img src="http://guangchuangyu.github.io/blog_images/2012/04/linearSVM.png" alt="" /></p>

<p>这和logistic regression有些相似，区别在于logistic
regression要求所有的点尽可能地远离中间那条线，而SVM是要求最接近中间线的点尽可能地远离中间线。也就是说SVM的主要目标是区分那些最难区分的点。</p>

<p>SVM对于hyperplane的定义，在形式上和logistic regression一样,logistic
regression的decision boundary由$\theta^TX=0$确定,SVM则用$w^TX+b=0$表示,其中b相当于logistic regression中的$\theta_0$，从形式上看，两者并无区别，当然如前面所说，两者的目标不一样，logistic regression着眼于全局，SVM着眼于support
vectors。有监督算法都有label变量y，logistic
regression取值是<code>{0,1}</code>，而SVM为了计算距离方便，取值为<code>{-1,1}</code>。</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2012_draw-chinese-character-jiong-ggplot2/">draw chinese character Jiong using ggplot2</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2012-02-17 09:26:50 &#43;0800 CST" itemprop="datePublished">Fri, Feb 17, 2012</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggplot2">ggplot2</a>
        
    </span>
    
    



</div>

        <p><p>The Chinese character Jiong (囧) is now widely used for expressing ideas
or feelings such as annoyance, shock, embarrassment, awkwardness, scorn.</p>

<p>The function plot of <code>y=1/(x^2-1)</code> looks very similar of this symbol.</p>

<p>I use ggplot2 to draw the symbol of Jiong.</p>
</p>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="http://guangchuangyu.github.io//post/page/5/">« Previous</a></li>
        
        
        <li><a href="http://guangchuangyu.github.io//post/page/7/">Next »</a></li>
        
    </ul>
</nav>



    <footer class="site-footer">
    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">
        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

</div>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

