<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="generator" content="Hugo 0.89.4 with theme Tranquilpeak 0.4.7-BETA">
<meta name="author" content="Guangchuang Yu">
<meta name="keywords" content="">
<meta name="description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。
这个纯属练习用，没啥实用价值。

Course Projects:
Project 1: Implementation of a simple gene finder
GOAL
Build a simple codon-usage based gene finder for finding genes in
E.coli.
Procedure
Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:
Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile
Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.
The output should be printed to the standard output as (xxx stands for
the likelihood)
ORF1: xxx ORF2: xxx
">


<meta property="og:description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。
这个纯属练习用，没啥实用价值。

Course Projects:
Project 1: Implementation of a simple gene finder
GOAL
Build a simple codon-usage based gene finder for finding genes in
E.coli.
Procedure
Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:
Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile
Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.
The output should be printed to the standard output as (xxx stands for
the likelihood)
ORF1: xxx ORF2: xxx
">
<meta property="og:type" content="article">
<meta property="og:title" content="a simple gene finder">
<meta name="twitter:title" content="a simple gene finder">
<meta property="og:url" content="https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
<meta property="twitter:url" content="https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
<meta property="og:site_name" content="Guangchuang Yu">
<meta property="og:description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。
这个纯属练习用，没啥实用价值。

Course Projects:
Project 1: Implementation of a simple gene finder
GOAL
Build a simple codon-usage based gene finder for finding genes in
E.coli.
Procedure
Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:
Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile
Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.
The output should be printed to the standard output as (xxx stands for
the likelihood)
ORF1: xxx ORF2: xxx
">
<meta name="twitter:description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。
这个纯属练习用，没啥实用价值。

Course Projects:
Project 1: Implementation of a simple gene finder
GOAL
Build a simple codon-usage based gene finder for finding genes in
E.coli.
Procedure
Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:
Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile
Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.
The output should be printed to the standard output as (xxx stands for
the likelihood)
ORF1: xxx ORF2: xxx
">
<meta property="og:locale" content="en-us">

  
    <meta property="article:published_time" content="2011-07-10T09:04:46">
  
  
    <meta property="article:modified_time" content="2011-07-10T09:04:46">
  
  
  
    
      <meta property="article:section" content="conference">
    
      <meta property="article:section" content="R">
    
  
  
    
      <meta property="article:tag" content="IBW">
    
  


<meta name="twitter:card" content="summary">

  <meta name="twitter:site" content="@guangchuangyu">


  <meta name="twitter:creator" content="@guangchuangyu">










  <meta property="og:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
  <meta property="twitter:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">


    <title>a simple gene finder</title>

    <link rel="icon" href="https://guangchuangyu.github.io/favicon.png">
    

    

    <link rel="canonical" href="https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/style-twzjdbqhmnnacqs0pwwdzcdbt8yhv8giawvjqjmyfoqnvazl0dalmnhdkvp7.min.css" />
    
    

    
      
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-77901140-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/#about">
    
    
    
      
        <img class="header-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/#about">
          <img class="sidebar-profile-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">Bioinformatics Professor @ SMU</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://twitter.com/guangchuangyu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-twitter"></i>
      
      <span class="sidebar-button-desc">Twitter</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/blog_images/biobabble.jpg">
    
      <i class="sidebar-button-icon fa fa-lg fa-weixin"></i>
      
      <span class="sidebar-button-desc">WeChat</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>


    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaIn
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-left">
  
    <h1 class="post-title" itemprop="headline">
      a simple gene finder
    </h1>
  
  
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2011-07-10T09:04:46&#43;08:00">
        
  July 10, 2011

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/conference">conference</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <p>上了几天的课，<a href="http://ibw2011.fmmu.edu.cn/schedule.htm">http://ibw2011.fmmu.edu.cn/schedule.htm</a>
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。</p>
<p>这个纯属练习用，没啥实用价值。</p>
<blockquote>
<p>Course Projects:</p>
<p>Project 1: Implementation of a simple gene finder</p>
<p>GOAL</p>
<p>Build a simple codon-usage based gene finder for finding genes in
E.coli.</p>
<p>Procedure</p>
<p>Collect 100 gene sequences from the bacterium E. coli in the genbank
(<a href="http://www.ncbi.nlm.nihh.gov">http://www.ncbi.nlm.nihh.gov</a>). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:</p>
<p>Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile</p>
<p>Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.</p>
<p>The output should be printed to the standard output as (xxx stands for
the likelihood)</p>
<p>ORF1: xxx ORF2: xxx</p>
</blockquote>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-r" data-lang="r">
<span style="color:#75715e">###########################################################</span>
<span style="color:#75715e">##                                                       ##</span>
<span style="color:#75715e">##  Dragonstar Course Project (IBW 2011)                 ##</span>
<span style="color:#75715e">##                                                       ##</span>
<span style="color:#75715e">##  Project 1: Implementation of a simple gene finder    ##</span>
<span style="color:#75715e">##  Author Guangchuang your                              ##</span>
<span style="color:#75715e">##  Xi&#39;an (2011-07-08)                                   ##</span>
<span style="color:#75715e">##                                                       ##</span>
<span style="color:#75715e">###########################################################</span>



<span style="color:#75715e">##&#39; .. content for \description{} (ECgnfinder: An R program to predict prokaryotic genes (e.g. E.coli) using 1st Markov Chain) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title ECgnfinder</span>
<span style="color:#75715e">##&#39; @param traningSequence : A FASTA file containing multiple sequences for training set</span>
<span style="color:#75715e">##&#39; @param testingSequence : A FASTA file containing multiple sequences for testing</span>
<span style="color:#75715e">##&#39; @param backgroundSequence : A FASTA file containing multiple sequences for building the background model</span>
<span style="color:#75715e">##&#39; @return log likelihood ratio</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
ECgnfinder <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(trainingSequence, testingSequence, backgroundSequence, print<span style="color:#f92672">=</span><span style="color:#66d9ef">FALSE</span>) {
    <span style="color:#a6e22e">print </span>(<span style="color:#e6db74">&#34;reading training sequence file...&#34;</span>)
    trainingFasta <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readFASTA</span>(trainingSequence)
    trainSeq <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sapply</span>(trainingFasta, <span style="color:#a6e22e">function</span>(i) i<span style="color:#f92672">$</span>seq)

    <span style="color:#a6e22e">print</span>(<span style="color:#e6db74">&#34;building Transition Probability Matrix...&#34;</span>)
    TPM <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getTPM</span>(trainSeq)

    <span style="color:#a6e22e">print</span>(<span style="color:#e6db74">&#34;building codon usage table...&#34;</span>)
    codonUsage <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getCodonUsage</span>(trainSeq)

    <span style="color:#a6e22e">print</span>(<span style="color:#e6db74">&#34;building NucleoTide table...&#34;</span>)
    NtUsage <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getNtUsage</span>(backgroundSequence)


    <span style="color:#a6e22e">print</span>(<span style="color:#e6db74">&#34;reading testing sequence file...&#34;</span>)
    testing <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readFASTA</span>(testingSequence)
    testingSeq <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sapply</span>(testing, <span style="color:#a6e22e">function</span>(i) i<span style="color:#f92672">$</span>seq)

    x <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">lapply</span>(testingSeq, logLikelihood, transition<span style="color:#f92672">=</span>TPM, codonusage<span style="color:#f92672">=</span>codonUsage, background<span style="color:#f92672">=</span>NtUsage)
    <span style="color:#a6e22e">names</span>(x) <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">paste</span>(<span style="color:#e6db74">&#34;Sequence&#34;</span>, <span style="color:#a6e22e">seq_along</span>(testingSeq), sep<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>)
    <span style="color:#a6e22e">if </span>(print) {
        x
    }
    <span style="color:#a6e22e">return</span>(x)
}


<span style="color:#75715e">##&#39; .. content for \description{} (read FASTA file) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title readFASTA</span>
<span style="color:#75715e">##&#39; @param file: A FASTA file</span>
<span style="color:#75715e">##&#39; @return a list of list which containing desc and seq.</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
readFASTA <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(file) {
    x<span style="color:#f92672">=</span><span style="color:#a6e22e">scan</span>(file, what<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>, sep<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;\n&#34;</span>)
    idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">grep</span>(<span style="color:#e6db74">&#34;&gt;&#34;</span>, x)
    fasta <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">list</span>()
    <span style="color:#a6e22e">for </span>(i in <span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#a6e22e">length</span>(idx)) {
            desc <span style="color:#f92672">&lt;-</span> x[idx[i]]
            <span style="color:#a6e22e">if </span>(i <span style="color:#f92672">==</span> <span style="color:#a6e22e">length</span>(idx)) {
                stop <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">length</span>(x)
            } else {
                stop <span style="color:#f92672">&lt;-</span> idx[i<span style="color:#ae81ff">+1</span>]<span style="color:#ae81ff">-1</span>
            }
            seq.idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">seq</span>(idx[i]<span style="color:#ae81ff">+1</span>, stop)
            s <span style="color:#f92672">&lt;-</span> x[seq.idx]
            s <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">paste</span>(s, collapse<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>)
            ss <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">list</span>(desc<span style="color:#f92672">=</span>desc, seq<span style="color:#f92672">=</span>s)
            fasta[i] <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">list</span>(ss)
    }
    <span style="color:#a6e22e">return</span>(fasta)
    }


<span style="color:#75715e">##&#39; Build a transition probability matrix from the given Sequence set.</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; @title getTPM</span>
<span style="color:#75715e">##&#39; @param sequences : A vector of Sequences.</span>
<span style="color:#75715e">##&#39; @return transition probability matrix</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
getTPM <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(sequences) {
    codonLength <span style="color:#f92672">&lt;-</span> <span style="color:#ae81ff">3</span>
    codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getCodon</span>()
    
    <span style="color:#75715e">## calculate a transition count matrix</span>
    transition <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">matrix</span>(<span style="color:#ae81ff">1</span>, nrow<span style="color:#f92672">=</span><span style="color:#a6e22e">length</span>(codons), ncol<span style="color:#f92672">=</span><span style="color:#a6e22e">length</span>(codons))  <span style="color:#75715e">## pseudo count</span>
    <span style="color:#a6e22e">colnames</span>(transition) <span style="color:#f92672">&lt;-</span> codons
    <span style="color:#a6e22e">rownames</span>(transition) <span style="color:#f92672">&lt;-</span> codons

    <span style="color:#a6e22e">for </span>(x in <span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#a6e22e">length</span>(sequences)) {
        s <span style="color:#f92672">&lt;-</span> sequences[x]
        len <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">nchar</span>(s)
        <span style="color:#75715e">## drop the last alphabets which is too short (</span>

        idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">seq</span>(<span style="color:#ae81ff">1</span>, len, codonLength)
        <span style="color:#a6e22e">for </span>(i in <span style="color:#a6e22e">seq_along</span>(idx)) {
            <span style="color:#75715e">## codonI--codonJ</span>
            codonI <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(s,idx[i], idx[i]<span style="color:#f92672">+</span>codonLength<span style="color:#ae81ff">-1</span>)
            <span style="color:#a6e22e">if </span>(i <span style="color:#f92672">!=</span> <span style="color:#a6e22e">length</span>(idx)) {
                codonJ <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(s,idx[i<span style="color:#ae81ff">+1</span>], idx[i<span style="color:#ae81ff">+1</span>]<span style="color:#f92672">+</span>codonLength<span style="color:#ae81ff">-1</span>)
            }
            <span style="color:#a6e22e">if </span>(codonI <span style="color:#f92672">%in%</span> codons <span style="color:#f92672">&amp;</span> codonJ <span style="color:#f92672">%in%</span> codons) {
                transition[codonI, codonJ] <span style="color:#f92672">&lt;-</span> transition[codonI, codonJ] <span style="color:#ae81ff">+1</span>
            } else {
                <span style="color:#a6e22e">warning</span>(<span style="color:#e6db74">&#34;codon &#34;</span>, codonI, <span style="color:#e6db74">&#34; or &#34;</span>, codonJ, <span style="color:#e6db74">&#34; which does not defined was detected...&#34;</span>)
            }
        }
    }

    <span style="color:#75715e">## Calculate Transition Probability Matrix..</span>
    rowsum <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rowSums</span>(transition)
    TPM <span style="color:#f92672">&lt;-</span> transition<span style="color:#f92672">/</span>rowsum
    <span style="color:#a6e22e">return</span>(TPM)
}

getCodon <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>() {
    codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;TTT&#34;</span>, <span style="color:#e6db74">&#34;TTC&#34;</span>, <span style="color:#e6db74">&#34;TTA&#34;</span>, <span style="color:#e6db74">&#34;TTG&#34;</span>,
                <span style="color:#e6db74">&#34;CTT&#34;</span>, <span style="color:#e6db74">&#34;CTC&#34;</span>, <span style="color:#e6db74">&#34;CTA&#34;</span>, <span style="color:#e6db74">&#34;CTG&#34;</span>,
                <span style="color:#e6db74">&#34;ATT&#34;</span>, <span style="color:#e6db74">&#34;ATC&#34;</span>, <span style="color:#e6db74">&#34;ATA&#34;</span>, <span style="color:#e6db74">&#34;ATG&#34;</span>,
                <span style="color:#e6db74">&#34;GTT&#34;</span>, <span style="color:#e6db74">&#34;GTC&#34;</span>, <span style="color:#e6db74">&#34;GTA&#34;</span>, <span style="color:#e6db74">&#34;GTG&#34;</span>,
                <span style="color:#e6db74">&#34;TCT&#34;</span>, <span style="color:#e6db74">&#34;TCC&#34;</span>, <span style="color:#e6db74">&#34;TCA&#34;</span>, <span style="color:#e6db74">&#34;TCG&#34;</span>,
                <span style="color:#e6db74">&#34;CCT&#34;</span>, <span style="color:#e6db74">&#34;CCC&#34;</span>, <span style="color:#e6db74">&#34;CCA&#34;</span>, <span style="color:#e6db74">&#34;CCG&#34;</span>,
                <span style="color:#e6db74">&#34;ACT&#34;</span>, <span style="color:#e6db74">&#34;ACC&#34;</span>, <span style="color:#e6db74">&#34;ACA&#34;</span>, <span style="color:#e6db74">&#34;ACG&#34;</span>,
                <span style="color:#e6db74">&#34;GCT&#34;</span>, <span style="color:#e6db74">&#34;GCC&#34;</span>, <span style="color:#e6db74">&#34;GCA&#34;</span>, <span style="color:#e6db74">&#34;GCG&#34;</span>,
                <span style="color:#e6db74">&#34;TAT&#34;</span>, <span style="color:#e6db74">&#34;TAC&#34;</span>, <span style="color:#e6db74">&#34;TAA&#34;</span>, <span style="color:#e6db74">&#34;TAG&#34;</span>,
                <span style="color:#e6db74">&#34;CAT&#34;</span>, <span style="color:#e6db74">&#34;CAC&#34;</span>, <span style="color:#e6db74">&#34;CAA&#34;</span>, <span style="color:#e6db74">&#34;CAG&#34;</span>,
                <span style="color:#e6db74">&#34;AAT&#34;</span>, <span style="color:#e6db74">&#34;AAC&#34;</span>, <span style="color:#e6db74">&#34;AAA&#34;</span>, <span style="color:#e6db74">&#34;AAG&#34;</span>,
                <span style="color:#e6db74">&#34;GAT&#34;</span>, <span style="color:#e6db74">&#34;GAC&#34;</span>, <span style="color:#e6db74">&#34;GAA&#34;</span>, <span style="color:#e6db74">&#34;GAG&#34;</span>,
                <span style="color:#e6db74">&#34;TGT&#34;</span>, <span style="color:#e6db74">&#34;TGC&#34;</span>, <span style="color:#e6db74">&#34;TGA&#34;</span>, <span style="color:#e6db74">&#34;TGG&#34;</span>,
                <span style="color:#e6db74">&#34;CGT&#34;</span>, <span style="color:#e6db74">&#34;CGC&#34;</span>, <span style="color:#e6db74">&#34;CGA&#34;</span>, <span style="color:#e6db74">&#34;CGG&#34;</span>,
                <span style="color:#e6db74">&#34;AGT&#34;</span>, <span style="color:#e6db74">&#34;AGC&#34;</span>, <span style="color:#e6db74">&#34;AGA&#34;</span>, <span style="color:#e6db74">&#34;AGG&#34;</span>,
                <span style="color:#e6db74">&#34;GGT&#34;</span>, <span style="color:#e6db74">&#34;GGC&#34;</span>, <span style="color:#e6db74">&#34;GGA&#34;</span>, <span style="color:#e6db74">&#34;GGG&#34;</span>
                )
    <span style="color:#a6e22e">return</span>(codons)
}


<span style="color:#75715e">##&#39; .. content for \description{} (estimate codon usage table based on training sequences) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title getCodonUsage</span>
<span style="color:#75715e">##&#39; @param sequences : A vector of sequence set.</span>
<span style="color:#75715e">##&#39; @return A data.frame containing two columns of codons and Freq (frequency probabilities)</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
getCodonUsage <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(sequences) {
    codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">getCodon</span>()
    codon.table <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">data.frame</span>(codons<span style="color:#f92672">=</span>codons, Freq<span style="color:#f92672">=</span><span style="color:#a6e22e">rep</span>(<span style="color:#ae81ff">0</span>, <span style="color:#a6e22e">length</span>(codons)))
    <span style="color:#a6e22e">rownames</span>(codon.table) <span style="color:#f92672">=</span> codons

    <span style="color:#a6e22e">for </span>(i in <span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#a6e22e">length</span>(sequences)) {
        codon <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">codonCount</span>(sequences[i])
        rowname.idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.character</span>(codon[,<span style="color:#ae81ff">1</span>])
        codon.table[rowname.idx,<span style="color:#ae81ff">2</span>] <span style="color:#f92672">&lt;-</span> codon.table[rowname.idx,<span style="color:#ae81ff">2</span>] <span style="color:#f92672">+</span> codon[,<span style="color:#ae81ff">2</span>]
    }
    codon.table[,<span style="color:#ae81ff">2</span>] <span style="color:#f92672">&lt;-</span> codon.table[,<span style="color:#ae81ff">2</span>]<span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(codon.table[,<span style="color:#ae81ff">2</span>])
                                        <span style="color:#75715e">#print(codon.table)  #### print result.</span>
    <span style="color:#a6e22e">return</span>(codon.table)
}

<span style="color:#75715e">##&#39; .. content for \description{} (no empty lines) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title codonCount</span>
<span style="color:#75715e">##&#39; @param seqStr : A sequence string</span>
<span style="color:#75715e">##&#39; @return A data.frame containing two columns of codons and their counts</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
codonCount <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(seqStr) {
    codonLength<span style="color:#f92672">=</span><span style="color:#ae81ff">3</span>
    len <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">nchar</span>(seqStr)
    <span style="color:#75715e">## drop the last alphabets which is too short (</span>

    idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">seq</span>(<span style="color:#ae81ff">1</span>, len,codonLength)
    codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>()
    <span style="color:#a6e22e">for </span>( i in <span style="color:#ae81ff">1</span><span style="color:#f92672">:</span><span style="color:#a6e22e">length</span>(idx)) {
        codon <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(seqStr, idx[i], idx[i]<span style="color:#f92672">+</span>codonLength<span style="color:#ae81ff">-1</span>)
        codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(codons, codon)
    }
    codons.df <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.data.frame</span>(<span style="color:#a6e22e">table</span>(codons))
    <span style="color:#a6e22e">return</span>(codons.df)
}

<span style="color:#75715e">##&#39; .. content for \description{} (reverse complementary sequence) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title revcom</span>
<span style="color:#75715e">##&#39; @param seqStr : A sequence string</span>
<span style="color:#75715e">##&#39; @return A reverse complementary sequence</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
revcom <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(seqStr) {
    seqStr.str <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">unlist</span>(<span style="color:#a6e22e">strsplit</span>(seqStr, split<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>))
    idxT <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">which</span>(seqStr.str<span style="color:#f92672">==</span><span style="color:#e6db74">&#34;T&#34;</span>)
    idxG <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">which</span>(seqStr.str<span style="color:#f92672">==</span><span style="color:#e6db74">&#34;G&#34;</span>)
    idxA <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">which</span>(seqStr.str<span style="color:#f92672">==</span><span style="color:#e6db74">&#34;A&#34;</span>)
    idxC <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">which</span>(seqStr.str<span style="color:#f92672">==</span><span style="color:#e6db74">&#34;C&#34;</span>)
    seqStr.str[idxT] <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;A&#34;</span>
    seqStr.str[idxA] <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;T&#34;</span>
    seqStr.str[idxG] <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;C&#34;</span>
    seqStr.str[idxC] <span style="color:#f92672">&lt;-</span> <span style="color:#e6db74">&#34;G&#34;</span>
    seqStr.rc <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rev</span>(seqStr.str)
    seqStr.rc <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">paste</span>(seqStr.rc, collapse<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>)
    <span style="color:#a6e22e">return</span>(seqStr.rc)
}

<span style="color:#75715e">##&#39; .. content for \description{} (calculate NT frequency probabilities of a given sequence set) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{use to estimate the background/random NT usage} ..</span>
<span style="color:#75715e">##&#39; @title getNtUsage</span>
<span style="color:#75715e">##&#39; @param file : A FASTA file</span>
<span style="color:#75715e">##&#39; @return A data.frame containing NTs and their corresponding frequency probabilities.</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
getNtUsage <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(file) {
    <span style="color:#a6e22e">print</span>(<span style="color:#e6db74">&#34;reading FASTA file&#34;</span>)
    fasta <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readFASTA</span>(file)
    sequences <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sapply</span>(fasta, <span style="color:#a6e22e">function</span>(i) i<span style="color:#f92672">$</span>seq)
    x <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sapply</span>(sequences, strsplit, split<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>)
    x <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">unlist</span>(x)
    NTprob <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as.data.frame</span>(<span style="color:#a6e22e">table</span>(x))
    NTprob[,<span style="color:#ae81ff">2</span>] <span style="color:#f92672">&lt;-</span> NTprob[,<span style="color:#ae81ff">2</span>]<span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(NTprob[,<span style="color:#ae81ff">2</span>])
    <span style="color:#a6e22e">rownames</span>(NTprob) <span style="color:#f92672">&lt;-</span> NTprob[,<span style="color:#ae81ff">1</span>]
    <span style="color:#a6e22e">return</span>(NTprob)
}

<span style="color:#75715e">##&#39; .. content for \description{} (calculate the log likelihood of the six possible reading frame of a given sequence) ..</span>
<span style="color:#75715e">##&#39;</span>
<span style="color:#75715e">##&#39; .. content for \details{} ..</span>
<span style="color:#75715e">##&#39; @title logLikelihood</span>
<span style="color:#75715e">##&#39; @param sequence : A sequence string</span>
<span style="color:#75715e">##&#39; @param transition : Transition Probatility Matrix</span>
<span style="color:#75715e">##&#39; @param codonusage : Codon Usage data.frame</span>
<span style="color:#75715e">##&#39; @param background : NT Usage data.frame for background/random model</span>
<span style="color:#75715e">##&#39; @return Log likelihood</span>
<span style="color:#75715e">##&#39; @author Guangchuang Yu</span>
logLikelihood <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(sequence, transition, codonusage, background) {
    codonLength <span style="color:#f92672">&lt;-</span> <span style="color:#ae81ff">3</span>
    orf1 <span style="color:#f92672">&lt;-</span> sequence
    orf2 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(sequence, <span style="color:#ae81ff">2</span>, <span style="color:#a6e22e">nchar</span>(sequence))
    orf3 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(sequence, <span style="color:#ae81ff">3</span>, <span style="color:#a6e22e">nchar</span>(sequence))
    seq.rc <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">revcom</span>(sequence)
    orf4 <span style="color:#f92672">&lt;-</span> seq.rc
    orf5 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(seq.rc, <span style="color:#ae81ff">2</span>, <span style="color:#a6e22e">nchar</span>(seq.rc))
    orf6 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(seq.rc, <span style="color:#ae81ff">3</span>, <span style="color:#a6e22e">nchar</span>(seq.rc))
    ORF <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(orf1, orf2, orf3, orf4, orf5, orf6)
    <span style="color:#a6e22e">names</span>(ORF) <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">c</span>(<span style="color:#e6db74">&#34;ORF1&#34;</span>, <span style="color:#e6db74">&#34;ORF2&#34;</span>, <span style="color:#e6db74">&#34;ORF3&#34;</span>, <span style="color:#e6db74">&#34;ORF4&#34;</span>, <span style="color:#e6db74">&#34;ORF5&#34;</span>, <span style="color:#e6db74">&#34;ORF6&#34;</span>)

    result <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">data.frame</span>(ORF<span style="color:#f92672">=</span><span style="color:#a6e22e">names</span>(ORF), logLikelihood<span style="color:#f92672">=</span><span style="color:#a6e22e">rep</span>(<span style="color:#ae81ff">0</span>,<span style="color:#ae81ff">6</span>))
    <span style="color:#a6e22e">for </span>(j in <span style="color:#a6e22e">seq_along</span>(ORF)) {
        firstCodon <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(ORF[j], <span style="color:#ae81ff">1</span>, <span style="color:#ae81ff">3</span>)
        <span style="color:#a6e22e">if </span>(firstCodon <span style="color:#f92672">%in%</span> codonusage[,<span style="color:#ae81ff">1</span>]) {
            initProb <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">log</span>(codonusage[firstCodon,<span style="color:#ae81ff">2</span>])
        } else {
            initProb <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">log</span>(<span style="color:#ae81ff">1</span><span style="color:#f92672">/</span><span style="color:#ae81ff">64</span>)
        }

        transitionCount <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">matrix</span>(<span style="color:#ae81ff">0</span>, nrow<span style="color:#f92672">=</span><span style="color:#a6e22e">nrow</span>(transition), ncol<span style="color:#f92672">=</span><span style="color:#a6e22e">ncol</span>(transition))
        <span style="color:#a6e22e">rownames</span>(transitionCount) <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rownames</span>(transition)
        <span style="color:#a6e22e">colnames</span>(transitionCount) <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">colnames</span>(transition)

        codons <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">rownames</span>(transition)

        len <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">nchar</span>(ORF[j])
        len <span style="color:#f92672">&lt;-</span> len <span style="color:#f92672">-</span> len <span style="color:#f92672">%%</span> codonLength <span style="color:#f92672">-</span> codonLength

        idx <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">seq</span>(<span style="color:#ae81ff">1</span>, len, codonLength)
        <span style="color:#a6e22e">for </span>(i in <span style="color:#a6e22e">seq_along</span>(idx)) {
            <span style="color:#75715e">## codonI--codonJ</span>
            codonI <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(ORF[j],idx[i], idx[i]<span style="color:#f92672">+</span>codonLength<span style="color:#ae81ff">-1</span>)
            <span style="color:#a6e22e">if </span>(i <span style="color:#f92672">!=</span> <span style="color:#a6e22e">length</span>(idx)) {
                codonJ <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">substr</span>(ORF[j],idx[i<span style="color:#ae81ff">+1</span>], idx[i<span style="color:#ae81ff">+1</span>]<span style="color:#f92672">+</span>codonLength<span style="color:#ae81ff">-1</span>)
            }
            <span style="color:#a6e22e">if </span>(codonI <span style="color:#f92672">%in%</span> codons <span style="color:#f92672">&amp;</span> codonJ <span style="color:#f92672">%in%</span> codons) {
                transitionCount[codonI, codonJ] <span style="color:#f92672">&lt;-</span> transitionCount[codonI, codonJ] <span style="color:#ae81ff">+1</span>
            } else {
                <span style="color:#a6e22e">warning</span>(<span style="color:#e6db74">&#34;codon &#34;</span>, codonI, <span style="color:#e6db74">&#34; or &#34;</span>, codonJ, <span style="color:#e6db74">&#34; which does not defined was detected...&#34;</span>)
            }
        }

        postTransition <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">log</span>(transition) <span style="color:#f92672">*</span> transitionCount
        posterior <span style="color:#f92672">&lt;-</span> initProb <span style="color:#f92672">+</span> <span style="color:#a6e22e">sum</span>(postTransition)

        NTseq <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">unlist</span>(<span style="color:#a6e22e">strsplit</span>(ORF[j], split<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;&#34;</span>))
        orf.len <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">nchar</span>(ORF[j]) <span style="color:#f92672">-</span> <span style="color:#a6e22e">nchar</span>(ORF[j]) <span style="color:#f92672">%%</span> codonLength
        NTseq <span style="color:#f92672">&lt;-</span> NTseq[1<span style="color:#f92672">:</span>orf.len]
        randomModel <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">sum</span>(<span style="color:#a6e22e">log</span>(background[NTseq,<span style="color:#ae81ff">2</span>]))

        likelihood <span style="color:#f92672">&lt;-</span> posterior <span style="color:#f92672">-</span> randomModel
        result[j, <span style="color:#ae81ff">2</span>] <span style="color:#f92672">&lt;-</span> likelihood
    }
    <span style="color:#a6e22e">return</span>(result)
}
</code></pre></div>
              
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/tags/ibw/">IBW</a>

                  </div>
                
              
            
            <div class="post-actions-wrap">
  
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/2011/07/ibw2011/" data-tooltip="IBW2011">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/2011/06/screen-shot/" data-tooltip="screen shot">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  
</div>

            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2021 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        <div class="post-actions-wrap">
  
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/2011/07/ibw2011/" data-tooltip="IBW2011">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/2011/06/screen-shot/" data-tooltip="screen shot">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-facebook-official"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-twitter"></i>
            </a>
          </li>
        
          <li class="post-action hide-xs">
            <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https://guangchuangyu.github.io/2011/07/a-simple-gene-finder/">
              <i class="fa fa-google-plus"></i>
            </a>
          </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  
</div>

      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <i id="btn-close-shareoptions" class="fa fa-close"></i>
  <ul class="share-options">
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fguangchuangyu.github.io%2F2011%2F07%2Fa-simple-gene-finder%2F">
          <i class="fa fa-facebook-official"></i><span>Share on Facebook</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://twitter.com/intent/tweet?text=https%3A%2F%2Fguangchuangyu.github.io%2F2011%2F07%2Fa-simple-gene-finder%2F">
          <i class="fa fa-twitter"></i><span>Share on Twitter</span>
        </a>
      </li>
    
      <li class="share-option">
        <a class="share-option-btn" target="new" href="https://plus.google.com/share?url=https%3A%2F%2Fguangchuangyu.github.io%2F2011%2F07%2Fa-simple-gene-finder%2F">
          <i class="fa fa-google-plus"></i><span>Share on Google&#43;</span>
        </a>
      </li>
    
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">Bioinformatics Professor @ SMU</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Bioinformatics Professor
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Guangzhou
      </div>
    
  </div>
</div>

    

    
  
    
      <div id="cover" style="background-image:url('https://guangchuangyu.github.io/images/cover.jpg');"></div>
    
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.7/js/jquery.fancybox.min.js" integrity="sha256-GEAnjcTqVP+vBp3SSc8bEDQqvWAZMiHyUSIorrWwH50=" crossorigin="anonymous"></script>


<script src="https://guangchuangyu.github.io/js/script-pcw6v3xilnxydl1vddzazdverrnn9ctynvnxgwho987mfyqkuylcb1nlt.min.js"></script>


<script lang="javascript">
window.onload = updateMinWidth;
window.onresize = updateMinWidth;
document.getElementById("sidebar").addEventListener("transitionend", updateMinWidth);
function updateMinWidth() {
  var sidebar = document.getElementById("sidebar");
  var main = document.getElementById("main");
  main.style.minWidth = "";
  var w1 = getComputedStyle(main).getPropertyValue("min-width");
  var w2 = getComputedStyle(sidebar).getPropertyValue("width");
  var w3 = getComputedStyle(sidebar).getPropertyValue("left");
  main.style.minWidth = `calc(${w1} - ${w2} - ${w3})`;
}
</script>

<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight > code, pre > code').each(function(i, block) {
    if (!$(this).hasClass('codeblock')) {
      $(this).addClass('codeblock');
    }
    hljs.highlightBlock(block);
  });
});
</script>


  
    
      <script>
        var disqus_config = function () {
          this.page.url = 'https:\/\/guangchuangyu.github.io\/2011\/07\/a-simple-gene-finder\/';
          
            this.page.identifier = '\/2011\/07\/a-simple-gene-finder\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'gcyu';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  




    
  </body>
</html>

