<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>enrichment analysis</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            enrichment analysis 
          </h1>
          <p class="meta">Apr 28, 2012 - 2 minute read -  <a href="http://guangchuangyu.github.io/2012/04/enrichment-analysis/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/statistics">Statistics</a>
          </p>
        </header>
        <div class="entry-content">
	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
	  	  
	  <p>经常看到一些饼图，描述某些事物的组成，比如说有钱人的学历分布，然后我们可以看到高学历所占比例并不高，根据这个比例下结论通常是错的，这些比例说明不了问题，如果把各种学历在总体人口中的分布做为背景进行考虑的话，你就会发现学历还是有点用的。
当我们用组学测定了一大堆分子之后，我们希望站在更高的角度去看这些分子和那些生物学过程相关。那么通常各种注释，对这些基因/蛋白进行分类，那么从分类的比例上，是不能草率下结论，正如上面有钱人学历分布的例子一样。我们需要把总体的分布考虑进去。
和某个注释/分类是否有相关性，把基因分成属于这一类，和不属于这一类两种，这就好比经典统计学中的白球和黑球的抽样问题。也可以列一个２x２的表，进行独立性分析。
以文章<a href="http://cancerres.aacrjournals.org/content/62/16/4722">Gene Expression in Ovarian Cancer Reflects Both Morphology and Biological Behavior, Distinguishing Clear Cell from Other Poor-Prognosis Ovarian Carcinomas</a>所鉴定的差异基因为例。</p>

<p>73个差异基因的Symbol，我把它转为 entrezgene
ID得到57个（漏掉的不管它，只是做为一个例子）:</p>

<pre><code>&gt; eg
 [1] &quot;7980&quot;   &quot;3081&quot;   &quot;3162&quot;   &quot;3059&quot;   &quot;1545&quot;   &quot;1917&quot;   &quot;6696&quot;   &quot;5797&quot;  
 [9] &quot;6648&quot;   &quot;10397&quot;  &quot;6781&quot;   &quot;5817&quot;   &quot;1282&quot;   &quot;1284&quot;   &quot;6948&quot;   &quot;7077&quot;  
[17] &quot;5744&quot;   &quot;8566&quot;   &quot;1368&quot;   &quot;1474&quot;   &quot;11015&quot;  &quot;3306&quot;   &quot;728441&quot; &quot;2678&quot;  
[25] &quot;4286&quot;   &quot;3929&quot;   &quot;5095&quot;   &quot;2064&quot;   &quot;1428&quot;   &quot;6590&quot;   &quot;3569&quot;   &quot;2745&quot;  
[33] &quot;3912&quot;   &quot;978&quot;    &quot;5950&quot;   &quot;6539&quot;   &quot;9445&quot;   &quot;5004&quot;   &quot;9971&quot;   &quot;7453&quot;  
[41] &quot;2719&quot;   &quot;1410&quot;   &quot;1311&quot;   &quot;4653&quot;   &quot;4162&quot;   &quot;5358&quot;   &quot;3484&quot;   &quot;3486&quot;  
[49] &quot;2261&quot;   &quot;307&quot;    &quot;1672&quot;   &quot;4837&quot;   &quot;22795&quot;  &quot;486&quot;    &quot;4118&quot;   &quot;3915&quot;  
[57] &quot;10140&quot;
</code></pre>

<p>下面测试一下这些基因和化学刺激响应的相关性。</p>

<pre><code>goid &lt;- “GO:0042221” # response to chemical stimulus
</code></pre>

<p>那么做为背景，总体基因为N，属于“化学刺激响应”这个分类的基因有M个。</p>

<pre><code>allgeneInCategory &lt;- unique(get(goid, org.Hs.egGO2ALLEGS))
M &lt;- length(allgeneInCategory)
N &lt;- length(mappedkeys(org.Hs.egGO))
</code></pre>

<p>样本的大小是n，属于“化学刺激响应”这个分类的基因有k个。</p>

<pre><code>n &lt;- length(eg)
k &lt;- sum(eg %in% allgeneInCategory)
</code></pre>

<p>白球黑球问题，最简单的莫过于用二项式分布，从总体上看，要拿到一个基因属于“化学刺激响应”这个分类的概率是M/N。那么现在抽了n个基因，里面有k个基于这个分类，p值为：</p>

<pre><code>&gt; 1-sum(sapply(0:k-1, function(i) choose(n,i) * (M/N)^i * (1-M/N)^(n-i)))
[1] 8.561432e-10
</code></pre>

<p>二项式分布，是有放回的抽样，你可以多次抽到同一基因，这是不符合的。所以这个计算只能说是做为近似的估计值，无放回的抽样，符合超几何分布，通过超几何分布的计算，p值为：</p>

<pre><code>&gt; phyper(k-1,M, N-M, n, lower.tail=FALSE)
[1] 7.879194e-10
</code></pre>

<p>如果用2x2表做独立性分析，表如下：</p>

<pre><code>&gt; d &lt;- data.frame(gene.not.interest=c(M-k, N-M-n+k), gene.in.interest=c(k, n-k))
&gt; row.names(d) &lt;- c(&quot;In_category&quot;, &quot;not_in_category&quot;)
&gt; d
                gene.not.interest gene.in.interest
In_category                  2613               28
not_in_category             15310               29
</code></pre>

<p>这个也有很多方法可以做检验，经典的有卡方检验和fisher&rsquo;s exact test。
如果用卡方检验来做。</p>

<pre><code>&gt; chisq.test(d, )

    Pearson's Chi-squared test with Yates' continuity correction

data:  d 
X-squared = 51.3849, df = 1, p-value = 7.592e-13
</code></pre>

<p>对于２x２表来说，卡方检验通常也只能做为近似估计值，特别是当sample
size或expected all count比较小的时候，计算并不准确。 fisher&rsquo;s exact
test，名副其实，真的就比较exact，因为它使用的是超几何分布来计算p值。这也是为什么fisher&rsquo;s
exact test和超几何模式计算的p-值是一样的，</p>

<pre><code>&gt; fisher.test(d)

    Fisher's Exact Test for Count Data

data:  d 
p-value = 7.879e-10
alternative hypothesis: true odds ratio is not equal to 1 
95 percent confidence interval:
 0.1013210 0.3089718 
sample estimates:
odds ratio 
 0.1767937 
</code></pre>

<p>通常各种软件做GO富集性分析，都是使用超几何分布进行计算。
IPA软件则是使用fisher&rsquo;s exact test来检验基因在某个网络中是否富集。
从这个例子上看，chi-squared
test的表现最差，binomial做为p值的近似估计还是不错的，因为计算较为简单。
富集性分析应用范围非常广，从<a href="http://bioconductor.org/packages/devel/bioc/html/DOSE.html">Disease Ontology</a>,
<a href="http://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html">Gene Ontology, KEGG</a>,
到<a href="http://bioconductor.org/packages/devel/bioc/html/ReactomePA.html">Reactome Pathway</a>等等。</p>


	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
	
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Apr 28, 2012</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/hypergeometric">hypergeometric</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2012/04/support-vector-machine/" title="Support Vector Machine">Support Vector Machine</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters/" title="clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters">clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>



  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/04/kegg-module-enrichment-analysis/">KEGG Module Enrichment Analysis</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/google-drive--hku/">Google Drive @ HKU</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a>
      </li>
      
    </ul>
  </section>

  
  
  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


