<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Guangchuang Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="Guangchuang Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
 <div id="content">
  <div class="blog-index">
    
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2010/05/gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products/">GOSemSim an R package for measuring semantic similarity among GO terms and gene products</a>
        </h1>
        <p class="meta">May 28, 2010 - 1 minute read -  <a href="http://guangchuangyu.github.io/2010/05/gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a><a class="label" href="http://guangchuangyu.github.io/categories/publication/">Publication</a>
        </p>

      </header>

        
          <p>Abstract SUMMARY: The semantic comparisons of Gene Ontology (GO) annotations provide quantitative ways to compute similarities between genes and gene groups, and have became important basis for many bioinformatics analysis approaches. GOSemSim is an R package for semantic similarity computation among GO terms, sets of GO terms, gene products and gene clusters. Four information content (IC)- and a graph-based methods are implemented in the GOSemSim package, multiple species including human, rat, mouse, fly and yeast are also supported.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2010/05/gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2009/07/bootstrap-method/">Bootstrap Method</a>
        </h1>
        <p class="meta">Jul 31, 2009 - 1 minute read -  <a href="http://guangchuangyu.github.io/2009/07/bootstrap-method/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/statistics/">statistics</a>
        </p>

      </header>

        
          <p>bootstrap是对观测数据集进行有放回（replacement）的随机抽样，以评估总体的各项统计指标。可以用于假设检验、参数估计。好处是并不要求大样本，也不要求正态数据，并且对于不同的统计指标使用的是同样的计算方法。结果也更为可靠，坏处是计算量大。 统计推断(statistical inference)是基于样本统计值的抽样分布来计算的，抽样分布需要从总体中许多的样本来计算，在只有一个样本的情况下，bootstrap对这一随机样本进行有放回的重复抽样，每一个重抽样本与原始随机样本一样大，每次计算相应的抽样的统计值，重复了N次之后，就可以计算统计值的bootstrap分布。 下面做一个小小的试验： a &lt;- c(seq(1:10), rnorm(50)) #创建一个样本，60个数据，非正态分布的，如下图 b &lt;- rep(0, 1000) for (i in 1:1000) { b[i] &lt;- mean(sample(a, replace=TRUE)) } 对它进行1000次有放回重复抽样，计算均值，均值分布的柱状图和qq图如下： 分布是正态的。 &gt; mean(a) #样本均值 [1] 0.947186 &gt; mean(b) #重抽样1000个样本均值的均值 [1] 0.9358049 &gt; sd(b) [1] 0.3245479 &gt; sd(a)/sqrt(60) [1] 0.3318863 由bootstrap方法得到的标准误是0.325，而由原先随机样本所估算的是0.332，两者是相当接近的。由此可见呢，bootstrap方法与理论上的从总体中抽样所得到的样本分布是一致的，这样子就可以基于一个样本来计算样本的分布，而不是要从总体中抽N个样本。 还用上面的例子，使用样本均值$ \bar{x}$去估算总体的均值$ \mu$，虽然我们的样本不是正态的，但如果bootstrap distribution是正态的话，就可以使用类似于单样本t置信区间（要求样本是正态的）的公式:$ \bar{x} \pm t \cdot SE = \bar{x} \pm t \cdot \frac{s}{\sqrt{n}}$ 相应的bootstrap t置信区间是：$ statistic \pm t \cdot SE_{boot}$ 上面例子，总体均值$ \mu$ 95%的置信度，就会落在：$ mean(a) \pm 2.009 \cdot sd(b)$ 可以使用它，计算所有统计指标的置信区间。 用于层次聚类分析的一个例子，使用Pvclust这个包，用于层次聚类，并通过multiscale bootstrap resampling给出相应的p-value用于评估聚类结果的不确定性。提供了两种p-values，AU（approximately unbiased p-value, 通过multiscale bootstrap resampling计算）和BP（bootstrap probability, 通过normal bootstrap resampling计算），AU比BP较为unbiased。 Pvclust使用*hclust*（来自于stats包）进行层次聚类，并自动计算所有的子类的p-value。 library(pvclust) # loads package a &lt;- matrix(rnorm(1000), 100, 10, dimnames=list(paste(&quot;g&quot;, 1:100, sep=&quot;&quot;), paste(&quot;t&quot;,1:10, sep=&quot;&quot;))) # creats sample data at &lt;- t(a) # transposes of a cl &lt;- pvclust(at, nboot=1000) #performs hierarchical cluster analysis with multiscale bootstrap with 1000 repetitions.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2009/07/bootstrap-method/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2008/09/%E9%80%9A%E8%BF%87%E5%B1%80%E5%9F%9F%E7%BD%91%E5%9C%A8linux%E4%B8%8B%E4%BD%BF%E7%94%A8windows%E7%9A%84%E7%A8%8B%E5%BA%8F/">通过局域网在linux下使用windows的程序</a>
        </h1>
        <p class="meta">Sep 22, 2008 - 1 minute read -  <a href="http://guangchuangyu.github.io/2008/09/%E9%80%9A%E8%BF%87%E5%B1%80%E5%9F%9F%E7%BD%91%E5%9C%A8linux%E4%B8%8B%E4%BD%BF%E7%94%A8windows%E7%9A%84%E7%A8%8B%E5%BA%8F/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/linux/">Linux</a><a class="label" href="http://guangchuangyu.github.io/categories/windows/">Windows</a>
        </p>

      </header>

        
          <p>用了好多年linux，现在用windows越来越少了，但是还是得用。没办法，上taobao买个东西，上网银都得用ie，有时候还得用office等其它一些软件。 实验室里有台机子是做文件服务的，装的是windows 2003 server。之前使用rdesktop远程桌面，最近发现配合seamlessrdp可以让rdesktop实现更好的效果，运行windows程序，就像linux中原生程序一样。。 下载 seamlessrdp.zip 解压丢在C盘。 在linux下运行这样的指令 rdesktop -A -s &ldquo;c:\seamlessrdp/seamlessrdpshell.exe C:\Program Files\Internet Explorer\iexplore.exe&rdquo; ip:3389 -u administrator -p password 就可以在linux下运行服务器端的ie了。windows下所有程序都没问题。 这个指令有点长，最好是写个shell脚本，每次运行实际上是进行一次远程连接，windows2003默认只能支持两个远程的连接，所以只能运行两个。正确的做法是运行explorer，然后在里面双击程序，这样可以运行N个程序。运行的程序跟linux下native的程序是一样的。但是读写都在远程机子上，最好架个samba，这样两个机器就可以互相读写文件。 ps: tsclient做为远程桌面客户端更强大点，不单支持RDP，还支持VNC,XDMCP。</p>

          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2008/08/gosemsim/">GOSemSim</a>
        </h1>
        <p class="meta">Aug 11, 2008 - 1 minute read -  <a href="http://guangchuangyu.github.io/2008/08/gosemsim/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics/">Bioinformatics</a>
        </p>

      </header>

        
          <p>第一次写软件包，计算基因和GO的语义相似性。 实现Wang的方法，基于GO的图结构。 以及四种基于信息含量的计算方法（Resnik, Jiang, Lin, Rel）。 可以从Bioconductor下载到。 http://bioconductor.org/packages/release/bioc/html/GOSemSim.html http://bioconductor.org/packages/devel/bioc/html/GOSemSim.html 支持human, rat, mouse, fly, yeast, zebrafish,worm, E coli, arabidopsis等物种。 下载量统计如下：</p>

          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2007/06/screenshot/">screenshot</a>
        </h1>
        <p class="meta">Jun 16, 2007 - 1 minute read -  <a href="http://guangchuangyu.github.io/2007/06/screenshot/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/bsd/">BSD</a><a class="label" href="http://guangchuangyu.github.io/categories/linux/">Linux</a><a class="label" href="http://guangchuangyu.github.io/categories/mac-os/">Mac OS</a>
        </p>

      </header>

        
          <p>OS 9.2 xfce4.2 xfce4.4 qvwm icewm fluxbox wmaker 越来越喜欢windowmaker&hellip;&hellip;喜欢它最小化到图标 wmii 最后是wmii，全屏的感觉就是爽。</p>

          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2007/06/newton-raphson-method%E4%BC%B0%E7%AE%97%E5%87%BD%E6%95%B0%E7%9A%84%E6%A0%B9/">Newton-Raphson Method估算函数的根</a>
        </h1>
        <p class="meta">Jun 2, 2007 - 1 minute read -  <a href="http://guangchuangyu.github.io/2007/06/newton-raphson-method%E4%BC%B0%E7%AE%97%E5%87%BD%E6%95%B0%E7%9A%84%E6%A0%B9/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/math/">Math</a><a class="label" href="http://guangchuangyu.github.io/categories/programming/">Programming</a><a class="label" href="http://guangchuangyu.github.io/categories/c/">C</a>
        </p>

      </header>

        
          <p>Newton-Raphson Method 曲线f(x)有根c,取曲线上一点$(x_1,f(x_1))$, 过此点的切线交x轴$x_2$,过曲线上$(x_2,f(x_2))$的切线交x轴$x_3$，如此反复得到一个序列 $x_1,x_2,\cdot \cdot \cdot,x_n$ 逼近c值. 过$(x_n,f(x_n))$的切线方程为 $y-f(x_n) = f'(x_n)\,(x-x_n)$,假设此方程与x轴的交点为$x_{n+1}$, 即有: $0 - f(x_n) = f'(x_n)(x_n+1 - x_n)$， 即$x_{n+1} = x_n - \frac{f(x_n)}{f'(x_n)}$ &lt;Eq. 1&gt;. 下面利用此法来求一个数的开方。 $f(x) = x^2 - a$ 有根$\sqrt{a}$, 由$f'(x_n) = 2x_n$, 代入式&lt;Eq. 1&gt;可得$x_{n+1} = (x_n + a/x_n)/2$; 当i -&gt; INF 时, $x_i$ -&gt; $\sqrt{a}$; C implementation #include int main(void){ int a,error; double x0,x1 = 1; do { printf(&quot;Input a positive integer: &quot;); scanf(&quot;%d&quot;,&amp;a); if</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2007/06/newton-raphson-method%E4%BC%B0%E7%AE%97%E5%87%BD%E6%95%B0%E7%9A%84%E6%A0%B9/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    
    






<div class="pagination">

    
        
        
        
        <a href="/" aria-label="First" class="label-pagination"><i class="fa fa-angle-double-left fa-lg"></i></a>
    

    
    
        <a href="/page/11/" aria-label="Previous" class="label-pagination"><i class="fa fa-angle-left fa-lg"></i></a>
    

    
        <a href="/" class="label-pagination">1</a>
    
        <a href="/page/2/" class="label-pagination">2</a>
    
        <a href="/page/3/" class="label-pagination">3</a>
    
        <a href="/page/4/" class="label-pagination">4</a>
    
        <a href="/page/5/" class="label-pagination">5</a>
    
        <a href="/page/6/" class="label-pagination">6</a>
    
        <a href="/page/7/" class="label-pagination">7</a>
    
        <a href="/page/8/" class="label-pagination">8</a>
    
        <a href="/page/9/" class="label-pagination">9</a>
    
        <a href="/page/10/" class="label-pagination">10</a>
    
        <a href="/page/11/" class="label-pagination">11</a>
    
        <a href="/page/12/" class="label-pagination">12</a>
    
        <a href="/page/13/" class="label-pagination">13</a>
    
        <a href="/page/14/" class="label-pagination">14</a>
    

    
    
        <a href="/page/13/" aria-label="Next" class="label-pagination"><i class="fa fa-angle-right fa-lg"></i></a>
     
    
    
    
        <a href="/page/14/" aria-label="Last"><i class="fa fa-angle-double-right fa-lg"></i></a>
    

</div>


   

  </div>

  


<aside class="sidebar thirds">
  <section class="first odd">
    <h1>Header in sidebar</h1>
    <p>This is the rest of the sidebar - here's a link to <a target="_blank" href="http://google.com" title="Google">Google</a>.</p>


    
    <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>

<ins class="adsbygoogle"
     style="display:inline-block;width:336px;height:280px"
     data-ad-client="ca-pub-9874493018367687"
     data-ad-slot="8676429051"></ins>
<script>
(adsbygoogle = window.adsbygoogle || []).push({});
</script>


    <p>If links are set in config file, there will be links to keybase/github/twitter/linkedin etc.</p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/go-analysis-using-clusterprofiler/">GO analysis using clusterProfiler</a>
      </li>
      
    </ul>
  </section>

</aside>
    
 </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>
   