<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="generator" content="Hugo 0.89.4 with theme Tranquilpeak 0.4.7-BETA">
<meta name="author" content="Guangchuang Yu">
<meta name="keywords" content="">
<meta name="description" content="Guangchuang&#39;s Blog">


<meta property="og:description" content="Guangchuang&#39;s Blog">
<meta property="og:type" content="website">
<meta property="og:title" content="clusterProfiler">
<meta name="twitter:title" content="clusterProfiler">
<meta property="og:url" content="https://guangchuangyu.github.io/tags/clusterprofiler/">
<meta property="twitter:url" content="https://guangchuangyu.github.io/tags/clusterprofiler/">
<meta property="og:site_name" content="Guangchuang Yu">
<meta property="og:description" content="Guangchuang&#39;s Blog">
<meta name="twitter:description" content="Guangchuang&#39;s Blog">
<meta property="og:locale" content="en-us">


<meta name="twitter:card" content="summary">

  <meta name="twitter:site" content="@guangchuangyu">


  <meta name="twitter:creator" content="@guangchuangyu">










  <meta property="og:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
  <meta property="twitter:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">


    <title>clusterProfiler</title>

    <link rel="icon" href="https://guangchuangyu.github.io/favicon.png">
    
      <link rel="alternate" type="application/rss+xml" title="RSS" href="https://guangchuangyu.github.io/tags/clusterprofiler/index.xml">
    

    

    <link rel="canonical" href="https://guangchuangyu.github.io/tags/clusterprofiler/">

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/style-twzjdbqhmnnacqs0pwwdzcdbt8yhv8giawvjqjmyfoqnvazl0dalmnhdkvp7.min.css" />
    
    

    
      
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-77901140-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="1">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/#about">
    
    
    
      
        <img class="header-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="1">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/#about">
          <img class="sidebar-profile-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">Bioinformatics Professor @ SMU</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://twitter.com/guangchuangyu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-twitter"></i>
      
      <span class="sidebar-button-desc">Twitter</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/blog_images/biobabble.jpg">
    
      <i class="sidebar-button-icon fa fa-lg fa-weixin"></i>
      
      <span class="sidebar-button-desc">WeChat</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>


    </ul>
  </div>
</nav>

      
        

      
      <div id="main" data-behavior="1"
        class="
               hasCoverMetaIn
               ">
        
          <section class="postShorten-group main-content-wrap">
            
            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/">
          [Bioc 32] NEWS of my BioC packages
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-10-15T16:13:57&#43;08:00">
        
  October 15, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>
<h3 id="gosemsim">GOSemSim</h3>
<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">
          functional enrichment analysis with NGS data
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-08-21T03:22:21&#43;08:00">
        
  August 21, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/genomics">Genomics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>
<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt=""></p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">
          functional enrichment for GTEx paper
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-08-13T05:44:27&#43;08:00">
        
  August 13, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>
<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt=""></p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">
          dotplot for enrichment result
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-06-23T05:37:09&#43;08:00">
        
  June 23, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/visualization">Visualization</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">
          use clusterProfiler as an universal enrichment analysis tool
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-05-11T08:32:04&#43;08:00">
        
  May 11, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/systems-biology">Systems Biology</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>
for human) and <a href="http://www.reactome.org/">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>
<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages/">
          [Bioc 31] NEWS of my BioC packages
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-04-20T11:44:01&#43;08:00">
        
  April 20, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt=""></p>
<h3 id="gosemsim">GOSemSim</h3>
<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>
<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/GOSemSim/gosemsim_stats.png" alt=""></p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">
          DAVID functional analysis with clusterProfiler
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-03-16T06:33:34&#43;08:00">
        
  March 16, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p><a href="https://github.com/GuangchuangYu/clusterProfiler">clusterProfiler</a> was
used to visualize <a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID</a>
results in a paper published in <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3907017/">BMC
Genomics</a>.</p>
<p><img src="http://guangchuangyu.github.io/blog_images/2015/clusterProfiler_david.png" alt=""></p>
<p>Some users told me that they may want to use DAVID at some
circumstances. I think it maybe a good idea to make clusterProfiler
supports DAVID, so that
<a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID</a> users can use
visualization functions provided by
<a href="https://github.com/GuangchuangYu/clusterProfiler">clusterProfiler</a>.</p>
<pre><code>require(DOSE)
require(clusterProfiler)
data(geneList)
gene = names(geneList)[abs(geneList) &gt; 2]
david = enrichDAVID(gene = gene, idType=&quot;ENTREZ_GENE_ID&quot;, 
listType=&quot;Gene&quot;, annotation=&quot;KEGG_PATHWAY&quot;)



&gt; summary(david)
               ID            Description GeneRatio  BgRatio       pvalue
hsa04110 hsa04110             Cell cycle     11/68 125/5085 4.254437e-06
hsa04114 hsa04114         Oocyte meiosis     10/68 110/5085 1.119764e-05
hsa03320 hsa03320 PPAR signaling pathway      7/68  69/5085 2.606715e-04
             p.adjust qvalue                                             geneID
hsa04110 0.0003998379     NA 9133/4174/890/991/1111/891/7272/8318/4085/983/9232
hsa04114 0.0005261534     NA    9133/5241/51806/3708/991/891/4085/983/9232/6790
hsa03320 0.0081354974     NA                 4312/2167/5346/5105/3158/9370/9415
         Count
hsa04110    11
hsa04114    10
hsa03320     7
</code></pre>
<p>There are only 5085 human genes annotated by KEGG, this is due to
out-of-date DAVID data.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
            
  <div class="pagination-bar">
    <ul class="pagination">
      
        
          <li class="pagination-prev">
            <a class="btn btn--default btn--small" href="https://guangchuangyu.github.io/tags/clusterprofiler/page/5/">
              <i class="fa fa-angle-left text-base icon-mr"></i>
              <span>NEWER POSTS</span>
            </a>
          </li>
        
        
          <li class="pagination-next">
            <a class="btn btn--default btn--small" href="https://guangchuangyu.github.io/tags/clusterprofiler/page/7/">
              <span>OLDER POSTS</span>
              <i class="fa fa-angle-right text-base icon-ml"></i>
            </a>
          </li>
        
      
      <li class="pagination-number">page 6 of 7</li>
    </ul>
  </div>


          </section>
        
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2021 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">Bioinformatics Professor @ SMU</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Bioinformatics Professor
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Guangzhou
      </div>
    
  </div>
</div>

    

    
  
    
      <div id="cover" style="background-image:url('https://guangchuangyu.github.io/images/cover.jpg');"></div>
    
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.7/js/jquery.fancybox.min.js" integrity="sha256-GEAnjcTqVP+vBp3SSc8bEDQqvWAZMiHyUSIorrWwH50=" crossorigin="anonymous"></script>


<script src="https://guangchuangyu.github.io/js/script-pcw6v3xilnxydl1vddzazdverrnn9ctynvnxgwho987mfyqkuylcb1nlt.min.js"></script>


<script lang="javascript">
window.onload = updateMinWidth;
window.onresize = updateMinWidth;
document.getElementById("sidebar").addEventListener("transitionend", updateMinWidth);
function updateMinWidth() {
  var sidebar = document.getElementById("sidebar");
  var main = document.getElementById("main");
  main.style.minWidth = "";
  var w1 = getComputedStyle(main).getPropertyValue("min-width");
  var w2 = getComputedStyle(sidebar).getPropertyValue("width");
  var w3 = getComputedStyle(sidebar).getPropertyValue("left");
  main.style.minWidth = `calc(${w1} - ${w2} - ${w3})`;
}
</script>

<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight > code, pre > code').each(function(i, block) {
    if (!$(this).hasClass('codeblock')) {
      $(this).addClass('codeblock');
    }
    hljs.highlightBlock(block);
  });
});
</script>





    
  </body>
</html>

