<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/categories/r/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/categories/r/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/categories/r/">

  

  <title>R | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>R</h1>

  

  
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a></h2>
    <div class="article-style">
      
      <p>In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf" target="_blank">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145" target="_blank">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno" target="_blank">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/" target="_blank">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/" target="_blank">PeakAnalyzer</a>.</p>

<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/">ggtree annotate phylogenetic tree with local images</a></h2>
    <div class="article-style">
      
      <p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/annotation_image/Screenshot%202015-08-01%2020.53.50.png" alt="" /></p>

<p>In <a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>,
we provide a function <em>annotation_image</em> for annotating phylogenetic
tree with images.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a></h2>
    <div class="article-style">
      
      <p><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/" target="_blank">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a></h2>
    <div class="article-style">
      
      <p>My R/Bioconductor package,
<a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a>, published
in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145" target="_blank">Bioinformatics</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a> had been
cited by <a href="http://www.biomedcentral.com/1471-2164/16/292" target="_blank">http://www.biomedcentral.com/1471-2164/16/292</a> and
<a href="http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short" target="_blank">http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short</a>, and
was used (not cited) in
<a href="http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract" target="_blank">http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract</a>
and
<a href="http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract" target="_blank">http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract</a>.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/flip-and-rotate-branches-in-ggtree/">flip and rotate branches in ggtree</a></h2>
    <div class="article-style">
      
      <p><strong><em>flip</em></strong> function is a feature request from ggtree user. It accepts a
tree view and two node numbers and exchange the positions of the
selected clades.</p>

<pre><code>require(ggtree)
set.seed(2015-07-01)
tr &lt;- rtree(30)
p &lt;- ggtree(tr) + geom_text(aes(label=node))
gridExtra::grid.arrange(p, flip(p, 38, 33), ncol=2)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
    <div class="article-style">
      
      This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/">ggtree with funny fonts</a></h2>
    <div class="article-style">
      
      <p><a href="http://cran.r-project.org/web/packages/showtext/index.html" target="_blank">showtext</a>
is a neat solution to use various types of fonts in R graphs and make it
easy to use funny fonts. With
<a href="http://cran.r-project.org/web/packages/showtext/index.html" target="_blank">showtext</a>,
we can draw phylogenetic tree with different types of fonts even with
symbolic/icon fonts.</p>

<pre><code>require(showtext)

font.add.google(&quot;Gochi Hand&quot;, &quot;gochi&quot;)
font.add.google(&quot;Rock Salt&quot;, &quot;rock&quot;)

link = &quot;http://img.dafont.com/dl/?f=wm_people_1&quot;;
download.file(link, &quot;wmpeople1.zip&quot;, mode = &quot;wb&quot;);
unzip(&quot;wmpeople1.zip&quot;);
font.add(&quot;wmpeople1&quot;, &quot;wmpeople1.TTF&quot;);

link = &quot;http://img.dafont.com/dl/?f=emoticons&quot;;
download.file(link, &quot;emoticons.zip&quot;, mode = &quot;wb&quot;);
unzip(&quot;emoticons.zip&quot;);
font.add(&quot;emoticons&quot;, &quot;emoticons.ttf&quot;);


showtext.auto()
set.seed(2015-05-14)

library(ggplot2)
library(ggtree)

tree &lt;- rtree(30)
ggtree(tree, color=&quot;darkgreen&quot;) + geom_tiplab(family=&quot;gochi&quot;) + theme_classic() +
theme(axis.text.x=element_text(size=rel(4), family=&quot;emoticons&quot;, color=&quot;firebrick&quot;)) +
scale_y_continuous(breaks=seq(0, 30, 4), labels=letters[c(12:17, 20, 21)]) +
theme(axis.text.y=element_text(size=rel(4), family=&quot;wmpeople1&quot;, color=&quot;#FDAC4F&quot;)) +
annotate(&quot;text&quot;, x=2, y=18, label=&quot;Have fun with ggtree!&quot;,
family=&quot;rock&quot;, angle=30, size=12, color=&quot;steelblue&quot;)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/">BMC favors source code plagiarism</a></h2>
    <div class="article-style">
      
      <p>I found source code plagiarism a year ago and reported this case to <a href="http://www.biomedcentral.com/bmcsystbiol" target="_blank">BMC Systems Biology</a>:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot 2015-05-27 19.56.58.png" alt="" /></p>

<p>I listed source code of many functions that are exactly copied from
<a href="http://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" target="_blank">GOSemSim</a>
with only function name changed in my email. The detail of source code
plagiarism can also be found at <strong><a href="http://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/" target="_blank">Proper use of
GOSemSim</a></strong>.</p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a></h2>
    <div class="article-style">
      
      <p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/" target="_blank">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/" target="_blank">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp" target="_blank">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/" target="_blank">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/" target="_blank">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
for human) and <a href="http://www.reactome.org/" target="_blank">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide" target="_blank">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/" target="_blank">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/" target="_blank">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/">using GOSemSim to rank proteins obtained by co-IP</a></h2>
    <div class="article-style">
      
      Co-IP is usually used to identified interactions among specific
proteins. It is widely used in detecting protein complex. Unfortunately,
an identified protein may not be an interactor, and sometimes can be a
background contaminant. Ranking proteins can help us to focus a study on
a few high quality candidates for subsequent interaction investigation.
My R package <a href="https://github.com/GuangchuangYu/GOSemSim" target="_blank">GOSemSim</a> has
been used in many studies to investigate protein protein interaction,
see <a href="http://pubs.rsc.org/en/content/articlehtml/2014/mb/c3mb70486f" target="_blank">1</a>,
<a href="http://www.sciencedirect.com/science/article/pii/S1874391915000937" target="_blank">2</a>,
<a href="http://www.sciencedirect.com/science/article/pii/S1476927115000493" target="_blank">3</a>,
<a href="http://www.mcponline.org/content/13/5/1318.short" target="_blank">4</a> and
<a href="http://scholar.google.com/scholar?oi=bibs&amp;hl=en&amp;cites=9484177541993722322,16048119560581430155,18126401808149291947" target="_blank">more</a>.

      
    </div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/categories/r/page/5/">&laquo; Previous</a></li>
    
    
    <li><a href="/categories/r/page/7/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

