<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/categories/r/">

    <title>R | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno">Bug of R package ChIPpeakAnno</a></h2>
        <div class="article-style"><p><p>I used R package <strong>ChIPpeakAnno</strong> for annotating peaks, and found that
it handle the DNA strand in the wrong way. Maybe the developers were
from the computer science but not biology background.</p>

<pre><code>&gt; require(ChIPpeakAnno)
&gt; packageVersion(&quot;ChIPpeakAnno&quot;)
[1] '2.10.0'
&gt; peak &lt;- RangedData(space=&quot;chr1&quot;, IRanges(24736757, 24737528))
&gt; data(TSS.human.GRCh37)
&gt; ap &lt;- annotatePeakInBatch(peak, Annotation=TSS.human.GRCh37)
&gt; ap
RangedData with 1 row and 9 value columns across 1 space
                     space               ranges |        peak      strand
                               |  
1 ENSG00000001461        1 [24736757, 24737528] |           1           +
                          feature start_position end_position insideFeature
                                   
1 ENSG00000001461 ENSG00000001461       24742284     24799466      upstream
                  distancetoFeature shortestDistance fromOverlappingOrNearest
                                                
1 ENSG00000001461             -5527             4756             NearestStart
</code></pre>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters">clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</a></h2>
        <div class="article-style"><p><p>Increasing quantitative data generated from transcriptomics and
proteomics require integrative strategies for analysis. Here, we present
an R package, clusterProfiler that automates the process of
biological-term classification and the enrichment analysis of gene
clusters. The analysis module and visualization module were combined
into a reusable workflow. Currently, clusterProfiler supports three
species, including humans, mice, and yeast. Methods provided in this
package can be easily extended to other species and ontologies. The
clusterProfiler package is released under Artistic-2.0 License within
Bioconductor project. The source code and vignette are freely available
at
<a href="http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html</a></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2012/02/draw-chinese-character-jiong-using-ggplot2">draw chinese character Jiong using ggplot2</a></h2>
        <div class="article-style"><p><p>The Chinese character Jiong (囧) is now widely used for expressing ideas
or feelings such as annoyance, shock, embarrassment, awkwardness, scorn.</p>

<p>The function plot of <code>y=1/(x^2-1)</code> looks very similar of this symbol.</p>

<p>I use ggplot2 to draw the symbol of Jiong.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/08/ggplot2-version-of-figures-in-25-recipes-for-getting-started-with-r">ggplot2 Version of Figures in &lt;25 Recipes for Getting Started with R&gt;</a></h2>
        <div class="article-style"><p><p>In order to provide an option to compare graphs produced by basic
internal plot function and ggplot2, I have recreated the figures in the
book, <a href="http://book.douban.com/subject/5992544/">25 Recipes for Getting Started with
R</a>, with ggplot2.</p>

<p>The code used to create the images is in separate paragraphs, allowing
easy comparison. <strong>1.16 Creating a Scatter Plot</strong></p>

<pre><code>plot(cars)
</code></pre>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/08/the-batman-equation">the batman equation</a></h2>
        <div class="article-style"><p><p><a href="http://www.hardocp.com/news/2011/07/29/batman_equation/">HardOCP</a> has
an image with an equation which apparently draws the Batman logo.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/06/r-package-dose-released">R package DOSE released</a></h2>
        <div class="article-style"><p><p>Disease Ontology (DO) provides an open source ontology for the
integration of biomedical data that is associated with human disease. DO
analysis can lead to interesting discoveries that deserve further
clinical investigation.</p>

<p>DOSE was designed for semantic similarity measure and enrichment
analysis.</p>

<p>Four information content (IC)-based methods, proposed by Resnik, Jiang,
Lin and Schlicker, and one graph structure-based method, proposed by
Wang, were implemented. The calculation details can be referred to the
vignette of R package
<a href="http://www.bioconductor.org/packages/2.9/bioc/html/GOSemSim.html" title="GOSemSim">GOSemSim</a>.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/06/r-package-dose-released">R package DOSE released</a></h2>
        <div class="article-style"><p></p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8">clusterProfiler in Bioconductor 2.8</a></h2>
        <div class="article-style"><p><p>In recently years, high-throughput experimental techniques such as
microarray and mass spectrometry can identify many lists of genes and
gene products. The most widely used strategy for high-throughput data
analysis is to identify different gene clusters based on their
expression profiles. Another commonly used approach is to annotate these
genes to biological knowledge, such as Gene Ontology (GO) and Kyoto
Encyclopedia of Genes and Genomes (KEGG), and identify the statistically
significantly enriched categories. These two different strategies were
implemented in many bioconductor packages, such as <em>Mfuzz</em> and <em>BHC</em> for
clustering analysis and <em>GOstats</em> for GO enrichment analysis.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2010/05/gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products">GOSemSim an R package for measuring semantic similarity among GO terms and gene products</a></h2>
        <div class="article-style"><p><p><a href="http://www.ncbi.nlm.nih.gov/pubmed/20179076"><img src="http://bioinformatics.oxfordjournals.org/content/26/18.cover.gif" alt="" title="bioinfo" /></a></p>

<h3 id="abstract:c41832c6f53a1eda12f563057a868592">Abstract</h3>

<p><strong>SUMMARY</strong>: The semantic comparisons of Gene Ontology (GO) annotations
provide quantitative ways to compute similarities between genes and gene
groups, and have became important basis for many bioinformatics analysis
approaches. GOSemSim is an R package for semantic similarity computation
among GO terms, sets of GO terms, gene products and gene clusters. Four
information content (IC)- and a graph-based methods are implemented in
the GOSemSim package, multiple species including human, rat, mouse, fly
and yeast are also supported. The functions provided by the GOSemSim
offer flexibility for applications, and can be easily integrated into
high-throughput analysis pipelines. AVAILABILITY: GOSemSim is released
under the GNU General Public License within Bioconductor project, and
freely available at
<a href="http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html">http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html</a>.</p>
</p></div>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="https://guangchuangyu.github.io//categories/r/page/5/">« Previous</a></li>
        
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

