<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2015 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">use emoji font in R</a>
      </h1>
      <span class="post-date">Dec 16, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggplot2">ggplot2</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/emoji">emoji</a>
      </span>
      
      I have played with emoji in R for a while. My solution of using it is different from what implemented in emoGG. emoGG is a good attemp to add emoji in ggplot2. It render emoji picture (png) and creat a layer, geom_emoji, to add emoji. In my opinion, emoji should be treated as ordinary font in user interface, albeit it maynot be true internally. It would be more flexible if we can use emoji as ordinary font and in this way user don&rsquo;t need to learn extra stuff.
      <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/">å–œæ–°åŒæ—§ä¹‹hello Hugo</a>
      </h1>
      <span class="post-date">Dec 4, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/blog">blog</a><a class="label" href="http://guangchuangyu.github.io/categories/hugo">hugo</a><a class="label" href="http://guangchuangyu.github.io/categories/wordpress">wordpress</a>
      </span>
      
      Installation å®‰è£…hugoå¯¹äºMacç”¨æˆ·æ¥è¯´ï¼Œå¤ªå®¹æ˜“ï¼Œbrewå¤§æ³•æå®šã€‚ brew install hubo MathJax support æŠŠä¸‹é¢çš„ä»£ç : &lt;script type=&quot;text/javascript&quot; src=&quot;https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML&quot;&gt; &lt;/script&gt; åŠ å…¥åˆ°æ–‡ä»¶themes/hyde-x/layouts/partials/foot.htmlä¸­ï¼Œå°±å¯ä»¥å®ç°$\LaTeX$å…¬å¼çš„è§£æã€‚ migrate wordpress to hugo Disqus Comment System plugin ç‚¹å‡»Export comments to Disqusæå®š convert Wordpress posts to markdown files è¯•ç”¨äº†å¤šä¸ªæ’ä»¶æˆ–è„šæœ¬ï¼Œéƒ½æ˜¯å„ç§å‡ºé”™ã€‚æœ€åå‘ç°äº†ç¥å™¨potter-wordpressï¼Œé€šè¿‡npmå®‰è£…ï¼š npm install -g pottercms-wordpress ç„¶åä¸€ä¸ªè¯­å¥: potter-wordpress -h WORDPRESS_SITE_URL -u USER_NAME -p PASSWORD --dir PATH_TO_STORE_MD_FILES è‡ªåŠ¨ä»çº¿ä¸ŠæŠŠpostsæŠ“ä¸‹æ¥ï¼Œå¹¶è½¬åŒ–ä¸ºmarkdownã€‚åœ¨ä¹‹å‰å„ç§æä¸å®šä¹‹åï¼Œçœ‹åˆ°ä¸‹é¢çš„è¾“å‡ºæœ‰ç§è«åçš„å¿«æ„Ÿã€‚ Fetching articles... Got : [Sun Nov 29 2015] R kernel in Jupyter notebook 3 Got : [Mon Nov 09 2015] phylomoji with ggtree Got : [Tue Nov 17 2015] åˆä½“ä¸­çš„æ“ä½œç³»ç»Ÿ Got : [Mon Nov 02 2015] Comparison of clusterProfiler and GSEA-P Got : [Thu Oct 01 2015] ChIPseq data mining with ChIPseeker Got : [Tue Oct 06 2015] TRIE Got : [Thu Oct 15 2015] [BioC 3.2] NEWS of my BioC packages Got : [Wed Oct 21 2015] use simplify to remove redundancy of enriched GO terms ...
      <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/">R kernel in Jupyter notebook 3</a>
      </h1>
      <span class="post-date">Nov 29, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/jupyter">jupyter</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>
      
      I followed the post, Installing an R kernel for IPython/jupyter notebook 3 on OSX, to install jupyter with python3 and R kernels in my iMac. I have elementaryOS on my Macbook Pro and also want to have jupyter on it. The installation process is quite similar. Install Jupyter sudo apt-get install python3-pip sudo pip3 install jupyter Then we can use the following command to start jupyter: ipython notebook Install IRkernel To compile IRkernel, we should firstly have zmq lib installed.
      <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/">åˆä½“ä¸­çš„æ“ä½œç³»ç»Ÿ</a>
      </h1>
      <span class="post-date">Nov 17, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/mac-os">Mac OS</a><a class="label" href="http://guangchuangyu.github.io/categories/windows">Windows</a>
      </span>
      
      æ›´æ–°EI Capitanå·²ç»è¿‘ä¸€ä¸ªæœˆï¼Œç¬¬ä¸€æ„Ÿè§‰å°±æ˜¯æ˜æ˜¾å˜æ…¢äº†ã€‚è¿™æ˜¯å»å¹´ä¹°çš„27&rsquo; iMacï¼Œå¿…é¡»ä¸æ…¢ã€‚MS officeç¨‹åºå¯ä»¥æ‰“å¼€ï¼Œä½†æ‰“å¼€æ–‡æ¡£ç›´æ¥é—ªé€€ã€‚å‡çº§å®‰è£…æ€»ä¼šæœ‰äº›é—®é¢˜ï¼Œå°±åƒæˆ‘å»å¹´åœ¨ç¬”è®°æœ¬ä¸Šå‡çº§yosemiteä¸€æ ·ï¼Œæœ€åä¸¤åˆ†é’Ÿå§‹ç»ˆè¿‡ä¸å»ï¼Œè¿«ä½¿æˆ‘å…¨æ–°å®‰è£…ï¼Œå½“æ—¶æˆ‘çœ‹åå°è®°å½•å°±å‘ç°äº†æ˜¯MacTexæ‹·ä¸è¿‡å»ï¼Œåæ¥MacTexçš„ç½‘ç«™ä¸Šä¹Ÿå†™äº†å‡çº§Yosemiteæœ‰è¿™ä¸ªé—®é¢˜ã€‚å‡çº§æœ‰äº›å°é—®é¢˜æ˜¯å¯ä»¥é¢„è§çš„ï¼Œä½†å¾ˆå°‘æœ‰åƒEI Capitanè¿™ä¹ˆå¤šçš„é—®é¢˜ã€‚ Macworld EI Capitan reviewæ–‡ç« ä¸­æœ‰äººè¿™æ ·è¯„è®ºï¼š haha nice review ! did you tried the update ?? slow connection safari freezes on launch outlook crash printer does not work anymore great job from Apple . My confidence in Apple has been seriously dented! æœ€ä¸å¯æ€è®®çš„æ˜¯å¼€å‘è€…å‘ç°å‡çº§åå¼€å‘å·¥å…·éƒ½æœ‰é—®é¢˜äº†ï¼š Last night I installed OSXâ€™s new â€œEl Capitainâ€ update, and it broke most of the tools I use in my daily life as a developer, including homebrew and git.
      <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">Comparison of clusterProfiler and GSEA-P</a>
      </h1>
      <span class="post-date">Nov 2, 2015 &middot; 5 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/gsea">GSEA</a>
      </span>
      
      Thanks @mevers for raising the issue to me and his efforts in benchmarking clusterProfiler. He pointed out two issues: outputs from gseGO and GSEA-P are poorly overlap. pvalues from gseGO are generally smaller and don&rsquo;t show a lot of variation For GSEA analysis, we have two inputs, a ranked gene list and gene set collections. First of all, the gene set collections are very different. The GMT file used in his test is c5.cc.v5.0.symbols.gmt, which is a tiny subset of GO CC, while clusterProfiler used the whole GO CC corpus.
      <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">use simplify to remove redundancy of enriched GO terms</a>
      </h1>
      <span class="post-date">Oct 21, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a>
      </span>
      
      To simplify enriched GO result, we can use slim version of GO and use enricher function to analyze. Another strategy is to use GOSemSim to calculate similarity of GO terms and remove those highly similar terms by keeping one representative term. To make this feature available to clusterProfiler users, I develop a simplify method to reduce redundant GO terms from output of enrichGO function. require(clusterProfiler) data(geneList, package=&quot;DOSE&quot;) de &lt;- names(geneList)[abs(geneList) &gt; 2] bp &lt;- enrichGO(de, ont=&quot;BP&quot;) enrichMap(bp) The enrichMap doesn&rsquo;t display the whole picture as we use the default value n=50 to only show 50 highly significant terms.
      <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/">BioC 3.2: NEWS of my BioC packages</a>
      </h1>
      <span class="post-date">Oct 15, 2015 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/bioc">BioC</a><a class="label" href="http://guangchuangyu.github.io/categories/gosemsim">GOSemSim</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>
      
      In BioC 3.2 release, all my packages including GOSemSim, clusterProfiler, DOSE, ReactomePA, and ChIPseeker switch from Sweave to R Markdown for package vignettes. GOSemSim To make it consistent between GOSemSim and clusterProfiler, &lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual, information content data was updated. clusterProfiler Enrichment results may contains terms that are very general (less informative) and we do not want to use them. In this release, we provide dropGO function that can be used to drop selected GO terms or specific level of GO terms.
      <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">ChIPseq data mining with ChIPseeker</a>
      </h1>
      <span class="post-date">Oct 1, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/systems-biology">Systems Biology</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>
      
      ChIP-seq is rapidly becoming a common technique and there are a large number of dataset available in the public domain. Results from individual experiments provide a limited understanding of chromatin interactions, as there is many factors cooperate to regulate transcription. Unlike other tools that designed for single dataset, ChIPseeker is designed for comparing profiles of ChIP-seq datasets at different levels. We provide functions to compare profiles of peaks binding to TSS regions, annotation, and enriched functional profiles.
      <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">subsetting data in ggtree</a>
      </h1>
      <span class="post-date">Sep 24, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>
      
      Subsetting is commonly used in ggtree as we would like to for example separating internal nodes from tips. We may also want to display annotation to specific node(s)/tip(s). Some software may stored clade information (e.g. bootstrap value) as internal node labels. Indeed we want to manipulate such information and taxa labels separately. In current ggplot2 (version=1.0.1, access date:2015-09-23), it support subset. For instance: library(ggplot2) library(ggtree) tree=read.tree(text=&quot;((A:2,B:2)95:2,(C:2,D:2)100:2);&quot;) ggtree(tree) + geom_text(aes(label=label), subset=.(!isTip), hjust=-.2) But this feature was removed in github version of ggplot2 and will not be available in next release.
      <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">Read On &rarr;</a>
      
    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">comic phylogenetic tree with ggtree and comicR</a>
      </h1>
      <span class="post-date">Sep 8, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>
      
      ggtree applies the concepts of grammar of graphic in phylogenetic tree presentation and make it easy to add multiple layers of text and even figures above a ğŸŒ². Here, I cartoonize a phylogenetic tree generated by ggtree with comicR, which is a funny package to generate comic (xkcd-like) graph in R. Have fun with ggtree and comicR. library(htmltools) library(XML) library(gridSVG) library(comicR) library(ggplot2) library(ggtree) p &lt;- ggtree(rtree(30), layout=&quot;fan&quot;) + geom_text(aes(label=label, angle=angle),
      <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">Read On &rarr;</a>
      
    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        class="disabled">
        <a href="" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        class="active"><a href="/">1</a></li>
        
        <li
        ><a href="/page/2/">2</a></li>
        
        <li
        ><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        ><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        >
        <a href="/page/2/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/8/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    
  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>



</body>
</html>

