<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Guangchuang Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="Guangchuang Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
 <div id="content">
  <div class="blog-index">
    
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
        </h1>
        <p class="meta">Feb 23, 2016 - 4 minute read -  <a href="http://guangchuangyu.github.io/2016/02/tweets-of-ggtree/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Treespace visualization of a heuristic best tree search in <a href="https://twitter.com/hashtag/R?src=hash">#R</a> package <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> <a href="https://twitter.com/hashtag/phylogenetics?src=hash">#phylogenetics</a> <a href="https://t.co/hnSJpIABql">pic.twitter.com/hnSJpIABql</a></p>&mdash; Alex Dami√°n (@antropoteuthis) <a href="https://twitter.com/antropoteuthis/status/693123105707986944">January 29, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr"><a href="https://twitter.com/guangchuangyu">@guangchuangyu</a> Updated to latest <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> and <a href="https://twitter.com/hashtag/ggplot2?src=hash">#ggplot2</a> and very impressed! Any ideas on using geom_label with tip labels to add box/fill?</p>&mdash; Jo Williams-Newkirk (@ajwnewkirk) <a href="https://twitter.com/ajwnewkirk/status/691762302492790788">January 25, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">ggtree looks really good! <a href="https://twitter.com/hashtag/rstats?src=hash">#rstats</a><a href="https://t.co/7rtexYV3I5">https://t.co/7rtexYV3I5</a></p>&mdash; Paul Staab (@paulstaab) <a href="https://twitter.com/paulstaab/status/687933344219312128">January 15, 2016</a></blockquote>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/02/tweets-of-ggtree/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
        </h1>
        <p class="meta">Feb 16, 2016 - 1 minute read -  <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/publication/">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a><a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics/">Bioinformatics</a>
        </p>

      </header>

        
          <p>My R/Bioconductor package, ReactomePA, published in Molecular BioSystems. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization G Yu, QY He. Molecular BioSystems 2016, 12:477-479 Received: 05 Oct 2015 Accepted: 20 Nov 2015 Online: 23 Nov 2015 DOI: 10.1039/C5MB00663E Reactome is a manually curated pathway annotation database for unveiling high-order biological pathways from high-throughput data. ReactomePA is an R/Bioconductor package providing enrichment analyses, including hypergeometric test and gene set enrichment analyses.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
        </h1>
        <p class="meta">Jan 15, 2016 - 1 minute read -  <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>Phylip is also a widely used tree format, which contains taxa sequences with Newick tree text.</p>

<p>In <code>ggtree</code>, we can use <code>read.phylip()</code> function to parse the file and use <code>ggtree()</code> to visualize the tree.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
        </h1>
        <p class="meta">Jan 13, 2016 - 1 minute read -  <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p>This is a question from ggtree user. In ape and phytools, it&rsquo;s easy to label edge using the edgelabels function. set.seed(1) tr = rtree(30) library(ape) plot(tr, main=&quot;ape&quot;) edgelabels() I don&rsquo;t see any necessity to label edge numbers, as they are meaningless. The number is labeled as the row index of tr$edge, and edge can be uniquely mapped to child node. If we need to relate something to edge, we can relate them to corresponding child node.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
        </h1>
        <p class="meta">Jan 11, 2016 - 3 minute read -  <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p>ggtree implemented a function, subview, that can add subplots on a ggplot2 object. It had successful applied to plot pie graphs on map. set.seed(2016-01-04) tr &lt;- rtree(30) tr &lt;- groupClade(tr, node=45) p &lt;- ggtree(tr, aes(color=group)) + geom_tippoint() cpos &lt;- get_clade_position(p, node=45) p1 &lt;- p + geom_hilight(node=45) p2 &lt;- with(cpos, p+xlim(xmin, xmax*1.01)+ylim(ymin, ymax)) with(cpos, subview(p2+geom_tiplab(), p1+theme_transparent(), x=xmin+(xmax-xmin)*.15, y=ymin+(ymax-ymin)*.85)) To make it more easy to use subview function for annotating taxa with subplots, ggtree provides a function, inset, for adding subplots to a phylogenetic tree.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/">GO analysis using clusterProfiler</a>
        </h1>
        <p class="meta">Jan 4, 2016 - 9 minute read -  <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p>clusterProfiler supports over-representation test and gene set enrichment analysis of Gene Ontology. It supports GO annotation from OrgDb object, GMT file and user&rsquo;s own data. support many species In github version of clusterProfiler, enrichGO and gseGO functions removed the parameter organism and add another parameter OrgDb, so that any species that have OrgDb object available can be analyzed in clusterProfiler. Bioconductor have already provide OrgDb for about 20 species, see http://bioconductor.org/packages/release/BiocViews.html#___OrgDb, and users can build OrgDb via AnnotationHub.</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    
    






<div class="pagination">

    
        
        
        
        <a href="/" aria-label="First" class="label-pagination"><i class="fa fa-angle-double-left fa-lg"></i></a>
    

    

    
        <a href="/" class="label-pagination">1</a>
    
        <a href="/page/2/" class="label-pagination">2</a>
    
        <a href="/page/3/" class="label-pagination">3</a>
    
        <a href="/page/4/" class="label-pagination">4</a>
    
        <a href="/page/5/" class="label-pagination">5</a>
    
        <a href="/page/6/" class="label-pagination">6</a>
    
        <a href="/page/7/" class="label-pagination">7</a>
    
        <a href="/page/8/" class="label-pagination">8</a>
    
        <a href="/page/9/" class="label-pagination">9</a>
    
        <a href="/page/10/" class="label-pagination">10</a>
    
        <a href="/page/11/" class="label-pagination">11</a>
    
        <a href="/page/12/" class="label-pagination">12</a>
    
        <a href="/page/13/" class="label-pagination">13</a>
    
        <a href="/page/14/" class="label-pagination">14</a>
    

    
    
        <a href="/page/2/" aria-label="Next" class="label-pagination"><i class="fa fa-angle-right fa-lg"></i></a>
     
    
    
    
        <a href="/page/14/" aria-label="Last"><i class="fa fa-angle-double-right fa-lg"></i></a>
    

</div>


   

  </div>

  


<aside class="sidebar thirds">
  <section class="first odd">
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
     
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </li>
      
    </ul>
  </section>

  <img src="img/wechat.jpg" />
  
</aside>
    
 </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>
   