<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/douban/">douban</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2016 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </h1>
      <span class="post-date">Feb 16, 2016 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/publication">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a><a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>
      </span>


      
      My R/Bioconductor package, ReactomePA, published in Molecular BioSystems. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization G Yu, QY He. Molecular BioSystems 2016, 12:477-479 Received: 05 Oct 2015 Accepted: 20 Nov 2015 Online: 23 Nov 2015 DOI: 10.1039/C5MB00663E Reactome is a manually curated pathway annotation database for unveiling high-order biological pathways from high-throughput data. ReactomePA is an R/Bioconductor package providing enrichment analyses, including hypergeometric test and gene set enrichment analyses.
      <a href="http://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </h1>
      <span class="post-date">Jan 15, 2016 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      Phylip is also a widely used tree format, which contains taxa sequences with Newick tree text. In ggtree, we can use read.phylip() function to parse the file and use ggtree() to visualize the tree. library(ggtree) phyfile &lt;- system.file(&quot;extdata&quot;, &quot;sample.phy&quot;, package=&quot;ggtree&quot;) phylip &lt;- read.phylip(phyfile) phylip ## 'phylip' S4 object that stored information of ## '/Users/guangchuangyu/Library/R/3.2/library/ggtree/extdata/sample.phy'. ## ## ...@ tree: ## Phylogenetic tree with 15 tips and 13 internal nodes. ## ## Tip labels: ## K, N, D, L, J, G, ...
      <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </h1>
      <span class="post-date">Jan 13, 2016 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      This is a question from ggtree user. In ape and phytools, it&rsquo;s easy to label edge using the edgelabels function. set.seed(1) tr = rtree(30) library(ape) plot(tr, main=&quot;ape&quot;) edgelabels() I don&rsquo;t see any necessity to label edge numbers, as they are meaningless. The number is labeled as the row index of tr$edge, and edge can be uniquely mapped to child node. If we need to relate something to edge, we can relate them to corresponding child node.
      <a href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </h1>
      <span class="post-date">Jan 11, 2016 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      ggtree implemented a function, subview, that can add subplots on a ggplot2 object. It had successful applied to plot pie graphs on map. set.seed(2016-01-04) tr &lt;- rtree(30) tr &lt;- groupClade(tr, node=45) p &lt;- ggtree(tr, aes(color=group)) + geom_tippoint() cpos &lt;- get_clade_position(p, node=45) p1 &lt;- p + geom_hilight(node=45) p2 &lt;- with(cpos, p+xlim(xmin, xmax*1.01)+ylim(ymin, ymax)) with(cpos, subview(p2+geom_tiplab(), p1+theme_transparent(), x=xmin+(xmax-xmin)*.15, y=ymin+(ymax-ymin)*.85)) To make it more easy to use subview function for annotating taxa with subplots, ggtree provides a function, inset, for adding subplots to a phylogenetic tree.
      <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/">GO analysis using clusterProfiler</a>
      </h1>
      <span class="post-date">Jan 4, 2016 &middot; 9 minute read &middot; <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/go">GO</a>
      </span>


      
      clusterProfiler supports over-representation test and gene set enrichment analysis of Gene Ontology. It supports GO annotation from OrgDb object, GMT file and user&rsquo;s own data. support many species In github version of clusterProfiler, enrichGO and gseGO functions removed the parameter organism and add another parameter OrgDb, so that any species that have OrgDb object available can be analyzed in clusterProfiler. Bioconductor have already provide OrgDb for about 20 species, see http://bioconductor.org/packages/release/BiocViews.html#___OrgDb, and users can build OrgDb via AnnotationHub.
      <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/">News of ggtree</a>
      </h1>
      <span class="post-date">Dec 30, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      A new version of ggtree that works with ggplot2 (version &gt;= 2.0.0) is now availabel. new layers Some functions, add_legend, hilight, annotation_clade and annotation_clade2 were removed. Instead we provide layer functions, geom_treescale, geom_hilight and geom_cladelabel. You can use + operator to add layers using these layer functions. In addtion, we provide geom_point2, geom_text2 and geom_segment2 which works exactly as geom_point, geom_text and geom_segment except they allow ggtree users to do subsetting.
      <a href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">use emoji font in R</a>
      </h1>
      <span class="post-date">Dec 16, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggplot2">ggplot2</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/emoji">emoji</a>
      </span>


      
      I have played with emoji in R for a while. My solution of using it is different from what implemented in emoGG. emoGG is a good attemp to add emoji in ggplot2. It render emoji picture (png) and creat a layer, geom_emoji, to add emoji. In my opinion, emoji should be treated as ordinary font in user interface, albeit it maynot be true internally. It would be more flexible if we can use emoji as ordinary font and in this way user don&rsquo;t need to learn extra stuff.
      <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/">喜新厌旧之hello Hugo</a>
      </h1>
      <span class="post-date">Dec 4, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/blog">blog</a><a class="label" href="http://guangchuangyu.github.io/categories/hugo">hugo</a><a class="label" href="http://guangchuangyu.github.io/categories/wordpress">wordpress</a>
      </span>


      
      Installation 安装hugo对于Mac用户来说，太容易，brew大法搞定。 brew install hubo MathJax support 把下面的代码: &lt;script type=&quot;text/javascript&quot; src=&quot;https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML&quot;&gt; &lt;/script&gt; 加入到文件themes/hyde-x/layouts/partials/foot.html中，就可以实现$\LaTeX$公式的解析。 migrate wordpress to hugo Disqus Comment System plugin 点击Export comments to Disqus搞定 convert Wordpress posts to markdown files 试用了多个插件或脚本，都是各种出错。最后发现了神器potter-wordpress，通过npm安装： npm install -g pottercms-wordpress 然后一个语句: potter-wordpress -h WORDPRESS_SITE_URL -u USER_NAME -p PASSWORD --dir PATH_TO_STORE_MD_FILES 自动从线上把posts抓下来，并转化为markdown。在之前各种搞不定之后，看到下面的输出有种莫名的快感。 Fetching articles... Got : [Sun Nov 29 2015] R kernel in Jupyter notebook 3 Got : [Mon Nov 09 2015] phylomoji with ggtree Got : [Tue Nov 17 2015] 合体中的操作系统 Got : [Mon Nov 02 2015] Comparison of clusterProfiler and GSEA-P Got : [Thu Oct 01 2015] ChIPseq data mining with ChIPseeker Got : [Tue Oct 06 2015] TRIE Got : [Thu Oct 15 2015] [BioC 3.2] NEWS of my BioC packages Got : [Wed Oct 21 2015] use simplify to remove redundancy of enriched GO terms ...
      <a href="http://guangchuangyu.github.io/2015/12/%E5%96%9C%E6%96%B0%E5%8E%8C%E6%97%A7%E4%B9%8Bhello-hugo/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/">R kernel in Jupyter notebook 3</a>
      </h1>
      <span class="post-date">Nov 29, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/jupyter">jupyter</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I followed the post, Installing an R kernel for IPython/jupyter notebook 3 on OSX, to install jupyter with python3 and R kernels in my iMac. I have elementaryOS on my Macbook Pro and also want to have jupyter on it. The installation process is quite similar. Install Jupyter sudo apt-get install python3-pip sudo pip3 install jupyter Then we can use the following command to start jupyter: ipython notebook Install IRkernel To compile IRkernel, we should firstly have zmq lib installed.
      <a href="http://guangchuangyu.github.io/2015/11/r-kernel-in-jupyter-notebook-3/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/">合体中的操作系统</a>
      </h1>
      <span class="post-date">Nov 17, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/mac-os">Mac OS</a><a class="label" href="http://guangchuangyu.github.io/categories/windows">Windows</a>
      </span>


      
      更新EI Capitan已经近一个月，第一感觉就是明显变慢了。这是去年买的27&rsquo; iMac，必须不慢。MS office程序可以打开，但打开文档直接闪退。升级安装总会有些问题，就像我去年在笔记本上升级yosemite一样，最后两分钟始终过不去，迫使我全新安装，当时我看后台记录就发现了是MacTex拷不过去，后来MacTex的网站上也写了升级Yosemite有这个问题。升级有些小问题是可以预见的，但很少有像EI Capitan这么多的问题。 Macworld EI Capitan review文章中有人这样评论： haha nice review ! did you tried the update ?? slow connection safari freezes on launch outlook crash printer does not work anymore great job from Apple . My confidence in Apple has been seriously dented! 最不可思议的是开发者发现升级后开发工具都有问题了： Last night I installed OSX’s new “El Capitain” update, and it broke most of the tools I use in my daily life as a developer, including homebrew and git.
      <a href="http://guangchuangyu.github.io/2015/11/%E5%90%88%E4%BD%93%E4%B8%AD%E7%9A%84%E6%93%8D%E4%BD%9C%E7%B3%BB%E7%BB%9F/">Read On &rarr;</a>
      

    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        class="disabled">
        <a href="" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        class="active"><a href="/">1</a></li>
        
        <li
        ><a href="/page/2/">2</a></li>
        
        <li
        ><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        ><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        ><a href="/page/9/">9</a></li>
        
        <li
        >
        <a href="/page/2/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/9/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    

  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

