<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="http://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="http://guangchuangyu.github.io/categories/visualization/">

    <title>Visualization | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#contact">Contact</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2014_multiple-annotation-in-chipseeker/">multiple annotation in ChIPseeker</a></h2>
        <p><h3 id="nearest-gene-annotation:211662300f8731372460bacaae7937b5">Nearest gene annotation</h3>

<p>Almost all annotation software calculate the distance of a peak to the
nearest TSS and assign the peak to that gene. This can be misleading, as
binding sites might be located between two start sites of different
genes or hit different genes which have the same TSS location in the
genome.</p>

<p>The function <em>annotatePeak</em> provides option to assign genes with a max
distance cutoff and all genes within this distance were reported for
each peak.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/clusterprofiler/2014_enrichment-map/">enrichment map</a></h2>
        <p><p>In <a href="http://www.plob.org/">PLOB</a>&rsquo;s QQ group, someone asked how to change
the color of enrichment map in <a href="http://cytoscape.org/">Cytoscape</a>. I am
very curious how enrichment map can helps to interpret enrichment
results. It took me 2 hours to implement it using R and I am surprised
that the enrichment map is better than anticipated.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2014_use-ggplot2/">Use ggplot2</a></h2>
        <p><h3 id="why-use-ggplot2:c38c63fac86d2290b6bc01da39216614">Why use ggplot2</h3>

<p>ggplot2是我见过最human friendly的画图软件，这得益于Leland
Wilkinson在他的著作《<a href="http://book.douban.com/subject/10123863/">The Grammar of
Graphics</a>》中提出了一套图形语法，把图形元素抽象成可以自由组合的成分，Hadley
Wickham把这套想法在R中实现。</p>

<p>为什么要学习ggplot2，可以参考<a href="http://book.douban.com/subject/24527091/">ggplot2: 数据分析与图形艺术</a>的<a href="https://github.com/cosname/ggplot2-translation/blob/master/preface.md">序言</a>（btw:
在序言的最后，我被致谢了）。</p>

<p>Hadley
Wickham也给出<a href="https://github.com/hadley/ggplot2/wiki/Why-use-ggplot2">一堆理由</a>让我们说服自己，我想再补充一点，Hadley
Wickham是学医出身的，做为学生物出身的人有什么理由不支持呢:)</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2014_visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a></h2>
        <p><p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes:6a8a65de687a162a83f53f5c5a317279">ChIP peaks over Chromosomes</h3>

<pre><code>&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/2014_boxplot/">boxplot</a></h2>
        <p><p>生物坑很多人画图只会直方图，统计只会T检验，在暨大见过太多的学生连T检验都不会，分不清SEM和SD的差别，也不清楚T检验那几个简单参数的含义。我写统计笔记也是因为不想重复性地跟学生讲解T检验。</p>

<p>Barplot和T test一样普遍而流行，barplot适合于表示计数数据和比例，显示比例也可以用pie
plot，但直方图比饼图要好，因为人类的眼睛适合于比较高度，而不是弧度。</p>

<p>多半时候生物学数据并非简单的计数数据，对于测量数据，在展示数据分布时，很多人会使用他们熟悉的barplot，用高度来表示mean，然后再加上errorbar，这样展示数据，信息量是非常低的，使用boxplot能够提供更多的数据分布信息，能更好地展现数据，但可能很多人只会在excel里画barplot，Nature
Methods
2013年的文章中有100个barplot图，而只有20个boxplot图，从这里就可以看出来，用boxplot的人远远没有barplot多，于是NPG怒了，写了两篇专栏文章<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2807.html">Points of View: Bar charts and box plots</a>和<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2813.html">Points of Significance: Visualizing samples with box plots</a>并且发表了一篇<a href="http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2811.html">BoxPlotR: a web tool for generation of box plots</a>方便大家画boxplot，如此简单的web tool能够发Nature Methods，实在是让人羡慕妒忌恨啊。</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2012_draw-chinese-character-jiong-ggplot2/">draw chinese character Jiong using ggplot2</a></h2>
        <p><p>The Chinese character Jiong (囧) is now widely used for expressing ideas
or feelings such as annoyance, shock, embarrassment, awkwardness, scorn.</p>

<p>The function plot of <code>y=1/(x^2-1)</code> looks very similar of this symbol.</p>

<p>I use ggplot2 to draw the symbol of Jiong.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2012_violin-plot/">violin plot</a></h2>
        <p><p>ggplot2 0.9.0版本将在3月1号发布，改动较大，最近bioc mailing
list也有诸多讨论，因为它导致了某些包编译出错（估计是NAMESPACE引起的）。</p>

<p>按照hadley的说法是这是为了把他个人的项目变成一个社区项目。</p>

<p>大多数的改动对用户来说是invisible的，用roxygen2去注释函数，核心代码使用S3重写，分离出一些特性（scales包）等。</p>

<p>最明显的是现在画图速度变快了，不过我觉得最让人兴奋的是<code>guide_legend()</code>和<code>guide_colorbar()</code>两个函数，可以修改legend.</p>

<p>画图的话，加了四个函数<code>geom_map()</code>, <code>geom_raster()</code>,
<code>geom_dotplot()</code>和<code>geom_violin()</code>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2011_ggplot2-version-figures-25-recipes-started-r/">ggplot2 Version of Figures in &lt;25 Recipes for Getting Started with R&gt;</a></h2>
        <p><p>In order to provide an option to compare graphs produced by basic
internal plot function and ggplot2, I have recreated the figures in the
book, <a href="http://book.douban.com/subject/5992544/">25 Recipes for Getting Started with
R</a>, with ggplot2.</p>

<p>The code used to create the images is in separate paragraphs, allowing
easy comparison. <strong>1.16 Creating a Scatter Plot</strong></p>

<pre><code>plot(cars)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2011_bat-man/">the batman equation</a></h2>
        <p><p><a href="http://www.hardocp.com/news/2011/07/29/batman_equation/">HardOCP</a> has
an image with an equation which apparently draws the Batman logo.</p>
</p>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="http://guangchuangyu.github.io//categories/visualization/">« Previous</a></li>
        
        
    </ul>
</nav>



    <footer class="site-footer">
    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">
        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

</div>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

