<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/categories/visualization/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/categories/visualization/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/categories/visualization/">

  

  <title>Visualization | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Visualization</h1>

  

  
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/09/ggtree-for-outbreak-data/">ggtree for outbreak data</a></h2>
    <div class="article-style">
      
      <p><a href="https://cran.r-project.org/package=OutbreakTools" target="_blank">OutbreakTools</a> implements basic tools for the analysis of Disease Outbreaks.</p>

<p>It defines <code>S4</code> class <code>obkData</code> to store case-base outbreak data. It also provides a function, <code>plotggphy</code>, to visualize such data on the phylogenetic tree.</p>

<pre><code>library(OutbreakTools)
data(FluH1N1pdm2009)
attach(FluH1N1pdm2009)


x &lt;- new(&quot;obkData&quot;, individuals = individuals, dna = FluH1N1pdm2009$dna,
         dna.individualID = samples$individualID, dna.date = samples$date,
         trees = FluH1N1pdm2009$trees)

plotggphy(x, ladderize = TRUE, branch.unit = &quot;year&quot;,
          tip.color = &quot;location&quot;, tip.size = 3, tip.alpha = 0.75)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/09/ggtree-for-microbiome-data/">ggtree for microbiome data</a></h2>
    <div class="article-style">
      
      <p><a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree</a> can parse many software outputs and the evolution evidences inferred by these software can be used directly for tree annotation. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree</a> not only works as an infrastructure that enables evolutionary data that inferred by commonly used software packages to be used in R, but also serves as a general tree visualization and annotation tool for the R community as it supports many S3/S4 objects defined by other R packages.</p>

<h2 id="phyloseq-for-microbiome-data">phyloseq for microbiome data</h2>

<p><code>phyloseq</code> class defined in the <a href="https://www.bioconductor.org/package/phyloseq" target="_blank">phyloseq</a> package was designed for microbiome data. <code>phyloseq</code> package implemented <code>plot_tree</code> function using <code>ggplot2</code>. Although the function was implemented by <code>ggplot2</code> and we can use <code>theme</code>, <code>scale_color_manual</code> etc for customization, the most valuable part of <code>ggplot2</code>, adding layer, is missing. <code>plot_tree</code> only provides limited parameters to control the output graph and it is hard to add layer unless user has expertise in both <code>phyloseq</code> and <code>ggplot2</code>.</p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a></h2>
    <div class="article-style">
      
      <p>I extended the <a href="http://guangchuangyu.github.io/2015/08/subview/" target="_blank">subview</a> function to support embed image file in a <code>ggplot</code> object.</p>

<pre><code>set.seed(123)
d = data.frame(x=rnorm(10), y=rnorm(10))

imgfile &lt;- tempfile(, fileext=&quot;.png&quot;)
download.file(&quot;https://avatars1.githubusercontent.com/u/626539?v=3&amp;u=e731426406dd3f45a73d96dd604bc45ae2e7c36f&amp;s=140&quot;,
              destfile=imgfile, mode='wb')

p = ggplot(d, aes(x, y))
subview(p, imgfile, x=d$x[1], y=d$y[1]) + geom_point(size=5)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a></h2>
    <div class="article-style">
      
      <p><a href="http://cran.r-project.org/package=emojifont" target="_blank"><img src="http://www.r-pkg.org/badges/version/emojifont" alt="CRAN_Status_Badge" /></a></p>

<p><a href="https://cran.r-project.org/web/packages/emojifont/index.html" target="_blank"><code>emojifont</code></a> is available in CRAN, you can use the following command to install it.</p>

<pre><code>install.packages(&quot;emojifont&quot;)
</code></pre>

<p>An example of using emoji font in R plot is showed below:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2016/douban_emoji.jpg" alt="" /></p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/02/covplot-supports-grangeslist/">covplot supports GRangesList</a></h2>
    <div class="article-style">
      
      <p>To answer the <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/27" target="_blank">issue</a>, I extend the <code>covplot</code> function to support viewing coverage of a list of <code>GRanges</code> objects or <code>bed</code> files.</p>

<pre><code>library(ChIPseeker)
files &lt;- getSampleFiles()
peak=GenomicRanges::GRangesList(CBX6=readPeakFile(files[[4]]),
                                CBX7=readPeakFile(files[[5]]))

p &lt;- covplot(peak)
print(p)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">use emoji font in R</a></h2>
    <div class="article-style">
      
      <p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot 2015-12-16 10.55.49.png" alt="" /></p>

<p>I have played with emoji in <code>R</code> for a while. My solution of using it is different from what implemented in <a href="https://github.com/dill/emoGG/" target="_blank">emoGG</a>.</p>

<p><code>emoGG</code> is a good attemp to add <code>emoji</code> in <code>ggplot2</code>. It render <code>emoji</code> picture (png) and creat a layer, <code>geom_emoji</code>, to add emoji.</p>

<p>In my opinion, <code>emoji</code> should be treated as ordinary font in user interface, albeit it maynot be true internally.</p>

<p>It would be more flexible if we can use emoji as ordinary font and in this way user don&rsquo;t need to learn extra stuff.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">use simplify to remove redundancy of enriched GO terms</a></h2>
    <div class="article-style">
      
      <p>To simplify enriched GO result, we can use slim version of GO and use
<em><a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/" target="_blank">enricher</a></em>
function to analyze.</p>

<p>Another strategy is to use
<a href="http://www.bioconductor.org/packages/GOSemSim" target="_blank">GOSemSim</a> to calculate
similarity of GO terms and remove those highly similar terms by keeping
one representative term. To make this feature available to
<a href="http://www.bioconductor.org/packages/clusterProfiler" target="_blank">clusterProfiler</a>
users, I develop a <em>simplify</em> method to reduce redundant GO terms from
output of <em>enrichGO</em> function.</p>

<pre><code>require(clusterProfiler)
data(geneList, package=&quot;DOSE&quot;)
de &lt;- names(geneList)[abs(geneList) &gt; 2]
bp &lt;- enrichGO(de, ont=&quot;BP&quot;)
enrichMap(bp)
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/subview/">subview</a></h2>
    <div class="article-style">
      
      <p>I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
    geom_bar(stat=&quot;identity&quot;, width=1) + 
    coord_polar(theta=&quot;y&quot;) + theme_tree() + 
    xlab(NULL) + ylab(NULL) + 
    theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
    p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
    print(p)
}
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a></h2>
    <div class="article-style">
      
      <p><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/" target="_blank">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
    <div class="article-style">
      
      This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.

      
    </div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/categories/visualization/">&laquo; Previous</a></li>
    
    
    <li><a href="/categories/visualization/page/3/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:350px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

