<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2016 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a>
      </h1>
      <span class="post-date">Jun 23, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a>
      </span>


      
      This is a feature request from clusterProfiler user. It&rsquo;s similar to what I implemented in clusterProfiler for comparing biological themes. For comparing different enrichment results, the x-axis represent different gene clusters while for a single enrichment result, the x-axis can be gene count or gene ratio. This is actually similar to traditional barplot, with dot position as bar height and dot color as bar color. But dotplot can represent one more feature nicely by dot size and it can be a good alternative to barplot.
      <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/">ggtree with funny fonts</a>
      </h1>
      <span class="post-date">Jun 12, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      showtext is a neat solution to use various types of fonts in R graphs and make it easy to use funny fonts. With showtext, we can draw phylogenetic tree with different types of fonts even with symbolic/icon fonts. require(showtext) font.add.google(&quot;Gochi Hand&quot;, &quot;gochi&quot;) font.add.google(&quot;Rock Salt&quot;, &quot;rock&quot;) link = &quot;http://img.dafont.com/dl/?f=wm_people_1&quot;; download.file(link, &quot;wmpeople1.zip&quot;, mode = &quot;wb&quot;); unzip(&quot;wmpeople1.zip&quot;); font.add(&quot;wmpeople1&quot;, &quot;wmpeople1.TTF&quot;); link = &quot;http://img.dafont.com/dl/?f=emoticons&quot;; download.file(link, &quot;emoticons.zip&quot;, mode = &quot;wb&quot;); unzip(&quot;emoticons.zip&quot;); font.add(&quot;emoticons&quot;, &quot;emoticons.ttf&quot;); showtext.auto() set.seed(2015-05-14) library(ggplot2) library(ggtree) tree &lt;- rtree(30) ggtree(tree, color=&quot;darkgreen&quot;) + geom_tiplab(family=&quot;gochi&quot;) + theme_classic() + theme(axis.text.x=element_text(size=rel(4), family=&quot;emoticons&quot;, color=&quot;firebrick&quot;)) + scale_y_continuous(breaks=seq(0, 30, 4), labels=letters[c(12:17, 20, 21)]) + theme(axis.text.y=element_text(size=rel(4), family=&quot;wmpeople1&quot;, color=&quot;#FDAC4F&quot;)) + annotate(&quot;text&quot;, x=2, y=18, label=&quot;Have fun with ggtree!&quot;, family=&quot;rock&quot;, angle=30, size=12, color=&quot;steelblue&quot;) download.file(&quot;http://dl.dafont.com/dl/?f=people_freak&quot;, &quot;people_freak.zip&quot;, mode=&quot;wb&quot;) unzip(&quot;people_freak.zip&quot;) font.add(&quot;people_freak&quot;, &quot;People_freak.ttf&quot;) ggtree(rtree(10)) + geom_text(aes(label=node, color=isTip), family=&quot;people_freak&quot;, hjust=0) Funny fonts can be downloaded from 1, 2 and 3.
      
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/06/bmc-favor-plagiarism/">BMC Favor Plagiarism</a>
      </h1>
      <span class="post-date">Jun 8, 2015 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2015/06/bmc-favor-plagiarism/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/biology">Biology</a><a class="label" href="http://guangchuangyu.github.io/categories/research">Research</a>
      </span>


      
      ppiPre抄袭了GOSemSim的代码，证据当然非常充分，比对一下代码就知道了，我在Proper use of GOSemSim一文中，做出了一些比较，另外也可以参考github页面，github记录了ppiPre被暴光抄袭之后所做的修改。 从我给BMC Systems Biology的编辑反馈这件事开始，在这铁板钉钉的事实面前，编辑拖了整整一年，而这一年时间过去了，ppiPre仍没有被编辑部受理。从最早反应这件事情，编辑信誓旦旦说他们很重视这种事情，到后面对我的邮件视而不见，我愿意相信编辑部处理这些事情，需要时间，他们有自己的规则，但一年的时间，不回邮件，冷处理以淡化此事，这绝对不是应该有的规则。 在编辑一直无视我的情况下，我写出了Proper use of GOSemSim一文，列举了一些一模一样的代码，并告知CRAN，当ppiPre被CRAN移除时，我写信给编辑，这时候，编辑告诉我说他们准备要去联系作者了，这时候已经过去半年了，是的！你没有看错，半年过去了，编辑说他们还没去联系作者！我是不相信的。必然是联系了之后，有某些不为人知的原因，所以编辑态度反常，对抄袭这种打鸡血的事情，不断在打太极。 再过二个月，ppiPre的作者邓岳给我写了信： 按照我在Proper use of GOSemSim中说的:&ldquo;你们可以抄我的代码，但要注明出处，引用我的文章&rdquo;，进行了修改。我给他回信就是我可以接受这样的版本。事实上如果他们基于现在的版本发表文章，我没有任何问题。但现在的版本是迫于抄袭暴光所做出的修改，而发表文章时，抄代码不注出处，更改函数名，刻意回避我的文章，而且对抄袭部分在他们的文章中宣称是自己的实现，这是恶意抄袭。 回信说现在的版本OK之后，我觉得这会给出错误信息，于是我再写一封信，强调抄袭的严重性： 从他的回信中，我感觉得出，态度并不好。他觉得只是对open source的版权不够重视而已，这不单单是版权的问题，还是出版伦理（publication ethics）的问题，再怎么对版权无知，你不能发表文章说别人做的东西是你做的。 事实上邓岳并没有意识到问题，2015年继续发表抄袭的R包HPOSim，这个包的代码抄袭了哈医大李霞实验室的DOSim，而DOSim又是抄了GOSim，我手贱对比了一下代码，竟然发现抄袭链条，而且抄袭者都发了文章，但发完文章，相应的包也就被CRAN移除了，要么是不维护，要么是主动要求移除。这个圈子好乱有木有！ 我也没必要去跟他掐，而是等待编辑的处理，我期望编辑给他上一课，做科研没有这样的捷径。 最终编辑给出的邮件是： Dear Guangchuang, Looking back through my emails, I did receive a response back from the authors previously. They in turn said that the email address I had originally been trying to contact them with earlier last year was no longer in use, so they only received my emails when I found an additional alternative email address for the authors in my more recent emails.
      <a href="http://guangchuangyu.github.io/2015/06/bmc-favor-plagiarism/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/">BMC favors source code plagiarism</a>
      </h1>
      <span class="post-date">May 27, 2015 &middot; 10 minute read &middot; <a href="http://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/research">Research</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I found source code plagiarism a year ago and reported this case to BMC Systems Biology: I listed source code of many functions that are exactly copied from GOSemSim with only function name changed in my email. The detail of source code plagiarism can also be found at Proper use of GOSemSim. I got reply from the Editor 3 days after (see screenshot). MS: 1069569029107167 Research ppiPre: predicting protein-protein interactions by combining heterogeneous features BMC Systems Biology 2013, 7:S8 Dear Dr Yu, Thank you for your message and bringing this to our attention.
      <a href="http://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a>
      </h1>
      <span class="post-date">May 11, 2015 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/systems-biology">Systems Biology</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a>
      </span>


      
      clusterProfiler supports enrichment analysis of both hypergeometric test and gene set enrichment analysis. It internally supports Gene Ontology analysis of about 20 species, Kyoto Encyclopedia of Genes and Genomes (KEGG) with all species that have annotation available in KEGG database, DAVID annotation (only hypergeometric test supported), Disease Ontology and Network of Cancer Genes (via DOSE for human) and Reactome Pathway (via ReactomePA for several species). This is still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (eg GO via blastgo and KEGG via KAAS), unsupported ontology/pathway or customized annotation.
      <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/">using GOSemSim to rank proteins obtained by co-IP</a>
      </h1>
      <span class="post-date">May 4, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/proteomics">Proteomics</a><a class="label" href="http://guangchuangyu.github.io/categories/publication">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/gosemsim">GOSemSim</a>
      </span>


      
      Co-IP is usually used to identified interactions among specific proteins. It is widely used in detecting protein complex. Unfortunately, an identified protein may not be an interactor, and sometimes can be a background contaminant. Ranking proteins can help us to focus a study on a few high quality candidates for subsequent interaction investigation. My R package GOSemSim has been used in many studies to investigate protein protein interaction, see 1, 2, 3, 4 and more.
      <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/">Bioc 3.1: NEWS of my BioC packages</a>
      </h1>
      <span class="post-date">Apr 20, 2015 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/bioc">BioC</a><a class="label" href="http://guangchuangyu.github.io/categories/gosemsim">GOSemSim</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      GOSemSim GOSemSim was first implemented in 2008 and published in Bioinformatics in 2010. It&rsquo;s now a mature package with no bugs found in the past half year. Only vignette and Information content data were updated. clusterProfiler clusterProfiler was developed in 2011 and published in OMICS in 2012. I am very glad to find that clusterProfiler was presented in the book &lsquo;Bioinformatics with R Cookbook&lsquo;: In this release, KEGG enrichment analysis (both hypergeometric and GSEA) now supports using latest online data and DAVID functional analysis is also supported.
      <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/">an example of drawing beast tree using ggtree</a>
      </h1>
      <span class="post-date">Apr 1, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/evolution">Evolution</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      FigTree is designed for viewing beast output as demonstrated by their example data: BEAST output is well supported by ggtree and it&rsquo;s easy to reproduce such a tree view. ggtree supports parsing beast output by read.beast function. We can visualize the tree directly by using ggtree function. Since this is a time scale tree, we can set the parameter mrsd to most recent sampling date. require(ggplot2) require(ggtree) x &lt;- read.beast(&quot;/Applications/FigTree/influenza.tree&quot;)
      <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">DAVID functional analysis with clusterProfiler</a>
      </h1>
      <span class="post-date">Mar 16, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/david">DAVID</a><a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a>
      </span>


      
      clusterProfiler was used to visualize DAVID results in a paper published in BMC Genomics. Some users told me that they may want to use DAVID at some circumstances. I think it maybe a good idea to make clusterProfiler supports DAVID, so that DAVID users can use visualization functions provided by clusterProfiler. require(DOSE) require(clusterProfiler) data(geneList) gene = names(geneList)[abs(geneList) &gt; 2] david = enrichDAVID(gene = gene, idType=&quot;ENTREZ_GENE_ID&quot;, listType=&quot;Gene&quot;, annotation=&quot;KEGG_PATHWAY&quot;) &gt; summary(david) ID Description GeneRatio BgRatio pvalue hsa04110 hsa04110 Cell cycle 11/68 125/5085 4.254437e-06 hsa04114 hsa04114 Oocyte meiosis 10/68 110/5085 1.119764e-05 hsa03320 hsa03320 PPAR signaling pathway 7/68 69/5085 2.606715e-04 p.adjust qvalue geneID hsa04110 0.0003998379 NA 9133/4174/890/991/1111/891/7272/8318/4085/983/9232 hsa04114 0.0005261534 NA 9133/5241/51806/3708/991/891/4085/983/9232/6790 hsa03320 0.0081354974 NA 4312/2167/5346/5105/3158/9370/9415 Count hsa04110 11 hsa04114 10 hsa03320 7 There are only 5085 human genes annotated by KEGG, this is due to out-of-date DAVID data.
      <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/03/dose-an-r/bioconductor-package-for-disease-ontology-semantic-and-enrichment-analysis/">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>
      </h1>
      <span class="post-date">Mar 1, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/03/dose-an-r/bioconductor-package-for-disease-ontology-semantic-and-enrichment-analysis/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/publication">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>
      </span>


      
      My R/Bioconductor package, DOSE, published in Bioinformatics. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis G Yu^^, LG Wang, GR Yan, QY He^^. Bioinformatics 2015, 31(4):608-609 Received: 7 May 2014 Accepted: 14 October 2014 Online: 17 October 2014 DOI: 10.1093/bioinformatics/btu684 Summary: Disease ontology (DO) annotates human genes in the context of disease. DO is important annotation in translating molecular findings from high-throughput data to clinical relevance. DOSE is an R package providing semantic similarity computations among DO terms and genes which allows biologists to explore the similarities of diseases and of gene functions in disease perspective.
      <a href="http://guangchuangyu.github.io/2015/03/dose-an-r/bioconductor-package-for-disease-ontology-semantic-and-enrichment-analysis/">Read On &rarr;</a>
      

    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        >
        <a href="/page/2/" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        ><a href="/">1</a></li>
        
        <li
        ><a href="/page/2/">2</a></li>
        
        <li
        class="active"><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        ><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        >
        <a href="/page/4/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/8/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    

  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

