<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Bug of R package ChIPpeakAnno</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            Bug of R package ChIPpeakAnno 
          </h1>
          <p class="meta">Jan 14, 2014 - 5 minute read -  <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          <p>I used R package <strong>ChIPpeakAnno</strong> for annotating peaks, and found that
it handle the DNA strand in the wrong way. Maybe the developers were
from the computer science but not biology background.</p>

<pre><code>&gt; require(ChIPpeakAnno)
&gt; packageVersion(&quot;ChIPpeakAnno&quot;)
[1] '2.10.0'
&gt; peak &lt;- RangedData(space=&quot;chr1&quot;, IRanges(24736757, 24737528))
&gt; data(TSS.human.GRCh37)
&gt; ap &lt;- annotatePeakInBatch(peak, Annotation=TSS.human.GRCh37)
&gt; ap
RangedData with 1 row and 9 value columns across 1 space
                     space               ranges |        peak      strand
                               |  
1 ENSG00000001461        1 [24736757, 24737528] |           1           +
                          feature start_position end_position insideFeature
                                   
1 ENSG00000001461 ENSG00000001461       24742284     24799466      upstream
                  distancetoFeature shortestDistance fromOverlappingOrNearest
                                                
1 ENSG00000001461             -5527             4756             NearestStart
</code></pre>

<p>In this example, I defined a peak ranging from chr1:24736757 to
chr1:24737528 and annotated the peak using ChIPpeakAnno package.</p>

<p>It returns that the nearest gene is ENSG00000001461, whose gene symbol
is NIPAL3.</p>

<pre><code>&gt; require(org.Hs.eg.db)
&gt; gene.ChIPpeakAnno &lt;- select(org.Hs.eg.db, key=ap$feature, keytype=&quot;ENSEMBL&quot;, columns=c(&quot;ENSEMBL&quot;, &quot;ENTREZID&quot;, &quot;SYMBOL&quot;))
&gt; gene.ChIPpeakAnno
          ENSEMBL ENTREZID SYMBOL
1 ENSG00000001461    57185 NIPAL3
</code></pre>

<p>When looking at the peak in Genome Browser, I found the nearest gene is
STPG1.
<img src="http://guangchuangyu.github.io/blog_images/2014/01/Screenshot-2014-01-13-22.00.46.png" alt="" />
The gene symbol, STPG1, was converted to ENTREZID for future processing.</p>

<pre><code>&gt; gene.nearest &lt;- select(org.Hs.eg.db, key=&quot;STPG1&quot;, keytype=&quot;SYMBOL&quot;, columns=c(&quot;ENSEMBL&quot;, &quot;ENTREZID&quot;, &quot;SYMBOL&quot;))
&gt; gene.nearest
  SYMBOL         ENSEMBL ENTREZID
1  STPG1 ENSG00000001460    90529
</code></pre>

<p>We can query the coordination of these two genes, and compare their
distances to the peak.</p>

<pre><code>&gt; require(TxDb.Hsapiens.UCSC.hg19.knownGene)
&gt; knownGene &lt;- transcriptsBy(TxDb.Hsapiens.UCSC.hg19.knownGene, by=&quot;gene&quot;)
&gt; 
&gt; gene.ChIPpeakAnno.info &lt;- knownGene[[gene.ChIPpeakAnno$ENTREZID]]
&gt; gene.ChIPpeakAnno.info
GRanges with 4 ranges and 2 metadata columns:
      seqnames               ranges strand |     tx_id     tx_name
                        |  
  [1]     chr1 [24742245, 24781314]      + |       618  uc010oek.2
  [2]     chr1 [24742245, 24799473]      + |       619  uc001bjh.3
  [3]     chr1 [24742245, 24799473]      + |       620  uc009vrc.3
  [4]     chr1 [24782628, 24792864]      + |       621  uc001bji.3
  ---
  seqlengths:
                    chr1                  chr2 ...        chrUn_gl000249
               249250621             243199373 ...                 38502
</code></pre>

<p>After getting the information of the gene annotated by ChIPpeakAnno, I
also found that the ranges of the gene is slightly different from the
one returned by <em>annotatePeakInBatch</em> function. This may due to the
variation of different versions and it&rsquo;s not a big deal.</p>

<p>As the gene, NIPAL3, is encoded in + strand, the nearest distance is:</p>

<pre><code>&gt; min(abs(start(peak) - start(gene.ChIPpeakAnno.info)))
[1] 5488
</code></pre>

<p>While the gene, STPG1, is encoded in - strand, the end of the gene
coordination is actually the start position of the gene and the start of
the gene coordination is the end position. So the distance should be
calculated by the end coordination.</p>

<pre><code>&gt; gene.nearest.info &lt;- knownGene[[gene.nearest$ENTREZID]]
&gt; gene.nearest.info
GRanges with 6 ranges and 2 metadata columns:
      seqnames               ranges strand |     tx_id     tx_name
                        |  
  [1]     chr1 [24683489, 24700300]      - |      4706  uc010oej.2
  [2]     chr1 [24683489, 24740262]      - |      4707  uc001bja.3
  [3]     chr1 [24683489, 24740262]      - |      4708  uc001bjb.3
  [4]     chr1 [24683489, 24740262]      - |      4709  uc001bjc.3
  [5]     chr1 [24683489, 24741587]      - |      4710  uc001bjd.3
  [6]     chr1 [24695211, 24718169]      - |      4711  uc001bjf.3
  ---
  seqlengths:
                    chr1                  chr2 ...        chrUn_gl000249
               249250621             243199373 ...                 38502
&gt; min(abs(end(peak) - end(knownGene[[gene.nearest$ENTREZID]])))
[1] 2734
</code></pre>

<p>STPG1 is more close to the peak than NIPAL3. Those genes encoded in -
strand can&rsquo;t be well-handled by <strong>ChIPpeakAnno</strong> package.</p>

<p>I look careful to the source code of <em>annotatePeakInBatch</em> function.</p>

<pre><code>    TSS.ordered &lt;- AnnotationData
    rm(AnnotationData)
    if (!length(rownames(TSS.ordered))) {
        rownames(TSS.ordered) = formatC(1:dim(TSS.ordered)[1],
            width = nchar(dim(TSS.ordered)[1]), flag = &quot;0&quot;)
    }
    if (length(TSS.ordered$strand) == length(start(TSS.ordered))) {
        r2 = cbind(rownames(TSS.ordered), start(TSS.ordered),
            end(TSS.ordered), as.character(TSS.ordered$strand))
    }
    else {
        TSS.ordered$strand = rep(&quot;+&quot;, length(start(TSS.ordered)))
        r2 = cbind(rownames(TSS.ordered), start(TSS.ordered),
            end(TSS.ordered), rep(&quot;+&quot;, length(start(TSS.ordered))))
    }
    colnames(r2) = c(&quot;feature_id&quot;, &quot;start_position&quot;, &quot;end_position&quot;,
        &quot;strand&quot;)
</code></pre>

<p>The <em>AnnotationData</em> object is provided by the package, or query from
biomaRt, the length(TSS.ordered\$strand) == length(start(TSS.ordered)
should be TRUE, and this code should works fine. But when the expression
return FALSE, the function should complain this, with warning message
showing in the screen or even stop running. But the author just simply
assign all the genes encoded in + strand, this is nonsense.</p>

<p>Then for calculating the feauture location:</p>

<pre><code>    if (FeatureLocForDistance == &quot;middle&quot; || FeatureLocForDistance ==
        &quot;m&quot;) {
        FeatureLoc = unlist(lapply(1:dim(r2)[1], function(i) {
            round(mean(c(as.numeric(r2[i, 2]), as.numeric(r2[i,
                3]))))
        }))
        FeatureLocForDistance = &quot;middle&quot;
    }
    else if (FeatureLocForDistance == &quot;start&quot; || FeatureLocForDistance ==
        &quot;s&quot;) {
        FeatureLoc = r2[, 2]
        FeatureLocForDistance = &quot;start&quot;
    }
    else if (FeatureLocForDistance == &quot;end&quot; || FeatureLocForDistance ==
        &quot;e&quot;) {
        FeatureLoc = r2[, 3]
        FeatureLocForDistance = &quot;end&quot;
    }
    else if (FeatureLocForDistance == &quot;geneEnd&quot; || FeatureLocForDistance ==
        &quot;g&quot;) {
        FeatureLoc = unlist(lapply(1:dim(r2)[1], function(i) {
            if (as.character(r2[i, 4]) == &quot;+&quot; || as.character(r2[i,
                4]) == &quot;1&quot; || as.character(r2[i, 4]) == &quot;*&quot;) {
                r2[i, 3]
            }
            else {
                r2[i, 2]
            }
        }))
        FeatureLocForDistance = &quot;geneEnd&quot;
    }
    else {
        FeatureLoc = unlist(lapply(1:dim(r2)[1], function(i) {
            if (as.character(r2[i, 4]) == &quot;+&quot; || as.character(r2[i,
                4]) == &quot;1&quot; || as.character(r2[i, 4]) == &quot;*&quot;) {
                r2[i, 2]
            }
            else {
                r2[i, 3]
            }
        }))
        FeatureLocForDistance = &quot;TSS&quot;
    }
</code></pre>

<p>Only &ldquo;geneEnd&rdquo; and &ldquo;TSS&rdquo; types consider the strand information. For
&ldquo;middle&rdquo;, omits the strand information should be fine, but for &ldquo;start&rdquo;
and &ldquo;end&rdquo;, the &ldquo;start&rdquo; should be &ldquo;end&rdquo; and &ldquo;end&rdquo; should be &ldquo;start&rdquo; when
the gene is located at - strand. Again, the <em>annotatePeakInBatch</em>
function omits the strand information of genes/features.</p>

<pre><code>        TSS.ordered$FeatureLoc = FeatureLoc
        myPeakList$PeakLoc = PeakLoc
        plusAnno = TSS.ordered[as.character(TSS.ordered$strand) %in%
            c(&quot;+&quot;, &quot;*&quot;, &quot;1&quot;), ]
        minusAnno = TSS.ordered[as.character(TSS.ordered$strand) %in%
            c(&quot;-1&quot;, &quot;-&quot;), ]
        r1 = do.call(rbind, lapply(seq_len(numberOfChromosome),
            function(i) {
                chr = allChr[i]
                if (chr %in% allChr.Anno) {
                  featureStart = as.numeric(TSS.ordered[chr]$FeatureLoc)
                  peakLoc = as.numeric(myPeakList[chr]$PeakLoc)
                  peakStart = as.numeric(start(myPeakList[chr]))
                  peakEnd = as.numeric(end(myPeakList[chr]))
                  name = rownames(myPeakList[chr])
                  peakRanges = IRanges(start = peakStart, end = peakEnd,
                    names = name)
                  featureID = rownames(TSS.ordered[chr])
                  featureRanges = IRanges(start = featureStart,
                    end = featureStart, names = featureID)
                  nearestFeature = featureRanges[nearest(peakRanges,
                    featureRanges)]
                  data.frame(name = name, feature_id = names(nearestFeature))
                }
            }))
</code></pre>

<p>For identifying the nearest feature, the author use:</p>

<pre><code>nearestFeature = featureRanges[nearest(peakRanges,
                    featureRanges)]
</code></pre>

<p>The &ldquo;start&rdquo; of the features should be used when it were encoded in +
strand and &ldquo;end&rdquo; when features were encoded in - strand. But the author
use <em>nearest</em> function which use both &ldquo;start&rdquo; and &ldquo;end&rdquo; of the interval
to calculate the distance by default.</p>

<p>Considering the &ldquo;start&rdquo; and &ldquo;end&rdquo; records may reverse when the
genes/features were encoded in - strand and the way ChIPpeakAnno
calcultes the distance is too simple by using <em>nearest</em> function, I
can&rsquo;t trust their results. It&rsquo;s not hard to implement a function to
annotate peaks. Preparing the gene annotation and calculating the
distances among genes and peaks for finding the nearest gene of the
peak. That&rsquo;s it.</p>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Jan 14, 2014</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>  <a class="category" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2013/09/install-emacs-on-mac/" title="Install Emacs on Mac">Install Emacs on Mac</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2014/03/boxplot/" title="boxplot">boxplot</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/03/%E4%BD%BF%E7%94%A8ggtree%E5%AE%9E%E7%8E%B0%E8%BF%9B%E5%8C%96%E6%A0%91%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E5%92%8C%E6%B3%A8%E9%87%8A/">使用ggtree实现进化树的可视化和注释</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/covplot-supports-grangeslist/">covplot supports GRangesList</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
    </ul>
  </section>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


