<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>ChIPseeker for ChIP peak annotation</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body class="theme-base-0d">


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">G. Yu</a></h1>
    <h2>Bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/link/" target="_blank" title="Link">Link</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban/" target="_blank" title="douban">douban</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            ChIPseeker for ChIP peak annotation 
          </h1>
          <p class="meta">Apr 13, 2014 - 4 minute read -  <a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/chipseq">ChIPseq</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          <p><a href="http://www.bioconductor.org/packages/2.14/bioc/html/ChIPpeakAnno.html">ChIPpeakAnno</a>
WAS the only R package for ChIP peak annotation. I used it for
annotating peak in my recent study.</p>

<p>I found it does not consider the strand information of genes. I
<a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">reported the
bug</a> to the
authors, but they are reluctant to change.</p>

<p>So I decided to develop my own package,
<a href="http://www.bioconductor.org/packages/2.14/bioc/html/ChIPseeker.html">ChIPseeker</a>,
and it&rsquo;s now available in Bioconductor.</p>

<pre><code class="language-lang=&quot;rsplus&quot;">&gt; require(TxDb.Hsapiens.UCSC.hg19.knownGene)
&gt; txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
&gt; require(ChIPseeker)
&gt; peakfile = getSampleFiles()[[4]]
&gt; peakfile
[1] &quot;/usr/local/Cellar/r/3.1.0/R.framework/Versions/3.1/Resources/library/ChIPseeker/extdata//GEO_sample_data/GSM1295076_CBX6_BF_ChipSeq_mergedReps_peaks.bed.gz&quot;
&gt; peakAnno &lt;- annotatePeak(peakfile, tssRegion=c(-3000, 1000), TxDb=txdb, annoDb=&quot;org.Hs.eg.db&quot;)
&gt;&gt; loading peak file...              2014-04-13 12:03:45 PM 
&gt;&gt; preparing features information...         2014-04-13 12:03:45 PM 
&gt;&gt; identifying nearest features...       2014-04-13 12:03:45 PM 
&gt;&gt; calculating distance from peak to TSS...  2014-04-13 12:03:46 PM 
&gt;&gt; assigning genomic annotation...       2014-04-13 12:03:46 PM 
&gt;&gt; adding gene annotation...             2014-04-13 12:03:50 PM 
&gt;&gt; assigning chromosome lengths          2014-04-13 12:03:50 PM 
&gt;&gt; done...                   2014-04-13 12:03:50 PM 
&gt; head(as.GRanges(peakAnno))
GRanges object with 6 ranges and 14 metadata columns:
      seqnames             ranges strand |          V4        V5
                      |     
  [1]     chr1 [ 815092,  817883]      * | MACS_peak_1    295.76
  [2]     chr1 [1243287, 1244338]      * | MACS_peak_2     63.19
  [3]     chr1 [2979976, 2981228]      * | MACS_peak_3    100.16
  [4]     chr1 [3566181, 3567876]      * | MACS_peak_4    558.89
  [5]     chr1 [3816545, 3818111]      * | MACS_peak_5     57.57
  [6]     chr1 [6304864, 6305704]      * | MACS_peak_6     54.57
             annotation  geneChr geneStart   geneEnd geneLength geneStrand
                    
  [1] Distal Intergenic     chr1    803451    812182       8732          -
  [2]  Promoter (&lt;=1kb)     chr1   1243994   1247057       3064          +
  [3]  Promoter (&lt;=1kb)     chr1   2976181   2980350       4170          -
  [4]  Promoter (&lt;=1kb)     chr1   3547331   3566671      19341          -
  [5]  Promoter (&lt;=1kb)     chr1   3816968   3832011      15044          +
  [6]            3  UTR     chr1   6304252   6305638       1387          +
           geneId transcriptId distanceToTSS         ENSEMBL      SYMBOL
                   
  [1]      284593   uc001abt.4         -5701 ENSG00000230368      FAM41C
  [2]      126789   uc001aed.3             0 ENSG00000169972       PUSL1
  [3]      440556   uc001aka.3             0 ENSG00000177133   LINC00982
  [4]       49856   uc001ako.3             0 ENSG00000116213      WRAP73
  [5]   100133612   uc001alg.3             0 ENSG00000236423   LINC01134
  [6]      390992   uc009vly.2          1452 ENSG00000173673        HES3
                                          GENENAME
                                       
  [1] family with sequence similarity 41, member C
  [2]              pseudouridylate synthase-like 1
  [3]   long intergenic non-protein coding RNA 982
  [4]      WD repeat containing, antisense to TP73
  [5]  long intergenic non-protein coding RNA 1134
  [6]       hes family bHLH transcription factor 3
  ---
  seqlengths:
        chr1     chr10     chr11     chr12 ...      chr8      chr9      chrX
   249250621 135534747 135006516 133851895 ... 146364022 141213431 155270560
</code></pre>

<p><em>annotatePeak</em> function can accept peak file or GRanges object that
contained peak information. The sample peakfile shown above is a sample
output from MACS. All the information contained in peakfile will be kept
in the output of annotatePeak.</p>

<p>The <em>annotation</em> column annotates genomic features of the peak, that is
whether peak is located in Promoter, Exon, UTR, Intron or Intergenic. If
the peak is annotated by Exon or Intron, more detail information will be
provided. For instance, Exon (38885 exon 3 of 11) indicates that the
peak is located at the 3th exon of gene 38885 (entrezgene ID) which
contain 11 exons in total.</p>

<p>All the names of column that contains information of nearest gene
annotated will start with gene in camelCaps style.</p>

<p>If parameter annoDb=&ldquo;org.Hs.eg.db&rdquo; is provided, extra columns of
ENSEMBL, SYMBOL and GENENAME will be provided in the final output.</p>

<p>For more detail, please refer to the package
<a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.pdf">vignette</a>.
Comments or suggestions are welcome.</p>

<p>&mdash;&mdash;- below was added at April 30, 2014 &mdash;&mdash;&mdash;&ndash;</p>

<p>After two weeks of ChIPseeker release in Bioconductor, I am very glad to
receive an email from <a href="http://www.flychip.org.uk/">Cambridge Systems Biology
Centre</a> that they are using ChIPseeker in
annotating Drosophila ChIP data.</p>

<p>Although I only illustrate examples in human, ChIPseeker do supports all
species only if genomic information was available. :mrgreen: :mrgreen:</p>

<pre><code class="language-lang=&quot;rsplus&quot;">&gt; require(TxDb.Dmelanogaster.UCSC.dm3.ensGene)
&gt; require(ChIPseeker)
&gt; txdb &lt;- TxDb.Dmelanogaster.UCSC.dm3.ensGene
&gt;
&gt; chr &lt;- c(&quot;chrX&quot;, &quot;chrX&quot;, &quot;chrX&quot;)
&gt; start &lt;- c(71349, 77124, 86364)
&gt; end &lt;- c(72835, 77836, 88378)
&gt; peak &lt;- GRanges(seqnames = chr, ranges = IRanges(start, end))
&gt; peakAnno &lt;- annotatePeak(peak,
+                          tssRegion=c(-3000,100),
+                          TxDb = txdb,
+                          annoDb=&quot;org.Dm.eg.db&quot;)
&gt;&gt; preparing features information...         2014-04-30 21:28:27
&gt;&gt; identifying nearest features...       2014-04-30 21:28:30
&gt;&gt; calculating distance from peak to TSS...  2014-04-30 21:28:31
&gt;&gt; assigning genomic annotation...       2014-04-30 21:28:31
&gt;&gt; adding gene annotation...             2014-04-30 21:28:42
Loading required package: org.Dm.eg.db
Loading required package: DBI

&gt;&gt; assigning chromosome lengths           2014-04-30 21:28:43
&gt;&gt; done...                   2014-04-30 21:28:43
&gt; as.GRanges(peakAnno)
GRanges object with 3 ranges and 12 metadata columns:
      seqnames         ranges strand |
                  |
  [1]     chrX [71349, 72835]      * |
  [2]     chrX [77124, 77836]      * |
  [3]     chrX [86364, 88378]      * |
                                           annotation  geneChr geneStart
                                            
  [1]                                Promoter (&lt;=1kb)     chrX     72458
  [2] Intron (FBtr0100135/FBgn0029128, intron 1 of 7)     chrX     72458
  [3] Intron (FBtr0100135/FBgn0029128, intron 2 of 7)     chrX     72458
        geneEnd geneLength geneStrand      geneId transcriptId distanceToTSS
                   
  [1]     96056      23599          + FBgn0029128  FBtr0100135             0
  [2]     96056      23599          + FBgn0029128  FBtr0100135          5378
  [3]     96056      23599          + FBgn0029128  FBtr0100135         15920
         ENTREZID      SYMBOL    GENENAME
        
  [1]     3354987         tyn     trynity
  [2]     3354987         tyn     trynity
  [3]     3354987         tyn     trynity
  ---
  seqlengths:
       chrX
   22422827
&gt;
</code></pre>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Apr 13, 2014</time>
    <span class="categories">
      Tags: 
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2014/03/boxplot/" title="boxplot">boxplot</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" title="visualization methods in ChIPseeker">visualization methods in ChIPseeker</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    <h1>Header in sidebar</h1>
    <p>This is the rest of the sidebar - here's a link to <a target="_blank" href="http://google.com" title="Google">Google</a>.</p>
    <p>If links are set in config file, there will be links to keybase/github/twitter/linkedin etc.</p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    
    
    
    
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    
     

    
    
    </li>
  </ul>

  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </li>
      
      <li class="post">
        <a href="/douban/">douban</a>
      </li>
      
    </ul>
  </section>

</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


