<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.21" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="PhD student">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/tags/clusterprofiler/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/tags/clusterprofiler/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/tags/clusterprofiler/">

  

  <title>Clusterprofiler | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Clusterprofiler</h1>

  

  
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/">[Bioc 32] NEWS of my BioC packages</a></h2>
    <div class="article-style">
      
      In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim" target="_blank">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler" target="_blank">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim" target="_blank">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler" target="_blank">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a></h2>
    <div class="article-style">
      
      I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/" target="_blank">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" />

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a></h2>
    <div class="article-style">
      
      The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002" target="_blank">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57" target="_blank">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" />

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
    <div class="article-style">
      
      This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a></h2>
    <div class="article-style">
      
      <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/" target="_blank">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/" target="_blank">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp" target="_blank">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/" target="_blank">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/" target="_blank">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
for human) and <a href="http://www.reactome.org/" target="_blank">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide" target="_blank">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/" target="_blank">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/" target="_blank">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages/">[Bioc 31] NEWS of my BioC packages</a></h2>
    <div class="article-style">
      
      <img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" target="_blank">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/" target="_blank">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>

<p>
      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">DAVID functional analysis with clusterProfiler</a></h2>
    <div class="article-style">
      
      <a href="https://github.com/GuangchuangYu/clusterProfiler" target="_blank">clusterProfiler</a> was
used to visualize <a href="http://david.abcc.ncifcrf.gov/summary.jsp" target="_blank">DAVID</a>
results in a paper published in <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3907017/" target="_blank">BMC
Genomics</a>.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/clusterProfiler_david.png" alt="" /></p>

<p>Some users told me that they may want to use DAVID at some
circumstances. I think it maybe a good idea to make clusterProfiler
supports DAVID, so that
<a href="http://david.abcc.ncifcrf.gov/summary.jsp" target="_blank">DAVID</a> users can use
visualization functions provided by
<a href="https://github.com/GuangchuangYu/clusterProfiler" target="_blank">clusterProfiler</a>.</p>

<pre><code>require(DOSE)
require(clusterProfiler)
data(geneList)
gene = names(geneList)[abs(geneList) &gt; 2]
david = enrichDAVID(gene = gene, idType=&quot;ENTREZ_GENE_ID&quot;, 
listType=&quot;Gene&quot;, annotation=&quot;KEGG_PATHWAY&quot;)



&gt; summary(david)
               ID            Description GeneRatio  BgRatio       pvalue
hsa04110 hsa04110             Cell cycle     11/68 125/5085 4.254437e-06
hsa04114 hsa04114         Oocyte meiosis     10/68 110/5085 1.119764e-05
hsa03320 hsa03320 PPAR signaling pathway      7/68  69/5085 2.606715e-04
             p.adjust qvalue                                             geneID
hsa04110 0.0003998379     NA 9133/4174/890/991/1111/891/7272/8318/4085/983/9232
hsa04114 0.0005261534     NA    9133/5241/51806/3708/991/891/4085/983/9232/6790
hsa03320 0.0081354974     NA                 4312/2167/5346/5105/3158/9370/9415
         Count
hsa04110    11
hsa04114    10
hsa03320     7
</code></pre>

<p>There are only 5085 human genes annotated by KEGG, this is due to
out-of-date DAVID data.</p>

<p>
      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/">KEGG enrichment analysis with latest online data using clusterProfiler</a></h2>
    <div class="article-style">
      
      <em>KEGG.db</em> is not updated since 2012. The data is now pretty old, but
many of the Bioconductor packages still using it for KEGG annotation and
enrichment analysis. As pointed out in &lsquo;<a href="http://www.acgt.me/blog/2015/1/16/are-there-too-many-biological-databases" target="_blank">Are there too many biological databases</a>&rsquo;,
there is a problem that many out of date biological databases often
don&rsquo;t get offline. This issue also exists in web-server or software that
using out-of-date data. For example, the
<a href="http://wego.genomics.org.cn/cgi-bin/wego/index.pl" target="_blank">WEGO</a> web-server
stopped updating GO annotation data since 2009, and WEGO still online
with <a href="https://scholar.google.com.hk/scholar?as_ylo=2015&amp;hl=en&amp;as_sdt=0,5&amp;sciodt=0,5&amp;cites=13025429171853466135&amp;scipsc=" target="_blank">many people using it</a>.
The biological story may changed totally if using a recently updated
data. Seriously, We should keep an eye on this issue.</p>

<p>Now <em>enrichKEGG</em> function is reloaded with a new parameter <em>use_internal_data</em>. This
parameter is by default setting to FALSE, and <em>enrichKEGG</em> function will
download the latest KEGG data for enrichment analysis. If the parameter
<em>use_internal_data</em> is explicitly setting to TRUE, it will use the
<em>KEGG.db</em> which is still supported but not recommended. With this new
feature, supported species is unlimited if only there are KEGG
annotations available in KEGG database. You can access the full list of
species supported by KEGG via:
<a href="http://www.genome.jp/kegg/catalog/org_list.html" target="_blank">http://www.genome.jp/kegg/catalog/org_list.html</a> Now the <em>organism</em>
parameter in <em>enrichKEGG</em> should be abbreviation of academic name, for
example &lsquo;hsa&rsquo; for human and &lsquo;mmu&rsquo; for mouse. It accepts any species
listed in <a href="http://www.genome.jp/kegg/catalog/org_list.html" target="_blank">http://www.genome.jp/kegg/catalog/org_list.html</a>. In the
current release version of
<a href="http://bioconductor.org/packages/3.0/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
(in Bioconductor 3.0), <em>enrichKEGG</em> supports about 20 species, and the
<em>organism</em> parameter accept common name of species, for instance &ldquo;human&rdquo;
and &ldquo;mouse&rdquo;. For these previously supported species, common name is also
supported. So that you script is still working with new version of
<a href="http://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>.
For other species, common name is not supported, since I don&rsquo;t want to
maintain such a long mapping list with many species have no common name
available and it may also introduce unexpected bugs.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">why clusterProfiler fails</a></h2>
    <div class="article-style">
      
      <a href="https://github.com/kaji331" target="_blank">@kaji331</a>
compared <code>cluserProfiler</code> with <code>GeneAnswers</code> and found that clusterProfiler
gives larger p values.</p>

<p>It eventually came out that he passed the input
gene as numeric vector, which was supposed to be character and he used an <strong>old</strong> version of <code>clusterProfiler</code> which didn&rsquo;t use <code>as.character</code> to coerce the input</p>

<p>But his comment forces me to test it.</p>

<p>
      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/08/enrichment-map/">enrichment map</a></h2>
    <div class="article-style">
      
      In <a href="http://www.plob.org/" target="_blank">PLOB</a>&rsquo;s QQ group, someone asked how to change
the color of enrichment map in <a href="http://cytoscape.org/" target="_blank">Cytoscape</a>. I am
very curious how enrichment map can helps to interpret enrichment
results. It took me 2 hours to implement it using R and I am surprised
that the enrichment map is better than anticipated.

<img src="http://guangchuangyu.github.io/blog_images/2014/08/Screenshot-2014-07-30-22.20.07.png" alt="" />
      
    </div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/tags/clusterprofiler/">&laquo; Previous</a></li>
    
    
    <li><a href="/tags/clusterprofiler/page/3/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:350px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

