<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>BioC 3.3: NEWS of my packages</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            BioC 3.3: NEWS of my packages 
          </h1>
          <p class="meta">May 5, 2016 - 10 minute read -  <a href="http://guangchuangyu.github.io/2016/05/bioc-3.3-news-of-my-packages/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
	  	  
	  

<p><img src="http://guangchuangyu.github.io/blog_images/MISC/ke.jpg" width=600></p>

<p>Today is my birthday and it happened to be the release day of Bioconductor 3.3. It&rsquo;s again the time to reflect what I&rsquo;ve done in the past year.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE">DOSE</a></li>
<li><a href="(http://guangchuangyu.github.io/ggtree">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA">ReactomePA</a></li>
</ul>

<h2 id="chipseeker:de4254a6abb9a1c947b7d959d55455bd">ChIPseeker</h2>

<p>Although ChIPseeker was designed for ChIP-seq annotation, I am very glad to find that someone else use it to annotate other data including <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/32">copy number variants</a> and <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/17">DNA breakpoints</a>.</p>

<h3 id="annotatepeak:de4254a6abb9a1c947b7d959d55455bd">annotatePeak</h3>

<p>Several parameters including <code>sameStrand</code>,<code>ignoreOverlap</code>, <code>ignoreUpstream</code> and <code>ignoreDownstream</code> were added in <code>annotatePeak</code> requested by <a href="https://github.com/crazyhottommy">@crazyhottommy</a> for using ChIPseeker to annotate breakpoints from whole genome sequencing data.</p>

<p>Another parameter <code>overlap</code> was also introduced. By default <code>overlap=&quot;TSS&quot;</code> and only overlap with TSS will be reported as the nearest gene. If <code>overlap=&quot;all&quot;</code>, then gene overlap with peak will be reported as nearest gene, no matter the overlap is at TSS region or not.</p>

<p>Now <code>annotatePeak</code> also support using user&rsquo;s customize regions to annotate their data by passing <code>TxDb=user_defined_GRanges</code>.</p>

<h3 id="getbioregion:de4254a6abb9a1c947b7d959d55455bd">getBioRegion</h3>

<p><code>getPromoters()</code> function prepare a <code>GRanges</code> object of promoter regions by user specific upstream and downstream distance from Transcript Start Site (TSS). Then we can align the peaks that are mapping to these regions and visualize the profile or heatmap of ChIP binding to the TSS regions.</p>

<p>Users (<a href="https://support.bioconductor.org/p/73069/#73157">1</a> and <a href="https://support.bioconductor.org/p/73377/">2</a>) are interesting in the intensity of peaks binding to the start of intron/exon, and ChIPseeker provides a new function <code>getBioRegion</code> to output <code>GRanges</code> object of Intron/Exon start regions.</p>

<h3 id="visualization:de4254a6abb9a1c947b7d959d55455bd">visualization</h3>

<ul>
<li><p><code>upsetplot</code> was implemented to visualize <a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">ChIP annotation overlap</a>.</p></li>

<li><p><code>covplot</code> now support <a href="http://guangchuangyu.github.io/2016/02/covplot-supports-grangeslist/">GRangesList</a>.</p></li>
</ul>

<h3 id="geo-data-mining:de4254a6abb9a1c947b7d959d55455bd">GEO data mining</h3>

<p>ChIPseeker incorporates GEO database and supports data mining to infer cooperative regulation. The data was updated and now ChIPseeker contains 19348 bed file information.</p>

<h2 id="clusterprofiler:de4254a6abb9a1c947b7d959d55455bd">clusterProfiler</h2>

<p>We compare clusterProfiler with GSEA-P (which released by broad institute), the p-values calculated by these two software is <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">almost identical</a>.</p>

<p>For comparing biological themes, <code>clusterProfiler</code> supports formula to express complex conditions and facet is supported to <a href="https://github.com/GuangchuangYu/clusterProfiler/issues/32">visualize complex result</a>.</p>

<p><code>read.gmt</code> function for parsing GMT file format from <a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a>, so that gene set collections in this database can be used in <code>clusterProfiler</code> for both hypergeometric test and GSEA.</p>

<p><a href="http://guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis/">KEGG Module</a> was supported just like the KEGG Pathway, <code>clusterProfiler</code> will query the online annotation data which keep the annotation data alwasy updated.</p>

<p>The KEGG database was updated quite frequently. The <code>KEGG.db</code> which was not updated since 2012, it contains annotation of <a href="http://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">5894 human genes</a>. In Feb. 2015, when <code>clusterProfiler</code> first supports querying online KEGG data, KEGG contains annotation of <a href="http://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/">6861 human genes</a> and today it has <strong>7018</strong> human genes annotated. Most of the tools/webservers used out-dated data (e.g. DAVID not updated since 2010), the analyzed result may totally changed if we use a recently updated data. Indeed <code>clusterProfiler</code> is more reliable as we always use the latest data.</p>

<p>In addition to <code>bitr</code> function that can translate biological ID using <code>OrgDb</code> object, we provides <a href="http://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler/">bitr_kegg</a> that uses KEGG API for translating biological ID. It supports more than 4000 species (can be search via the <code>search_kegg_species</code> function) as in KEGG Pathway and Module analysis.</p>

<p>The function called of <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/">enrichGO and gseGO</a> was changed. Now not only species that have OrgDb available in Bioconductor can be analyzed but also all species that have an OrgDb can be analyzed which can be query online via <code>AnnotationHub</code> or build with user&rsquo;s own data. With this update, <code>enrichGO</code> and <code>gseGO</code> can input any gene ID type if only the ID type was supported in the <code>OrgDb</code>.</p>

<p>GO enrichment analysis alwasy output redundant terms, we implemented a <code>simplify</code> function to remove redundant terms by calculating GO semantic similarity using <code>GOSemSim</code>. Several useful utilities include <code>dropGO</code>, <code>go2ont</code>, <code>go2term</code>, <code>gofilter</code> and <code>gsfilter</code> are also provided.</p>

<p>I bump the version to 3.0.0 the following three reasons:</p>

<ul>
<li>the changes of function calls</li>
<li>can analyze any ontology/pathway annotation</li>
<li>can analyze all speices</li>
</ul>

<p>Although the package is very simple when I published it, I keep update and add new features from my own idea or user&rsquo;s request. Now this package is indeed in good shape. Here is the summary.</p>

<p>This package implements methods to analyze and visualize functional profiles of genomic coordinates (supported by <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>), gene and gene clusters.</p>

<p>clusterProfiler supports both <em>hypergeometric test</em> and <em>Gene Set Enrichment Analysis</em> for many ontologies/pathways, including:</p>

<ul>
<li>Disease Ontology (via <a href="http://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (via <a href="http://www.bioconductor.org/packages/DOSE">DOSE</a>)</li>
<li>Gene Ontology (supports many species with GO annotation query online via <a href="https://bioconductor.org/packages/AnnotationHub/">AnnotationHub</a>)</li>
<li>KEGG Pathway and Module with latest online data (supports more than 2000 species listed in <a href="http://www.genome.jp/kegg/catalog/org_list.html">http://www.genome.jp/kegg/catalog/org_list.html</a>)</li>
<li>Reactome Pathway (via <a href="http://www.bioconductor.org/packages/ReactomePA">ReactomePA</a>)</li>
<li>DAVID (via <a href="http://www.bioconductor.org/packages/RDAVIDWebService">RDAVIDWebService</a>)</li>
<li><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a>

<ul>
<li>hallmark gene sets</li>
<li>positional gene sets</li>
<li>curated gene sets</li>
<li>motif gene sets</li>
<li>computational gene sets</li>
<li>GO gene sets</li>
<li>oncogenic signatures</li>
<li>immunologic signatures</li>
</ul></li>
<li>Other Annotations

<ul>
<li>from other sources (e.g. <a href="http://www.disgenet.org/web/DisGeNET/menu/home">DisGeNET</a> as <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">an example</a>)</li>
<li>user&rsquo;s annotation</li>
<li>customized ontology</li>
<li>and many others</li>
</ul></li>
</ul>

<p><em>clusterProfiler</em> also provides several visualization methods to help interpreting enriched results, including:</p>

<ul>
<li>barplot</li>
<li>cnetplot</li>
<li>dotplot</li>
<li>enrichMap</li>
<li>gseaplot</li>
<li>plotGOgraph (via <a href="http://www.bioconductor.org/packages/topGO">topGO</a> package)</li>
<li>upsetplot</li>
</ul>

<p>and several useful utilities:</p>

<ul>
<li>bitr (Biological Id TranslatoR)</li>
<li>bitr_kegg (bitr using KEGG source)</li>
<li>compareCluster (biological theme comparison)</li>
<li>dropGO (screen out GO term of specific level or specific term)</li>
<li>go2ont (convert GO ID to Ontology)</li>
<li>go2term (convert GO ID to descriptive term)</li>
<li>gofilter (restrict result at specific GO level)</li>
<li>gsfilter (restrict result by gene set size)</li>
<li>search_kegg_organism (search kegg supported organism)</li>
<li>setReadable (convert IDs stored <code>enrichResult</code> object to gene symbol)</li>
<li>simplify (remove redundant GO terms, supported via <a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>)</li>
</ul>

<h2 id="dose:de4254a6abb9a1c947b7d959d55455bd">DOSE</h2>

<p>DOSE now test bimodal separately in GSEA and the output pvalues are <a href="(http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">more conserved</a>.</p>

<p><code>maxGSSize</code> parameter was added, with default value of <code>500</code>. Usually if the geneset &gt; 500, its probability of being called significant by GSEA rises quite dramatically.</p>

<p><code>gsfilter</code> function for restricting enriched results with minimal and maximal gene set sizes.</p>

<p><code>upsetplot</code> was implemented to visualize overlap of enriched gene sets.</p>

<p>The dot sizes in <code>enrichMap</code> now scaled by category sizes</p>

<h2 id="ggtree:de4254a6abb9a1c947b7d959d55455bd">ggtree</h2>

<p>I put more efforts to extend <code>ggtree</code> than the sum of all other packages. Here listed the major new features while small improvement and bug fixed can be found in the NEWS file.</p>

<h3 id="io:de4254a6abb9a1c947b7d959d55455bd">IO</h3>

<ul>
<li>support <a href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/">NHX</a> file format via <code>read.nhx</code> function</li>
<li>support <a href="http://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format/">phylip</a> tree format via <code>read.phylip</code> function</li>
<li>raxml2nwk for converting raxml bootstrap tree to newick text</li>
<li>all parser functions support passing <code>textConnection(text_string)</code> as a file</li>
<li>support <code>ape</code> bootstrap analysis</li>
<li>support annotating tree with ancestral sequences inferred by <code>phangorn</code></li>
<li>support obkData object defined by OutbreakTools package</li>
<li>support phyloseq object defined by phyloseq package</li>
</ul>

<h3 id="layers:de4254a6abb9a1c947b7d959d55455bd">layers</h3>

<ul>
<li>geom_point2,geom_text2, geom_segment2 and geom_label2 to <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">support subsetting</a></li>
<li>geom_treescale for adding scale of branch length</li>
<li>geom_cladelabel for labeling selected clade</li>
<li>geom_tiplab2 for adding tiplab of circular tree</li>
<li>geom_taxalink for connecting related taxa</li>
<li>geom_range for adding range to present uncertainty of branch lengths</li>
<li>subview and inset now support annotating with <a href="http://guangchuangyu.github.io/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">image files</a></li>
</ul>

<h3 id="utilities:de4254a6abb9a1c947b7d959d55455bd">utilities</h3>

<ul>
<li><a href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/">rescale_tree</a> function to rescale branch lengths using numerical variable</li>
<li>MRCA for finding Most Recent Common Ancestor among a vector of tips</li>
<li>viewClade to zoom in a selected clade</li>
</ul>

<h3 id="vignettes:de4254a6abb9a1c947b7d959d55455bd">vignettes</h3>

<p>Split the long vignette to several small ones and add more examples.</p>

<ul>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/ggtree.html">ggtree</a></li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/treeImport.html">Tree Data Import</a></li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/treeVisualization.html">Tree Visualization</a></li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/treeAnnotation.html">Tree Annotation</a></li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/treeManipulation.html">Tree Manipulation</a></li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/ggtree/inst/doc/advanceTreeAnnotation.html">Advance Tree Annotation</a></li>
</ul>

<p>Here is the NEWS record:</p>

<pre><code>CHANGES IN VERSION 1.3.16
------------------------
 o geom_treescale() supports family argument &lt;2016-04-27, Wed&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/56
 o update fortify.phylo to work with phylo that has missing value of edge length &lt;2016-04-21, Thu&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/54
 o support passing textConnection(text_string) as a file &lt;2016-04-21, Thu&gt;
   + contributed by Casey Dunn &lt;casey_dunn@brown.edu&gt;
   + https://github.com/GuangchuangYu/ggtree/pull/55#issuecomment-212859693
 
CHANGES IN VERSION 1.3.15
------------------------
 o geom_tiplab2 supports parameter hjust &lt;2016-04-18, Mon&gt;
 o geom_tiplab and geom_tiplab2 support using geom_label2 by passing geom=&quot;label&quot; &lt;2016-04-07, Thu&gt;
 o geom_label2 that support subsetting &lt;2016-04-07, Thu&gt;
 o geom_tiplab2 for adding tip label of circular layout &lt;2016-04-06, Wed&gt;
 o use plot$plot_env to access ggplot2 parameter &lt;2016-04-06, Wed&gt;
 o geom_taxalink for connecting related taxa &lt;2016-04-01, Fri&gt; 
 o geom_range for adding range of HPD to present uncertainty of evolutionary inference &lt;2016-04-01, Fri&gt;
 
CHANGES IN VERSION 1.3.14
------------------------
 o geom_tiplab works with NA values, compatible with collapse &lt;2016-03-05, Sat&gt;
 o update theme_tree2 due to the issue of https://github.com/hadley/ggplot2/issues/1567 &lt;2016-03-05, Sat&gt;
 o offset works in `align=FFALSE` with `annotation_image` function &lt;2016-02-23, Tue&gt;
   + see https://github.com/GuangchuangYu/ggtree/issues/46
 o subview and inset now supports annotating with img files &lt;2016-02-23, Tue&gt;
 
CHANGES IN VERSION 1.3.13
------------------------
 o add example of rescale_tree function in treeAnnotation.Rmd &lt;2016-02-07, Sun&gt;
 o geom_cladelabel works with collapse &lt;2016-02-07, Sun&gt;
   + see https://github.com/GuangchuangYu/ggtree/issues/38 

CHANGES IN VERSION 1.3.12
------------------------
 o exchange function name of geom_tree and geom_tree2  &lt;2016-01-25, Mon&gt;
 o solved issues of geom_tree2 &lt;2016-01-25, Mon&gt;
   + https://github.com/hadley/ggplot2/issues/1512
 o colnames_level parameter in gheatmap &lt;2016-01-25, Mon&gt;
 o raxml2nwk function for converting raxml bootstrap tree to newick format &lt;2016-01-25, Mon&gt; 
 
CHANGES IN VERSION 1.3.11
------------------------
 o solved issues of geom_tree2 &lt;2016-01-25, Mon&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/36
 o change compute_group() to compute_panel in geom_tree2() &lt;2016-01-21, Thu&gt;
   + fixed issue, https://github.com/GuangchuangYu/ggtree/issues/36
 o support phyloseq object &lt;2016-01-21, Thu&gt;
 o update geom_point2, geom_text2 and geom_segment2 to support setup_tree_data &lt;2016-01-21, Thu&gt;
 o implement geom_tree2 layer that support duplicated node records via the setup_tree_data function &lt;2016-01-21, Thu&gt;
 o rescale_tree function for rescaling branch length of tree object &lt;2016-01-20, Wed&gt;
 o upgrade set_branch_length, now branch can be rescaled using feature in extraInfo slot &lt;2016-01-20, Wed&gt;
 
CHANGES IN VERSION 1.3.10
------------------------
 o remove dependency of gridExtra by implementing multiplot function instead of using grid.arrange &lt;2016-01-20, Wed&gt;
 o remove dependency of colorspace &lt;2016-01-20, Wed&gt;
 o support phylip tree format and update vignette of phylip example &lt;2016-01-15, Fri&gt;

CHANGES IN VERSION 1.3.9
------------------------
 o optimize getYcoord &lt;2016-01-14, Thu&gt;
 o add 'multiPhylo' example in 'Tree Visualization' vignette &lt;2016-01-13, Wed&gt;
 o viewClade, scaleClade, collapse, expand, rotate, flip, get_taxa_name and scale_x_ggtree accepts input tree_view=NULL.
   these function will access the last plot if tree_view=NULL. &lt;2016-01-13, Wed&gt;
   + &gt; ggtree(rtree(30)); viewClade(node=35) works. no need to pipe.
 
CHANGES IN VERSION 1.3.8
------------------------
 o add example of viewClade in 'Tree Manipulation' vignette &lt;2016-01-13, Wed&gt;
 o add viewClade function &lt;2016-01-12, Tue&gt;
 o support obkData object defined by OutbreakTools &lt;2016-01-12, Tue&gt;
 o update vignettes &lt;2016-01-07, Thu&gt;
 o 05 advance tree annotation vignette &lt;2016-01-04, Mon&gt;
 o export theme_inset &lt;2016-01-04, Mon&gt;
 o inset, nodebar, nodepie functions &lt;2015-12-31, Thu&gt;
 
CHANGES IN VERSION 1.3.7
------------------------
 o split the long vignette to several vignettes
   + 00 ggtree &lt;2015-12-29, Tue&gt;
   + 01 tree data import &lt;2015-12-28, Mon&gt;
   + 02 tree visualization &lt;2015-12-28, Mon&gt;
   + 03 tree manipulation &lt;2015-12-28, Mon&gt;
   + 04 tree annotation &lt;2015-12-29, Tue&gt;
 
CHANGES IN VERSION 1.3.6
------------------------
 o MRCA function for finding Most Recent Common Ancestor among a vector of tips &lt;2015-12-22, Tue&gt;
 o geom_cladelabel: add bar and label to annotate a clade &lt;2015-12-21, Mon&gt;
   - remove annotation_clade and annotation_clade2 functions.
 o geom_treescale: tree scale layer. (add_legend was removed) &lt;2015-12-21, Mon&gt;
 
CHANGES IN VERSION 1.3.5
------------------------
 o bug fixed, read.nhx now works with scientific notation &lt;2015-11-30, Mon&gt;
   + see https://github.com/GuangchuangYu/ggtree/issues/30

CHANGES IN VERSION 1.3.4
------------------------
 o rename beast feature when name conflict with reserve keywords (label, branch, etc) &lt;2015-11-27, Fri&gt;
 o get_clade_position function &lt;2015-11-26, Thu&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/28
 o get_heatmap_column_position function &lt;2015-11-25, Wed&gt;
   + see https://github.com/GuangchuangYu/ggtree/issues/26
 o support NHX (New Hampshire X) format via read.nhx function &lt;2015-11-17, Tue&gt;
 o bug fixed in extract.treeinfo.jplace &lt;2015-11-17, Thu&gt;
 
CHANGES IN VERSION 1.3.3
------------------------
 o support color=NULL in gheatmap, then no colored line will draw within the heatmap &lt;2015-10-30, Fri&gt;
 o add `angle` for also rectangular, so that it will be available for layout='rectangular' following by coord_polar() &lt;2015-10-27, Tue&gt;
 
CHANGES IN VERSION 1.3.2
------------------------
 o update vignette, add example of ape bootstrap and phangorn ancestral sequences &lt;2015-10-26, Mon&gt;
 o add support of ape bootstrap analysis &lt;2015-10-26, Mon&gt;
   see https://github.com/GuangchuangYu/ggtree/issues/20
 o add support of ancestral sequences inferred by phangorn &lt;2015-10-26, Mon&gt;
   see https://github.com/GuangchuangYu/ggtree/issues/21
 
CHANGES IN VERSION 1.3.1
------------------------
 o change angle to angle + 90, so that label will in radial direction &lt;2015-10-22, Thu&gt;
   + see https://github.com/GuangchuangYu/ggtree/issues/17
 o na.rm should be always passed to layer(), fixed it in geom_hilight and geom_text2 &lt;2015-10-21, Wed&gt;
   + see https://github.com/hadley/ggplot2/issues/1380
 o matching beast stats with tree using internal node number instead of label &lt;2015-10-20, Tue&gt;
</code></pre>

<h2 id="gosemsim:de4254a6abb9a1c947b7d959d55455bd">GOSemSim</h2>

<p>update IC data using update OrgDb packages.</p>

<h2 id="reactomepa:de4254a6abb9a1c947b7d959d55455bd">ReactomePA</h2>

<p>Internal implementation was updated according to the change of <code>DOSE</code>.</p>

<p>We published ReactomePA in <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E">Molecular BioSystems</a>.</p>


	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
	
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>May 5, 2016</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>  <a class="category" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>  <a class="category" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>  <a class="category" href="http://guangchuangyu.github.io/tags/dose">DOSE</a>  <a class="category" href="http://guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>  <a class="category" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>  <a class="category" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler/" title="convert biological ID with KEGG API using clusterProfiler">convert biological ID with KEGG API using clusterProfiler</a>
    

    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>



  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/05/bioc-3.3-news-of-my-packages/">BioC 3.3: NEWS of my packages</a>
      </li>
      
      <li class="post">
        <a href="/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler/">convert biological ID with KEGG API using clusterProfiler</a>
      </li>
      
      <li class="post">
        <a href="/2016/04/kegg-module-enrichment-analysis/">KEGG Module Enrichment Analysis</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/google-drive--hku/">Google Drive @ HKU</a>
      </li>
      
    </ul>
  </section>

  
  
  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


