<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2015 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="post">
    <h1>ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</h1>
    <span class="post-date">Jul 13, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/#disqus_thread">Comments</a>
    
    <br/>
    <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/publication">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
    </span>
    <p>My R/Bioconductor package,
<a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>, published
in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145">Bioinformatics</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> had been
cited by <a href="http://www.biomedcentral.com/1471-2164/16/292">http://www.biomedcentral.com/1471-2164/16/292</a> and
<a href="http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short">http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short</a>, and
was used (not cited) in
<a href="http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract">http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract</a>
and
<a href="http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract">http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract</a>.</p>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/master/Bioconductor/ChIPseeker/bioinfo_2015.png" alt="" />
<strong>ChIPseeker: an R/Bioconductor package for ChIP peak annotation,
comparison and visualization</strong> <strong>G Yu</strong>, LG Wang, QY He.
<strong>Bioinformatics</strong> 2015, 31(14):2382-2383 DOI:
<a href="http://doi.org/10.1093/bioinformatics/btv145">10.1093/bioinformatics/btv145</a></p>

<p><strong>Summary:</strong> ChIPseeker is an R package for annotating ChIP-seq data
analysis. It supports annotating ChIP peaks and provides functions to
visualize ChIP peaks coverage over chromosomes and profiles of peaks
binding to TSS regions. Comparison of ChIP peak profiles and annotation
are also supported. Moreover, it supports evaluating significant overlap
among ChIP-seq datasets. Currently, ChIPseeker contains 15 000 bed file
information from GEO database. These datasets can be downloaded and
compare with userâ€™s own data to explore significant overlap datasets for
inferring co-regulation or transcription factor complex for further
investigation. <strong>Availability and implementation:</strong> ChIPseeker is
released under Artistic-2.0 License. The source code and documents are
freely available through Bioconductor
(<a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html">http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html</a>).</p>

  </div>
  <div id="disqus_thread"></div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>



<script type="text/javascript">
    var disqus_shortname = "gcyu";
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>



</body>
</html>

