<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/2016/10/bioc-34-news-of-my-bioc-packages/">

  

  <title>[Bioc 34] NEWS of my BioC packages | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  

  <div class="article-container">
    <h1 itemprop="name">[Bioc 34] NEWS of my BioC packages</h1>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-10-19 18:13:57 &#43;0800 HKT" itemprop="datePublished">
      Wed, Oct 19, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >, 
    
    <a href="/tags/meshes">meshes</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2f2016%2f10%2fbioc-34-news-of-my-bioc-packages%2f"
         target="_blank">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%5bBioc%2034%5d%20NEWS%20of%20my%20BioC%20packages&amp;url=https%3a%2f%2fguangchuangyu.github.io%2f2016%2f10%2fbioc-34-news-of-my-bioc-packages%2f"
         target="_blank">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fguangchuangyu.github.io%2f2016%2f10%2fbioc-34-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2034%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fguangchuangyu.github.io%2f2016%2f10%2fbioc-34-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2034%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%5bBioc%2034%5d%20NEWS%20of%20my%20BioC%20packages&amp;body=https%3a%2f%2fguangchuangyu.github.io%2f2016%2f10%2fbioc-34-news-of-my-bioc-packages%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    <div class="article-style" itemprop="articleBody">
        <p>I have 7 packages published within the <code>Bioconductor</code> project.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker" target="_blank">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler" target="_blank">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE" target="_blank">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree" target="_blank">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim" target="_blank">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/meshes" target="_blank">meshes</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA" target="_blank">ReactomePA</a></li>
</ul>

<p>A new package <a href="http://guangchuangyu.github.io/meshes" target="_blank">meshes</a> was included in BioC 3.4 release.</p>

<p></p>

<h2 id="chipseeker">ChIPseeker</h2>

<p>For <code>ChIPseeker</code>, this release fixed several minor bugs.</p>

<p>GEO data was updated, now <code>ChIPseeker</code> contains <code>20947</code> bed file information.</p>

<pre><code>CHANGES IN VERSION 1.9.8
------------------------
 o plotAvgProf/plotAvgProf2 order of panel by names of input tagMatrix List &lt;2016-09-25, Sun&gt;
 o test ENSEMBL ID using '^ENS' instead of '^ENSG' &lt;2016-09-20, Tue&gt;
   + https://github.com/GuangchuangYu/ChIPseeker/issues/41

CHANGES IN VERSION 1.9.7
------------------------
 o unit test &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 1.9.6
------------------------
 o update vignette &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 1.9.5
------------------------
 o when TxDb doesn't have gene_id information, converting gene ID (ensembl/entrez and symbol) will be omitted instead of throw error. &lt;2016-08-02, Tue&gt;
   + https://www.biostars.org/p/204142
 o bug fixed if testing targetPeak is a list of GRanges objects in enrichPeakOverlap function &lt;2016-07-20, Wed&gt;
   + https://github.com/GuangchuangYu/ChIPseeker/issues/37
   + https://github.com/GuangchuangYu/ChIPseeker/issues/36
 o fixed typo in determine gene ID type &lt;2016-06-21, Tue&gt;
   + https://github.com/GuangchuangYu/ChIPseeker/issues/28#issuecomment-227212519
 o move upsetplot generics to DOSE and import from DOSE to prevent function name conflict &lt;2016-06-14, Tue&gt;

CHANGES IN VERSION 1.9.4
------------------------
 o bug fixed &lt;2016-06-08, Wed&gt;
   + https://github.com/GuangchuangYu/ChIPseeker/issues/17#issuecomment-224407402
   + https://github.com/GuangchuangYu/ChIPseeker/pull/24/files

CHANGES IN VERSION 1.9.3
------------------------
 o use byte compiler &lt;2016-05-18, Wed&gt;
 o https://github.com/Bioconductor-mirror/ChIPseeker/commit/f1ada57b9c66a1a44355bbbbdaf5b0a88e10cf7d

CHANGES IN VERSION 1.9.2
------------------------
 o name tagMatrix in plotAvgProf automatically if missing &lt;2016-05-12, Thu&gt;
 o https://github.com/Bioconductor-mirror/ChIPseeker/commit/d5f16b2bc01725e30282c3acb33007ef521a514c

CHANGES IN VERSION 1.9.1
------------------------
 o bug fixed in getNearestFeatureIndicesAndDistances &lt;2016-05-11, Wed&gt;
   + correct metadata in dummy NA feature
</code></pre>

<h2 id="clusterprofiler">clusterProfiler</h2>

<p>Thanks to the update of <code>GOSemSim</code>, now <code>simplify</code> method for removing redundent GO terms can be applied to all species that have <code>OrgDb</code> object.</p>

<p><code>summary</code> for <code>compareClusterResult</code> object was deprecated, user need to use <code>as.data.frame</code>. Several accesor functions including <code>$</code>, <code>[</code> and <code>[[</code> were supported as well as <code>head</code>, <code>tail</code> and <code>dim</code> for <code>compareClusterResult</code> object.</p>

<p><code>browseKEGG</code> function was added, which can open <code>browser</code> and highligh enriched genes in the pathway.</p>

<p>Updates of <code>DOSE</code> also affect <code>clusterProfiler</code>, especially the leading edge analysis was incorporated.</p>

<pre><code>CHANGES IN VERSION 3.1.9
------------------------
 o bug fixed of simplify method for compareCluster object &lt;2016-08-17, Mon&gt;
   + use semData parameter according to the change of GOSemSim

CHANGES IN VERSION 3.1.8
------------------------
 o as.data.frame method for compareClusterResult &lt;2016-09-29, Thu&gt;
 o geneID and geneInCategory methods &lt;2016-09-19, Mon&gt;

CHANGES IN VERSION 3.1.7
------------------------
 o [, [[, head, tail, dim methods for compareClusterResult &lt;2016-08-28, Sun&gt;
 o fixed R-devel check &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 3.1.6
------------------------
 o update vignette &lt;2016-08-16, Tue&gt;
 o browseKEGG &lt;2016-08-15, Mon&gt;
 o unit test &lt;2016-08-15, Mon&gt;

CHANGES IN VERSION 3.1.5
------------------------
 o move enrichMeSH &amp; gseMeSH to meshes &lt;2016-08-11, Thu&gt;

CHANGES IN VERSION 3.1.4
------------------------
 o MeSH term enrichment analysis (enrichMeSH and gseMeSH) &lt;2016-07-28, Thu&gt;

CHANGES IN VERSION 3.1.3
------------------------
 o export download_KEGG &lt;2016-07-25, Mon&gt;
 o according to the changes of GOSemSim and DOSE &lt;2016-07-05, Tue&gt;

CHANGES IN VERSION 3.1.2
------------------------
 o 'by' parameter for GSEA analysis &lt;2016-07-04, Mon&gt;
 o duplicated KEGG path id for KEGG orthology, only use ko &lt;2016-06-28, Tue&gt;
   + e.g. map00010 and ko00010
   + http://www.kegg.jp/dbget-bin/www_bget?ko+ko00010
   + http://www.kegg.jp/dbget-bin/www_bget?ko+map00010

CHANGES IN VERSION 3.1.1
------------------------
 o getGOLevel and dropGO support using a vector of GO level &lt;2016-05-20, Fri&gt;
   + https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/clusterprofiler/zNYHLI6RpJI/1u4XRKfNAQAJ
 o use byte compiler &lt;2016-05-18, Wed&gt;
 o https://github.com/Bioconductor-mirror/clusterProfiler/commit/2949fe8db5464c419c9d1665cb24f43c49ac54ec
</code></pre>

<h2 id="dose">DOSE</h2>

<p>I am glad to release <code>DOSE</code> v3.0, the most important enhancement is now GSEA supports <a href="https://guangchuangyu.github.io/2016/07/leading-edge-analysis/" target="_blank">leading edge analysis</a>.</p>

<p>Thanks to the invitation from Alexey Sergishichev, I switch to <code>fgsea</code> in <code>DOSE</code> for calculating enrichment score of GSEA analysis. User can still use my implementation by specifying <code>by=&quot;DOSE&quot;</code>.</p>

<p>Now <code>enrichMap</code> and <code>cnetplot</code> output <code>igraph</code> object, so that it can be viewed by other software (e.g. networkD3). <code>title</code> parameter is supported in <code>gseaplot</code>.</p>

<p>Several accessor functions, including <code>$</code>, <code>[</code>, and <code>[[</code> were implemented for <code>enrichResult</code> and <code>gseaResult</code> object. <code>head</code>, <code>tail</code> and <code>dim</code> is also supported. <code>summary</code> method was deprecated, instead user should use <code>as.data.frame</code> to convert the <code>enrichResult</code> or <code>gseaResult</code> object to <code>data.frame</code>.</p>

<p>In additional to <code>DO</code> and <code>NCG</code>, <code>DisGeNET</code> data was incorporated and we can identify disease association by using this annotation.</p>

<pre><code>CHANGES IN VERSION 2.11.12
------------------------
 o deprecate summary method and use as.data.frame instead &lt;2016-09-29, Thu&gt;

CHANGES IN VERSION 2.11.11
------------------------
 o export geneID and geneInCategory &lt;2016-09-19, Mon&gt;

CHANGES IN VERSION 2.11.10
------------------------
 o geneID and geneInCategory accessor functions &lt;2016-09-19, Mon&gt;

CHANGES IN VERSION 2.11.9
------------------------
 o enrichMap works with result contains only one term &lt;2016-08-15, Wed&gt;
   + https://github.com/GuangchuangYu/DOSE/issues/15
 o build_Anno now works with tibble &lt;2016-08-18, Tue&gt;

CHANGES IN VERSION 2.11.8
------------------------
 o add unit test &lt;2016-08-15, Mon&gt;

CHANGES IN VERSION 2.11.7
------------------------
 o change for meshes packages &lt;2016-08-11, Thu&gt;

CHANGES IN VERSION 2.11.6
------------------------
 o user can use options(DOSE_workers = x) to set using x cores for GSEA analysis &lt;2016-08-02, Tue&gt;
 o support DisGeNET enrichment analyses &lt;2016-08-01, Mon&gt;
   + enrichDGN, enrichDGNv, gseDGN
 o update vignettes &lt;2016-07-29, Fri&gt;

CHANGES IN VERSION 2.11.5
------------------------
 o enrichMap now output igraph object that can be viewed using other software like networkD3 &lt;2016-07-25, Mon&gt;
 o dim methods for enrichResult and gseaResult &lt;2016-07-22, Fri&gt;
 o $ methods for enrichResult and gseaResult &lt;2016-07-20, Wed&gt;
 o switch from parallel to BiocParallel &lt;2017-07-07, Thu&gt;
 o [, head and tail methods for enrichResult and gseaResult &lt;2016-07-06, Wed&gt;
 o change according to GOSemSim &lt;2016-07-05, Tue&gt;

CHANGES IN VERSION 2.11.4
------------------------
 o 'title' parameter for gseaplot &lt;2016-07-04, Mon&gt;
   + contributed by https://github.com/pedrostrusso
   + https://github.com/GuangchuangYu/DOSE/pull/13
 o 'by' parameter in GSEA_internal, by default by = 'fgsea' &lt;2016-07-04, Mon&gt;
   + by = 'fgsea', use GSEA algorithm implemented in fgsea
   + by = 'DOSE', use GSEA algorithm implemented in DOSE
 o leading edge analysis for GSEA &lt;2016-07-04, Mon&gt;

CHANGES IN VERSION 2.11.3
------------------------
 o output igraph object in cnetplot &lt;2016-06-21, Tue&gt;
 o upsetplot generics &lt;2016-06-14, Tue&gt;
 o [[ methods for enrichResult and gseaResult for accessing gene set &lt;2016-06-14, Tue&gt;

CHANGES IN VERSION 2.11.2
------------------------
 o use byte compiler &lt;2016-05-18, Wed&gt;
 o https://github.com/Bioconductor-mirror/DOSE/commit/6c508c6a6816f465bb372f30f4ab99c839d81767

CHANGES IN VERSION 2.11.1
------------------------
 o https://github.com/Bioconductor-mirror/DOSE/commit/7e87d01e671ce1b5fbe974c06b796b1a2970f11c
</code></pre>

<h2 id="ggtree">ggtree</h2>

<p>I am very happy that <code>ggtree</code> was published in <code>MEE</code>.</p>

<p><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. <a href="http://dx.doi.org/10.1111/2041-210X.12628" target="_blank"><code>doi:10.1111/2041-210X.12628</code></a>.</p>

<p>In this release, more functionalities were added and of course several bug fixed.</p>

<h3 id="file-parsing">File parsing</h3>

<p><code>read.phyloT</code> for parsing newick format of <a href="https://www.biostars.org/p/210401/#216128" target="_blank">phyloT output</a>.</p>

<p><code>read.nhx</code> supports newick file as NHX format allow tree without <code>nhx</code> tags.</p>

<p>Support reading BEAST MCC trees by <code>read.beast</code> function.</p>

<h3 id="tree-object-support">tree object support</h3>

<p><code>phylo4d</code> defined in <code>phylobase</code> package is supported, and associated data stored in <code>phylo4d</code> object can be used to annotate the tree directly in <code>ggtree</code>.</p>

<h3 id="geom">Geom</h3>

<p>New geoms including <code>geom_strip</code>, <code>geom_range</code>, <code>geom_balance</code> (contributed by Justin Silverman, the author of <a href="https://www.bioconductor.org/packages/philr" target="_blank">philr</a>).</p>

<p><code>geom_cladelabel</code> now supports parsing expression to label clades.</p>

<h3 id="layout">Layout</h3>

<p><code>fan</code> layout is supported, and user can use <code>open_tree</code> function to transform a tree from <code>rectangular</code> or <code>circular</code> to <code>fan</code> layout.</p>

<h3 id="enhancement-and-utilities">Enhancement and utilities</h3>

<p><a href="https://guangchuangyu.github.io/2016/06/identify-method-for-ggtree" target="_blank">identify</a> method support annotating trees interactively.</p>

<p><code>facet_plot</code> supports aligning other plots to phylogenetic trees.</p>

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">associate tree with different type of data by <a href="https://twitter.com/hashtag/ggtree?src=hash&amp;ref_src=twsrc%5Etfw">#ggtree</a> <a href="https://t.co/6w755VWytZ">https://t.co/6w755VWytZ</a> <a href="https://t.co/K8WViEi13E">pic.twitter.com/K8WViEi13E</a></p>&mdash; Guangchuang Yu (@guangchuangyu) <a href="https://twitter.com/guangchuangyu/status/773359054924095488?ref_src=twsrc%5Etfw">September 7, 2016</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>


<p><code>as.polytomy</code> allow collapsing binary tree to polytomy by applying &lsquo;fun&rsquo; to selected &lsquo;feature&rsquo; (e.g. bootstrap value less than 70).</p>

<p><code>rotate_tree</code> allow user to rotating <code>circular</code> layout tree.</p>

<pre><code>CHANGES IN VERSION 1.5.17
------------------------
 o read.nhx support newick file &lt;2016-10-17, Mon&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/79

CHANGES IN VERSION 1.5.16
------------------------
 o read.phyloT for parsing newick format of phyloT output &lt;2016-10-11, Tue&gt;
   + https://www.biostars.org/p/210401/#216128
 o fixed aes mapping in geom_strip &lt;2016-10-11, Tue&gt;
 o fixed R check &lt;2016-10-10, Mon&gt;
   + check.aes parameter is not available in release version of ggplot2 yet

CHANGES IN VERSION 1.5.15
------------------------
 o check.aes for layers defined in ggtree &lt;2016-10-07, Fri&gt;
 o recalculate 'angle' when collapse, expand and rotate clade &lt;2016-10-06, Thu&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/78

CHANGES IN VERSION 1.5.14
------------------------
 o subset tip in geom_tiplab2 &lt;2016-10-05, Wed&gt;
 o add `compute_group` according to ggplot (v2.1.0) &lt;2016-09-29, Thu&gt;
   + https://github.com/hadley/ggplot2/issues/1797
 o unit test for groupOTU and groupClade &lt;2016-09-22, Thu&gt;
 o groupOTU label groups by input group names (when input is a named list) &lt;2016-09-22, Thu&gt;
 o update angle calculation for geom_tiplab &lt;2016-09-13, Thu&gt;
 o as.polytomy to collapse binary tree to polytomy by applying 'fun' to selected 'feature' (e.g. bootstrap value less than 70). &lt;2016-09-13, Tue&gt;
   + currently only phylo object supported.
   + add test for as.polytomy

CHANGES IN VERSION 1.5.13
------------------------
 o facet_plot for plotting data with tree &lt;2016-09-06, Tue&gt;
 o more parameters for column names in gheatmap &lt;2016-09-06, Tue&gt;
   + colnames_angle
   + colnames_offset_x
   + colnames_offset_y
   + hjust
 o offset parameter in geom_tiplab and geom_tiplab2 &lt;2016-09-05, Mon&gt;

CHANGES IN VERSION 1.5.12
------------------------
 o use data in all layers instead of the base layer for coordination calculation in subview &lt;2016-09-01, Thu&gt;
 o bug fixed in subview, width &amp; height should be width/2 &amp; height/2 &lt;2016-09-01, Thu&gt;

CHANGES IN VERSION 1.5.11
------------------------
 o gheatmap works with matrix &lt;2016-08-28, Sun&gt;
   + https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/bioc-ggtree/2YLvXHMJJ6U/c4zS7yfGCAAJ
 o support parsing expression in geom_strip &lt;2016-08-18, Thu&gt;
 o bug fixed in geom_tiplab &lt;2016-08-17, Wed&gt;
   + https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/bioc-ggtree/Tm9ULK7hd9E/HviXEh3CBwAJ
 o update citation info, add doi. &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 1.5.10
------------------------
 o fixed issue #72 for label of geom_treescale not displayed &lt;2016-08-16, Tue&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/72

CHANGES IN VERSION 1.5.9
------------------------
 o update citation info &lt;2016-08-12, Fri&gt;

CHANGES IN VERSION 1.5.8
------------------------
 o add color parameter in geom_cladelabel, color should be of length 1 or 2 &lt;2016-08-11, Thu&gt;
 o geom_cladelabel support parsing expression &lt;2016-08-11, Thu&gt;

CHANGES IN VERSION 1.5.7
------------------------
 o geom_strip can accept taxa name as input but labeling strip will not supported.
   To support labeling strip, user need to input node id  &lt;2016-07-27, Wed&gt;
 o nodeid function for converting node label(s) to node id(s) &lt;2016-07-27, Wed&gt;

CHANGES IN VERSION 1.5.6
------------------------
 o remove dependency of Biostring for installing ggtree &lt;2016-07-21, Thu&gt;
   + still needed for building vignette and for processing FASTA file
 o remove dependency of EBImage for building &amp; installing ggtree &lt;2016-07-21, Thu&gt;
   + the package is still needed if user want to annotate tree with image file
 o `%&lt;+%` now works with tbl_df &lt;2016-07-21, Thu&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/66
 o identify method for ggtree &lt;2016-06-28, Tue&gt;
   + see https://guangchuangyu.github.io/2016/06/identify-method-for-ggtree
 o geom_balance contributed by Justin Silverman &lt;2016-06-22, Wed&gt;
   + see https://github.com/GuangchuangYu/ggtree/pull/64

CHANGES IN VERSION 1.5.5
------------------------
 o update geom_tiplab2 according to angle change introduced by open_tree &lt;2016-06-20, Mon&gt;
 o bug fixed in collapse, now work with collapse a clade that contain a subclade that was already collapsed &lt;2016-06-02-Thu&gt;
 o bug fixed if time-scaled tree extend into the BCE. &lt;2016-06-02, Thu&gt;
   + as.Date won't work for BCE time.
   + as.Date=FALSE by default in fortify method, just use the time in decimal format (real number, not Date object).

CHANGES IN VERSION 1.5.4
------------------------
 o reroot method for raxml object &lt;2016-05-22, Sun&gt;
 o bug fixed in scaleClade, now y positions are (hopefully) always correct. &lt;2016-05-20, Fri&gt;
 o bug fixed in collapse &lt;2016-05-20, Fri&gt;
   + if user collapse a node that is an offspring of a collapsed node, print warning msg and return the tree directly
 o use byte compiler &lt;2016-05-18, Wed&gt;
 o change any(is.na()) to anyNA() which is more efficient &lt;2016-05-18, Wed&gt;
 o https://github.com/Bioconductor-mirror/ggtree/commit/559548c66b51253e8ccb983d353385838a81f106

CHANGES IN VERSION 1.5.3
------------------------
 o add examples in vignettes &lt;2016-05-13, Fri&gt;
   + add fan layout example in treeVisualization vignette
   + add open_tree and rotate_tree example in treeManipulation vignette
 o add angle in ggtree function, fan layout supported &lt;2016-05-12, Thu&gt;
 o rotate_tree and open_tree function &lt;2016-05-12, Thu&gt;
 o support reading BEAST MCC trees (multiple trees in one file) via the read.beast function &lt;2016-05-12, Thu&gt;
 o https://github.com/Bioconductor-mirror/ggtree/commit/51eec4721595c274c24dc4df2f1fdf40700cb1a5

CHANGES IN VERSION 1.5.2
------------------------
 o add multiplot in ggtreeUtilities vignette &lt;2016-05-12, Thu&gt;
 o add example of integrate user's data using phylo4d in treeAnnotation vignette &lt;2016-05-11, Wed&gt;
 o add extend, extendto parameter in geom_hilight &lt;2016-05-10, Tue&gt;
 o geom_hilight now supports hilight tips &lt;2016-05-10, Tue&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/53
 o more accurate ylim &amp; angle for circular layout &lt;2016-05-10, Tue&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/40
 o supports phylo4d object &lt;2016-05-10, Tue&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/47

CHANGES IN VERSION 1.5.1
------------------------
 o update vignettes &lt;2016-05-10, Tue&gt;
   + add geom_range example in treeImport
   + add geom_strip and geom_taxalink example in treeAnnotation
   + add ggtreeUtilities vignette
 o gheatmap now works with data.frame of only one column &lt;2016-05-09, Mon&gt;
   + contributed by Justin Silverman &lt;jsilve24@gmail.com&gt;
   + https://github.com/GuangchuangYu/ggtree/pull/57
 o geom_strip for associated taxa &lt;2016-05-09, Mon&gt;
   + https://github.com/GuangchuangYu/ggtree/issues/52
</code></pre>

<h2 id="gosemsim">GOSemSim</h2>

<p>After 7.5 years in Bioconductor, I bump up the version of <code>GOSemSim</code> to 2.0 for optimization of <code>Wang</code> method and extend <code>GOSemSim</code> to all species that have <code>OrgDb</code> object which can be queried by <code>AnnotationHub</code>.</p>

<pre><code>CHANGES IN VERSION 1.99.4
------------------------
 o fixed NOTE in R check &lt;2016-08-12, Fri&gt;
 o add unit test using testthat &lt;2016-08-11, Thu&gt;

CHANGES IN VERSION 1.99.3
------------------------
 o changes to satisfy meshes package &lt;2016-08-05, Fri&gt;

CHANGES IN VERSION 1.99.2
------------------------
 o fixed Rcpp issue &lt;2016-07-19, Tue&gt;
   + https://github.com/GuangchuangYu/GOSemSim/issues/6

CHANGES IN VERSION 1.99.1
------------------------
 o update vignette &lt;2016-07-14, Thu&gt;

CHANGES IN VERSION 1.99.0
------------------------
 o support all organisms that have OrgDb object &lt;2016-07-05, Tue&gt;
 o optimize Wang method &lt;2016-07-04, Mon&gt;
</code></pre>

<h2 id="meshes">meshes</h2>

<blockquote>
<p>MeSH (Medical Subject Headings) is the NLM controlled vocabulary
         used to manually index articles for MEDLINE/PubMed. MeSH terms were
         associated by Entrez Gene ID by three methods, gendoo, gene2pubmed and RBBH.
         This association is fundamental for enrichment and semantic analyses.
         meshes supports enrichment analysis (over-representation and gene set
         enrichment analysis) of gene list or whole expression profile.
         The semantic comparisons of MeSH terms provide quantitative ways to compute
         similarities between genes and gene groups. meshes implemented five methods
         proposed by Resnik, Schlicker, Jiang, Lin and Wang respectively and supports
         more than 70 species.</p>
</blockquote>

<p><code>meshes</code> was newly introduced in this BioC release branch. It supports semantic similarity measurement and enrichment analysis of more than 70 species.</p>

<pre><code>CHANGES IN VERSION 0.99.7
------------------------
 o Depends DOSE &lt;2016-09-29, Thu&gt;
 
CHANGES IN VERSION 0.99.6
------------------------
 o update docs &lt;2016-09-19, Mon&gt;
 
CHANGES IN VERSION 0.99.3
------------------------
 o remove non-ASCII character in vignettes/meshes.bib &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 0.99.2
------------------------
 o all exported functions have runnable examples required by BiocCheck &lt;2016-08-16, Tue&gt;

CHANGES IN VERSION 0.99.1
------------------------
 o R check passed &lt;2016-08-16, Tue&gt;
 
CHANGES IN VERSION 0.99.0
------------------------
 o add vignette &lt;2016-08-11, Thu&gt;
 o enrichMeSH and gseMeSH &lt;2016-08-11, Thu&gt;
   + move from clusterProfiler
 
CHANGES IN VERSION 0.0.1
------------------------
 o geneSim &lt;2016-08-10, Wed&gt;
 o meshSim &lt;2016-08-05, Fri&gt;
</code></pre>

<h2 id="reactomepa">ReactomePA</h2>

<p>For <code>ReactomePA</code> itself, the change is minor. I only add unit test to this package.</p>

<p>But there are some changes of <code>DOSE</code> also affect this package as it internally use <code>DOSE</code> for enrichment analysis, including leading edge analysis for GSEA and other minor changes.</p>

<pre><code>CHANGES IN VERSION 1.17.4
------------------------
 o unit test &lt;2016-08-15, Mon&gt;
 
CHANGES IN VERSION 1.17.3
------------------------
 o move getDb from GOSemSim (no longer need this function) to ReactomePA and remove dependency of GOSemSim &lt;2016-07-06, Wed&gt;
 o 'by' parameter for GSEA analysis &lt;2016-07-04, Mon&gt;

CHANGES IN VERSION 1.17.2
------------------------
 o use byte compiler &lt;2016-05-18, Wed&gt;
 o https://github.com/Bioconductor-mirror/ReactomePA/commit/6ce32c8e03e1b252662a07901cce022fab038086

CHANGES IN VERSION 1.17.1
------------------------
 o https://github.com/Bioconductor-mirror/ReactomePA/commit/5d150f5fe545cfa3983872bf5485af1b9ba3129d
</code></pre>

<h2 id="citation">Citation</h2>

<p><strong><em>Yu G</em></strong>, Wang LG and He QY<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/31/14/2382" target="_blank">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(14):2382-2383.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="3781087"></div></p>

<p><strong><em>Yu G</em></strong>, Wang L, Han Y and He Q<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118" target="_blank">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="681089"></div></p>

<p><strong><em>G Yu</em></strong>, LG Wang, GR Yan, QY He. <a href="http://bioinformatics.oxfordjournals.org/content/31/4/608" target="_blank">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(4):608-609.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="2788597"></div></p>

<p><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. <a href="http://dx.doi.org/10.1111/2041-210X.12628" target="_blank"><code>doi:10.1111/2041-210X.12628</code></a>.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1111/2041-210X.12628" class="altmetric-embed"></div></p>

<p><strong>Yu G</strong><sup>†</sup>, Li F<sup>†</sup>, Qin Y, Bo X<sup>*</sup>, Wu Y and Wang S<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="100979"></div></p>

<p><strong><em>G Yu</em></strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E#!divAbstract" target="_blank">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong> 2016, 12(2):477-479.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="4796667"></div></p>
    </div>
  </div>

  <center><span class="zaln"><a href="https://guangchuangyu.github.io/2017/11/bmc-journal-protects-paper-of-plagiarism/" class="zb">Plea to against BMC</a></span></center>


</article>

<div class="container">
  <nav>
  <ul class="pager">
    
    <li class="previous"><a href="https://guangchuangyu.github.io/2016/10/reproducible-logo-generated-by-ggtree/"><span
      aria-hidden="true">&larr;</span> reproducible logo generated by ggtree</a></li>
    

    
    <li class="next"><a href="https://guangchuangyu.github.io/2016/10/facet_plot-a-general-solution-to-associate-data-with-phylogenetic-tree/">facet_plot: a general solution to associate data with phylogenetic tree <span
      aria-hidden="true">&rarr;</span></a></li>
    
  </ul>
</nav>

</div>

<div class="article-container">
  
<section id="comments">
  <div id="disqus_thread">
    <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "gcyu" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
  </div>
</section>


</div>

<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

