<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Guangchuang Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="Guangchuang Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
 <div id="content">
  <div class="blog-index">
    
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a>
        </h1>
        <p class="meta">Aug 31, 2015 - 1 minute read -  <a href="http://guangchuangyu.github.io/2015/08/subview/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/visualization/">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
geom_bar(stat=&quot;identity&quot;, width=1) + 
coord_polar(theta=&quot;y&quot;) + theme_tree() + 
xlab(NULL) + ylab(NULL) + 
theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
print(p)
}
</code></pre>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/08/subview/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a>
        </h1>
        <p class="meta">Aug 21, 2015 - 2 minute read -  <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics/">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/genomics/">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" /></p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a>
        </h1>
        <p class="meta">Aug 13, 2015 - 2 minute read -  <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" /></p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a>
        </h1>
        <p class="meta">Aug 7, 2015 - 1 minute read -  <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics/">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/">PeakAnalyzer</a>.</p>

<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/">ggtree annotate phylogenetic tree with local images</a>
        </h1>
        <p class="meta">Aug 1, 2015 - 2 minute read -  <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/annotation_image/Screenshot%202015-08-01%2020.53.50.png" alt="" /></p>

<p>In <a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>,
we provide a function <em>annotation_image</em> for annotating phylogenetic
tree with images.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    
    






<div class="pagination">

    
        
        
        
        <a href="/" aria-label="First" class="label-pagination"><i class="fa fa-angle-double-left fa-lg"></i></a>
    

    
    
        <a href="/page/4/" aria-label="Previous" class="label-pagination"><i class="fa fa-angle-left fa-lg"></i></a>
    

    
        <a href="/" class="label-pagination">1</a>
    
        <a href="/page/2/" class="label-pagination">2</a>
    
        <a href="/page/3/" class="label-pagination">3</a>
    
        <a href="/page/4/" class="label-pagination">4</a>
    
        <a href="/page/5/" class="label-pagination">5</a>
    
        <a href="/page/6/" class="label-pagination">6</a>
    
        <a href="/page/7/" class="label-pagination">7</a>
    
        <a href="/page/8/" class="label-pagination">8</a>
    
        <a href="/page/9/" class="label-pagination">9</a>
    
        <a href="/page/10/" class="label-pagination">10</a>
    
        <a href="/page/11/" class="label-pagination">11</a>
    
        <a href="/page/12/" class="label-pagination">12</a>
    
        <a href="/page/13/" class="label-pagination">13</a>
    
        <a href="/page/14/" class="label-pagination">14</a>
    
        <a href="/page/15/" class="label-pagination">15</a>
    
        <a href="/page/16/" class="label-pagination">16</a>
    
        <a href="/page/17/" class="label-pagination">17</a>
    

    
    
        <a href="/page/6/" aria-label="Next" class="label-pagination"><i class="fa fa-angle-right fa-lg"></i></a>
     
    
    
    
        <a href="/page/17/" aria-label="Last"><i class="fa fa-angle-double-right fa-lg"></i></a>
    

</div>


   

  </div>

  


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/%E4%BD%BF%E7%94%A8ggtree%E5%AE%9E%E7%8E%B0%E8%BF%9B%E5%8C%96%E6%A0%91%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E5%92%8C%E6%B3%A8%E9%87%8A/">使用ggtree实现进化树的可视化和注释</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/covplot-supports-grangeslist/">covplot supports GRangesList</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
    </ul>
  </section>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
  
</aside>
    
 </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>
   