<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/douban/">douban</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2016 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">multiple annotation in ChIPseeker</a>
      </h1>
      <span class="post-date">Oct 1, 2014 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      Nearest gene annotation Almost all annotation software calculate the distance of a peak to the nearest TSS and assign the peak to that gene. This can be misleading, as binding sites might be located between two start sites of different genes or hit different genes which have the same TSS location in the genome. The function annotatePeak provides option to assign genes with a max distance cutoff and all genes within this distance were reported for each peak.
      <a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">why clusterProfiler fails</a>
      </h1>
      <span class="post-date">Aug 7, 2014 &middot; 4 minute read &middot; <a href="http://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>
      </span>


      
      Recently, there are some comments said that sometimes clusterProfiler failed in KEGG enrichment analysis. kaji331 compared cluserProfiler with GeneAnswers and found that clusterProfiler gives larger p values. The result forces me to test it. it eventually came out that he passed the input gene as numeric vector, which was supposed to be character and he used an old version of clusterProfiler which didn&rsquo;t use as.character to coerce the input require(GeneAnswers) data('humanGeneInput') y &lt;- geneAnswersBuilder(humanGeneInput, 'org.Hs.eg.db', categoryType='KEGG', testType='hyperG', pvalueT=0.1, geneExpressionProfile=humanExpr, verbose=FALSE) yy &lt;- y@enrichmentInfo require(clusterProfiler) x &lt;- enrichKEGG(humanGeneInput$GeneID, pvalueCutoff=0.2, qvalueCutoff=0.2, minGSSize=1) xx &lt;- summary(x) id &lt;- sub(&quot;hsa&quot;, &quot;&quot;, xx$ID) idx &lt;- id %in% rownames(yy) p.clusterProfiler &lt;- xx$pvalue[idx] p.GeneAnswers &lt;- yy[id[idx],]$&quot;p value&quot; &gt; cor(p.clusterProfiler, p.GeneAnswers) [1] 0.9996165 &gt; p.clusterProfiler - p.GeneAnswers [1] 1.029789e-04 -3.588252e-05 -4.623010e-05 1.079117e-04 -1.075746e-04 [6] -1.077398e-04 -3.774637e-04 -2.849278e-04 -4.197993e-04 7.588155e-04 [11] -3.702141e-04 2.314721e-03 -5.695641e-04 -5.940830e-04 -4.923697e-04 [16] -5.560738e-04 -5.884079e-04 2.011138e-03 Here, I used the dataset, humanGeneInput, provided by GeneAnswers.
      <a href="http://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/08/enrichment-map/">enrichment map</a>
      </h1>
      <span class="post-date">Aug 3, 2014 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2014/08/enrichment-map/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a>
      </span>


      
      In PLOB&rsquo;s QQ group, someone asked how to change the color of enrichment map in Cytoscape. I am very curious how enrichment map can helps to interpret enrichment results. It took me 2 hours to implement it using R and I am surprised that the enrichment map is better than anticipated. Now in the development version of clusterProfiler, DOSE, and ReactomePA, you can use enrichmap function to generate the enrichment map of enrichment results obtained by hypergeometric test or gene set enrichment analysis.
      
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/06/use-linux-desktop-again/">use linux desktop again</a>
      </h1>
      <span class="post-date">Jun 1, 2014 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2014/06/use-linux-desktop-again/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/linux">Linux</a>
      </span>


      
      上个月写了《使用ggplot画图》一文，图片太多，一下子把博客的12G流量给用光了，月中博客就被停了，小伙伴们太给力了，blog不能访问，真是件不爽的事情。 言归正传，从2002年使用redhat 7.1开始，就没中止过使用linux，为什么要说again呢，因为在过去几年的工作时间里，使用的是OS X和win7做为桌面的，linux只在服务器端使用。自从用了OS X之后，就不想再用linux当桌面了，OS X绝对是节省生命的系统，本身是unix-like系统，对各种桌面软件支持又好，特别是中华大地，linux还是小众，用来当桌面实在是各种坑。 不过现在又回到学生时代了，所以还得继续学生屌丝的折腾，本科时从redhat,debian到gentoo，硕士时实验室的机器用ubuntu，而笔记本用NetBSD。这么多年的使用，该有的坑都填了，但是时代在发展，新坑还是不断有的，使用百度云来同步化文献库，就是最近的新坑，装上linux之后，发现度娘没有客户端，首先想到的当然是wine，wine出来的偶尔能行，时常崩溃，跟它死磕我也会崩溃，github上寻找开源实现，各种功能缺失和限制，那就只能虚拟机了，把home目录做为共享盘，挂到虚拟机中，但是度娘一直都是占100% CPU，然后数据无法同步，只有两个原因，不是硬盘就是网络，感觉好像我一个分区800G太大了，但在虚拟中对硬盘读写正常，网络因为使用NAT，虚拟成内网，有可能是这个原因，于是换成bridge，把虚拟机当成现实网络中的另一台主机，但度娘依然只占CPU，不干活。那只能还是硬盘问题，最后问题解决了，通过在linux中提供samba服务，然后在虚拟机中通过网上邻居把共享目录映射成网络驱动器，这时候度娘终于肯干活了。都不知道是virtualbox的驱动有问题，还是度娘太挑，这太TM坑了。 不想自己太geek，假装我在用windows 7，不过我还是喜欢当年的windows 2000的主题，可惜qvwm已经很久不更新了，现在想编译成功已经非常困难了。 假装了一星期，我就觉得用着很不爽，屏幕太小，又老用鼠标，手臂不舒服，于是想起了当年的wmii，这次我装的是wmii的增强版i3-wm，有了tab这些元素，用着更舒服一点，于是窗口都全屏了，这样小屏幕也还凑合了。最主要的是提高效率，减少鼠标操作，我现在把鼠标放在左上角，因为使用频率非常少，偶尔要用，就用左手搞一下。告别鼠标手，这酸爽，谁用谁知道！有了OS X之后，linux桌面依然有吸引我的地方，最主要的就是tiling windows，另外fcitx的五笔是我最喜欢的输入法。 小屏毕竟还是不爽，准备把宿舍的显示器搬去实验室用了，囧rz&hellip; 搬来显示器之后，无比爽。
      
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/05/use-ggplot2/">Use ggplot2</a>
      </h1>
      <span class="post-date">May 11, 2014 &middot; 5 minute read &middot; <a href="http://guangchuangyu.github.io/2014/05/use-ggplot2/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggplot2">ggplot2</a>
      </span>


      
      Why use ggplot2 ggplot2是我见过最human friendly的画图软件，这得益于Leland Wilkinson在他的著作《The Grammar of Graphics》中提出了一套图形语法，把图形元素抽象成可以自由组合的成分，Hadley Wickham把这套想法在R中实现。 为什么要学习ggplot2，可以参考ggplot2: 数据分析与图形艺术的序言（btw: 在序言的最后，我被致谢了）。 Hadley Wickham也给出一堆理由让我们说服自己，我想再补充一点，Hadley Wickham是学医出身的，做为学生物出身的人有什么理由不支持呢:) ggplot2基本要素 数据（Data）和映射（Mapping） 几何对象（Geometric） 标尺（Scale） 统计变换（Statistics） 坐标系统（Coordinante） 图层（Layer） 分面（Facet） 主题（Theme） 这里将从这些基本要素对ggplot2进行介绍。 数据（Data）和映射（Mapping) 下面以一份钻石的数据为例，这份数据非常大，随机取一个子集来画图。 require(ggplot2) data(diamonds) set.seed(42) small &lt;- diamonds[sample(nrow(diamonds), 1000), ] head(small) ## carat cut color clarity depth table price x y z ## 49345 0.71 Very Good H SI1 62.5 60 2096 5.68 5.75 3.57 ## 50545 0.79 Premium H SI1 61.8 59 2275 5.97 5.91 3.67 ## 15434 1.03 Ideal F SI1 62.4 57 6178 6.48 6.44 4.03 ## 44792 0.50 Ideal E VS2 62.2 54 1624 5.08 5.11 3.17 ## 34614 0.27 Ideal E VS1 61.6 56 470 4.14 4.17 2.56 ## 27998 0.30 Premium E VS2 61.7 58 658 4.32 4.34 2.67 summary(small) ## carat cut color clarity depth ## Min.
      <a href="http://guangchuangyu.github.io/2014/05/use-ggplot2/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/05/%E6%9B%B4%E6%94%B9%E7%99%BE%E5%BA%A6%E4%BA%91%E5%90%8C%E6%AD%A5%E7%9B%98%E8%B7%AF%E5%BE%84/">更改百度云同步盘路径</a>
      </h1>
      <span class="post-date">May 5, 2014 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2014/05/%E6%9B%B4%E6%94%B9%E7%99%BE%E5%BA%A6%E4%BA%91%E5%90%8C%E6%AD%A5%E7%9B%98%E8%B7%AF%E5%BE%84/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/mac-os">Mac OS</a><a class="label" href="http://guangchuangyu.github.io/categories/software">Software</a>
      </span>


      
      从2007年写了第一篇文章之后，我发现管理文献真不是人干的，一直以来使用Zotero来管理文献，自从有了dropbox之后，就想把文献库放在dropbox上，一来有个云备份，不怕硬盘坏，二来嘛，实验室电脑和个人电脑可以实时同步化，无奈dropbox空间太小，而我的zotero早已超过10G，因为电子书也放在里面，只能做罢。 度娘出手还是很大方的，我现在的网盘已经有3T，包括之前在推广阶段用1元买的1T。 于是我要把zotero放在度娘的网盘里，有可能是因为我用英文版，firefox不认中文路径。 第一反应就是给个英文路径的软链接，OS X自带的Terminal和iTerm都不支持我输入中文，只能在Emacs里跑eshell来搞。事实证明软链接是不行的，会被还原回实际路径，依然是中文。 那么就用硬链接，但是系统不干，真是邪门，搜索一下发现这是OS X的问题，在stackoverflow上还有人提供了一段C代码来做硬链接，不过我编译后运行依然是失败，无法创建硬链接。 于是只能土办法了，软件的路径并不是死的，那么总得有个配置文件来定位，通过修改配置文件总应该可以的吧。于是在下面的路径下，找到了macosnetdisk.plist user_workpath /Volumes/DATA/Users/ygc/百度云同步盘 把百度云同步盘换成个英文文件夹，整个世界清静了，firefox终于找到了它的好基友zotero。 TIPS：有了度娘网盘，再也不用担心下载问题了，找个BT种子，让度娘下载，速度极快，因为通常的东西度娘的网盘上是有的，它做的不过是链接一份给你而已，越多的人保留同一份东西，度娘就越省空间。度娘同步到电脑上速度也是极快的，秒杀其它下载工具。
      
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a>
      </h1>
      <span class="post-date">Apr 30, 2014 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseq">ChIPseq</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      After two weeks developed, I have added/updated some plot functions in ChIPseeker (version &gt;=1.0.1). ChIP peaks over Chromosomes &gt; files=getSampleFiles() &gt; peak=readPeakFile(files[[4]]) &gt; peak GRanges object with 1331 ranges and 2 metadata columns: seqnames ranges strand | V4 V5 | [1] chr1 [ 815092, 817883] * | MACS_peak_1 295.76 [2] chr1 [1243287, 1244338] * | MACS_peak_2 63.19 [3] chr1 [2979976, 2981228] * | MACS_peak_3 100.16 [4] chr1 [3566181, 3567876] * | MACS_peak_4 558.89 [5] chr1 [3816545, 3818111] * | MACS_peak_5 57.57 ...
      <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/">ChIPseeker for ChIP peak annotation</a>
      </h1>
      <span class="post-date">Apr 13, 2014 &middot; 4 minute read &middot; <a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/chipseq">ChIPseq</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      ChIPpeakAnno WAS the only R package for ChIP peak annotation. I used it for annotating peak in my recent study. I found it does not consider the strand information of genes. I reported the bug to the authors, but they are reluctant to change. So I decided to develop my own package, ChIPseeker, and it&rsquo;s now available in Bioconductor. &gt; require(TxDb.Hsapiens.UCSC.hg19.knownGene) &gt; txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene &gt; require(ChIPseeker) &gt; peakfile = getSampleFiles()[[4]] &gt; peakfile [1] &quot;/usr/local/Cellar/r/3.1.0/R.framework/Versions/3.1/Resources/library/ChIPseeker/extdata//GEO_sample_data/GSM1295076_CBX6_BF_ChipSeq_mergedReps_peaks.bed.gz&quot; &gt; peakAnno &lt;- annotatePeak(peakfile, tssRegion=c(-3000, 1000), TxDb=txdb, annoDb=&quot;org.Hs.eg.db&quot;) &gt;&gt; loading peak file...
      <a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/03/boxplot/">boxplot</a>
      </h1>
      <span class="post-date">Mar 4, 2014 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2014/03/boxplot/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a>
      </span>


      
      生物坑很多人画图只会直方图，统计只会T检验，在暨大见过太多的学生连T检验都不会，分不清SEM和SD的差别，也不清楚T检验那几个简单参数的含义。我写统计笔记也是因为不想重复性地跟学生讲解T检验。 Barplot和T test一样普遍而流行，barplot适合于表示计数数据和比例，显示比例也可以用pie plot，但直方图比饼图要好，因为人类的眼睛适合于比较高度，而不是弧度。 多半时候生物学数据并非简单的计数数据，对于测量数据，在展示数据分布时，很多人会使用他们熟悉的barplot，用高度来表示mean，然后再加上errorbar，这样展示数据，信息量是非常低的，使用boxplot能够提供更多的数据分布信息，能更好地展现数据，但可能很多人只会在excel里画barplot，Nature Methods 2013年的文章中有100个barplot图，而只有20个boxplot图，从这里就可以看出来，用boxplot的人远远没有barplot多，于是NPG怒了，写了两篇专栏文章Points of View: Bar charts and box plots和Points of Significance: Visualizing samples with box plots并且发表了一篇BoxPlotR: a web tool for generation of box plots方便大家画boxplot，如此简单的web tool能够发Nature Methods，实在是让人羡慕妒忌恨啊。 Boxplot从1970年由John Tukey提示，到现在已经40多年了，他有很多的变种，比如violin plot和beanplot，violin plot和beanplot是很像的，都是把box换成density curve，beanplot另外把数据点在中间用线条表示出来，这个很容易加上去，正如我在violin plot中把数据点加进去一样。同样可以把数据点加进boxplot中，还有很多手法来展示boxplot，比如可以用缺口(notch)形式来画boxplot，可以把中间的宽度表示为median值的置信空间。boxplot的宽度可以用样本量来归一化，如果sample size足够大的话，$mean\pm errorbar$ 不会超出box，可以很清晰把这个信息也展示在boxplot中。 如果数据量比较少，可以用scatterplot+errorbar的形式展示数据，正如我在Phosphoproteome profile of human lung cancer cell line A549中所画的图7。
      
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">Bug of R package ChIPpeakAnno</a>
      </h1>
      <span class="post-date">Jan 14, 2014 &middot; 5 minute read &middot; <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseq">ChIPseq</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/bioc">BioC</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      I used R package ChIPpeakAnno for annotating peaks, and found that it handle the DNA strand in the wrong way. Maybe the developers were from the computer science but not biology background. &gt; require(ChIPpeakAnno) &gt; packageVersion(&quot;ChIPpeakAnno&quot;) [1] '2.10.0' &gt; peak &lt;- RangedData(space=&quot;chr1&quot;, IRanges(24736757, 24737528)) &gt; data(TSS.human.GRCh37) &gt; ap &lt;- annotatePeakInBatch(peak, Annotation=TSS.human.GRCh37) &gt; ap RangedData with 1 row and 9 value columns across 1 space space ranges | peak strand | 1 ENSG00000001461 1 [24736757, 24737528] | 1 + feature start_position end_position insideFeature 1 ENSG00000001461 ENSG00000001461 24742284 24799466 upstream distancetoFeature shortestDistance fromOverlappingOrNearest 1 ENSG00000001461 -5527 4756 NearestStart In this example, I defined a peak ranging from chr1:24736757 to chr1:24737528 and annotated the peak using ChIPpeakAnno package.
      <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">Read On &rarr;</a>
      

    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        >
        <a href="/page/4/" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        ><a href="/">1</a></li>
        
        <li
        ><a href="/page/2/">2</a></li>
        
        <li
        ><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        class="active"><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        ><a href="/page/9/">9</a></li>
        
        <li
        >
        <a href="/page/6/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/9/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    

  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

