<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>BMC favors source code plagiarism</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            BMC favors source code plagiarism 
          </h1>
          <p class="meta">May 27, 2015 - 10 minute read -  <a href="http://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/research">Research</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          <p>I found source code plagiarism a year ago and reported this case to <a href="http://www.biomedcentral.com/bmcsystbiol">BMC Systems Biology</a>:</p>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/0f22f6ad87427adb59941160130b016ee1e41004/2015/plagiarism/Screenshot%202015-05-27%2019.56.58.png" alt="" /></p>

<p>I listed source code of many functions that are exactly copied from
<a href="http://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
with only function name changed in my email. The detail of source code
plagiarism can also be found at <strong><a href="http://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/">Proper use of
GOSemSim</a></strong>.</p>

<p>I got reply from the Editor 3 days after (see
<a href="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/0f22f6ad87427adb59941160130b016ee1e41004/2015/plagiarism/Screenshot%202015-05-27%2019.58.16.png">screenshot</a>).</p>

<blockquote>
<p>MS: 1069569029107167
Research
ppiPre: predicting protein-protein interactions by combining heterogeneous features
BMC Systems Biology 2013, 7:S8</p>

<p>Dear Dr Yu,</p>

<p>Thank you for your message and bringing this to our attention. Firstly I would like to assure you that we do take cases of plagiarism very seriously. We are members of the Committee on Publication Ethics (<a href="http://publicationethics.org">http://publicationethics.org</a>) follow their guidelines for circumstances such as this.</p>

<p>As a first step we will be investigating the overlap between this publication in BMC Systems Biology and your previously published work, to establish the extent of the duplication.</p>

<p>Could I just check, have you compared all the code in the ppiPRE article for similarity with your own code? That is, is there any possibility that more of your code has been duplicated without attribution? We will of course have to examine the code for unattributed use of other authors code in addition to your own</p>

<p>Please do let us know if there are more relevant details you are aware of. We will be in contact again in due course as our investigations proceed, but do not hesitate to contact me in the meantime if you would like an update</p>

<p>Best wishes,
Tim</p>
</blockquote>

<p>The editor assured me that they &lsquo;<strong>do take cases of plagiarism very
seriously</strong>&rsquo; and will investigate the overlap. I was waiting for their
update. But 5 months passed, I never receive any reply from them even I
sent several emails to ask for updates. I have no idea how long their
&lsquo;<strong>due course</strong>&rsquo; is. As I did not receive any reply, I wrote a blog post
(see <a href="http://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/">Proper use of
GOSemSim</a>) to bring
this to the R community and CRAN did remove the ppiPre package at that
time. I told the Editor that this package had been removed from CRAN and
again asked for update. This time, he did reply me (see
<a href="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot%202015-05-27%2020.11.54.png">screenshot</a>):</p>

<blockquote>
<p>MS: 1069569029107167
Research
ppiPre: predicting protein-protein interactions by combining heterogeneous features
BMC Systems Biology 2013, 7:S8</p>

<p>Dear Guangchuang,</p>

<p>Thank you for your message and my apologies for the delay in this investigation. We had been trying to identify if any other code had also been plagiarised in the publication, although we did not find any evidence of this. Thank you also for drawing our attention to the removal of ppiPre from CRAN. We will be in touch with the article authors and will let you know when there is a further update.</p>

<p>Regards,
Tim</p>
</blockquote>

<p>As we can find in the editor&rsquo;s email, &lsquo;<strong>We will be in touch with the
article authors</strong>&rsquo;, they never contact the authors after 5 months passed
from my report. In the following 2 months, I never receive any reply
again. I was angry and sent an email to several Editors of BMC to
complain that they are ruinning the reputation of BMC. Then again as
expected, I got email (see
<a href="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot%202015-05-27%2020.33.30.png">screenshot</a>):</p>

<blockquote>
<p>Dear Guangchuang,</p>

<p>We are continuing to pursue this case. As you will understand, as well as the clear evidence of plagiarism of your code, we are concerned about possible unacknowledged reuse of other code as well. We have been trying to contact the authors for a full explanation, as this will affect how we can address the issue. Until very recently we had not been successful in contacting them, although they have very recently got back to us  and have said that they are considering their response, so we expect an update soon. From our perspective it is highly preferable for the authors to come clean about the issues and for any correction or retraction notice to come from them rather than from the editors, and for this reason it is often a drawn out process.</p>

<p>Thank you for your patience, and I can again assure you that this will be addressed formally in due course.</p>

<p>Yours sincerely,
Tim</p>
</blockquote>

<p>I am very happy to see this as the editor told me they have contacted
the authors and will have an update soon. I wait another 3 months
without any update as usual. I asked for update again and this time they
did give me an update (see
<a href="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot%202015-04-23%2021.31.55.png">screenshot</a>):</p>

<blockquote>
<p>Dear Guangchuang,</p>

<p>Looking back through my emails, I did receive a response back from the authors previously. They in turn said that the email address I had originally been trying to contact them with earlier last year was no longer in use, so they only received my emails when I found an additional alternative email address for the authors in my more recent emails.</p>

<p>Their explanation for the use of your code was that as the similarity measures were not their main focus of their study they had intended to either implement existing methods themselves or as in the case of GoSemSim, import the packages. However they had some problems with this and instead utilised the source code directly. Their rationale was that as the code was GPL licenced that this was acceptable, without realising that it was also required to cite the original source.</p>

<p>They say they have now created an updated version which they say has the following changes:</p>

<p>(1) ppiPre imports GOSemSim.
(2) ppiPre calls function geneSim() exported by GOSemSim to calculate Wang’s measure, instead of deriving code from GOSemSim directly.
(3) Several internal functions of ppiPre (TCSSGetAncestors, GetOntology, GetGOParents, GetLatestCommonAncestor, TCSSCompute_ICA) are derived from GOSemSim since GOSemSim don’t export them. In source code (.R) and manuals (.RD), acknowledgement has been added including the information of author and publication of GOSemSim.
(4) The author of GOSemSim (Guangchuang Yu) has been added in the ‘Author’ filed as contributor, as required by CRAN Repository Policy.
(5) All the information content data in \data directory of ppiPre have been removed. ppiPre directly obtained data from GOSemSim.</p>

<p>This version has been available in SourceForge (<a href="https://sourceforge.net/p/ppipre/">https://sourceforge.net/p/ppipre/</a>). And we are submitting ppiPre to CRAN. We will notify you after ppiPre is available on CRAN.</p>

<p>They also apologise for failing to cite you correctly and have asked if they may submit a correction to their article to rectify this. This would be within our policy if this sounds acceptable to you. Please let me know if you would need more details or if you need to get in touch with the authors</p>

<p>Best wishes,
Tim</p>
</blockquote>

<p>After the changes, ppiPre is back to CRAN. Now in CRAN, what we can find
is that ppiPre use source code from GOSemSim properly with
acknowledgement. They removed all previous source tarball so that their
source code plagiarism was gone. Fortunately, github recorded all the
<a href="https://github.com/cran/ppiPre/commit/c267700ca09a599dab77326959a0aa6a2980ceed">changes</a>.
The editor delay the processing to give the authors enough time to
remove their bad records. I think CRAN shouldn&rsquo;t accept re-submitted
package if it was removed for plagiarism issue. In the email, the editor
not only conveyed explanation from the author, but also have expressed
their opinion: &lsquo;<strong>They also apologise for failing to cite you correctly
and have asked if they may submit a correction to their article to
rectify this. This would be within our policy if this sounds acceptable
to you</strong>&rsquo;. This is why I think BMC do favor source code plagiarism.
Their judgement/opinion is not fair! I can&rsquo;t accept their explanation
and replied to the editor (see
<a href="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot%202015-04-23%2021.45.33.png">screenshot</a>):</p>

<blockquote>
<p>Dear Prof. Sands,</p>

<p>Thanks for your email. That makes it more clear.</p>

<p>I can&rsquo;t accept their explanation.</p>

<p>The ppiPre is a die package, they didn&rsquo;t update any source code or IC data after publication, with the fact that annotation packages for calculating IC data have updated several times. The recent changes were forced by my report of plagiarism.</p>

<p>The recent update follows the requirement of GPL license which requires the derivative work also GPL licensed and acknowledge original works. That&rsquo;s good, but remember it was forced to happen after my report of source code plagiarism.</p>

<p>In the version they published the paper, it is indeed source codes copied without any acknowledgement and the proportion is very large (about <sup>2</sup>&frasl;<sub>3</sub> of ppiPre was copied from GOSemSim).</p>

<p>In their explanation, the author claimed that similarity measures were not their main focus. Can I ask what&rsquo;s their main focus? SVM for protein-pretein interaction prediction as indicated in the title of the paper and in the package name? The SVM in that package is less than 80 lines, <a href="https://github.com/cran/ppiPre/blob/master/R/SVMPredict.r">https://github.com/cran/ppiPre/blob/master/R/SVMPredict.r</a> with more than half in calling similarity measures, and several empty lines, paramter checking and preparing input and output. They did nothing in svm; they just call e1071::svm in line 28 and and e1071::predict in line 16 with default parameters (users even don&rsquo;t have chance to change kernel function in their package). If that&rsquo;s their main focus, what&rsquo;s their contribution of their paper? In the Method session, their only use a small paragraph, Prediction framework, that only mention they used svm for prediction but with many paragraphs emphasizing they have implemented many GO semantic similarity measures. They call svm implemented in e1071 package to train features calculated by source codes from GOSemSim. How do they explain this?  I do have no idea of their so call main focus.</p>

<p>They also mentioned that they had problems in importing GOSemSim. This is also un-acceptable. As they copied most of the functions without any change, it is easy to import these functions using GOSemSim:::un-exported_function if the function is not exported. If they don&rsquo;t know this feature and did not google search how to do it. They should contact me for help or ask for my permission of re-using my codes. But they didn&rsquo;t. They may believe they can use my source code freely as it is GPL licensed, they should know that open source license also have requirement of acknowledgement. Even that&rsquo;s due to their ignorance, why they only change the function names to pretend that&rsquo;s their own functions? If these are not their main focus, why more than half of the text in the paper explaining the calculation of these algorithms and claim they were implemented in their package?</p>

<p>If they are not intended to do this, they will have acknowledgement in their source code, they will have citation in their paper and they will not just change function name that was copied without other changes. They intended to do so to cheat editor, reviewers and readers.</p>

<blockquote>
<p>(2) ppiPre calls function geneSim() exported by GOSemSim to calculate Wang’s measure, instead of deriving code from GOSemSim directly.</p>
</blockquote>

<p>Besides, most of the source code is not &lsquo;derived&rsquo; but copied from GOSemSim with only function name changed which can still be detected in the updated version, <a href="https://github.com/cran/ppiPre/blob/master/R/GOKEGGSims.r">https://github.com/cran/ppiPre/blob/master/R/GOKEGGSims.r</a>.</p>

<p>Best Regards,
Guangchuang</p>
</blockquote>

<p><strong>AND again, the editor just ignore my email!</strong> 1 year passed and the
case is very clear that they copied my source code and intended to
pretend that&rsquo;s their original work and use it to publish a paper. That&rsquo;s
definitely plagiarism. The editor replied me very quickly at the
beginning and assured me that BMC do take cases like this seriously. I
don&rsquo;t know what happened after they contacted the authors. The editor
always ignore my email except I put some pressure to them, they delay
the process and try to persuade me that this is due to the author&rsquo;s
ignorance of open source license and an apology from the author is
enough to solve this issue. If you see this post, you are welcome to
comment and please help share it to social media. This will helps to
give BMC some pressure and I will contact COPE if they won&rsquo;t addressing
this.</p>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>May 27, 2015</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/bmc">BMC</a>  <a class="category" href="http://guangchuangyu.github.io/tags/plagiarism">plagiarism</a>  <a class="category" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/" title="use clusterProfiler as an universal enrichment analysis tool">use clusterProfiler as an universal enrichment analysis tool</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2015/06/bmc-favor-plagiarism/" title="BMC Favor Plagiarism">BMC Favor Plagiarism</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>



  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/google-drive--hku/">Google Drive @ HKU</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/%E4%BD%BF%E7%94%A8ggtree%E5%AE%9E%E7%8E%B0%E8%BF%9B%E5%8C%96%E6%A0%91%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E5%92%8C%E6%B3%A8%E9%87%8A/">使用ggtree实现进化树的可视化和注释</a>
      </li>
      
    </ul>
  </section>

  <div class="alignleft">  
<script type='text/javascript'>
 amzn_assoc_ad_type = 'banner';
 amzn_assoc_tracking_id = 'guanyu08-20';
 amzn_assoc_marketplace = 'amazon';
 amzn_assoc_region = 'US';
 amzn_assoc_placement = 'assoc_banner_placement_default';
 amzn_assoc_linkid = 'KM54UT7KCDBVUBUJ';
 amzn_assoc_campaigns = 'books';
 amzn_assoc_p = '12';
 amzn_assoc_banner_type = 'category';
 amzn_assoc_isresponsive = 'false';
 amzn_assoc_banner_id = '052F8JZYSHABHK7MK482';
 amzn_assoc_width = '290';
 amzn_assoc_height = '250';
</script>
<script src='//z-na.amazon-adsystem.com/widgets/q?ServiceVersion=20070822&Operation=GetScript&ID=OneJS&WS=1'></script>
</div>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


