

<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 3.1.1">
  <meta name="generator" content="Hugo 0.57.2" />
  <meta name="author" content="Guangchuang YU">

  
  
  
  
    
  
  <meta name="description" content="Professor of Bioinformatics">

  
  <link rel="alternate" hreflang="en-us" href="https://guangchuangyu.github.io/categories/r/">

  


  

  

  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha256-eSi1q2PG6J7g7ib17yAaWMcrr5GrtohYChqibrV7PBE=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.4.1/css/all.css" integrity="sha384-5sAR7xN1Nv6T6+dT2mhtzEpVJvfS3NScPQTrOxhwjIuvcA67KV2R5Jz6kr4abQsz" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" crossorigin="anonymous">
        
      
    

    

    

  

  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono">
  

  <link rel="stylesheet" href="/styles.css">
  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-77901140-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  
  <link rel="alternate" href="https://guangchuangyu.github.io/categories/r/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/categories/r/index.xml" type="application/rss+xml" title="Guangchuang YU">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/categories/r/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="Guangchuang YU">
  <meta property="og:url" content="https://guangchuangyu.github.io/categories/r/">
  <meta property="og:title" content="R | Guangchuang YU">
  <meta property="og:description" content="Professor of Bioinformatics">
  
  
    
  <meta property="og:image" content="https://guangchuangyu.github.io/img/icon-192.png">
  <meta property="og:locale" content="en-us">
  
  <meta property="og:updated_time" content="2018-09-10T11:00:08&#43;08:00">
  

  

  

  <title>R | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#TableOfContents" data-offset="71" >
  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header fixed-top">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Guangchuang YU</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#posts">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#teaching">
            
            <span>Teaching</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/resume">
            
            <span>Resume</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#contact">
            
            <span>Contact</span>
            
          </a>
        </li>

        
        

        

        
        
        

        <li class="nav-item">
          <a class="nav-link" href="/cn">
            
            <span>中文入口</span>
            
          </a>
        </li>

        
        

      

        

        

        
      </ul>

    </div>
  </div>
</nav>















  

  
  
  
    
  
<div class="universal-wrapper pt-3">
  <h1 itemprop="name">R</h1>

  

  
</div>



<div class="universal-wrapper">
  

  
  
  <div>
    <h2><a href="/2015/01/ggtree-in-bioconductor-3.1/">ggtree in Bioconductor 3.1</a></h2>
    <div class="article-style">
      
      I am very glad that ggtree is now available via Bioconductor. This is my 6th Bioconductor package. ggtree now supports parsing output files from BEAST, PAML, HYPHY, EPA and PPLACER and can annotate phylogenetic tree directly using plot methods.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2015/01/"></a></h2>
    <div class="article-style">
      
      I like running fortune every time when the terminal was started. A screenshot is shown below:

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/">viewing and annotating phylogenetic tree with ggtree</a></h2>
    <div class="article-style">
      
      When I need to annotate nucleotide substitutions in the phylogenetic tree, I found that all the software are designed to display the tree but not annotating it. Some of them may support annotating the tree with specific data such as bootstrap values, but they are restricted to a few supported data types. It is hard/impossible to inject user specific data.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/11/proper-use-of-gosemsim/">proper use of GOSemSim</a></h2>
    <div class="article-style">
      
      One day, I am looking for R packages that can analyze PPI and after searching, I found the ppiPre package in CRAN.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/10/sir-model-of-epidemics/">SIR Model of Epidemics</a></h2>
    <div class="article-style">
      
      The SIR model divides the population to three compartments: Susceptible, Infected and Recovered. If the disease dynamic fits the SIR model, then the flow of individuals is one direction from the susceptible group to infected group and then to the recovered group. All individuals are assumed to be identical in terms of their susceptibility to infection, infectiousness if infected and mixing behaviour associated with disease transmission.
We defined: $S_t$ = the number of susceptible individuals at time t
$ I_t $ = the number of infected individuals at time t
$R_t$ = the number of recovered individuals at time t
Suppose on average every infected individual will contact $\gamma$ person, and $\kappa$ percent of these $\gamma$ person will be infected. Then on average there are $\beta = \gamma \times \kappa$ person will be infected an infected individual.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/10/multiple-annotation-in-chipseeker/">multiple annotation in ChIPseeker</a></h2>
    <div class="article-style">
      
      Nearest gene annotation Almost all annotation software calculate the distance of a peak to the nearest TSS and assign the peak to that gene. This can be misleading, as binding sites might be located between two start sites of different genes or hit different genes which have the same TSS location in the genome.
The function annotatePeak provides option to assign genes with a max distance cutoff and all genes within this distance were reported for each peak.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/08/the-spread-of-new-mutations/">The spread of new mutations</a></h2>
    <div class="article-style">
      
       Genetic drift is the term used in population genetics to refer to the statistical drift over time of gene frequencies in a population due to random sampling effects in the formation of successive generations. In a narrower sense, genetic drift refers to the expected population dynamics of neutral alleles (those defined as having no positive or negative impact on reproductive fitness), which are predicted to eventually become fixed at zero or 100% frequency in the absence of other mechanisms affecting allele distributions.
 The most important keyword in the definition of genetic drift is random sampling effects. The figure belowed illustrates this idea. The surviving individuals do not necessarily have selection advantage. They are randomly selected.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/08/why-clusterprofiler-fails/">why clusterProfiler fails</a></h2>
    <div class="article-style">
      
      @kaji331 compared cluserProfiler with GeneAnswers and found that clusterProfiler gives larger p values.
It eventually came out that he passed the input gene as numeric vector, which was supposed to be character and he used an old version of clusterProfiler which didn&rsquo;t use as.character to coerce the input
But his comment forces me to test it.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/08/enrichment-map/">enrichment map</a></h2>
    <div class="article-style">
      
      In PLOB&rsquo;s QQ group, someone asked how to change the color of enrichment map in Cytoscape. I am very curious how enrichment map can helps to interpret enrichment results. It took me 2 hours to implement it using R and I am surprised that the enrichment map is better than anticipated.

      
    </div>
  </div>
  
  <div>
    <h2><a href="/2014/04/visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a></h2>
    <div class="article-style">
      
      After two weeks developed, I have added/updated some plot functions in ChIPseeker (version &gt;=1.0.1).
ChIP peaks over Chromosomes &gt; files=getSampleFiles() &gt; peak=readPeakFile(files[[4]]) &gt; peak GRanges object with 1331 ranges and 2 metadata columns: seqnames ranges strand | V4 V5 | [1] chr1 [ 815092, 817883] * | MACS_peak_1 295.76 [2] chr1 [1243287, 1244338] * | MACS_peak_2 63.19 [3] chr1 [2979976, 2981228] * | MACS_peak_3 100.16 [4] chr1 [3566181, 3567876] * | MACS_peak_4 558.89 [5] chr1 [3816545, 3818111] * | MACS_peak_5 57.57 ... ... ... ... ... ... ... [1327] chrX [135244782, 135245821] * | MACS_peak_1327 55.54 [1328] chrX [139171963, 139173506] * | MACS_peak_1328 270.19 [1329] chrX [139583953, 139586126] * | MACS_peak_1329 918.73 [1330] chrX [139592001, 139593238] * | MACS_peak_1330 210.88 [1331] chrY [ 13845133, 13845777] * | MACS_peak_1331 58.39 --- seqlengths: chr1 chr10 chr11 chr12 chr13 chr14 ... chr6 chr7 chr8 chr9 chrX chrY NA NA NA NA NA NA ... NA NA NA NA NA NA &gt; covplot(peak, weightCol=&quot;V5&quot;) 
      
    </div>
  </div>
  

  
<nav>
  <ul class="pagination justify-content-center">
    
    <li class="page-item"><a class="page-link" href="/categories/r/page/7/">&laquo;</a></li>
    
    
    <li class="page-item"><a class="page-link" href="/categories/r/page/9/">&raquo;</a></li>
    
  </ul>
</nav>


</div>
<div class="container">
  <footer class="site-footer">
  

  <p class="powered-by">
    &copy; 2016 Guangchuang YU &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

</div>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha512-+NqPlbbtM1QqiK8ZAo4Yrj2c4lNQoGv8P79DPtKzj++l5jnN39rHA/xsqn8zE9l0uSoxaCdrOgFs6yjyfbBxSg==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha256-VsEqElsCHSGmnmHXGQzvoWjWwoznFSZc6hs7ARLRacQ=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
        
      

      
      
    

    
    

    
    
    
    <script id="dsq-count-scr" src="//gcyu.disqus.com/count.js" async></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    

    
    

    
    

    
    
    
    
    
    
    
    
    <script src="/js/academic.min.c45052ed2c5a2349273f02aeb6adb37d.js"></script>

    

  </body>
</html>



