<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.21" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/2013/11/local-blast/">

    <title>local blast | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    <article class="article" itemscope itemtype="http://schema.org/Article">
        <h1 itemprop="name">local blast</h1>

        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2013-11-21 10:40:22 &#43;0800 CST" itemprop="datePublished">Thu, Nov 21, 2013</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="https://guangchuangyu.github.io/tags/blast">blast</a>
        
    </span>
    
    

    
<div class="a2a_kit a2a_kit_size_32 a2a_default_style">
<a class="a2a_dd" href="//www.addtoany.com/share"></a>
<a class="a2a_button_facebook"></a>
<a class="a2a_button_twitter"></a>
<a class="a2a_button_google_plus"></a>
<a class="a2a_button_pinterest"></a>
<a class="a2a_button_reddit"></a>
<a class="a2a_button_sina_weibo"></a>
<a class="a2a_button_wechat"></a>
<a class="a2a_button_douban"></a>
</div>
<script async src="//static.addtoany.com/menu/page.js"></script>



 

</div>


        <div class="post" itemprop="articleBody">

	  
        <div class="article-style" itemprop="articleBody">

          <p>I was asked to set up a local blast for the lab. Blast can be installed
directly using apt in debian and it turns out to be easy.</p>

<pre><code class="language-lang=&quot;bash&quot;">root@jz:/ssd/genomes# apt-get install ncbi-blast+
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  ncbi-blast+
0 upgraded, 1 newly installed, 0 to remove and 26 not upgraded.
Need to get 11.2 MB of archives.
After this operation, 32.8 MB of additional disk space will be used.
Get:1 http://ftp.hk.debian.org/debian/ wheezy/main ncbi-blast+ amd64 2.2.26-3 [11.2 MB]
Fetched 11.2 MB in 1min 16s (146 kB/s)
Selecting previously unselected package ncbi-blast+.
(Reading database ... 252681 files and directories currently installed.)
Unpacking ncbi-blast+ (from .../ncbi-blast+_2.2.26-3_amd64.deb) ...
Processing triggers for man-db ...
Setting up ncbi-blast+ (2.2.26-3) ...
</code></pre>

<p>Before the program can be used for sequence alignment, we should prepare
the db files:</p>

<pre><code class="language-lang=&quot;bash&quot;">root@jz:/ssd/genomes/blast/db# makeblastdb -in ../../hg19.fa -out hg19 -dbtype nucl
Building a new DB, current time: 11/21/2013 16:03:05
New DB name:   hg19
New DB title:  ../../hg19.fa
Sequence type: Nucleotide
Keep Linkouts: T
Keep MBits: T
Maximum file size: 1073741824B
Adding sequences from FASTA; added 25 sequences in 27.7084 seconds.
</code></pre>

<p>That&rsquo;s it. Now blast is supported in the lab server. I tested blastn
with the <a href="http://www.ncbi.nlm.nih.gov/nuccore/NC_000011.9?report=fasta&amp;from=62184015&amp;to=62331329&amp;strand=true">AHANAK
sequence</a>.</p>

<pre><code class="language-lang=&quot;bash&quot;">ygc@jz:~$ date; blastn -num_threads 8 -query sequence.fasta -db /ssd/genomes/blast/db/hg19 &gt;&gt; testblast.txt ; date
Thu Nov 21 16:19:19 HKT 2013
Thu Nov 21 17:19:45 HKT 2013
ygc@jz:~$ wc -c sequence.fasta
149510 sequence.fasta
</code></pre>

<p>Blast search has three distinctive stages: word matching with database
scan, ungapped alignment, gapped alignment with traceback. Only word
matching stage is multi-threaded. So it tooks 1 hour to align this long
gene.</p>

<p>The most significant hit of the blast result indicated that the sequence
is located at chr11:62331329-62184015, which is consistent with the
information presented in the <a href="http://www.ncbi.nlm.nih.gov/nuccore/NC_000011.9?report=fasta&amp;from=62184015&amp;to=62331329&amp;strand=true">AHANAK sequence
link</a>.</p>

<pre><code>ygc@jz:~$ cat testblast.txt
BLASTN 2.2.26+


Reference: Zheng Zhang, Scott Schwartz, Lukas Wagner, and Webb
Miller (2000), &quot;A greedy algorithm for aligning DNA sequences&quot;, J
Comput Biol 2000; 7(1-2):203-14.



Database: ../../hg19.fa
           25 sequences; 3,095,693,983 total letters



Query= gi|224589802:c62331329-62184015 Homo sapiens chromosome 11,
GRCh37.p13 Primary Assembly

Length=147315
                                                                      Score        E
Sequences producing significant alignments:                          (Bits)     Value

  chr11                                                               2.720e+05  0.0
  chr14                                                                941       0.0

...

&gt; chr11
Length=135006516

 Score = 2.720e+05 bits (147315),  Expect = 0.0
 Identities = 147315/147315 (100%), Gaps = 0/147315 (0%)
 Strand=Plus/Minus

Query  1         CATTTGGAAAGGAACCTGGGACCTAGGGTTAAAGGACAGATCCTCTGCTTTGCTACAGTG  60
                 ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  62331329  CATTTGGAAAGGAACCTGGGACCTAGGGTTAAAGGACAGATCCTCTGCTTTGCTACAGTG  62331270

Query  61        TAATCTTGGACAGGTCATTCTACTTGGTCTCATCTGTAAAGAGAGGGGAGGGTCAGGGTA  120
                 ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  62331269  TAATCTTGGACAGGTCATTCTACTTGGTCTCATCTGTAAAGAGAGGGGAGGGTCAGGGTA  62331210

...

  Database: ../../hg19.fa
    Posted date:  Nov 21, 2013  4:03 PM
  Number of letters in database: 3,095,693,983
  Number of sequences in database:  25



Matrix: blastn matrix 1 -2
Gap Penalties: Existence: 0, Extension: 2.5
</code></pre>

<p>In addition to the species-specific sequence file, we should download
the non-redundent gene and protein sequences from
<a href="ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nt.gz">ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nt.gz</a> and
<a href="ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nr.gz">ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nr.gz</a> respectively.</p>

<p>nt contains all nucleotide sequences of GenBank, RefSeq Nucleotides,
EMBL, DDBJ and PDB, while nr contains all non-redundant peptide
sequences from GenBank CDS translations, RefSeq Proteins, PDB,
SwissProt, PIR and PRF. After generating the db files, we can blast a
sequence to multiple species.</p>

<pre><code class="language-lang=&quot;bash&quot;">makeblastdb -in ../../nr/nr -out nr -dbtype prot
makeblastdb -in ../../nr/nt -out nt -dbtype nucl
</code></pre>


	  

	  


	</div>
    </article>

    <nav>
    <ul class="pager">
        
        <li class="previous"><a href="https://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters/"><span aria-hidden="true">&larr;</span> clusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters</a></li>
        

        
        <li class="next"><a href="https://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">Bug of R package ChIPpeakAnno <span aria-hidden="true">&rarr;</span></a></li>
        
    </ul>
</nav>

    
<section id="comments">
  <div id="disqus_thread">
    <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    if (window.location.hostname == "localhost")
        return;

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = 'https://' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  </div>
</section>





</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

