<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="http://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="http://guangchuangyu.github.io/post/">

    <title>Posts | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#contact">Contact</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2012_violin-plot/">violin plot</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2012-02-07 06:31:55 &#43;0800 CST" itemprop="datePublished">Tue, Feb 7, 2012</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggplot2">ggplot2</a>
        
    </span>
    
    

    

</div>

        <p><p>ggplot2 0.9.0版本将在3月1号发布，改动较大，最近bioc mailing
list也有诸多讨论，因为它导致了某些包编译出错（估计是NAMESPACE引起的）。</p>

<p>按照hadley的说法是这是为了把他个人的项目变成一个社区项目。</p>

<p>大多数的改动对用户来说是invisible的，用roxygen2去注释函数，核心代码使用S3重写，分离出一些特性（scales包）等。</p>

<p>最明显的是现在画图速度变快了，不过我觉得最让人兴奋的是<code>guide_legend()</code>和<code>guide_colorbar()</code>两个函数，可以修改legend.</p>

<p>画图的话，加了四个函数<code>geom_map()</code>, <code>geom_raster()</code>,
<code>geom_dotplot()</code>和<code>geom_violin()</code>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2011_ggplot2-version-figures-25-recipes-started-r/">ggplot2 Version of Figures in &lt;25 Recipes for Getting Started with R&gt;</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2011-08-16 18:26:36 &#43;0800 CST" itemprop="datePublished">Tue, Aug 16, 2011</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggplot2">ggplot2</a>
        
    </span>
    
    

    

</div>

        <p><p>In order to provide an option to compare graphs produced by basic
internal plot function and ggplot2, I have recreated the figures in the
book, <a href="http://book.douban.com/subject/5992544/">25 Recipes for Getting Started with
R</a>, with ggplot2.</p>

<p>The code used to create the images is in separate paragraphs, allowing
easy comparison. <strong>1.16 Creating a Scatter Plot</strong></p>

<pre><code>plot(cars)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggplot2/2011_bat-man/">the batman equation</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2011-08-13 17:44:42 &#43;0800 CST" itemprop="datePublished">Sat, Aug 13, 2011</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggplot2">ggplot2</a>
        
    </span>
    
    

    

</div>

        <p><p><a href="http://www.hardocp.com/news/2011/07/29/batman_equation/">HardOCP</a> has
an image with an equation which apparently draws the Batman logo.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/stats/2011_five-things-biologists-should-know-about-statistics/">Five things biologists should know about statistics</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2011-06-24 06:10:37 &#43;0800 CST" itemprop="datePublished">Fri, Jun 24, 2011</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/bootstrap">bootstrap</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/power">power</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/nonparametric">nonparametric</a>
        
    </span>
    
    

    

</div>

        <p><p><a href="http://www.ebi.ac.uk/~birney/">Ewan Birney</a>最近的一篇博文（<a href="http://genomeinformatician.blogspot.com/2011/06/five-statistical-things-i-wished-i-had.html">Five statistical things I wished I had been taught 20 years ago</a>
）讲述了统计对于生物学的重要性。</p>

<p>一开始从RA
Fisher讲起，说生物压根就是统计。Fisher是个农业学家，他所建立的那些统计方法，都是从生物学问题出发。</p>

<p>Ewan所谈及的五个方面分别是：</p>

<blockquote>
<p>1. Non parametric statistics. These are statistical tests which make a
bare minimum of assumptions of underlying distributions; in biology we
are rarely confident that we know the underlying distribution, and hand
waving about central limit theorem can only get you so far. Wherever
possible you should use a non parameteric test. This is Mann-Whitney (or
Wilcoxon if you prefer) for testing &ldquo;medians&rdquo; (Medians is in quotes
because this is not quite true. They test something which is closely
related to the median) of two distributions, Spearman&rsquo;s Rho (rather
pearson&rsquo;s r2) for correlation, and the Kruskal test rather than ANOVAs
(though if I get this right, you can&rsquo;t in Kruskal do the more
sophisticated nested models you can do with ANOVA). Finally, don&rsquo;t
forget the rather wonderful Kolmogorov-Smirnov (I always think it sounds
like really good vodka) test of whether two sets of observations come
from the same distribution. All of these methods have a basic theme of
doing things on the rank of items in a distribution, not the actual
level. So - if in doubt, do things on the rank of metric, rather than
the metric itself.</p>
</blockquote>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/dose/2011_package-dose-released/">R package DOSE released</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2011-06-13 04:58:10 &#43;0800 CST" itemprop="datePublished">Mon, Jun 13, 2011</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/dose">DOSE</a>
        
    </span>
    
    

    

</div>

        <p><p>Disease Ontology (DO) provides an open source ontology for the
integration of biomedical data that is associated with human disease. DO
analysis can lead to interesting discoveries that deserve further
clinical investigation.</p>

<p>DOSE was designed for semantic similarity measure and enrichment
analysis.</p>

<p>Four information content (IC)-based methods, proposed by Resnik, Jiang,
Lin and Schlicker, and one graph structure-based method, proposed by
Wang, were implemented. The calculation details can be referred to the
vignette of R package
<a href="http://www.bioconductor.org/packages/2.9/bioc/html/GOSemSim.html" title="GOSemSim">GOSemSim</a>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/clusterprofiler/2011_clusterprofiler-bioconductor/">clusterProfiler in Bioconductor 2.8</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2011-03-27 04:12:46 &#43;0800 CST" itemprop="datePublished">Sun, Mar 27, 2011</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>
        
    </span>
    
    

    

</div>

        <p><p>In recently years, high-throughput experimental techniques such as
microarray and mass spectrometry can identify many lists of genes and
gene products. The most widely used strategy for high-throughput data
analysis is to identify different gene clusters based on their
expression profiles. Another commonly used approach is to annotate these
genes to biological knowledge, such as Gene Ontology (GO) and Kyoto
Encyclopedia of Genes and Genomes (KEGG), and identify the statistically
significantly enriched categories. These two different strategies were
implemented in many bioconductor packages, such as <em>Mfuzz</em> and <em>BHC</em> for
clustering analysis and <em>GOstats</em> for GO enrichment analysis.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/mypaper/2010_phosphoproteome-profile-of-human-lung-cancer-cell-line-a549/">Phosphoproteome profile of human lung cancer cell line A549</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2010-11-23 02:22:22 &#43;0800 CST" itemprop="datePublished">Tue, Nov 23, 2010</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/cancer">cancer</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/phosphorylation">phosphorylation</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/profile">profile</a>
        
    </span>
    
    

    

</div>

        <p>As an in vitro model for type II human lung cancer, A549 cells resist cytotoxicity via phosphorylation of proteins as demonstrated by many studies. However, to date, no large-scale phosphoproteome investigation has been conducted on A549. Here, we performed a systematical analysis of the phosphoproteome of A549 by using mass spectrometry (MS)-based strategies. This investigation led to the identification of 337 phosphorylation sites on 181 phosphoproteins. Among them, 67 phosphoproteins and 230 phosphorylation sites identified appeared to be novel with no previous characterization in lung cancer.</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/gosemsim/2010_gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products/">GOSemSim an R package for measuring semantic similarity among GO terms and gene products</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2010-05-28 05:06:26 &#43;0800 CST" itemprop="datePublished">Fri, May 28, 2010</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>
        
    </span>
    
    

    

</div>

        <p><p><a href="http://www.ncbi.nlm.nih.gov/pubmed/20179076"><img src="http://bioinformatics.oxfordjournals.org/content/26/18.cover.gif" alt="" title="bioinfo" /></a></p>

<h3 id="abstract:c41832c6f53a1eda12f563057a868592">Abstract</h3>

<p><strong>SUMMARY</strong>: The semantic comparisons of Gene Ontology (GO) annotations
provide quantitative ways to compute similarities between genes and gene
groups, and have became important basis for many bioinformatics analysis
approaches. GOSemSim is an R package for semantic similarity computation
among GO terms, sets of GO terms, gene products and gene clusters. Four
information content (IC)- and a graph-based methods are implemented in
the GOSemSim package, multiple species including human, rat, mouse, fly
and yeast are also supported. The functions provided by the GOSemSim
offer flexibility for applications, and can be easily integrated into
high-throughput analysis pipelines. AVAILABILITY: GOSemSim is released
under the GNU General Public License within Bioconductor project, and
freely available at
<a href="http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html">http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html</a>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/stats/2009_bootstrap-method/">Bootstrap Method</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2009-07-31 09:14:42 &#43;0800 CST" itemprop="datePublished">Fri, Jul 31, 2009</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/bootstrap">bootstrap</a>
        
    </span>
    
    

    

</div>

        <p><p><a href="http://en.wikipedia.org/wiki/Bootstrapping_(statistics)">bootstrap</a>是对观测数据集进行有放回（replacement）的随机抽样，以评估总体的各项统计指标。可以用于假设检验、参数估计。好处是并不要求大样本，也不要求正态数据，并且对于不同的统计指标使用的是同样的计算方法。结果也更为可靠，坏处是计算量大。</p>

<p>统计推断(statistical
inference)是基于样本统计值的抽样分布来计算的，抽样分布需要从总体中许多的样本来计算，在只有一个样本的情况下，bootstrap对这一随机样本进行有放回的重复抽样，每一个重抽样本与原始随机样本一样大，每次计算相应的抽样的统计值，重复了N次之后，就可以计算统计值的bootstrap分布。</p>

<p>下面做一个小小的试验：</p>

<pre><code>a &lt;- c(seq(1:10), rnorm(50))  
</code></pre>

<p>#创建一个样本，60个数据，非正态分布的，如下图</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2009/07/sample_a.png" alt="" title="sample_a" /></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/gosemsim/2008_gosemsim/">GOSemSim</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2008-08-11 14:21:00 &#43;0800 CST" itemprop="datePublished">Mon, Aug 11, 2008</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>
        
    </span>
    
    

    

</div>

        <p><p>第一次写软件包，计算基因和GO的语义相似性。</p>

<p>实现<a href="http://bioinformatics.oxfordjournals.org/cgi/content/full/23/10/1274">Wang</a>的方法，基于GO的图结构。
以及四种基于信息含量的计算方法（Resnik, Jiang, Lin, Rel）。</p>

<p>可以从Bioconductor下载到。
<a href="http://bioconductor.org/packages/release/bioc/html/GOSemSim.html">http://bioconductor.org/packages/release/bioc/html/GOSemSim.html</a>
<a href="http://bioconductor.org/packages/devel/bioc/html/GOSemSim.html">http://bioconductor.org/packages/devel/bioc/html/GOSemSim.html</a></p>

<p>支持human, rat, mouse, fly, yeast, zebrafish,worm, E coli,
arabidopsis等物种。</p>
</p>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="http://guangchuangyu.github.io//post/page/6/">« Previous</a></li>
        
        
    </ul>
</nav>



    <footer class="site-footer">
    <div class="inner">
        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

</div>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

    </body>
</html>

