<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="//guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="//guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="//guangchuangyu.github.io/post/">

    <title>Posts | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="//guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    <h1>Posts</h1>

    
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/05/news-of-my-bioc-packages">NEWS of my BioC packages</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-05-05 20:13:57 &#43;0800 HKT" itemprop="datePublished">Thu, May 5, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/bioc">BioC</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/dose">DOSE</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/MISC/ke.jpg" width=600></p>

<p>Today is my birthday and it happened to be the release day of Bioconductor 3.3. It&rsquo;s again the time to reflect what I&rsquo;ve done in the past year.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA">ReactomePA</a></li>
</ul>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler">convert biological ID with KEGG API using clusterProfiler</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-05-03 17:00:00 &#43;0800 HKT" itemprop="datePublished">Tue, May 3, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/kegg">KEGG</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><h2 id="bitr-kegg:ea0718166651864960b492e0d7486a4f">bitr_kegg</h2>

<p><code>clusterProfiler</code> can convert biological IDs using <code>OrgDb</code> object via the <code>bitr</code> function. Now I implemented another function, <code>bitr_kegg</code> for converting IDs through KEGG API.</p>

<pre><code>library(clusterProfiler)
data(gcSample)
hg &lt;- gcSample[[1]]
head(hg)

## [1] &quot;4597&quot;  &quot;7111&quot;  &quot;5266&quot;  &quot;2175&quot;  &quot;755&quot;   &quot;23046&quot;

eg2np &lt;- bitr_kegg(hg, fromType='kegg', toType='ncbi-proteinid', organism='hsa')

## Warning in bitr_kegg(hg, fromType = &quot;kegg&quot;, toType = &quot;ncbi-proteinid&quot;,
## organism = &quot;hsa&quot;): 3.7% of input gene IDs are fail to map...

head(eg2np)

##     kegg ncbi-proteinid
## 1   8326      NP_003499
## 2  58487   NP_001034707
## 3 139081      NP_619647
## 4  59272      NP_068576
## 5    993      NP_001780
## 6   2676      NP_001487

np2up &lt;- bitr_kegg(eg2np[,2], fromType='ncbi-proteinid', toType='uniprot', organism='hsa')

head(np2up)

##   ncbi-proteinid uniprot
## 1      NP_005457  O75586
## 2      NP_005792  P41567
## 3      NP_005792  Q6IAV3
## 4      NP_037536  Q13421
## 5      NP_006054  O60662
## 6   NP_001092002  O95398
</code></pre>

<p>The ID type (both fromType &amp; toType) should be one of &lsquo;kegg&rsquo;, &lsquo;ncbi-geneid&rsquo;, &lsquo;ncbi-proteinid&rsquo; or &lsquo;uniprot&rsquo;. The &lsquo;kegg&rsquo; is the primary ID used in KEGG database. The data source of KEGG was from NCBI. A rule of thumb for the &lsquo;kegg&rsquo; ID is <code>entrezgene</code> ID for eukaryote species and <code>Locus</code> ID for prokaryotes.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis">KEGG Module Enrichment Analysis</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-04-13 20:48:15 &#43;0800 HKT" itemprop="datePublished">Wed, Apr 13, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/kegg">KEGG</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><blockquote>
<p>KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for annotation and biological interpretation of sequenced genomes. There are four types of KEGG modules:</p>

<ul>
<li>pathway modules – representing tight functional units in KEGG metabolic pathway maps, such as M00002 (Glycolysis, core module involving three-carbon compounds)</li>
<li>structural complexes – often forming molecular machineries, such as M00072 (Oligosaccharyltransferase)</li>
<li>functional sets – for other types of essential sets, such as M00360 (Aminoacyl-tRNA synthases, prokaryotes)</li>
<li>signature modules – as markers of phenotypes, such as M00363 (EHEC pathogenicity signature, Shiga toxin)</li>
</ul>
</blockquote>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/03/yet-an-unofficial-bioedit-for-osx">yet an unofficial BioEdit for OSX</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-03-30 13:40:22 &#43;0800 HKT" itemprop="datePublished">Wed, Mar 30, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/bioedit">BioEdit</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/alignment">alignment</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/osx">OSX</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p><a href="http://www.mbio.ncsu.edu/bioedit/bioedit.html"><img src="http://www.mbio.ncsu.edu/bioedit/Banner1.GIF" alt="" /></a></p>

<p>To my knowledge, <a href="http://www.mbio.ncsu.edu/bioedit/bioedit.html">BioEdit</a> is the most comprehensive biological sequence alignment editor. Most of my labmates run this software using <code>Parallels Desktop</code>. For some of them, BioEdit is the only reason to install <code>Parallels Desktop</code>.</p>

<p>I need to edit my alignment recently, and install it in my iMac using <a href="https://www.winehq.org/">Wine</a>, which is a compatibility layer for running Windows applications on POSIX-compliant OS. Although it is famous in Linux community for many years, many OSX users never heard of it.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-03-20 10:50:00 &#43;0800 HKT" itemprop="datePublished">Sun, Mar 20, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggplot2">ggplot2</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>I extended the <a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a> function to support embed image file in a <code>ggplot</code> object.</p>

<pre><code>set.seed(123)
d = data.frame(x=rnorm(10), y=rnorm(10))

imgfile &lt;- tempfile(, fileext=&quot;.png&quot;)
download.file(&quot;https://avatars1.githubusercontent.com/u/626539?v=3&amp;u=e731426406dd3f45a73d96dd604bc45ae2e7c36f&amp;s=140&quot;,
              destfile=imgfile, mode='wb')

p = ggplot(d, aes(x, y))
subview(p, imgfile, x=d$x[1], y=d$y[1]) + geom_point(size=5)
</code></pre>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/03/font-awesome-supported-in-emojifont">Font Awesome supported in emojifont</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-03-14 16:21:11 &#43;0800 HKT" itemprop="datePublished">Mon, Mar 14, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggplot2">ggplot2</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/emoji">emoji</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/emojifont">emojifont</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p><a href="http://cran.r-project.org/package=emojifont"><img src="http://www.r-pkg.org/badges/version/emojifont" alt="CRAN_Status_Badge" /></a></p>

<p><a href="https://cran.r-project.org/web/packages/emojifont/index.html"><code>emojifont</code></a> is available in CRAN, you can use the following command to install it.</p>

<pre><code>install.packages(&quot;emojifont&quot;)
</code></pre>

<p>An example of using emoji font in R plot is showed below:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2016/douban_emoji.jpg" alt="" /></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/02/covplot-supports-grangeslist">covplot supports GRangesList</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-02-29 11:45:08 &#43;0800 HKT" itemprop="datePublished">Mon, Feb 29, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>To answer the <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/27">issue</a>, I extend the <code>covplot</code> function to support viewing coverage of a list of <code>GRanges</code> objects or <code>bed</code> files.</p>

<pre><code>library(ChIPseeker)
files &lt;- getSampleFiles()
peak=GenomicRanges::GRangesList(CBX6=readPeakFile(files[[4]]),
                                CBX7=readPeakFile(files[[5]]))

p &lt;- covplot(peak)
print(p)
</code></pre>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/02/tweets-of-ggtree">tweets of ggtree</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-02-23 11:10:00 &#43;0800 HKT" itemprop="datePublished">Tue, Feb 23, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/twitter">twitter</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Treespace visualization of a heuristic best tree search in <a href="https://twitter.com/hashtag/R?src=hash">#R</a> package <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> <a href="https://twitter.com/hashtag/phylogenetics?src=hash">#phylogenetics</a> <a href="https://t.co/hnSJpIABql">pic.twitter.com/hnSJpIABql</a></p>&mdash; Alex Damián (@antropoteuthis) <a href="https://twitter.com/antropoteuthis/status/693123105707986944">January 29, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr"><a href="https://twitter.com/guangchuangyu">@guangchuangyu</a> Updated to latest <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> and <a href="https://twitter.com/hashtag/ggplot2?src=hash">#ggplot2</a> and very impressed! Any ideas on using geom_label with tip labels to add box/fill?</p>&mdash; Jo Williams-Newkirk (@ajwnewkirk) <a href="https://twitter.com/ajwnewkirk/status/691762302492790788">January 25, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">ggtree looks really good! <a href="https://twitter.com/hashtag/rstats?src=hash">#rstats</a><a href="https://t.co/7rtexYV3I5">https://t.co/7rtexYV3I5</a></p>&mdash; Paul Staab (@paulstaab) <a href="https://twitter.com/paulstaab/status/687933344219312128">January 15, 2016</a></blockquote>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-02-16 13:10:21 &#43;0800 HKT" itemprop="datePublished">Tue, Feb 16, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>My R/Bioconductor package,
<a href="https://github.com/GuangchuangYu/ReactomePA">ReactomePA</a>, published in
<a href="http://pubs.rsc.org/en/content/articlelanding/2016/mb/c5mb00663e">Molecular BioSystems</a>.</p>

<p><strong>ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</strong></p>

<p><strong>G Yu</strong>, QY He. <strong><em>Molecular BioSystems</em></strong> 2016, 12:477-479</p>

<p>Received: 05 Oct 2015 Accepted: 20 Nov 2015 Online: 23 Nov 2015</p>

<p>DOI: <a href="http://dx.doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format">ggtree supports phylip tree format</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2016-01-15 13:19:00 &#43;0800 HKT" itemprop="datePublished">Fri, Jan 15, 2016</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>Phylip is also a widely used tree format, which contains taxa sequences with Newick tree text.</p>

<p>In <code>ggtree</code>, we can use <code>read.phylip()</code> function to parse the file and use <code>ggtree()</code> to visualize the tree.</p>
</p></div>
    </div>
    

    
<nav>
    <ul class="pager">
        
        
        <li><a href="//guangchuangyu.github.io//post/page/2/">Next »</a></li>
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="//guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="//guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="//guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

