<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.21" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="PhD student">

  
  
  
    
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/">

  

  <title>Posts | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Posts</h1>

  

  
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2017/06/pixel-art-of-ggplot2-faceting-using-geofacet/">pixel art of ggplot2 faceting using geofacet</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2017-06-26 20:13:35 &#43;0800 CST" itemprop="datePublished">
      Mon, Jun 26, 2017
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggplot2">ggplot2</a
    >, 
    
    <a href="/tags/geofacet">geofacet</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      I just discovered an interesting <code>ggplot2</code> extension, <code>geofacet</code>, that supports arranging facet panels that mimics geographic topoloty.</p>

<p>After playing with it, I realized that it is not only for visualizing <code>geo</code>-related data, but also can be fun for presenting data to mimics pixel art.</p>

<p><img src="https://guangchuangyu.github.io/blog_images/2017/Screenshot 2017-06-26 15.49.50.png" alt="" /></p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2017/05/bioc-3.5-news-of-my-bioc-packages/">[Bioc 3.5] NEWS of my BioC packages</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2017-05-19 18:13:57 &#43;0800 CST" itemprop="datePublished">
      Fri, May 19, 2017
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >, 
    
    <a href="/tags/meshes">meshes</a
    >, 
    
    <a href="/tags/treeio">treeio</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      I have 8 packages published within the <code>Bioconductor</code> project.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker" target="_blank">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler" target="_blank">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE" target="_blank">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree" target="_blank">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim" target="_blank">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/meshes" target="_blank">meshes</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA" target="_blank">ReactomePA</a></li>
<li><a href="http://guangchuangyu.github.io/treeio" target="_blank">treeio</a></li>
</ul>

<p>A new package <a href="http://guangchuangyu.github.io/treeio" target="_blank">treeio</a> was included in BioC 3.5 release.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2017/05/phylomoji-with-ggtree-and-emojifont/">Phylomoji with ggtree and emojifont</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2017-05-17 11:11:11 &#43;0800 CST" itemprop="datePublished">
      Wed, May 17, 2017
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggplot2">ggplot2</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/emoji">emoji</a
    >, 
    
    <a href="/tags/emojifont">emojifont</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      With ggtree (Yu et al. 2017), it is very easy to create phylomoji. Emoji is internally supported by ggtree.</p>

<h2 id="use-emoji-in-taxa-labels">Use emoji in taxa labels</h2>

<pre><code class="language-r">library(ggtree)
tree_text &lt;- &quot;(((((cow, (whale, dolphin)), (pig2, boar)), camel), fish), seedling);&quot;
x &lt;- read.tree(text=tree_text)
ggtree(x, linetype=&quot;dashed&quot;, color='firebrick') +
    xlim(NA, 7) + ylim(NA, 8.5) +
    geom_tiplab(aes(color=label), parse='emoji', size=14, vjust=0.25) +
    labs(title=&quot;phylomoji&quot;, caption=&quot;powered by ggtree + emojifont&quot;)
</code></pre>

<p><img src="https://guangchuangyu.github.io/blog_images/R/emojifont/phylomoji_files/figure-markdown_github/unnamed-chunk-2-1.png" alt="" /></p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2017/01/convert-graphic-object-to-tree-object-using-treeio/">convert graphic object to tree object using treeio</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2017-01-03 14:58:10 &#43;0800 CST" itemprop="datePublished">
      Tue, Jan 3, 2017
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/treeio">treeio</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      I have splitted <code>ggtree</code> to 2 packages, <code>treeio</code> and <code>ggtree</code>. Now <code>ggtree</code> is mainly focus on visualization and annotation, while <code>treeio</code> focus on parsing and exporting tree files. Here is a welcome message from <code>treeio</code> that you can convert <code>ggtree</code> output to tree object which can be exported as newick or nexus file if you want.</p>

<p>Thanks to <code>ggplot2</code>, output of <code>ggtree</code> is actually a <code>ggplot</code> object. The <code>ggtree</code> object can be rendered as graph by <code>print</code> method, but internally it is an object that contains data. <code>treeio</code> defines <code>as.phylo</code> and <code>as.treedata</code> to convert <code>ggtree</code> object to <code>phylo</code> or <code>treedata</code> object.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/12/dotplot-for-gsea-result/">dotplot for GSEA result</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-12-30 12:58:10 &#43;0800 CST" itemprop="datePublished">
      Fri, Dec 30, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/gsea">GSEA</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <img src="https://guangchuangyu.github.io/blog_images/Bioconductor/DOSE/gseaplot.png" alt="" /></p>

<p>For GSEA analysis, we are familar with the above figure which shows the running enrichment score. But for most of the software, it lack of visualization method to summarize the whole enrichment result.</p>

<p>In <a href="https://guangchuangyu.github.io/DOSE" target="_blank">DOSE</a> (and related tools including <a href="https://guangchuangyu.github.io/clusterProfiler" target="_blank">clusterProfiler</a>, <a href="https://guangchuangyu.github.io/ReactomePA" target="_blank">ReactomePA</a> and <a href="https://guangchuangyu.github.io/meshes" target="_blank">meshes</a>), we provide <code>enrichMap</code> and <code>cnetplot</code> to summarize GSEA result.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/12/add-layer-to-specific-panel-of-facet_plot-output/">add layer to specific panel of facet_plot output</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-12-28 20:30:00 &#43;0800 CST" itemprop="datePublished">
      Wed, Dec 28, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/evolution">Evolution</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/ggplot2">ggplot2</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      This is a question from <a href="https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/bioc-ggtree/uPgksYWkwCA/oMtHKW7KEgAJ" target="_blank">ggtree google group</a>:</p>

<blockquote>
<p>Dear ggtree team,</p>

<p>how can I apply a geom_xxx to only one facet panel? For example if i want to get <code>geom_hline(yintersect=1:30)</code> or a <code>geom_text()</code> in the dot panel?
I cant see the <code>facet_grid(. ~ var)</code> function call, so I don&rsquo;t know which subsetting to use. I have already read <a href="http://stackoverflow.com/questions/29873155/geom-text-and-facets-not-working" target="_blank">http://stackoverflow.com/questions/29873155/geom-text-and-facets-not-working</a></p>

<pre><code>  tr &lt;- rtree(30)
  
  d1 &lt;- data.frame(id=tr$tip.label, val=rnorm(30, sd=3))
  p &lt;- ggtree(tr)
  
  p2 &lt;- facet_plot(p, panel=&quot;dot&quot;, data=d1, geom=geom_point, aes(x=val), color='firebrick')
  d2 &lt;- data.frame(id=tr$tip.label, value = abs(rnorm(30, mean=100, sd=50)))
  
  p3 &lt;- facet_plot(p2, panel='bar', data=d2, geom=geom_segment, aes(x=0, xend=value, y=y, yend=y), size=3, color='steelblue') + theme_tree2()
</code></pre>

<p>Thanks!
Andreas</p>
</blockquote>

<p>If this can be done, we can create even more <a href="https://guangchuangyu.github.io/2016/12/ggtree-version-of-plottree/" target="_blank">comprehensive tree plots</a>.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/12/ggtree-version-of-plottree/">ggtree version of plotTree</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-12-17 13:53:00 &#43;0800 CST" itemprop="datePublished">
      Sat, Dec 17, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <a href="https://holtlab.net/2015/10/03/plotting-trees-data/" target="_blank">PLOTTING TREES + DATA</a> is difficult. <a href="http://holtlab.net/" target="_blank">@DrKatHolt</a> developed <a href="https://github.com/katholt/plotTree" target="_blank">plotTree</a> (R and Python scripts) to visualize associated data with trees, <em>e.g.</em> heatmap, horizontal bar <em>etc</em>.</p>

<p>I reproduced the examples presented in the <a href="https://github.com/katholt/plotTree" target="_blank">plotTree</a> repo using <a href="https://guangchuangyu.github.io" target="_blank">ggtree</a>. Source code is freely available in <a href="https://github.com/GuangchuangYu/plotTree-ggtree" target="_blank">https://github.com/GuangchuangYu/plotTree-ggtree</a>.</p>

<p>Here are the outputs produced by <code>ggtree</code>:</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/12/edge-coloring-with-user-data/">Edge coloring with user data</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-12-12 00:53:00 &#43;0800 CST" itemprop="datePublished">
      Mon, Dec 12, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      Coloring edges in ggtree is quite easy, as we can map the color to numerical or categorical values via the <code>aes(color=VAR)</code> syntax. For user&rsquo;s own data, it is also easy as <code>ggtree</code> provide the <code>%&lt;+%</code> operator to attach user data.</p>

<p>But as it seems not so obviously for <code>ggtree</code> users, see question <a href="https://groups.google.com/forum/#!topic/bioc-ggtree/4GgivKqVjB8" target="_blank">1</a>, <a href="https://github.com/GuangchuangYu/ggtree/issues/76" target="_blank">2</a>, and <a href="https://github.com/GuangchuangYu/ggtree/issues/105" target="_blank">3</a>, I will demonstrate how to color edges using user data here.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/12/scatterpie-for-plotting-pies-on-ggplot/">scatterpie for plotting pies on ggplot</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-12-07 00:53:00 &#43;0800 CST" itemprop="datePublished">
      Wed, Dec 7, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/scatterpie">scatterpie</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <img src="https://guangchuangyu.github.io/blog_images/R/scatterpie_files/so.png" alt="" /></p>

<p>Plotting pies on ggplot/ggmap is not an easy task, as ggplot2 doesn&rsquo;t provide native pie geom. The pie we produced in ggplot2 is actually a barplot transform to polar coordination. This make it difficult if we want to produce a map like the above screenshot, which was posted by Tyler Rinker, the author of R package pacman.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/11/align-genomic-features-with-phylogenetic-tree/">align genomic features with phylogenetic tree</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-11-15 20:10:48 &#43;0800 CST" itemprop="datePublished">
      Tue, Nov 15, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/ggbio">ggbio</a
    >, 
    
    <a href="/tags/tree">tree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <a href="https://www.biostars.org/p/218591/" target="_blank">A question</a> on biostars asking how to generate the following figure:</p>

<p><img src="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/ggtree_ggbio/218591.png" alt="" /></p>

<p>This can be quite easy to implement in <a href="https://guangchuangyu.github.io" target="_blank">ggtree</a>, I can write a <code>geom</code> layer to layout the alignment. As <a href="https://bioconductor.org/packages/ggbio/" target="_blank">ggbio</a> already provides many <code>geom</code> for genomic data and I don&rsquo;t want to re-invent the wheel, I decided to try <code>ggtree+ggbio</code>. This is also the beauty of <code>R</code> that packages complete each others.</p>

<p>
      
    </p></div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    
    <li><a href="/post/page/2/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:350px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    

    
    

  </body>
</html>

