<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Guangchuang Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="Guangchuang Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
 <div id="content">
  <div class="blog-index">
    
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a>
        </h1>
        <p class="meta">May 11, 2015 - 3 minute read -  <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics/">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/systems-biology/">Systems Biology</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>
for human) and <a href="http://www.reactome.org/">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/">using GOSemSim to rank proteins obtained by co-IP</a>
        </h1>
        <p class="meta">May 4, 2015 - 2 minute read -  <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/proteomics/">Proteomics</a><a class="label" href="http://guangchuangyu.github.io/categories/publication/">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>Co-IP is usually used to identified interactions among specific
proteins. It is widely used in detecting protein complex. Unfortunately,
an identified protein may not be an interactor, and sometimes can be a
background contaminant. Ranking proteins can help us to focus a study on
a few high quality candidates for subsequent interaction investigation.
My R package <a href="https://github.com/GuangchuangYu/GOSemSim">GOSemSim</a> has
been used in many studies to investigate protein protein interaction,
see <a href="http://pubs.rsc.org/en/content/articlehtml/2014/mb/c3mb70486f">1</a>,
<a href="http://www.sciencedirect.com/science/article/pii/S1874391915000937">2</a>,
<a href="http://www.sciencedirect.com/science/article/pii/S1476927115000493">3</a>,
<a href="http://www.mcponline.org/content/13/5/1318.short">4</a> and
<a href="http://scholar.google.com/scholar?oi=bibs&amp;hl=en&amp;cites=9484177541993722322,16048119560581430155,18126401808149291947">more</a>.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/">Bioc 3.1: NEWS of my BioC packages</a>
        </h1>
        <p class="meta">Apr 20, 2015 - 3 minute read -  <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim:d7b2a91b2c7afdf12777a4b5cd75c8f0">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/a9ed4c29e8f66c07de917937694d7f43933d4fe5/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/">an example of drawing beast tree using ggtree</a>
        </h1>
        <p class="meta">Apr 1, 2015 - 1 minute read -  <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/evolution/">Evolution</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization/">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p><a href="http://beast.bio.ed.ac.uk/figtree">FigTree</a> is designed for viewing
beast output as demonstrated by their example data:
<img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-10%2018.15.57.png" alt="" /></p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">DAVID functional analysis with clusterProfiler</a>
        </h1>
        <p class="meta">Mar 16, 2015 - 2 minute read -  <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="http://guangchuangyu.github.io/categories/r/">R</a>
        </p>

      </header>

        
          <p><p><a href="https://github.com/GuangchuangYu/clusterProfiler">clusterProfiler</a> was
used to visualize <a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID</a>
results in a paper published in <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3907017/">BMC
Genomics</a>.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/clusterProfiler_david.png" alt="" /></p>

<p>Some users told me that they may want to use DAVID at some
circumstances. I think it maybe a good idea to make clusterProfiler
supports DAVID, so that
<a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID</a> users can use
visualization functions provided by
<a href="https://github.com/GuangchuangYu/clusterProfiler">clusterProfiler</a>.</p>

<pre><code>require(DOSE)
require(clusterProfiler)
data(geneList)
gene = names(geneList)[abs(geneList) &gt; 2]
david = enrichDAVID(gene = gene, idType=&quot;ENTREZ_GENE_ID&quot;, 
listType=&quot;Gene&quot;, annotation=&quot;KEGG_PATHWAY&quot;)



&gt; summary(david)
               ID            Description GeneRatio  BgRatio       pvalue
hsa04110 hsa04110             Cell cycle     11/68 125/5085 4.254437e-06
hsa04114 hsa04114         Oocyte meiosis     10/68 110/5085 1.119764e-05
hsa03320 hsa03320 PPAR signaling pathway      7/68  69/5085 2.606715e-04
             p.adjust qvalue                                             geneID
hsa04110 0.0003998379     NA 9133/4174/890/991/1111/891/7272/8318/4085/983/9232
hsa04114 0.0005261534     NA    9133/5241/51806/3708/991/891/4085/983/9232/6790
hsa03320 0.0081354974     NA                 4312/2167/5346/5105/3158/9370/9415
         Count
hsa04110    11
hsa04114    10
hsa03320     7
</code></pre>

<p>There are only 5085 human genes annotated by KEGG, this is due to
out-of-date DAVID data.</p>
</p>

          
              <footer>
                  <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    
    






<div class="pagination">

    
        
        
        
        <a href="/" aria-label="First" class="label-pagination"><i class="fa fa-angle-double-left fa-lg"></i></a>
    

    
    
        <a href="/page/6/" aria-label="Previous" class="label-pagination"><i class="fa fa-angle-left fa-lg"></i></a>
    

    
        <a href="/" class="label-pagination">1</a>
    
        <a href="/page/2/" class="label-pagination">2</a>
    
        <a href="/page/3/" class="label-pagination">3</a>
    
        <a href="/page/4/" class="label-pagination">4</a>
    
        <a href="/page/5/" class="label-pagination">5</a>
    
        <a href="/page/6/" class="label-pagination">6</a>
    
        <a href="/page/7/" class="label-pagination">7</a>
    
        <a href="/page/8/" class="label-pagination">8</a>
    
        <a href="/page/9/" class="label-pagination">9</a>
    
        <a href="/page/10/" class="label-pagination">10</a>
    
        <a href="/page/11/" class="label-pagination">11</a>
    
        <a href="/page/12/" class="label-pagination">12</a>
    
        <a href="/page/13/" class="label-pagination">13</a>
    
        <a href="/page/14/" class="label-pagination">14</a>
    
        <a href="/page/15/" class="label-pagination">15</a>
    
        <a href="/page/16/" class="label-pagination">16</a>
    
        <a href="/page/17/" class="label-pagination">17</a>
    

    
    
        <a href="/page/8/" aria-label="Next" class="label-pagination"><i class="fa fa-angle-right fa-lg"></i></a>
     
    
    
    
        <a href="/page/17/" aria-label="Last"><i class="fa fa-angle-double-right fa-lg"></i></a>
    

</div>


   

  </div>

  


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/covplot-supports-grangeslist/">covplot supports GRangesList</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
    </ul>
  </section>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />
  
</aside>
    
 </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>
   