<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Annotate a phylogenetic tree with insets</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            Annotate a phylogenetic tree with insets 
          </h1>
          <p class="meta">Jan 11, 2016 - 3 minute read -  <a href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
	  	  
	  

<p><code>Ggtree</code> implemented a function, <code>subview</code>, that can add subplots on a
ggplot2 object. It had successful applied to <a href="http://stackoverflow.com/questions/10368180/plotting-pie-graphs-on-map-in-ggplot/32380396#32380396">plot pie graphs on map</a>.</p>

<pre><code>set.seed(2016-01-04)
tr &lt;- rtree(30)
tr &lt;- groupClade(tr, node=45)
p &lt;- ggtree(tr, aes(color=group)) + geom_tippoint()
cpos &lt;- get_clade_position(p, node=45)
p1 &lt;- p + geom_hilight(node=45)
p2 &lt;- with(cpos, p+xlim(xmin, xmax*1.01)+ylim(ymin, ymax))
with(cpos, subview(p2+geom_tiplab(), p1+theme_transparent(), x=xmin+(xmax-xmin)*.15, y=ymin+(ymax-ymin)*.85))
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-2-1.png" alt="" /></p>

<p>To make it more easy to use subview function for annotating taxa with
subplots, <em>ggtree</em> provides a function, <code>inset</code>, for adding subplots to
a phylogenetic tree. The input is a tree graphic object and a named list
of ggplot graphic objects (can be any kind of charts), these objects
should named by node numbers. To facilitate adding bar and pie charts
(e.g. summarized stats of results from ancestral reconstruction) to
phylogenetic tree, <em>ggtree</em> provides <code>nodepie</code> and <code>nodebar</code> functions
to create a list of pie or bar charts.</p>

<h2 id="annotate-with-bar-charts:94c165ad3f1a57afa6b70fe6932c98e6">Annotate with bar charts</h2>

<pre><code>set.seed(2015-12-31)
tr &lt;- rtree(15)
p &lt;- ggtree(tr)

a &lt;- runif(14, 0, 0.33)
b &lt;- runif(14, 0, 0.33)
c &lt;- runif(14, 0, 0.33)
d &lt;- 1 - a - b - c
dat &lt;- data.frame(a=a, b=b, c=c, d=d)
## input data should have a column of `node` that store the node number
dat$node &lt;- 15+1:14

## cols parameter indicate which columns store stats (a, b, c and d in this example)
bars &lt;- nodebar(dat, cols=1:4)

inset(p, bars)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-3-1.png" alt="" /><br />
 The sizes of the insets can be ajusted by the paramter <em>width</em> and
<em>height</em>.</p>

<pre><code>inset(p, bars, width=.03, height=.06)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-4-1.png" alt="" /><br />
 Users can set the color via the parameter <em>color</em>. The <em>x</em> position can
be one of &lsquo;node&rsquo; or &lsquo;branch&rsquo; and can be adjusted by the parameter
<em>hjust</em> and <em>vjust</em> for horizontal and vertical adjustment respecitvely.</p>

<pre><code>bars2 &lt;- nodebar(dat, cols=1:4, position='dodge',
                 color=c(a='blue', b='red', c='green', d='cyan'))
p2 &lt;- inset(p, bars2, x='branch', width=.03, vjust=-.3)
print(p2)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-5-1.png" alt="" /></p>

<h2 id="annotate-with-pie-charts:94c165ad3f1a57afa6b70fe6932c98e6">Annotate with pie charts</h2>

<p>Similarly, users can use <code>nodepie</code> function to generate a list of pie
charts and place these charts to annotate corresponding nodes. Both
<code>nodebar</code> and <code>nodepie</code> accepts parameter <em>alpha</em> to allow transparency.</p>

<pre><code>pies &lt;- nodepie(dat, cols=1:4, alpha=.6)
inset(p, pies)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-6-1.png" alt="" /></p>

<pre><code>inset(p, pies, hjust=-.06)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-7-1.png" alt="" /></p>

<h2 id="annotate-with-other-types-of-charts:94c165ad3f1a57afa6b70fe6932c98e6">Annotate with other types of charts</h2>

<p>The <code>inset</code> function accepts a list of ggplot graphic objects and these
input objects are not restricted to pie or bar charts. They can be any
kinds of charts and hybrid of these charts.</p>

<pre><code>pies_and_bars &lt;- bars2
pies_and_bars[9:14] &lt;- pies[9:14]
inset(p, pies_and_bars)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-8-1.png" alt="" /></p>

<pre><code>d &lt;- lapply(1:15, rnorm, n=100)
ylim &lt;- range(unlist(d))
bx &lt;- lapply(d, function(y) {
    dd &lt;- data.frame(y=y)
    ggplot(dd, aes(x=1, y=y))+geom_boxplot() + ylim(ylim) + theme_inset()
})
names(bx) &lt;- 1:15
inset(p, bx, width=.03, height=.1, hjust=-.05)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-9-1.png" alt="" /></p>

<p>After annotating with insets, users can further annotate the tree with
another layer of insets.</p>

<pre><code>p2 &lt;- inset(p, bars2, x='branch', width=.03, vjust=-.4)
p2 &lt;- inset(p2, pies, x='branch', vjust=.4)
bx2 &lt;- lapply(bx, function(g) g+coord_flip())
inset(p2, bx2, width=.2, height=.03, vjust=.04, hjust=p2$data$x[1:15]-4) + xlim(NA, 4.5)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/advanceTreeAnnotation_files/figure-markdown_strict/unnamed-chunk-10-1.png" alt="" /></p>


	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
	
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Jan 11, 2016</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/" title="GO analysis using clusterProfiler">GO analysis using clusterProfiler</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree/" title="label edge number in ggtree">label edge number in ggtree</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>



  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/05/bioc-3.3-news-of-my-packages/">BioC 3.3: NEWS of my packages</a>
      </li>
      
      <li class="post">
        <a href="/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler/">convert biological ID with KEGG API using clusterProfiler</a>
      </li>
      
      <li class="post">
        <a href="/2016/04/kegg-module-enrichment-analysis/">KEGG Module Enrichment Analysis</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/google-drive--hku/">Google Drive @ HKU</a>
      </li>
      
    </ul>
  </section>

  
  
  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


