<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/categories/visualization/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/categories/visualization/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/categories/visualization/">

  

  <title>Visualization | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Visualization</h1>

  

  
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/subview/">subview</a></h2>
    <div class="article-style">
      
      <p>I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
    geom_bar(stat=&quot;identity&quot;, width=1) + 
    coord_polar(theta=&quot;y&quot;) + theme_tree() + 
    xlab(NULL) + ylab(NULL) + 
    theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
    p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
    print(p)
}
</code></pre>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a></h2>
    <div class="article-style">
      
      <p><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/" target="_blank">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
    <div class="article-style">
      
      This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/">an example of drawing beast tree using ggtree</a></h2>
    <div class="article-style">
      
      <p><a href="http://beast.bio.ed.ac.uk/figtree" target="_blank">FigTree</a> is designed for viewing
beast output as demonstrated by their example data:
<img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-10%2018.15.57.png" alt="" /></p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/">viewing and annotating phylogenetic tree with ggtree</a></h2>
    <div class="article-style">
      
      <p>When I need to annotate nucleotide substitutions in the phylogenetic
tree, I found that all the software are designed to display the tree but
not annotating it. Some of them may support annotating the tree with
specific data such as bootstrap values, but they are restricted to a few
supported data types. It is hard/impossible to inject user specific
data.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/12/2014-12-21-145753_579x436_scrot.png" alt="" />
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/10/sir-model-of-epidemics/">SIR Model of Epidemics</a></h2>
    <div class="article-style">
      
      <p>The SIR model divides the population to three compartments:
<strong>S</strong>usceptible, <strong>I</strong>nfected and <strong>R</strong>ecovered. If the disease dynamic
fits the SIR model, then the flow of individuals is one direction from
the susceptible group to infected group and then to the recovered group.
All individuals are assumed to be identical in terms of their
susceptibility to infection, infectiousness if infected and mixing
behaviour associated with disease transmission.</p>

<p>We defined: $S_t$ = the number of susceptible individuals at time t</p>

<p>$ I_t $ = the number of infected individuals at time t</p>

<p>$R_t$ = the number of recovered individuals at time t</p>

<p>Suppose on average every infected individual will contact $\gamma$
person, and $\kappa$ percent of these $\gamma$ person will be
infected. Then on average there are $\beta = \gamma \times \kappa$
person will be infected an infected individual.</p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">multiple annotation in ChIPseeker</a></h2>
    <div class="article-style">
      
      <h3 id="nearest-gene-annotation">Nearest gene annotation</h3>

<p>Almost all annotation software calculate the distance of a peak to the
nearest TSS and assign the peak to that gene. This can be misleading, as
binding sites might be located between two start sites of different
genes or hit different genes which have the same TSS location in the
genome.</p>

<p>The function <em>annotatePeak</em> provides option to assign genes with a max
distance cutoff and all genes within this distance were reported for
each peak.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/08/enrichment-map/">enrichment map</a></h2>
    <div class="article-style">
      
      In <a href="http://www.plob.org/" target="_blank">PLOB</a>&rsquo;s QQ group, someone asked how to change
the color of enrichment map in <a href="http://cytoscape.org/" target="_blank">Cytoscape</a>. I am
very curious how enrichment map can helps to interpret enrichment
results. It took me 2 hours to implement it using R and I am surprised
that the enrichment map is better than anticipated.

<img src="http://guangchuangyu.github.io/blog_images/2014/08/Screenshot-2014-07-30-22.20.07.png" alt="" />
      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a></h2>
    <div class="article-style">
      
      <p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes">ChIP peaks over Chromosomes</h3>

<pre><code>&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2012/02/draw-chinese-character-jiong-using-ggplot2/">draw chinese character Jiong using ggplot2</a></h2>
    <div class="article-style">
      
      <p>The Chinese character Jiong (囧) is now widely used for expressing ideas
or feelings such as annoyance, shock, embarrassment, awkwardness, scorn.</p>

<p>The function plot of <code>y=1/(x^2-1)</code> looks very similar of this symbol.</p>

<p>I use ggplot2 to draw the symbol of Jiong.
</p>

      
    </div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/categories/visualization/page/2/">&laquo; Previous</a></li>
    
    
    <li><a href="/categories/visualization/page/4/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

