<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 3.1.1">
  <meta name="generator" content="Hugo 0.57.2" />
  

  
  
  
  
    
  
  <meta name="description" content="

GOSemSim

GOSemSim
was first implemented in 2008
and published in
Bioinformatics
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.

">

  
  <link rel="alternate" hreflang="en-us" href="https://guangchuangyu.github.io/post/r/bioc/2015_bioc-3.1-news-of-my-bioc-packages/">

  


  

  

  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha256-eSi1q2PG6J7g7ib17yAaWMcrr5GrtohYChqibrV7PBE=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.4.1/css/all.css" integrity="sha384-5sAR7xN1Nv6T6+dT2mhtzEpVJvfS3NScPQTrOxhwjIuvcA67KV2R5Jz6kr4abQsz" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" crossorigin="anonymous">
        
      
    

    

    

  

  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono">
  

  <link rel="stylesheet" href="/styles.css">
  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-77901140-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  
  <link rel="alternate" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/r/bioc/2015_bioc-3.1-news-of-my-bioc-packages/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@guangchuangyu">
  <meta property="twitter:creator" content="@guangchuangyu">
  
  <meta property="og:site_name" content="Guangchuang YU">
  <meta property="og:url" content="https://guangchuangyu.github.io/post/r/bioc/2015_bioc-3.1-news-of-my-bioc-packages/">
  <meta property="og:title" content="[Bioc 31] NEWS of my BioC packages | Guangchuang YU">
  <meta property="og:description" content="

GOSemSim

GOSemSim
was first implemented in 2008
and published in
Bioinformatics
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.

">
  
  
    
  <meta property="og:image" content="https://guangchuangyu.github.io/img/icon-192.png">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2015-04-20T11:44:01&#43;08:00">
  
  <meta property="article:modified_time" content="2015-04-20T11:44:01&#43;08:00">
  

  

  

  <title>[Bioc 31] NEWS of my BioC packages | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#TableOfContents" data-offset="71" >
  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header fixed-top">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" role="textbox" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Guangchuang YU</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#posts">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#teaching">
            
            <span>Teaching</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/resume">
            
            <span>Resume</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#contact">
            
            <span>Contact</span>
            
          </a>
        </li>

        
        

        

        
        
        

        <li class="nav-item">
          <a class="nav-link" href="/cn">
            
            <span>中文入口</span>
            
          </a>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">[Bioc 31] NEWS of my BioC packages</h1>

  

  
    

<div class="article-metadata">

  
  
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="">
  </span>
  

  <span class="article-date">
    
    <meta content="2015-04-20 11:44:01 &#43;0800 CST" itemprop="datePublished">
    <time datetime="2015-04-20 11:44:01 &#43;0800 CST" itemprop="dateModified">
      Apr 20, 2015
    </time>
  </span>
  <span itemscope itemprop="publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    3 min read
  </span>
  

  
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder"></i>
    
    <a href="/categories/r/">R</a>
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages&amp;url=https%3a%2f%2fguangchuangyu.github.io%2fpost%2fr%2fbioc%2f2015_bioc-3.1-news-of-my-bioc-packages%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fpost%2fr%2fbioc%2f2015_bioc-3.1-news-of-my-bioc-packages%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-facebook-f"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fguangchuangyu.github.io%2fpost%2fr%2fbioc%2f2015_bioc-3.1-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank" rel="noopener">
        <i class="fab fa-linkedin-in"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fguangchuangyu.github.io%2fpost%2fr%2fbioc%2f2015_bioc-3.1-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank" rel="noopener">
        <i class="fab fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages&amp;body=https%3a%2f%2fguangchuangyu.github.io%2fpost%2fr%2fbioc%2f2015_bioc-3.1-news-of-my-bioc-packages%2f">
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    















  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      <p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" target="_blank">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/" target="_blank">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>

<h3 id="clusterprofiler">clusterProfiler</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
was developed in
<a href="http://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8/" target="_blank">2011</a> and
published in
<a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118" target="_blank">OMICS</a> in
2012. I am very glad to find that
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
was presented in the book &lsquo;<a href="https://www.safaribooksonline.com/library/view/bioinformatics-with-r/9781783283132/ch02s08.html" target="_blank">Bioinformatics with R Cookbook</a>&lsquo;:
In this release, KEGG enrichment analysis (both hypergeometric and GSEA)
now supports <a href="http://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/" target="_blank">using latest online data</a>
and <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/" target="_blank">DAVID functional analysis</a>
is also supported. In addition, compareCluster function now supports
<a href="http://bioinfoblog.it/2015/02/a-formula-interface-for-geneontology-analysis/" target="_blank">formula interface</a>
like x ~ y + z; this feature was contributed by
<a href="https://twitter.com/dalloliogm" target="_blank">@dalloliogm</a>. Considering that many new
users have problem of converting biological IDs,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
contains a new function <em>bitr</em>, stands for Biological Id TranslatoR.
NOW,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
is not restricted to GO and KEGG analyses. All kinds of annotation can
be supported via <em>enricher</em> function for hypergeometric test and <em>GSEA</em>
function for gene set enrichment analysis. These two functions are not
yet documented in vignette and will soon.</p>

<h3 id="dose">DOSE</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
was first released in
<a href="http://guangchuangyu.github.io/2011/06/r-package-dose-released/" target="_blank">2011</a>, and has been
published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684" target="_blank">Bioinformatics</a>
on February 2015. An example of applying DOSE for disease ontology
enrichment analysis can be found in <a href="http://www.sciencedirect.com/science/article/pii/S0002929714002626" target="_blank">Genome-wide Identification and Characterization of Fixed Human-Specific Regulatory Regions</a>.
Besides,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
also supports <a href="http://bioinfoblog.it/2015/04/ncg-enrichment-implemented-in-dose/" target="_blank">Network of Cancer Genes database</a>.</p>

<h3 id="reactomepa">ReactomePA</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>
was developed in 2012 when KEGG is no longer available in public domain.
In BioC 3.0, reactome.db has <a href="https://support.bioconductor.org/p/63024/" target="_blank">a lot of issues</a>. Until the early of
this April, Bioconductor restored reactome.db in Bioc 3.0 to Bioc 2.14.
This issue affects ReactomePA several months and was removed in BioC
3.1.</p>

<h3 id="chipseeker">ChIPseeker</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
was developed in
<a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/" target="_blank">2014</a>,
after I found a <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/" target="_blank">bug of ChIPpeakAnno</a>
package. Many new features were added thereafter, including <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
<a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">multiple annotations</a>,
ChIP-peak comparison and data mining with GEO database. ChIPseeker has
been published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145" target="_blank">Bioinformatics</a>.
In this release, several minor bugs were fixed, a new parameter
<em>genomicAnnotationPriority</em> was added in <em>annotatePeak</em> and
<em>plotAvgProf</em> supports estimating confidence interval as shown below:
<img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/dd399ab1d2287d0c7fee1009abf3b1b95553c73a/2015/Screenshot%202015-03-30%2020.09.44.png" alt="" />
This CI estimation feature was contributed by <a href="https://github.com/Puriney" target="_blank">Yan Yun</a>.
<a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
works fine with other ChIP-seq related tools as demonstrated in the
<a href="http://compbio.ucdenver.edu/Hunter_lab/Phang/resources/Tzu-Software/ChIPseq.Analysis.html" target="_blank">ChIPseq Analysis workflow</a>.
It has also been presented in <a href="http://physiology.med.cornell.edu/faculty/elemento/lab/data/courses/2014/CSHLseq/ChIP-seq.pdf" target="_blank">2014 CSHLseq course</a>
and <a href="https://github.com/Bioconductor/UseBioconductor/blob/master/vignettes/F_ChIPSeq.Rmd" target="_blank">2015 Use Bioconductor course</a>.</p>

<h3 id="ggtree">ggtree</h3>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot-2015-02-10-10.27.27.png" alt="" /></p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
was developed on <a href="http://guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/" target="_blank">December 2014</a>,
when I got tired of annotating tree by modifying node labels. This
release (version=1.0.0) is the first release, output of many common
software in evolution were supported and many new features were
introduced in the past few months. Please view the
<a href="http://www.bioconductor.org/packages/3.1/bioc/vignettes/ggtree/inst/doc/ggtree.html" target="_blank">vignette</a>
to find out more and I can guarantee that you won&rsquo;t be disappointed.</p>

<h2 id="citation">Citation</h2>

<p><strong><em>Yu G</em></strong>, Wang LG and He QY<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/31/14/2382" target="_blank">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(14):2382-2383.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="3781087"></div></p>

<p><strong><em>Yu G</em></strong>, Wang L, Han Y and He Q<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118" target="_blank">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="681089"></div></p>

<p><strong><em>G Yu</em></strong>, LG Wang, GR Yan, QY He. <a href="http://bioinformatics.oxfordjournals.org/content/31/4/608" target="_blank">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(4):608-609.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="2788597"></div></p>

<p><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. <a href="http://dx.doi.org/10.1111/2041-210X.12628" target="_blank"><code>doi:10.1111/2041-210X.12628</code></a>.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1111/2041-210X.12628" class="altmetric-embed"></div></p>

<p><strong>Yu G</strong><sup>†</sup>, Li F<sup>†</sup>, Qin Y, Bo X<sup>*</sup>, Wu Y and Wang S<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="100979"></div></p>

<p><strong><em>G Yu</em></strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E#!divAbstract" target="_blank">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong> 2016, 12(2):477-479.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="4796667"></div></p>
    </div>

    


<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/bioc/">BioC</a>
  
  <a class="badge badge-light" href="/tags/gosemsim/">GOSemSim</a>
  
  <a class="badge badge-light" href="/tags/clusterprofiler/">clusterProfiler</a>
  
  <a class="badge badge-light" href="/tags/dose/">DOSE</a>
  
  <a class="badge badge-light" href="/tags/reactomepa/">ReactomePA</a>
  
  <a class="badge badge-light" href="/tags/ggtree/">ggtree</a>
  
  <a class="badge badge-light" href="/tags/chipseeker/">ChIPseeker</a>
  
</div>



    






<div class="media author-card" itemscope itemtype="http://schema.org/Person">
  
  <div class="media-body">
    <h5 class="card-title" itemprop="name"><a href="/"></a></h5>
    
    
    <ul class="network-icon" aria-hidden="true">
      
    </ul>
  </div>
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/r/ggtree/2015_ggtree-in-bioconductor-3-1/">ggtree in Bioconductor 3.1</a></li>
        
        <li><a href="/post/r/dose/2011_package-dose-released/">R package DOSE released</a></li>
        
        <li><a href="/post/r/clusterprofiler/2011_clusterprofiler-bioconductor/">clusterProfiler in Bioconductor 2.8</a></li>
        
        <li><a href="/post/r/ggtree/2015_an-example-of-drawing-beast-tree-using-ggtree/">an example of drawing beast tree using ggtree</a></li>
        
        <li><a href="/post/r/clusterprofiler/2015_david-functional-analysis-with-clusterprofiler/">DAVID functional analysis with clusterProfiler</a></li>
        
      </ul>
    </div>
    

    

    


  </div>
</article>

<div class="container">
  <footer class="site-footer">
  

  <p class="powered-by">
    &copy; {year} Guangchuang YU &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

</div>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha512-+NqPlbbtM1QqiK8ZAo4Yrj2c4lNQoGv8P79DPtKzj++l5jnN39rHA/xsqn8zE9l0uSoxaCdrOgFs6yjyfbBxSg==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha256-VsEqElsCHSGmnmHXGQzvoWjWwoznFSZc6hs7ARLRacQ=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/r.min.js"></script>
        
      

      
      
    

    
    

    
    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
    
    <script src="/js/academic.min.c45052ed2c5a2349273f02aeb6adb37d.js"></script>

    

  </body>
</html>

