<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/douban/">douban</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2016 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">Comparison of clusterProfiler and GSEA-P</a>
      </h1>
      <span class="post-date">Nov 2, 2015 &middot; 5 minute read &middot; <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/gsea">GSEA</a>
      </span>


      
      Thanks @mevers for raising the issue to me and his efforts in benchmarking clusterProfiler. He pointed out two issues: outputs from gseGO and GSEA-P are poorly overlap. pvalues from gseGO are generally smaller and don&rsquo;t show a lot of variation For GSEA analysis, we have two inputs, a ranked gene list and gene set collections. First of all, the gene set collections are very different. The GMT file used in his test is c5.cc.v5.0.symbols.gmt, which is a tiny subset of GO CC, while clusterProfiler used the whole GO CC corpus.
      <a href="http://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">use simplify to remove redundancy of enriched GO terms</a>
      </h1>
      <span class="post-date">Oct 21, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a>
      </span>


      
      To simplify enriched GO result, we can use slim version of GO and use enricher function to analyze. Another strategy is to use GOSemSim to calculate similarity of GO terms and remove those highly similar terms by keeping one representative term. To make this feature available to clusterProfiler users, I develop a simplify method to reduce redundant GO terms from output of enrichGO function. require(clusterProfiler) data(geneList, package=&quot;DOSE&quot;) de &lt;- names(geneList)[abs(geneList) &gt; 2] bp &lt;- enrichGO(de, ont=&quot;BP&quot;) enrichMap(bp) The enrichMap doesn&rsquo;t display the whole picture as we use the default value n=50 to only show 50 highly significant terms.
      <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/">BioC 3.2: NEWS of my BioC packages</a>
      </h1>
      <span class="post-date">Oct 15, 2015 &middot; 3 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/bioc">BioC</a><a class="label" href="http://guangchuangyu.github.io/categories/gosemsim">GOSemSim</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      In BioC 3.2 release, all my packages including GOSemSim, clusterProfiler, DOSE, ReactomePA, and ChIPseeker switch from Sweave to R Markdown for package vignettes. GOSemSim To make it consistent between GOSemSim and clusterProfiler, &lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual, information content data was updated. clusterProfiler Enrichment results may contains terms that are very general (less informative) and we do not want to use them. In this release, we provide dropGO function that can be used to drop selected GO terms or specific level of GO terms.
      <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">ChIPseq data mining with ChIPseeker</a>
      </h1>
      <span class="post-date">Oct 1, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/systems-biology">Systems Biology</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      ChIP-seq is rapidly becoming a common technique and there are a large number of dataset available in the public domain. Results from individual experiments provide a limited understanding of chromatin interactions, as there is many factors cooperate to regulate transcription. Unlike other tools that designed for single dataset, ChIPseeker is designed for comparing profiles of ChIP-seq datasets at different levels. We provide functions to compare profiles of peaks binding to TSS regions, annotation, and enriched functional profiles.
      <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">subsetting data in ggtree</a>
      </h1>
      <span class="post-date">Sep 24, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      Subsetting is commonly used in ggtree as we would like to for example separating internal nodes from tips. We may also want to display annotation to specific node(s)/tip(s). Some software may stored clade information (e.g. bootstrap value) as internal node labels. Indeed we want to manipulate such information and taxa labels separately. In current ggplot2 (version=1.0.1, access date:2015-09-23), it support subset. For instance: library(ggplot2) library(ggtree) tree=read.tree(text=&quot;((A:2,B:2)95:2,(C:2,D:2)100:2);&quot;) ggtree(tree) + geom_text(aes(label=label), subset=.(!isTip), hjust=-.2) But this feature was removed in github version of ggplot2 and will not be available in next release.
      <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">comic phylogenetic tree with ggtree and comicR</a>
      </h1>
      <span class="post-date">Sep 8, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      ggtree applies the concepts of grammar of graphic in phylogenetic tree presentation and make it easy to add multiple layers of text and even figures above a ðŸŒ². Here, I cartoonize a phylogenetic tree generated by ggtree with comicR, which is a funny package to generate comic (xkcd-like) graph in R. Have fun with ggtree and comicR. library(htmltools) library(XML) library(gridSVG) library(comicR) library(ggplot2) library(ggtree) p &lt;- ggtree(rtree(30), layout=&quot;fan&quot;) + geom_text(aes(label=label, angle=angle),
      <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a>
      </h1>
      <span class="post-date">Aug 31, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/subview/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I implemented a function, subview, in ggtree that make it easy to embed a subplot in ggplot. An example is shown below: library(ggplot2) library(ggtree) dd &lt;- data.frame(x=LETTERS[1:3], y=1:3) pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + geom_bar(stat=&quot;identity&quot;, width=1) + coord_polar(theta=&quot;y&quot;) + theme_tree() + xlab(NULL) + ylab(NULL) + theme_transparent() x &lt;- sample(2:9) y &lt;- sample(2:9) width &lt;- sample(seq(0.05, 0.15, length.out=length(x))) height &lt;- width p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank() print(p) for (i in seq_along(x)) { p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i]) print(p) } With this function, we can plot a specific clade and add a subplot of the whole topology; we can also add relative statistic graph above the tree or to a specific node.
      <a href="http://guangchuangyu.github.io/2015/08/subview/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a>
      </h1>
      <span class="post-date">Aug 21, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I found a Bioconductor package, seq2pathway, that can apply functional analysis to NGS data. It consists of two components, seq2gene and gene2pathway. seq2gene converts genomic coordination to genes while gene2pathway performs functional analysis at gene level. I think it would be interesting to incorporate seq2gene with clusterProfiler. But it fail to run due to it call absolute path of python installed in the author&rsquo;s computer. THIS should never be happened in a Bioconductor package since Bioconductor requires every package to have a least one vignette and runnable examples in man files.
      <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a>
      </h1>
      <span class="post-date">Aug 13, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a>
      </span>


      
      The ENCODE consortium has recently published a great paper on Gene Expression from the GTEx dataset. A criticism raised on pubpeer is that the gene ontology enrichment analysis was done with DAVID which has not been updated in the last five years. The result is shown below: It would be interest to see if results change by using clusterProfiler to reproduce GO &amp; KEGG analyses. For GO (BP) analysis: 1. clusterProfiler (194) annotate more genes than DAVID (168).
      <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a>
      </h1>
      <span class="post-date">Aug 7, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      In supplemental file of ChIPseeker paper, I compare distances to TSS reported by several ChIP annotation software, including ChIPseeker, ChIPpeakAnno, HOMER and PeakAnalyzer. Although I found that the chromStart positions in HOMER output have a +1 shift compare to other software, I did not realize this issue since all other software are consistent. Until recently, I found BAM, BCF, BED and PSL formats are using the 0-based coordinate system, while SAM, VCF, GFF and Wiggle formats are using the 1-based coordinate system.
      <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">Read On &rarr;</a>
      

    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        >
        <a href="/" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        ><a href="/">1</a></li>
        
        <li
        class="active"><a href="/page/2/">2</a></li>
        
        <li
        ><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        ><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        ><a href="/page/9/">9</a></li>
        
        <li
        >
        <a href="/page/3/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/9/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    

  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

