<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>G. Yu &middot; G. Yu</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/poole-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-overrides.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hyde-x.css">
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight/zenburn.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Sans:400,400italic,700|Abril+Fatface">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="http://guangchuangyu.github.io/touch-icon-144-precomposed.png">
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  <link href="http://guangchuangyu.github.io/index.xml" rel="alternate" type="application/rss+xml" title="G. Yu &middot; G. Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">
  
</head>
<body class="theme-base-0d">
<div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      
         <img src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=200"
             alt="gravatar" title="G. Yu">
	
       
      <h1>G. Yu</h1>
      <p class="lead">Bioinformatician</p>
    </div>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/">HOME</a></li>
      
      <li class="sidebar-nav-item"><a href="http://guangchuangyu.github.io/link/">Link</a></li>
      
      <li><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
    </ul>

    <ul class="sidebar-nav">
      <li class="sidebar-nav-item">
      <a href="https://github.com/GuangchuangYu"><i class="fa fa-github-square fa-3x"></i></a>
      
      
      
      
      <a href="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook-square fa-3x"></i></a>
      <a href="https://twitter.com/guangchuangyu"><i class="fa fa-twitter-square fa-3x"></i></a>
      
      <a href="http://guangchuangyu.github.io/index.xml" type="application/rss+xml"><i class="fa fa-rss-square fa-3x"></i></a>
      </li>
    </ul>

    

    <p>Copyright &copy; 2015 <a href="http://guangchuangyu.github.io/license/">License</a><br/>
       Powered by <a href="http://gohugo.io">Hugo</a> and <a href="https://github.com/zyro/hyde-x">Hyde-X</a></p>
  </div>
</div>


<div class="content container">
  <div class="posts">
    
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a>
      </h1>
      <span class="post-date">Aug 31, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/subview/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I implemented a function, subview, in ggtree that make it easy to embed a subplot in ggplot. An example is shown below: library(ggplot2) library(ggtree) dd &lt;- data.frame(x=LETTERS[1:3], y=1:3) pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + geom_bar(stat=&quot;identity&quot;, width=1) + coord_polar(theta=&quot;y&quot;) + theme_tree() + xlab(NULL) + ylab(NULL) + theme_transparent() x &lt;- sample(2:9) y &lt;- sample(2:9) width &lt;- sample(seq(0.05, 0.15, length.out=length(x))) height &lt;- width p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank() print(p) for (i in seq_along(x)) { p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i]) print(p) } With this function, we can plot a specific clade and add a subplot of the whole topology; we can also add relative statistic graph above the tree or to a specific node.
      <a href="http://guangchuangyu.github.io/2015/08/subview/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a>
      </h1>
      <span class="post-date">Aug 21, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      I found a Bioconductor package, seq2pathway, that can apply functional analysis to NGS data. It consists of two components, seq2gene and gene2pathway. seq2gene converts genomic coordination to genes while gene2pathway performs functional analysis at gene level. I think it would be interesting to incorporate seq2gene with clusterProfiler. But it fail to run due to it call absolute path of python installed in the author&rsquo;s computer. THIS should never be happened in a Bioconductor package since Bioconductor requires every package to have a least one vignette and runnable examples in man files.
      <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a>
      </h1>
      <span class="post-date">Aug 13, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/enrichment">Enrichment</a><a class="label" href="http://guangchuangyu.github.io/categories/dose">DOSE</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a><a class="label" href="http://guangchuangyu.github.io/categories/reactomepa">ReactomePA</a>
      </span>


      
      The ENCODE consortium has recently published a great paper on Gene Expression from the GTEx dataset. A criticism raised on pubpeer is that the gene ontology enrichment analysis was done with DAVID which has not been updated in the last five years. The result is shown below: It would be interest to see if results change by using clusterProfiler to reproduce GO &amp; KEGG analyses. For GO (BP) analysis: 1. clusterProfiler (194) annotate more genes than DAVID (168).
      <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a>
      </h1>
      <span class="post-date">Aug 7, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      In supplemental file of ChIPseeker paper, I compare distances to TSS reported by several ChIP annotation software, including ChIPseeker, ChIPpeakAnno, HOMER and PeakAnalyzer. Although I found that the chromStart positions in HOMER output have a +1 shift compare to other software, I did not realize this issue since all other software are consistent. Until recently, I found BAM, BCF, BED and PSL formats are using the 0-based coordinate system, while SAM, VCF, GFF and Wiggle formats are using the 1-based coordinate system.
      <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/">ggtree annotate phylogenetic tree with local images</a>
      </h1>
      <span class="post-date">Aug 1, 2015 &middot; 2 minute read &middot; <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      In ggtree, we provide a function annotation_image for annotating phylogenetic tree with images. To demonstrate the usage, I created a tree view from a random tree as shown below: set.seed(2015-08-01) p &lt;- ggtree(rtree(10)) + xlim(0, 5)+ ylim(0, 11) We need a data.frame that contains information of taxa labels and image paths. Here I created such a data.frame containing image files downloaded from phylopic. &gt; img_info V1 V2 1 t1 25f165fa-f279-4f7c-9869-c55be251ffb8.512.png 2 t2 5d7ab302-960b-4db7-9dfd-215175a55906.512.png 3 t3 6fbe723c-3a6b-4d06-8680-bb2a52113df4.512.png 4 t4 d83c02ca-76ed-436b-83ae-7f98d7297be9.512.png 5 t5 ee764929-c865-44f6-b5db-b4e7d5693d1a.512.png Annotating tree with images is simple in ggtree by using annotation_image function.
      <a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a>
      </h1>
      <span class="post-date">Jul 28, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      ChIPseeker is an R package for ChIP peak annotation, comparison and visualization. We have implemented several visualization methods, including vennpie that was designed for viewing annotation overlap as shown below: Not all overlap information can be demonstrated in vennpie. Upset is an effective way to visualize sets and intersections. We import upset function from UpSetR package and implement upsetplot function, that can visualize ChIP annotation overlap directly with output of annotatePeak.
      <a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a>
      </h1>
      <span class="post-date">Jul 13, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a><a class="label" href="http://guangchuangyu.github.io/categories/genomics">Genomics</a><a class="label" href="http://guangchuangyu.github.io/categories/publication">Publication</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/chipseeker">ChIPseeker</a>
      </span>


      
      My R/Bioconductor package, ChIPseeker, published in Bioinformatics. ChIPseeker had been cited by http://www.biomedcentral.com/1471-2164/16/292 and http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short, and was used (not cited) in http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract and http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization G Yu, LG Wang, QY He. Bioinformatics 2015, 31(14):2382-2383 DOI: 10.1093/bioinformatics/btv145 Summary: ChIPseeker is an R package for annotating ChIP-seq data analysis. It supports annotating ChIP peaks and provides functions to visualize ChIP peaks coverage over chromosomes and profiles of peaks binding to TSS regions.
      <a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/07/flip-and-rotate-branches-in-ggtree/">flip and rotate branches in ggtree</a>
      </h1>
      <span class="post-date">Jul 1, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/07/flip-and-rotate-branches-in-ggtree/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
      </span>


      
      flip function is a feature request from ggtree user. It accepts a tree view and two node numbers and exchange the positions of the selected clades. require(ggtree) set.seed(2015-07-01) tr &lt;- rtree(30) p &lt;- ggtree(tr) + geom_text(aes(label=node)) gridExtra::grid.arrange(p, flip(p, 38, 33), ncol=2) It&rsquo;s chainable and all possible position adjustment can be achieved via multiple flip operations. flip(p, 38, 33) %&gt;% flip(53, 58) %&gt;% flip(32, 51) It&rsquo;s tedious to rotate a clade by 180 degree with several flip operations, ggtree provides another function rotate that accepts a selected node and rotate that branch by 180 degree.
      <a href="http://guangchuangyu.github.io/2015/07/flip-and-rotate-branches-in-ggtree/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a>
      </h1>
      <span class="post-date">Jun 23, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/clusterprofiler">clusterProfiler</a>
      </span>


      
      This is a feature request from clusterProfiler user. It&rsquo;s similar to what I implemented in clusterProfiler for comparing biological themes. For comparing different enrichment results, the x-axis represent different gene clusters while for a single enrichment result, the x-axis can be gene count or gene ratio. This is actually similar to traditional barplot, with dot position as bar height and dot color as bar color. But dotplot can represent one more feature nicely by dot size and it can be a good alternative to barplot.
      <a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">Read On &rarr;</a>
      

    </div>
    
    <div class="post">
      <h1 class="post-title">
        <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/">ggtree with funny fonts</a>
      </h1>
      <span class="post-date">Jun 12, 2015 &middot; 1 minute read &middot; <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/#disqus_thread">Comments</a>
      
      <br/>
      <a class="label" href="http://guangchuangyu.github.io/categories/visualization">Visualization</a><a class="label" href="http://guangchuangyu.github.io/categories/r">R</a><a class="label" href="http://guangchuangyu.github.io/categories/ggtree">ggtree</a>
      </span>


      
      showtext is a neat solution to use various types of fonts in R graphs and make it easy to use funny fonts. With showtext, we can draw phylogenetic tree with different types of fonts even with symbolic/icon fonts. require(showtext) font.add.google(&quot;Gochi Hand&quot;, &quot;gochi&quot;) font.add.google(&quot;Rock Salt&quot;, &quot;rock&quot;) link = &quot;http://img.dafont.com/dl/?f=wm_people_1&quot;; download.file(link, &quot;wmpeople1.zip&quot;, mode = &quot;wb&quot;); unzip(&quot;wmpeople1.zip&quot;); font.add(&quot;wmpeople1&quot;, &quot;wmpeople1.TTF&quot;); link = &quot;http://img.dafont.com/dl/?f=emoticons&quot;; download.file(link, &quot;emoticons.zip&quot;, mode = &quot;wb&quot;); unzip(&quot;emoticons.zip&quot;); font.add(&quot;emoticons&quot;, &quot;emoticons.ttf&quot;); showtext.auto() set.seed(2015-05-14) library(ggplot2) library(ggtree) tree &lt;- rtree(30) ggtree(tree, color=&quot;darkgreen&quot;) + geom_tiplab(family=&quot;gochi&quot;) + theme_classic() + theme(axis.text.x=element_text(size=rel(4), family=&quot;emoticons&quot;, color=&quot;firebrick&quot;)) + scale_y_continuous(breaks=seq(0, 30, 4), labels=letters[c(12:17, 20, 21)]) + theme(axis.text.y=element_text(size=rel(4), family=&quot;wmpeople1&quot;, color=&quot;#FDAC4F&quot;)) + annotate(&quot;text&quot;, x=2, y=18, label=&quot;Have fun with ggtree!&quot;, family=&quot;rock&quot;, angle=30, size=12, color=&quot;steelblue&quot;) download.file(&quot;http://dl.dafont.com/dl/?f=people_freak&quot;, &quot;people_freak.zip&quot;, mode=&quot;wb&quot;) unzip(&quot;people_freak.zip&quot;) font.add(&quot;people_freak&quot;, &quot;People_freak.ttf&quot;) ggtree(rtree(10)) + geom_text(aes(label=node, color=isTip), family=&quot;people_freak&quot;, hjust=0) Funny fonts can be downloaded from 1, 2 and 3.
      
      

    </div>
    
    
    
    <ul class="pagination">
        
        <li>
            <a href="/" aria-label="First"><span aria-hidden="true">&laquo;&laquo;</span></a>
        </li>
        
        <li
        >
        <a href="/" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
        </li>
        
        <li
        ><a href="/">1</a></li>
        
        <li
        class="active"><a href="/page/2/">2</a></li>
        
        <li
        ><a href="/page/3/">3</a></li>
        
        <li
        ><a href="/page/4/">4</a></li>
        
        <li
        ><a href="/page/5/">5</a></li>
        
        <li
        ><a href="/page/6/">6</a></li>
        
        <li
        ><a href="/page/7/">7</a></li>
        
        <li
        ><a href="/page/8/">8</a></li>
        
        <li
        >
        <a href="/page/3/" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
        </li>
        
        <li>
            <a href="/page/8/" aria-label="Last"><span aria-hidden="true">&raquo;&raquo;</span></a>
        </li>
        
    </ul>
    

  </div>
</div>


<script type="text/javascript">
var disqus_shortname = "gcyu";
(function () {
    var s = document.createElement('script'); s.async = true;
    s.type = 'text/javascript';
    s.src = '//' + disqus_shortname + '.disqus.com/count.js';
    (document.getElementsByTagName('HEAD')[0] || document.getElementsByTagName('BODY')[0]).appendChild(s);
}());
</script>

<script src="http://guangchuangyu.github.io/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>


<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

