<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/categories/r/">

    <title>R | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2017/05/enrichdavid%E6%94%AF%E6%8C%81%E8%87%AA%E5%AE%9A%E4%B9%89%E8%83%8C%E6%99%AF">enrichDAVID支持自定义背景</a></h2>
        <div class="article-style"><p><p>虽然我不喜欢DAVID，但很多用户喜欢，所以clusterProfiler也支持了，最近github上又有人<a href="https://github.com/GuangchuangYu/clusterProfiler/issues/87">要求支持自定义背景</a>。</p>

<blockquote>
<p>Dear Guangchuang,</p>

<p>I am using clusterProfiler in Kegg pathway enrichment analysis, it is useful and nice. I am looking for a function which accept background and has ability to deal with Ensembl gene ID.</p>

<p>In a function enrichDAVID it can takes ensembl gene id as an input format, but not allows to enter background.
<code>enrichDAVID(gene = gene, idType=&quot;ENSEMBL_GENE_ID&quot;, annotation=&quot;KEGG_PATHWAY&quot;, species= &quot;hsa&quot;)</code></p>

<p>Other command enrichKEGG has a background input but only takes entrez gene id,
<code>enrichKEGG(gene, organism = &quot;hsa&quot;, keyType = &quot;kegg&quot;, universe)</code></p>

<p>I have tried to convert my ensembl gene IDs to entrez gene id, but some ensembl gene IDs represent more than one entrez gene ID.
I downloaded KEGG pathway dataset to apply fisher exact test. however, genes are in entrez ID and i am still dont know how to convert.</p>
</blockquote>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2017/01/convert-graphic-object-to-tree-object-using-treeio">convert graphic object to tree object using treeio</a></h2>
        <div class="article-style"><p><p>I have splitted <code>ggtree</code> to 2 packages, <code>treeio</code> and <code>ggtree</code>. Now <code>ggtree</code> is mainly focus on visualization and annotation, while <code>treeio</code> focus on parsing and exporting tree files. Here is a welcome message from <code>treeio</code> that you can convert <code>ggtree</code> output to tree object which can be exported as newick or nexus file if you want.</p>

<p>Thanks to <code>ggplot2</code>, output of <code>ggtree</code> is actually a <code>ggplot</code> object. The <code>ggtree</code> object can be rendered as graph by <code>print</code> method, but internally it is an object that contains data. <code>treeio</code> defines <code>as.phylo</code> and <code>as.treedata</code> to convert <code>ggtree</code> object to <code>phylo</code> or <code>treedata</code> object.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/dotplot-for-gsea-result">dotplot for GSEA result</a></h2>
        <div class="article-style"><p><p><img src="https://guangchuangyu.github.io/blog_images/Bioconductor/DOSE/gseaplot.png" alt="" /></p>

<p>For GSEA analysis, we are familar with the above figure which shows the running enrichment score. But for most of the software, it lack of visualization method to summarize the whole enrichment result.</p>

<p>In <a href="https://guangchuangyu.github.io/DOSE">DOSE</a> (and related tools including <a href="https://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a>, <a href="https://guangchuangyu.github.io/ReactomePA">ReactomePA</a> and <a href="https://guangchuangyu.github.io/meshes">meshes</a>), we provide <code>enrichMap</code> and <code>cnetplot</code> to summarize GSEA result.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/add-layer-to-specific-panel-of-facet_plot-output">add layer to specific panel of facet_plot output</a></h2>
        <div class="article-style"><p><p>This is a question from <a href="https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/bioc-ggtree/uPgksYWkwCA/oMtHKW7KEgAJ">ggtree google group</a>:</p>

<blockquote>
<p>Dear ggtree team,</p>

<p>how can I apply a geom_xxx to only one facet panel? For example if i want to get <code>geom_hline(yintersect=1:30)</code> or a <code>geom_text()</code> in the dot panel?
I cant see the <code>facet_grid(. ~ var)</code> function call, so I don&rsquo;t know which subsetting to use. I have already read <a href="http://stackoverflow.com/questions/29873155/geom-text-and-facets-not-working">http://stackoverflow.com/questions/29873155/geom-text-and-facets-not-working</a></p>

<pre><code>  tr &lt;- rtree(30)
  
  d1 &lt;- data.frame(id=tr$tip.label, val=rnorm(30, sd=3))
  p &lt;- ggtree(tr)
  
  p2 &lt;- facet_plot(p, panel=&quot;dot&quot;, data=d1, geom=geom_point, aes(x=val), color='firebrick')
  d2 &lt;- data.frame(id=tr$tip.label, value = abs(rnorm(30, mean=100, sd=50)))
  
  p3 &lt;- facet_plot(p2, panel='bar', data=d2, geom=geom_segment, aes(x=0, xend=value, y=y, yend=y), size=3, color='steelblue') + theme_tree2()
</code></pre>

<p>Thanks!
Andreas</p>
</blockquote>

<p>If this can be done, we can create even more <a href="https://guangchuangyu.github.io/2016/12/ggtree-version-of-plottree/">comprehensive tree plots</a>.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/ggtree-version-of-plottree">ggtree version of plotTree</a></h2>
        <div class="article-style"><p><p><a href="https://holtlab.net/2015/10/03/plotting-trees-data/">PLOTTING TREES + DATA</a> is difficult. <a href="http://holtlab.net/">@DrKatHolt</a> developed <a href="https://github.com/katholt/plotTree">plotTree</a> (R and Python scripts) to visualize associated data with trees, <em>e.g.</em> heatmap, horizontal bar <em>etc</em>.</p>

<p>I reproduced the examples presented in the <a href="https://github.com/katholt/plotTree">plotTree</a> repo using <a href="https://guangchuangyu.github.io">ggtree</a>. Source code is freely available in <a href="https://github.com/GuangchuangYu/plotTree-ggtree">https://github.com/GuangchuangYu/plotTree-ggtree</a>.</p>

<p>Here are the outputs produced by <code>ggtree</code>:</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/edge-coloring-with-user-data">Edge coloring with user data</a></h2>
        <div class="article-style"><p><p>Coloring edges in ggtree is quite easy, as we can map the color to numerical or categorical values via the <code>aes(color=VAR)</code> syntax. For user&rsquo;s own data, it is also easy as <code>ggtree</code> provide the <code>%&lt;+%</code> operator to attach user data.</p>

<p>But as it seems not so obviously for <code>ggtree</code> users, see question <a href="https://groups.google.com/forum/#!topic/bioc-ggtree/4GgivKqVjB8">1</a>, <a href="https://github.com/GuangchuangYu/ggtree/issues/76">2</a>, and <a href="https://github.com/GuangchuangYu/ggtree/issues/105">3</a>, I will demonstrate how to color edges using user data here.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/scatterpie-for-plotting-pies-on-ggplot">scatterpie for plotting pies on ggplot</a></h2>
        <div class="article-style"><p><p><img src="https://guangchuangyu.github.io/blog_images/R/scatterpie_files/so.png" alt="" /></p>

<p>Plotting pies on ggplot/ggmap is not an easy task, as ggplot2 doesn&rsquo;t provide native pie geom. The pie we produced in ggplot2 is actually a barplot transform to polar coordination. This make it difficult if we want to produce a map like the above screenshot, which was posted by Tyler Rinker, the author of R package pacman.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/11/align-genomic-features-with-phylogenetic-tree">align genomic features with phylogenetic tree</a></h2>
        <div class="article-style"><p><p><a href="https://www.biostars.org/p/218591/">A question</a> on biostars asking how to generate the following figure:</p>

<p><img src="https://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/ggtree_ggbio/218591.png" alt="" /></p>

<p>This can be quite easy to implement in <a href="https://guangchuangyu.github.io">ggtree</a>, I can write a <code>geom</code> layer to layout the alignment. As <a href="https://bioconductor.org/packages/ggbio/">ggbio</a> already provides many <code>geom</code> for genomic data and I don&rsquo;t want to re-invent the wheel, I decided to try <code>ggtree+ggbio</code>. This is also the beauty of <code>R</code> that packages complete each others.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output">showCategory parameter for visualizing compareCluster output</a></h2>
        <div class="article-style"><p><blockquote>
<p>I am using dotplot() to visualize results from enrichGO(), enrichDO(), enricher() and compareCluster() in clusterProfiler R package. When specifying showCategory, I get the right number of categories except with the results of compareCluser().</p>

<p>In my case, I use compareCluster() on a list of 3 elements:</p>

<pre><code>str(ClusterList)
List of 3
 $ All : chr [1:1450] &quot;89886&quot; &quot;29923&quot; &quot;100132891&quot; &quot;101410536&quot; ...
 $ g1  : chr [1:858] &quot;89886&quot; &quot;29923&quot; &quot;100132891&quot; &quot;101410536&quot; ...
 $ g2: chr [1:592] &quot;5325&quot; &quot;170691&quot; &quot;29953&quot; &quot;283392&quot; ...
CompareGO_BP=compareCluster(ClusterList, fun=&quot;enrichGO&quot;, pvalueCutoff=0.01, pAdjustMethod=&quot;BH&quot;, OrgDb=org.Hs.eg.db,ont=&quot;BP&quot;,readable=T)

dotplot(CompareGO_BP, showCategory=10, title=&quot;GO - Biological Process&quot;)
</code></pre>

<p>I ask for 10 categories, but I get 15 categories in All, 8 categories in g1 and 12 categories in g2. None of the categories, neither the sum of the categories are 10&hellip;</p>

<p>Is the option showCategory working in the case of comparison? Am I missing something here?</p>

<p>And which categories precisely will it plot? the most significant whatever my 3 cases or the most significant of each case?</p>
</blockquote>

<p>The question was <a href="https://support.bioconductor.org/p/88965/">posted</a> in Bioconductor support site. It seems quite confusing and I think I need to write a post to clarify it.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/10/xlim_tree-set-x-axis-limits-for-only-tree-panel">xlim_tree: set x axis limits for only Tree panel</a></h2>
        <div class="article-style"><p><p>A <a href="https://guangchuangyu.github.io/ggtree">ggtree</a> user recently asked me the following question in <a href="https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!msg/bioc-ggtree/6ec5krxpBMc/sd6jr53KBQAJ">google group</a>:</p>

<blockquote>
<p>I try to plot long tip labels in ggtree and usually adjust them using xlim(), however when creating a facet_plot xlim affects all plots and minimizes them.</p>

<p>Is it possible to work around this and only affect the tree and it&rsquo;s tip labels leaving the other plots in facet_plot unaffected?</p>
</blockquote>

<p>This is indeed a desire feature, as <code>ggplot2</code> can&rsquo;t automatically adjust <code>xlim</code> for text since the units are in two different spaces (data and pixel).</p>
</p></div>

    </div>
    

    
<nav>
    <ul class="pager">
        
        
        <li><a href="https://guangchuangyu.github.io//categories/r/page/2/">Next »</a></li>
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

