<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>BioC 3.2: NEWS of my BioC packages</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            BioC 3.2: NEWS of my BioC packages 
          </h1>
          <p class="meta">Oct 15, 2015 - 3 minute read -  <a href="http://guangchuangyu.github.io/2015/10/bioc-3.2-news-of-my-bioc-packages/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          

<p>In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim:35c90aeebec9a933fea1dafd42584f96">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.</p>

<h3 id="clusterprofiler:35c90aeebec9a933fea1dafd42584f96">clusterProfiler</h3>

<p>Enrichment results may contains terms that are very general (less
informative) and we do not want to use them. In this release, we provide
<em>dropGO</em> function that can be used to drop selected GO terms or specific
level of GO terms. It can be applied to output from both <em>enrichGO</em> and
<em>compareCluster</em>. This is a feature request from
<a href="https://github.com/GuangchuangYu/clusterProfiler/issues/26">@ahorvath</a>.</p>

<p>Another feature request is to visualize GO enrichment result with GO
topology. I implement <em>plotGOgraph</em> function by extending
<a href="http://www.bioconductor.org/packages/topGO">topGO</a> to support output of
both <em>enrichGO</em> or <em>gseGO</em>.</p>

<p><em><a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot</a></em>
is another feature request and was implemented in
<a href="http://www.bioconductor.org/packages/DOSE">DOSE</a> as a general function
for visualize enrichment result.
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
import this function.</p>

<p><em>merge_result</em> function was implemented for merging enrichment results
and then the results can be visualized simultaneously for comparison.
This function was developed for comparing <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment of GTEx
paper</a>.
An example of comparing results from
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
and DAVID can be found in
<a href="https://github.com/GuangchuangYu/enrichment4GTEx_clusterProfiler/blob/master/GO_BP.md">github</a>.</p>

<p>A section &lsquo;Functional analysis of NGS data&rsquo; was added in the vignette.
<a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">The blog
post</a>
illustrated using <em>enricher</em> and <em>GSEA</em> function to analyze user defined
annotation.</p>

<h3 id="dose:35c90aeebec9a933fea1dafd42584f96">DOSE</h3>

<p>Network of Cancer Genes data was updated to version 4.9.0, contributed
by <a href="https://github.com/GuangchuangYu/DOSE/pull/8">@dalloliogm</a>.</p>

<p><em>dotplot</em> function was implemented for visualizing enrichment result and
was imported to
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
and <a href="http://www.bioconductor.org/packages/ReactomePA">ReactomePA</a>.</p>

<p>With new release of ChIPseeker, DOSE can be applied to <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">NGS
data</a>.</p>

<p>Several bugs were removed and small improvement were incorporated.</p>

<h3 id="reactomepa:35c90aeebec9a933fea1dafd42584f96">ReactomePA</h3>

<p>Now fly (Drosophila melanogaster) is supported.</p>

<p><em>viewPathway</em> supports other organisms, which was contributed by
<a href="https://github.com/GuangchuangYu/ReactomePA/pull/1">@vladpetyuk</a>.</p>

<p>Import <em>dotplot</em> from DOSE and add &lsquo;Pathway analysis of NGS data&rsquo;
section in vignette.</p>

<h3 id="chipseeker:35c90aeebec9a933fea1dafd42584f96">ChIPseeker</h3>

<p>Implement
<em><a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">seq2gene</a></em>
function for linking genomic regions to genes by many-to-many mapping.
With this function, enrichment analysis of
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/DOSE">DOSE</a> and
<a href="http://www.bioconductor.org/packages/ReactomePA">ReactomePA</a> can be
applied to NGS data.</p>

<p><a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot</a>
function is implemented for viewing overlap of ChIP annotation.</p>

<p>Better implementation of internal function <em>getFirstHitIndex</em>
contributed by
<a href="https://support.bioconductor.org/p/70432/#70545">@hpages</a>. Now
annotating a BED file of size about 1GB only take several minutes.</p>

<p>An issue of BED file that using <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">the 0-based coordinate
system</a> was fixed.</p>

<p>GEO data information was updated, now ChIPseeker contains more than
18,000 BED file information. User can compare their own data with those
deposited in GEO to <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">identify co-occurrence binding
proteins</a>
that maybe cooperated with the one we are interested in.</p>

<h3 id="ggtree:35c90aeebec9a933fea1dafd42584f96">ggtree</h3>

<p>My major focus in the past half year is on developing
<a href="http://www.bioconductor.org/packages/ggtree">ggtree</a>.
<a href="http://www.bioconductor.org/packages/ggtree">ggtree</a> has a lot of
changes. Some of the updates can be found in <a href="http://bioconductor.org/help/newsletters/2015_October/">Bioconductor
Newsletter</a>
(Oct. 2015). In addition, I would like to emphasize the
<em><a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a></em> function which works
fine with all the ggplot objects and was applied to <a href="http://stackoverflow.com/questions/10368180/plotting-pie-graphs-on-map-in-ggplot/32380396#32380396">plot pie graph on a
map</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ggtree">ggtree</a> can be fun by
using <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/">symbolic fonts</a>, <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">cartoonize the tree</a>
or annotating the tree <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">with
emoji</a>.</p>

<p>There are many updates that based on the github version of ggplot2.
Since the new features of ggplot2 are currently not available in CRAN. I
do not commit the change to Bioconductor and will introduce these new
feature to ggtree user when new version of ggplot2 is available in CRAN.</p>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Oct 15, 2015</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>  <a class="category" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>  <a class="category" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>  <a class="category" href="http://guangchuangyu.github.io/tags/dose">DOSE</a>  <a class="category" href="http://guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>  <a class="category" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>  <a class="category" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/" title="ChIPseq data mining with ChIPseeker">ChIPseq data mining with ChIPseeker</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/" title="use simplify to remove redundancy of enriched GO terms">use simplify to remove redundancy of enriched GO terms</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    <h1>Header in sidebar</h1>
    <p>This is the rest of the sidebar - here's a link to <a target="_blank" href="http://google.com" title="Google">Google</a>.</p>
    <p>If links are set in config file, there will be links to keybase/github/twitter/linkedin etc.</p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    
    
    
    
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    
     

    
    
    </li>
  </ul>

  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/ggtree-supports-phylip-tree-format/">ggtree supports phylip tree format</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/label-edge-number-in-ggtree/">label edge number in ggtree</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/annotate-a-phylogenetic-tree-with-insets/">Annotate a phylogenetic tree with insets</a>
      </li>
      
      <li class="post">
        <a href="/2016/01/go-analysis-using-clusterprofiler/">GO analysis using clusterProfiler</a>
      </li>
      
    </ul>
  </section>

</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


