<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/tags/clusterprofiler/">

    <title>Clusterprofiler | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/05/news-of-my-bioc-packages">NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/MISC/ke.jpg" width=600></p>

<p>Today is my birthday and it happened to be the release day of Bioconductor 3.3. It&rsquo;s again the time to reflect what I&rsquo;ve done in the past year.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA">ReactomePA</a></li>
</ul>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler">convert biological ID with KEGG API using clusterProfiler</a></h2>
        <div class="article-style"><p><h2 id="bitr-kegg:ea0718166651864960b492e0d7486a4f">bitr_kegg</h2>

<p><code>clusterProfiler</code> can convert biological IDs using <code>OrgDb</code> object via the <code>bitr</code> function. Now I implemented another function, <code>bitr_kegg</code> for converting IDs through KEGG API.</p>

<pre><code>library(clusterProfiler)
data(gcSample)
hg &lt;- gcSample[[1]]
head(hg)

## [1] &quot;4597&quot;  &quot;7111&quot;  &quot;5266&quot;  &quot;2175&quot;  &quot;755&quot;   &quot;23046&quot;

eg2np &lt;- bitr_kegg(hg, fromType='kegg', toType='ncbi-proteinid', organism='hsa')

## Warning in bitr_kegg(hg, fromType = &quot;kegg&quot;, toType = &quot;ncbi-proteinid&quot;,
## organism = &quot;hsa&quot;): 3.7% of input gene IDs are fail to map...

head(eg2np)

##     kegg ncbi-proteinid
## 1   8326      NP_003499
## 2  58487   NP_001034707
## 3 139081      NP_619647
## 4  59272      NP_068576
## 5    993      NP_001780
## 6   2676      NP_001487

np2up &lt;- bitr_kegg(eg2np[,2], fromType='ncbi-proteinid', toType='uniprot', organism='hsa')

head(np2up)

##   ncbi-proteinid uniprot
## 1      NP_005457  O75586
## 2      NP_005792  P41567
## 3      NP_005792  Q6IAV3
## 4      NP_037536  Q13421
## 5      NP_006054  O60662
## 6   NP_001092002  O95398
</code></pre>

<p>The ID type (both fromType &amp; toType) should be one of &lsquo;kegg&rsquo;, &lsquo;ncbi-geneid&rsquo;, &lsquo;ncbi-proteinid&rsquo; or &lsquo;uniprot&rsquo;. The &lsquo;kegg&rsquo; is the primary ID used in KEGG database. The data source of KEGG was from NCBI. A rule of thumb for the &lsquo;kegg&rsquo; ID is <code>entrezgene</code> ID for eukaryote species and <code>Locus</code> ID for prokaryotes.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis">KEGG Module Enrichment Analysis</a></h2>
        <div class="article-style"><p><blockquote>
<p>KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for annotation and biological interpretation of sequenced genomes. There are four types of KEGG modules:</p>

<ul>
<li>pathway modules – representing tight functional units in KEGG metabolic pathway maps, such as M00002 (Glycolysis, core module involving three-carbon compounds)</li>
<li>structural complexes – often forming molecular machineries, such as M00072 (Oligosaccharyltransferase)</li>
<li>functional sets – for other types of essential sets, such as M00360 (Aminoacyl-tRNA synthases, prokaryotes)</li>
<li>signature modules – as markers of phenotypes, such as M00363 (EHEC pathogenicity signature, Shiga toxin)</li>
</ul>
</blockquote>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler">GO analysis using clusterProfiler</a></h2>
        <div class="article-style"><p><p><code>clusterProfiler</code> supports over-representation test and gene set
enrichment analysis of Gene Ontology. It supports GO annotation from
OrgDb object, GMT file and user&rsquo;s own data.</p>

<h2 id="support-many-species:bdc631894cab1e81ad84e5406d79b479">support many species</h2>

<p>In github version of clusterProfiler, <code>enrichGO</code> and <code>gseGO</code> functions
removed the parameter <em>organism</em> and add another parameter <em>OrgDb</em>, so
that any species that have <code>OrgDb</code> object available can be analyzed in
<strong>clusterProfiler</strong>. Bioconductor have already provide OrgDb for about
20 species, see
<a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">http://bioconductor.org/packages/release/BiocViews.html#___OrgDb</a>, and
users can build <code>OrgDb</code> via <code>AnnotationHub</code>.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/11/comparison-of-clusterprofiler-and-gsea-p">Comparison of clusterProfiler and GSEA-P</a></h2>
        <div class="article-style"><p><p>Thanks <a href="https://github.com/mevers">@mevers</a> for raising <a href="https://github.com/GuangchuangYu/clusterProfiler/issues/31">the issue</a> to me
and his efforts in benchmarking
<a href="https://github.com/GuangchuangYu/clusterProfiler">clusterProfiler</a>.</p>

<p>He pointed out two issues:</p>

<ul>
<li>outputs from gseGO and GSEA-P are poorly overlap.</li>
<li>pvalues from gseGO are generally smaller and don&rsquo;t show a lot of
variation</li>
</ul>

<p>For GSEA analysis, we have two inputs, a ranked gene list and gene set
collections.</p>

<p>First of all, the gene set collections are very different. The GMT file
used in his test is <em>c5.cc.v5.0.symbols.gmt</em>, which is a tiny subset of
GO CC, while clusterProfiler used the whole GO CC corpus.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms">use simplify to remove redundancy of enriched GO terms</a></h2>
        <div class="article-style"><p><p>To simplify enriched GO result, we can use slim version of GO and use
<em><a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">enricher</a></em>
function to analyze.</p>

<p>Another strategy is to use
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> to calculate
similarity of GO terms and remove those highly similar terms by keeping
one representative term. To make this feature available to
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
users, I develop a <em>simplify</em> method to reduce redundant GO terms from
output of <em>enrichGO</em> function.</p>

<pre><code>require(clusterProfiler)
data(geneList, package=&quot;DOSE&quot;)
de &lt;- names(geneList)[abs(geneList) &gt; 2]
bp &lt;- enrichGO(de, ont=&quot;BP&quot;)
enrichMap(bp)
</code></pre>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/10/news-of-my-bioc-packages">NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p>In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim:35c90aeebec9a933fea1dafd42584f96">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data">functional enrichment analysis with NGS data</a></h2>
        <div class="article-style"><p><p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" /></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper">functional enrichment for GTEx paper</a></h2>
        <div class="article-style"><p><p>The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" /></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result">dotplot for enrichment result</a></h2>
        <div class="article-style"><p><p>This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.</p>
</p></div>
    </div>
    

    
<nav>
    <ul class="pager">
        
        
        <li><a href="https://guangchuangyu.github.io//tags/clusterprofiler/page/2/">Next »</a></li>
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

