<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/">

  

  <title>Posts | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Posts</h1>

  

  
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/">viewing and annotating phylogenetic tree with ggtree</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-12-21 08:00:08 &#43;0800 HKT" itemprop="datePublished">
      Sun, Dec 21, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/evolution">Evolution</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>When I need to annotate nucleotide substitutions in the phylogenetic
tree, I found that all the software are designed to display the tree but
not annotating it. Some of them may support annotating the tree with
specific data such as bootstrap values, but they are restricted to a few
supported data types. It is hard/impossible to inject user specific
data.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/12/2014-12-21-145753_579x436_scrot.png" alt="" />
</p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/12/auto-complete-in-ess/">auto-complete in ESS</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-12-07 10:22:35 &#43;0800 HKT" itemprop="datePublished">
      Sun, Dec 7, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/software">Software</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/emacs">Emacs</a
    >, 
    
    <a href="/tags/ess">ESS</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      Auto complete is good, it can save you times in typing and prevent typo sometimes.
RStudio now supports function arguments in auto complete. ESS&rsquo;s auto complete is more advance, it supports help page.
We should use &lsquo;ess-use-auto-complete&rsquo; to activate auto complete in ESS buffers.
(setq ess-use-auto-complete t)  The effect is quite amazing. Auto-complete extension is needed for ESS, we need to install the auto-complete extension and load it at Emacs startup.
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/">proper use of GOSemSim</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-11-23 02:27:23 &#43;0800 HKT" itemprop="datePublished">
      Sun, Nov 23, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/research">Research</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/gosemsim">GOSemSim</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>One day, I am looking for R packages that can analyze PPI and after
searching, I found the
<a href="http://cran.r-project.org/web/packages/ppiPre/index.html" target="_blank">ppiPre</a>
package in CRAN.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/11/2014-11-22-215959_895x256_scrot.png" alt="" /></p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/11/hello-yosemite/">hello yosemite</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-11-14 04:34:48 &#43;0800 HKT" itemprop="datePublished">
      Fri, Nov 14, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/bsd">BSD</a
    >, 
    
    <a href="/categories/mac-os">Mac OS</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/osx">OSX</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><img src="http://guangchuangyu.github.io/blog_images/2014/11/Screenshot-2014-11-11-14.07.45.png" alt="" /></p>

<p>Installing OS is painful, you need to re-install all the software and
configure them to the way you want. We don&rsquo;t want to wast time in doing
this. To prevent doing this dirty job, we perform an upgrade install
instead of clean install. We all have experience of upgrading Windows
sucks, <a href="http://osxdaily.com/2014/10/17/os-x-yosemite-installation-stuck/" target="_blank">same as OS X</a>.
All the issues you have in old system will be remained, and sometimes
new issues will be introduced in the process of upgrading. The system
will be slower compare to clean install one.

Life is short, we want a clean system without wasting time in setting
software, but how can it be. The answer is yes and no. Firstly, you
should have two partitions, one for OS and the other one for your data.
Only in this way, you can formatting the system partition and keep your
data untouch.</p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/10/sir-model-of-epidemics/">SIR Model of Epidemics</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-10-13 05:57:22 &#43;0800 HKT" itemprop="datePublished">
      Mon, Oct 13, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/epidemics">Epidemics</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/sir">SIR</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>The SIR model divides the population to three compartments:
<strong>S</strong>usceptible, <strong>I</strong>nfected and <strong>R</strong>ecovered. If the disease dynamic
fits the SIR model, then the flow of individuals is one direction from
the susceptible group to infected group and then to the recovered group.
All individuals are assumed to be identical in terms of their
susceptibility to infection, infectiousness if infected and mixing
behaviour associated with disease transmission.</p>

<p>We defined: $S_t$ = the number of susceptible individuals at time t</p>

<p>$I_t$ = the number of infected individuals at time t</p>

<p>$R_t$ = the number of recovered individuals at time t</p>

<p>Suppose on average every infected individual will contact $\gamma$
person, and $\kappa$ percent of these $\gamma$ person will be
infected. Then on average there are $\beta = \gamma \times \kappa$
person will be infected an infected individual.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">multiple annotation in ChIPseeker</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-10-01 13:00:33 &#43;0800 HKT" itemprop="datePublished">
      Wed, Oct 1, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/genomics">Genomics</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <h3 id="nearest-gene-annotation">Nearest gene annotation</h3>

<p>Almost all annotation software calculate the distance of a peak to the
nearest TSS and assign the peak to that gene. This can be misleading, as
binding sites might be located between two start sites of different
genes or hit different genes which have the same TSS location in the
genome.</p>

<p>The function <em>annotatePeak</em> provides option to assign genes with a max
distance cutoff and all genes within this distance were reported for
each peak.
</p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/09/install-454-gs-data-analysis-software-on-ubuntu/">install 454 GS Data Analysis Software on ubuntu</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-09-04 15:38:44 &#43;0800 HKT" itemprop="datePublished">
      Thu, Sep 4, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/linux">Linux</a
    >, 
    
    <a href="/categories/software">Software</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/454">454</a
    >, 
    
    <a href="/tags/ubuntu">ubuntu</a
    >, 
    
    <a href="/tags/redhat">redhat</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      Usually Roche&rsquo;s installer is a catastrophe, they only provides rpm packages of the software for 454 GS FLX (version 2.9). Although the package contains setup.sh, the script is useless since it is actually a binary payload.
I run the setup.sh, and it throw error of not finding /sbin/lspci. In debian derived distribution, lspci command is located in /bin folder. This issue is easy to solve by adding a soft link to /sbin/lspci.
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">why clusterProfiler fails</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-08-07 05:46:35 &#43;0800 HKT" itemprop="datePublished">
      Thu, Aug 7, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/bioinformatics">Bioinformatics</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><a href="https://github.com/kaji331" target="_blank">@kaji331</a>
compared <code>cluserProfiler</code> with <code>GeneAnswers</code> and found that clusterProfiler
gives larger p values.</p>

<p>It eventually came out that he passed the input
gene as numeric vector, which was supposed to be character and he used an <strong>old</strong> version of <code>clusterProfiler</code> which didn&rsquo;t use <code>as.character</code> to coerce the input</p>

<p>But his comment forces me to test it.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/08/enrichment-map/">enrichment map</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-08-03 12:48:15 &#43;0800 HKT" itemprop="datePublished">
      Sun, Aug 3, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      In <a href="http://www.plob.org/" target="_blank">PLOB</a>&rsquo;s QQ group, someone asked how to change
the color of enrichment map in <a href="http://cytoscape.org/" target="_blank">Cytoscape</a>. I am
very curious how enrichment map can helps to interpret enrichment
results. It took me 2 hours to implement it using R and I am surprised
that the enrichment map is better than anticipated.

<img src="http://guangchuangyu.github.io/blog_images/2014/08/Screenshot-2014-07-30-22.20.07.png" alt="" />
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">visualization methods in ChIPseeker</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2014-04-30 06:33:41 &#43;0800 HKT" itemprop="datePublished">
      Wed, Apr 30, 2014
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/genomics">Genomics</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >, 
    
    <a href="/tags/chipseq">ChIPseq</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes">ChIP peaks over Chromosomes</h3>

<pre><code>&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>

      
    </p></div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/post/page/7/">&laquo; Previous</a></li>
    
    
    <li><a href="/post/page/9/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    

    
    

  </body>
</html>

