<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.21" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="PhD student">

  
  
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/">

  

  <title>[Bioc 32] NEWS of my BioC packages | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#top">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  

  <div class="article-container">
    <h1 itemprop="name">[Bioc 32] NEWS of my BioC packages</h1>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-10-15 16:13:57 &#43;0800 CST" itemprop="datePublished">
      Thu, Oct 15, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  
  

  

</div>

    <div class="article-style" itemprop="articleBody">
      <p>In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.
</p>

<h3 id="clusterprofiler">clusterProfiler</h3>

<p>Enrichment results may contains terms that are very general (less
informative) and we do not want to use them. In this release, we provide
<em>dropGO</em> function that can be used to drop selected GO terms or specific
level of GO terms. It can be applied to output from both <em>enrichGO</em> and
<em>compareCluster</em>. This is a feature request from
<a href="https://github.com/GuangchuangYu/clusterProfiler/issues/26">@ahorvath</a>.</p>

<p>Another feature request is to visualize GO enrichment result with GO
topology. I implement <em>plotGOgraph</em> function by extending
<a href="http://www.bioconductor.org/packages/topGO">topGO</a> to support output of
both <em>enrichGO</em> or <em>gseGO</em>.</p>

<p><em><a href="http://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot</a></em>
is another feature request and was implemented in
<a href="http://www.bioconductor.org/packages/DOSE">DOSE</a> as a general function
for visualize enrichment result.
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
import this function.</p>

<p><em>merge_result</em> function was implemented for merging enrichment results
and then the results can be visualized simultaneously for comparison.
This function was developed for comparing <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment of GTEx paper</a>.
An example of comparing results from
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
and DAVID can be found in
<a href="https://github.com/GuangchuangYu/enrichment4GTEx_clusterProfiler/blob/master/GO_BP.md">github</a>.</p>

<p>A section &lsquo;Functional analysis of NGS data&rsquo; was added in the vignette.
<a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">The blog post</a>
illustrated using <em>enricher</em> and <em>GSEA</em> function to analyze user defined
annotation.</p>

<h3 id="dose">DOSE</h3>

<p>Network of Cancer Genes data was updated to version 4.9.0, contributed
by <a href="https://github.com/GuangchuangYu/DOSE/pull/8">@dalloliogm</a>.</p>

<p><em>dotplot</em> function was implemented for visualizing enrichment result and
was imported to
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
and <a href="http://www.bioconductor.org/packages/ReactomePA">ReactomePA</a>.</p>

<p>With new release of ChIPseeker, DOSE can be applied to <a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">NGS data</a>.</p>

<p>Several bugs were removed and small improvement were incorporated.</p>

<h3 id="reactomepa">ReactomePA</h3>

<p>Now fly (Drosophila melanogaster) is supported.</p>

<p><em>viewPathway</em> supports other organisms, which was contributed by
<a href="https://github.com/GuangchuangYu/ReactomePA/pull/1">@vladpetyuk</a>.</p>

<p>Import <em>dotplot</em> from DOSE and add &lsquo;Pathway analysis of NGS data&rsquo;
section in vignette.</p>

<h3 id="chipseeker">ChIPseeker</h3>

<p>Implement
<em><a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">seq2gene</a></em>
function for linking genomic regions to genes by many-to-many mapping.
With this function, enrichment analysis of
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/DOSE">DOSE</a> and
<a href="http://www.bioconductor.org/packages/ReactomePA">ReactomePA</a> can be
applied to NGS data.</p>

<p><a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot</a>
function is implemented for viewing overlap of ChIP annotation.</p>

<p>Better implementation of internal function <em>getFirstHitIndex</em>
contributed by
<a href="https://support.bioconductor.org/p/70432/#70545">@hpages</a>. Now
annotating a BED file of size about 1GB only take several minutes.</p>

<p>An issue of BED file that using <a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">the 0-based coordinate system</a> was fixed.</p>

<p>GEO data information was updated, now ChIPseeker contains more than
18,000 BED file information. User can compare their own data with those
deposited in GEO to <a href="http://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">identify co-occurrence binding
proteins</a>
that maybe cooperated with the one we are interested in.</p>

<h3 id="ggtree">ggtree</h3>

<p>My major focus in the past half year is on developing
<a href="http://www.bioconductor.org/packages/ggtree">ggtree</a>.
<a href="http://www.bioconductor.org/packages/ggtree">ggtree</a> has a lot of
changes. Some of the updates can be found in <a href="http://bioconductor.org/help/newsletters/2015_October/">Bioconductor Newsletter</a>
(Oct. 2015). In addition, I would like to emphasize the
<em><a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a></em> function which works
fine with all the ggplot objects and was applied to <a href="http://stackoverflow.com/questions/10368180/plotting-pie-graphs-on-map-in-ggplot/32380396#32380396">plot pie graph on a map</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ggtree">ggtree</a> can be fun by using <a href="http://guangchuangyu.github.io/2015/06/ggtree-with-funny-fonts/">symbolic fonts</a>, <a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">cartoonize the tree</a>
or annotating the tree <a href="http://guangchuangyu.github.io/2015/12/use-emoji-font-in-r/">with emoji</a>.</p>

<p>There are many updates that based on the github version of ggplot2.
Since the new features of ggplot2 are currently not available in CRAN. I
do not commit the change to Bioconductor and will introduce these new
feature to ggtree user when new version of ggplot2 is available in CRAN.</p>

<h2 id="citation">Citation</h2>

<p><strong><em>Yu G</em></strong>, Wang LG and He QY<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/31/14/2382">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(14):2382-2383.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="3781087"></div></p>

<p><strong><em>Yu G</em></strong>, Wang L, Han Y and He Q<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="681089"></div></p>

<p><strong><em>G Yu</em></strong>, LG Wang, GR Yan, QY He. <a href="http://bioinformatics.oxfordjournals.org/content/31/4/608">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(4):608-609.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="2788597"></div></p>

<p><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. <a href="http://dx.doi.org/10.1111/2041-210X.12628"><code>doi:10.1111/2041-210X.12628</code></a>.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1111/2041-210X.12628" class="altmetric-embed"></div></p>

<p><strong>Yu G</strong><sup>†</sup>, Li F<sup>†</sup>, Qin Y, Bo X<sup>*</sup>, Wu Y and Wang S<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/26/7/976">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="100979"></div></p>

<p><strong><em>G Yu</em></strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E#!divAbstract">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong> 2016, 12(2):477-479.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="4796667"></div></p>
    </div>
  </div>

</article>

<div class="container">
  <nav>
  <ul class="pager">
    
    <li class="previous"><a href="https://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/"><span
      aria-hidden="true">&larr;</span> ChIPseq data mining with ChIPseeker</a></li>
    

    
    <li class="next"><a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">use simplify to remove redundancy of enriched GO terms <span
      aria-hidden="true">&rarr;</span></a></li>
    
  </ul>
</nav>

</div>

<div class="article-container">
  
<section id="comments">
  <div id="disqus_thread">
    <div id="disqus_thread"></div>
<script type="text/javascript">
    var disqus_shortname = 'gcyu';
    var disqus_identifier = 'https:\/\/guangchuangyu.github.io\/2015\/10\/bioc-32-news-of-my-bioc-packages\/';
    var disqus_title = '[Bioc 32] NEWS of my BioC packages';
    var disqus_url = 'https:\/\/guangchuangyu.github.io\/2015\/10\/bioc-32-news-of-my-bioc-packages\/';

    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
  </div>
</section>


</div>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    <script async src="//cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_CHTML"></script>
    

  </body>
</html>

