<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.21" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="PhD student">

  
  
  
    
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/">

  

  <title>Posts | Guangchuang YU</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>‰∏≠ÊñáÂÖ•Âè£</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Posts</h1>

  

  
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/">[Bioc 32] NEWS of my BioC packages</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-10-15 16:13:57 &#43;0800 CST" itemprop="datePublished">
      Thu, Oct 15, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim" target="_blank">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler" target="_blank">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim" target="_blank">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler" target="_blank">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/10/chipseq-data-mining-with-chipseeker/">ChIPseq data mining with ChIPseeker</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-10-01 09:08:06 &#43;0800 CST" itemprop="datePublished">
      Thu, Oct 1, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/genomics">Genomics</a
    >, 
    
    <a href="/categories/systems-biology">Systems Biology</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >, 
    
    <a href="/tags/interaction">interaction</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      ChIP-seq is rapidly becoming a common technique and there are a large
number of dataset available in the public domain. Results from
individual experiments provide a limited understanding of chromatin
interactions, as there is many factors cooperate to regulate
transcription. Unlike other tools that designed for single dataset,
<a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a> is
designed for comparing profiles of ChIP-seq datasets at different
levels.</p>

<p>We provide functions to compare profiles of peaks binding to TSS
regions, annotation, and enriched functional profiles. More importantly,
ChIPseeker incorporates statistical testing of co-occurrence of
different ChIP-seq datasets and can be used to identify co-factors.

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">subsetting data in ggtree</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-09-24 01:20:21 &#43;0800 CST" itemprop="datePublished">
      Thu, Sep 24, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      Subsetting is commonly used in ggtree as we would like to for example
separating internal nodes from tips. We may also want to display
annotation to specific node(s)/tip(s).</p>

<p>Some software may stored clade information (e.g. bootstrap value) as
internal node labels. Indeed we want to manipulate such information and
taxa labels separately.</p>

<p>In current ggplot2 (version=1.0.1, access date:2015-09-23), it support
<em>subset</em>. For instance:

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr/">comic phylogenetic tree with ggtree and comicR</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-09-08 01:15:26 &#43;0800 CST" itemprop="datePublished">
      Tue, Sep 8, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
applies the concepts of grammar of graphic in phylogenetic tree
presentation and make it easy to add multiple layers of text and even
figures above a üå≤.</p>

<p>Here, I cartoonize a phylogenetic tree generated by
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
with <a href="https://github.com/timelyportfolio/comicR" target="_blank">comicR</a>, which is a
funny package to generate comic (xkcd-like) graph in R. Have fun with
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
and <a href="https://github.com/timelyportfolio/comicR" target="_blank">comicR</a>.</p>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/08/subview/">subview</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-08-31 03:17:48 &#43;0800 CST" itemprop="datePublished">
      Mon, Aug 31, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
    geom_bar(stat=&quot;identity&quot;, width=1) + 
    coord_polar(theta=&quot;y&quot;) + theme_tree() + 
    xlab(NULL) + ylab(NULL) + 
    theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
    p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
    print(p)
}
</code></pre>

<p>
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-08-21 03:22:21 &#43;0800 CST" itemprop="datePublished">
      Fri, Aug 21, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/bioinformatics">Bioinformatics</a
    >, 
    
    <a href="/categories/genomics">Genomics</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/" target="_blank">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" />

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-08-13 05:44:27 &#43;0800 CST" itemprop="datePublished">
      Thu, Aug 13, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002" target="_blank">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57" target="_blank">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" />

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-08-07 03:03:08 &#43;0800 CST" itemprop="datePublished">
      Fri, Aug 7, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/bioinformatics">Bioinformatics</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf" target="_blank">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145" target="_blank">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno" target="_blank">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/" target="_blank">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/" target="_blank">PeakAnalyzer</a>.</p>

<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images/">ggtree annotate phylogenetic tree with local images</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-08-01 12:52:34 &#43;0800 CST" itemprop="datePublished">
      Sat, Aug 1, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/annotation_image/Screenshot%202015-08-01%2020.53.50.png" alt="" /></p>

<p>In <a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>,
we provide a function <em>annotation_image</em> for annotating phylogenetic
tree with images.

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-07-28 06:45:01 &#43;0800 CST" itemprop="datePublished">
      Tue, Jul 28, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/genomics">Genomics</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/" target="_blank">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:

      
    </p></div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/post/page/4/">&laquo; Previous</a></li>
    
    
    <li><a href="/post/page/6/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:350px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    

    
    

  </body>
</html>

