<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="//guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="//guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="//guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="//guangchuangyu.github.io/post/">

    <title>Posts | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="//guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="//guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    <h1>Posts</h1>

    
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler">KEGG enrichment analysis with latest online data using clusterProfiler</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-02-01 12:38:27 &#43;0800 CST" itemprop="datePublished">Sun, Feb 1, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/kegg">KEGG</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p><em>KEGG.db</em> is not updated since 2012. The data is now pretty old, but
many of the Bioconductor packages still using it for KEGG annotation and
enrichment analysis. As pointed out in &lsquo;<a href="http://www.acgt.me/blog/2015/1/16/are-there-too-many-biological-databases">Are there too many biological databases</a>&rsquo;,
there is a problem that many out of date biological databases often
don&rsquo;t get offline. This issue also exists in web-server or software that
using out-of-date data. For example, the
<a href="http://wego.genomics.org.cn/cgi-bin/wego/index.pl">WEGO</a> web-server
stopped updating GO annotation data since 2009, and WEGO still online
with <a href="https://scholar.google.com.hk/scholar?as_ylo=2015&amp;hl=en&amp;as_sdt=0,5&amp;sciodt=0,5&amp;cites=13025429171853466135&amp;scipsc=">many people using it</a>.
The biological story may changed totally if using a recently updated
data. Seriously, We should keep an eye on this issue.</p>

<p>Now <em>enrichKEGG</em> function is reloaded with a new parameter <em>use_internal_data</em>. This
parameter is by default setting to FALSE, and <em>enrichKEGG</em> function will
download the latest KEGG data for enrichment analysis. If the parameter
<em>use_internal_data</em> is explicitly setting to TRUE, it will use the
<em>KEGG.db</em> which is still supported but not recommended. With this new
feature, supported species is unlimited if only there are KEGG
annotations available in KEGG database. You can access the full list of
species supported by KEGG via:
<a href="http://www.genome.jp/kegg/catalog/org_list.html">http://www.genome.jp/kegg/catalog/org_list.html</a> Now the <em>organism</em>
parameter in <em>enrichKEGG</em> should be abbreviation of academic name, for
example &lsquo;hsa&rsquo; for human and &lsquo;mmu&rsquo; for mouse. It accepts any species
listed in <a href="http://www.genome.jp/kegg/catalog/org_list.html">http://www.genome.jp/kegg/catalog/org_list.html</a>. In the
current release version of
<a href="http://bioconductor.org/packages/3.0/bioc/html/clusterProfiler.html">clusterProfiler</a>
(in Bioconductor 3.0), <em>enrichKEGG</em> supports about 20 species, and the
<em>organism</em> parameter accept common name of species, for instance &ldquo;human&rdquo;
and &ldquo;mouse&rdquo;. For these previously supported species, common name is also
supported. So that you script is still working with new version of
<a href="http://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html">clusterProfiler</a>.
For other species, common name is not supported, since I don&rsquo;t want to
maintain such a long mapping list with many species have no common name
available and it may also introduce unexpected bugs.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2015/01/ggtree-in-bioconductor-3.1">ggtree in Bioconductor 3.1</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-01-19 03:50:23 &#43;0800 CST" itemprop="datePublished">Mon, Jan 19, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/bioc">BioC</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>I am very glad that <a href="http://guangchuangyu.github.io/2014/12/ggtree/">ggtree</a> is now
available via Bioconductor. This is my 6th Bioconductor package.
<a href="http://guangchuangyu.github.io/2014/12/ggtree/">ggtree</a> now supports parsing output
files from BEAST, PAML, HYPHY, EPA and PPLACER and can annotate
phylogenetic tree directly using plot methods.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree">viewing and annotating phylogenetic tree with ggtree</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-12-21 08:00:08 &#43;0800 CST" itemprop="datePublished">Sun, Dec 21, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>When I need to annotate nucleotide substitutions in the phylogenetic
tree, I found that all the software are designed to display the tree but
not annotating it. Some of them may support annotating the tree with
specific data such as bootstrap values, but they are restricted to a few
supported data types. It is hard/impossible to inject user specific
data.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/12/2014-12-21-145753_579x436_scrot.png" alt="" /></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/12/auto-complete-in-ess">auto-complete in ESS</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-12-07 10:22:35 &#43;0800 CST" itemprop="datePublished">Sun, Dec 7, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/emacs">Emacs</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/ess">ESS</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p>Auto complete is good, it can save you times in typing and prevent typo sometimes. RStudio now supports function arguments in auto complete. ESS&rsquo;s auto complete is more advance, it supports help page. We should use &lsquo;ess-use-auto-complete&rsquo; to activate auto complete in ESS buffers. (setq ess-use-auto-complete t) The effect is quite amazing. Auto-complete extension is needed for ESS, we need to install the auto-complete extension and load it at Emacs startup.</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/11/proper-use-of-gosemsim">proper use of GOSemSim</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-11-23 02:27:23 &#43;0800 CST" itemprop="datePublished">Sun, Nov 23, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>One day, I am looking for R packages that can analyze PPI and after
searching, I found the
<a href="http://cran.r-project.org/web/packages/ppiPre/index.html">ppiPre</a>
package in CRAN.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/11/2014-11-22-215959_895x256_scrot.png" alt="" /></p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/11/hello-yosemite">hello yosemite</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-11-14 04:34:48 &#43;0800 CST" itemprop="datePublished">Fri, Nov 14, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/osx">OSX</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/2014/11/Screenshot-2014-11-11-14.07.45.png" alt="" /></p>

<p>Installing OS is painful, you need to re-install all the software and
configure them to the way you want. We don&rsquo;t want to wast time in doing
this. To prevent doing this dirty job, we perform an upgrade install
instead of clean install. We all have experience of upgrading Windows
sucks, <a href="http://osxdaily.com/2014/10/17/os-x-yosemite-installation-stuck/">same as OS X</a>.
All the issues you have in old system will be remained, and sometimes
new issues will be introduced in the process of upgrading. The system
will be slower compare to clean install one.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker">multiple annotation in ChIPseeker</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-10-01 13:00:33 &#43;0800 CST" itemprop="datePublished">Wed, Oct 1, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><h3 id="nearest-gene-annotation:211662300f8731372460bacaae7937b5">Nearest gene annotation</h3>

<p>Almost all annotation software calculate the distance of a peak to the
nearest TSS and assign the peak to that gene. This can be misleading, as
binding sites might be located between two start sites of different
genes or hit different genes which have the same TSS location in the
genome.</p>

<p>The function <em>annotatePeak</em> provides option to assign genes with a max
distance cutoff and all genes within this distance were reported for
each peak.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/08/why-clusterprofiler-fails">why clusterProfiler fails</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-08-07 05:46:35 &#43;0800 CST" itemprop="datePublished">Thu, Aug 7, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>Recently, there are some
<a href="http://guangchuangyu.github.io/2012/05/clusterprofiler-an-r-package-for-comparing-biological-themes-among-gene-clusters/">comments</a>
said that sometimes clusterProfiler failed in KEGG enrichment analysis.</p>

<p><a href="http://guangchuangyu.github.io/2013/07/permanent-head-damage-to-be/">kaji331</a>
compared cluserProfiler with GeneAnswers and found that clusterProfiler
gives larger p values. The result forces me to test it.</p>

<blockquote>
<p>it eventually came out that he passed the input
gene as numeric vector, which was supposed to be character and he used
an old version of clusterProfiler which didn&rsquo;t use <em>as.character</em> to
coerce the input</p>
</blockquote>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/08/enrichment-map">enrichment map</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-08-03 12:48:15 &#43;0800 CST" itemprop="datePublished">Sun, Aug 3, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>In <a href="http://www.plob.org/">PLOB</a>&rsquo;s QQ group, someone asked how to change
the color of enrichment map in <a href="http://cytoscape.org/">Cytoscape</a>. I am
very curious how enrichment map can helps to interpret enrichment
results. It took me 2 hours to implement it using R and I am surprised
that the enrichment map is better than anticipated.</p>
</p></div>
    </div>
    
    <div>
        <h2><a href="//guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker">visualization methods in ChIPseeker</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2014-04-30 06:33:41 &#43;0800 CST" itemprop="datePublished">Wed, Apr 30, 2014</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>, 
        
        <a class="article-tag-link" href="//guangchuangyu.github.io/tags/chipseq">ChIPseq</a>
        
    </span>
    
    



</div>

        <div class="article-style"><p><p>After two weeks developed, I have added/updated some plot functions in
<a href="http://bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
(version &gt;=1.0.1).</p>

<h3 id="chip-peaks-over-chromosomes:6a8a65de687a162a83f53f5c5a317279">ChIP peaks over Chromosomes</h3>

<pre><code>&gt; files=getSampleFiles()
&gt; peak=readPeakFile(files[[4]])
&gt; peak
GRanges object with 1331 ranges and 2 metadata columns:
         seqnames                 ranges strand   |             V4        V5
                               |        
     [1]     chr1     [ 815092,  817883]      *   |    MACS_peak_1    295.76
     [2]     chr1     [1243287, 1244338]      *   |    MACS_peak_2     63.19
     [3]     chr1     [2979976, 2981228]      *   |    MACS_peak_3    100.16
     [4]     chr1     [3566181, 3567876]      *   |    MACS_peak_4    558.89
     [5]     chr1     [3816545, 3818111]      *   |    MACS_peak_5     57.57
     ...      ...                    ...    ... ...            ...       ...
  [1327]     chrX [135244782, 135245821]      *   | MACS_peak_1327     55.54
  [1328]     chrX [139171963, 139173506]      *   | MACS_peak_1328    270.19
  [1329]     chrX [139583953, 139586126]      *   | MACS_peak_1329    918.73
  [1330]     chrX [139592001, 139593238]      *   | MACS_peak_1330    210.88
  [1331]     chrY [ 13845133,  13845777]      *   | MACS_peak_1331     58.39
  ---
  seqlengths:
    chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
      NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
&gt; covplot(peak, weightCol=&quot;V5&quot;)
</code></pre>
</p></div>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="//guangchuangyu.github.io//post/page/4/">« Previous</a></li>
        
        
        <li><a href="//guangchuangyu.github.io//post/page/6/">Next »</a></li>
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="//guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="//guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="//guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

