<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/">

  

  <title>Posts | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Posts</h1>

  

  
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2011/06/r-package-dose-released/">R package DOSE released</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2011-06-13 04:58:10 &#43;0800 HKT" itemprop="datePublished">
      Mon, Jun 13, 2011
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>Disease Ontology (DO) provides an open source ontology for the
integration of biomedical data that is associated with human disease. DO
analysis can lead to interesting discoveries that deserve further
clinical investigation.</p>

<p>DOSE was designed for semantic similarity measure and enrichment
analysis.</p>

<p>Four information content (IC)-based methods, proposed by Resnik, Jiang,
Lin and Schlicker, and one graph structure-based method, proposed by
Wang, were implemented. The calculation details can be referred to the
vignette of R package
<a href="http://www.bioconductor.org/packages/2.9/bioc/html/GOSemSim.html" title="GOSemSim" target="_blank">GOSemSim</a>.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8/">clusterProfiler in Bioconductor 2.8</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2011-03-27 04:12:46 &#43;0800 HKT" itemprop="datePublished">
      Sun, Mar 27, 2011
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/bioc">BioC</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      In recently years, high-throughput experimental techniques such as
microarray and mass spectrometry can identify many lists of genes and
gene products. The most widely used strategy for high-throughput data
analysis is to identify different gene clusters based on their
expression profiles. Another commonly used approach is to annotate these
genes to biological knowledge, such as Gene Ontology (GO) and Kyoto
Encyclopedia of Genes and Genomes (KEGG), and identify the statistically
significantly enriched categories. These two different strategies were
implemented in many bioconductor packages, such as <em>Mfuzz</em> and <em>BHC</em> for
clustering analysis and <em>GOstats</em> for GO enrichment analysis.

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2010/12/one-dimensional-integrals/">one-dimensional integrals</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2010-12-25 10:40:22 &#43;0800 HKT" itemprop="datePublished">
      Sat, Dec 25, 2010
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/calculus">Calculus</a
    >, 
    
    <a href="/categories/programming">Programming</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/integrals">integrals</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      The foundamental idea of numerical integration is to estimate the area of the region in the xy-plane bounded by the graph of function f(x). The integral was esimated by dividing x into small intervals, then adds all the small approximations to give a total approximation.
Trapezoidal rule Numerical integration can be done by trapezoidal rule, simpson&rsquo;s rule and quadrature rules. R has a built-in function, integrate, which performs adaptive quadrature.
      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2010/12/root-finding/">Root finding</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2010-12-04 10:40:22 &#43;0800 HKT" itemprop="datePublished">
      Sat, Dec 4, 2010
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/calculus">Calculus</a
    >, 
    
    <a href="/categories/programming">Programming</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/root">root</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>Numerical root finding methods use iteration, producing a sequence of numbers that hopefully converge towards a limits which is a root. This post only focuses four basic algorithms on root finding, and covers bisection method, fixed point method, Newton-Raphson method, and secant method.</p>

<p>The simplest root finding algorithms is the <a href="https://guangchuangyu.github.io/cn/2008/11/bisect-to-solve-equation/" target="_blank">bisection method</a>. It works when <code>f</code> is a continuous function and it requires previous knowledge of two initial gueeses, <code>u</code> and <code>v</code>, such that <code>f(u)</code> and <code>f(v)</code> have opposite signs. This method is reliable, but converges slowly. For detail, see <a href="https://guangchuangyu.github.io/cn/2008/11/bisect-to-solve-equation/" target="_blank">https://guangchuangyu.github.io/cn/2008/11/bisect-to-solve-equation/</a> .</p>

<p>Root finding can be reduced to the problem of finding fixed points of the function <code>g(x) = c*f(x) +x</code>, where <code>c</code> is a non-zero constant. It is clearly that <code>f(a) = 0</code> if and only if <code>g(a) = a</code>. This is the so called fixed point algorithm.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2010/05/gosemsim-an-r-package-for-measuring-semantic-similarity-among-go-terms-and-gene-products/">GOSemSim an R package for measuring semantic similarity among GO terms and gene products</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2010-05-28 05:06:26 &#43;0800 HKT" itemprop="datePublished">
      Fri, May 28, 2010
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/publication">Publication</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/gosemsim">GOSemSim</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><a href="http://www.ncbi.nlm.nih.gov/pubmed/20179076" target="_blank"><img src="http://bioinformatics.oxfordjournals.org/content/26/18.cover.gif" alt="" title="bioinfo" /></a></p>

<h3 id="abstract">Abstract</h3>

<p><strong>SUMMARY</strong>: The semantic comparisons of Gene Ontology (GO) annotations
provide quantitative ways to compute similarities between genes and gene
groups, and have became important basis for many bioinformatics analysis
approaches. GOSemSim is an R package for semantic similarity computation
among GO terms, sets of GO terms, gene products and gene clusters. Four
information content (IC)- and a graph-based methods are implemented in
the GOSemSim package, multiple species including human, rat, mouse, fly
and yeast are also supported. The functions provided by the GOSemSim
offer flexibility for applications, and can be easily integrated into
high-throughput analysis pipelines. AVAILABILITY: GOSemSim is released
under the GNU General Public License within Bioconductor project, and
freely available at
<a href="http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html" target="_blank">http://bioconductor.org/packages/2.6/bioc/html/GOSemSim.html</a>.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2009/03/assignment-token-of-r/">Assignment token of R</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2009-03-09 13:35:24 &#43;0800 HKT" itemprop="datePublished">
      Mon, Mar 9, 2009
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/visualization">Visualization</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/syntax">syntax</a
    >, 
    
    <a href="/tags/semantic">semantic</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>when I recalled the <em>switch</em> function, it always gave the first element, no matter what the parameter is.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2009/03/r-29-dev.png" alt="r-29-dev" /></p>

<p>when <em>organism</em> changed to &ldquo;yeast&rdquo;, and called <em>switch</em> function, <em>species</em> supposed to be changed to &ldquo;Sc&rdquo;, but it remains it&rsquo;s original value.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2008/08/sequence-alignment-program-written-in-r/">sequence alignment program written in R</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2008-08-07 12:13:35 &#43;0800 HKT" itemprop="datePublished">
      Thu, Aug 7, 2008
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >, 
    
    <a href="/categories/biology">Biology</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/alignment">alignment</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>Sequence alignment by dynamic programming.</p>

<p></p>

      
    </p></div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/post/page/9/">&laquo; Previous</a></li>
    
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    

    
    

  </body>
</html>

