<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/categories/r/">

    <title>R | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/03/font-awesome-supported-in-emojifont">Font Awesome supported in emojifont</a></h2>
        <div class="article-style"><p><p><a href="http://cran.r-project.org/package=emojifont"><img src="http://www.r-pkg.org/badges/version/emojifont" alt="CRAN_Status_Badge" /></a></p>

<p><a href="https://cran.r-project.org/web/packages/emojifont/index.html"><code>emojifont</code></a> is available in CRAN, you can use the following command to install it.</p>

<pre><code>install.packages(&quot;emojifont&quot;)
</code></pre>

<p>An example of using emoji font in R plot is showed below:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2016/douban_emoji.jpg" alt="" /></p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/02/covplot-supports-grangeslist">covplot supports GRangesList</a></h2>
        <div class="article-style"><p><p>To answer the <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/27">issue</a>, I extend the <code>covplot</code> function to support viewing coverage of a list of <code>GRanges</code> objects or <code>bed</code> files.</p>

<pre><code>library(ChIPseeker)
files &lt;- getSampleFiles()
peak=GenomicRanges::GRangesList(CBX6=readPeakFile(files[[4]]),
                                CBX7=readPeakFile(files[[5]]))

p &lt;- covplot(peak)
print(p)
</code></pre>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/02/tweets-of-ggtree">tweets of ggtree</a></h2>
        <div class="article-style"><p><blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Treespace visualization of a heuristic best tree search in <a href="https://twitter.com/hashtag/R?src=hash">#R</a> package <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> <a href="https://twitter.com/hashtag/phylogenetics?src=hash">#phylogenetics</a> <a href="https://t.co/hnSJpIABql">pic.twitter.com/hnSJpIABql</a></p>&mdash; Alex Damián (@antropoteuthis) <a href="https://twitter.com/antropoteuthis/status/693123105707986944">January 29, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr"><a href="https://twitter.com/guangchuangyu">@guangchuangyu</a> Updated to latest <a href="https://twitter.com/hashtag/ggtree?src=hash">#ggtree</a> and <a href="https://twitter.com/hashtag/ggplot2?src=hash">#ggplot2</a> and very impressed! Any ideas on using geom_label with tip labels to add box/fill?</p>&mdash; Jo Williams-Newkirk (@ajwnewkirk) <a href="https://twitter.com/ajwnewkirk/status/691762302492790788">January 25, 2016</a></blockquote>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">ggtree looks really good! <a href="https://twitter.com/hashtag/rstats?src=hash">#rstats</a><a href="https://t.co/7rtexYV3I5">https://t.co/7rtexYV3I5</a></p>&mdash; Paul Staab (@paulstaab) <a href="https://twitter.com/paulstaab/status/687933344219312128">January 15, 2016</a></blockquote>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a></h2>
        <div class="article-style"><p><p>My R/Bioconductor package,
<a href="https://github.com/GuangchuangYu/ReactomePA">ReactomePA</a>, published in
<a href="http://pubs.rsc.org/en/content/articlelanding/2016/mb/c5mb00663e">Molecular BioSystems</a>.</p>

<p><strong>ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</strong></p>

<p><strong>G Yu</strong>, QY He. <strong><em>Molecular BioSystems</em></strong> 2016, 12:477-479</p>

<p>Received: 05 Oct 2015 Accepted: 20 Nov 2015 Online: 23 Nov 2015</p>

<p>DOI: <a href="http://dx.doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a></p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/01/ggtree-supports-phylip-tree-format">ggtree supports phylip tree format</a></h2>
        <div class="article-style"><p><p>Phylip is also a widely used tree format, which contains taxa sequences with Newick tree text.</p>

<p>In <code>ggtree</code>, we can use <code>read.phylip()</code> function to parse the file and use <code>ggtree()</code> to visualize the tree.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/01/label-edge-number-in-ggtree">label edge number in ggtree</a></h2>
        <div class="article-style"><p><p>This is a question from <code>ggtree</code> user. In <code>ape</code> and <code>phytools</code>, it&rsquo;s easy to label edge using the <code>edgelabels</code> function.</p>

<pre><code>set.seed(1)
tr = rtree(30)
library(ape)
plot(tr, main=&quot;ape&quot;)
edgelabels()
</code></pre>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets">Annotate a phylogenetic tree with insets</a></h2>
        <div class="article-style"><p><p><code>ggtree</code> implemented a function, <code>subview</code>, that can add subplots on a
ggplot2 object. It had successful applied to <a href="http://stackoverflow.com/questions/10368180/plotting-pie-graphs-on-map-in-ggplot/32380396#32380396">plot pie graphs on map</a>.</p>

<pre><code>set.seed(2016-01-04)
tr &lt;- rtree(30)
tr &lt;- groupClade(tr, node=45)
p &lt;- ggtree(tr, aes(color=group)) + geom_tippoint()
cpos &lt;- get_clade_position(p, node=45)
p1 &lt;- p + geom_hilight(node=45)
p2 &lt;- with(cpos, p+xlim(xmin, xmax*1.01)+ylim(ymin, ymax))
with(cpos, subview(p2+geom_tiplab(), p1+theme_transparent(), x=xmin+(xmax-xmin)*.15, y=ymin+(ymax-ymin)*.85))
</code></pre>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler">GO analysis using clusterProfiler</a></h2>
        <div class="article-style"><p><p><code>clusterProfiler</code> supports over-representation test and gene set
enrichment analysis of Gene Ontology. It supports GO annotation from
OrgDb object, GMT file and user&rsquo;s own data.</p>

<h2 id="support-many-species:bdc631894cab1e81ad84e5406d79b479">support many species</h2>

<p>In github version of clusterProfiler, <code>enrichGO</code> and <code>gseGO</code> functions
removed the parameter <em>organism</em> and add another parameter <em>OrgDb</em>, so
that any species that have <code>OrgDb</code> object available can be analyzed in
<strong>clusterProfiler</strong>. Bioconductor have already provide OrgDb for about
20 species, see
<a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">http://bioconductor.org/packages/release/BiocViews.html#___OrgDb</a>, and
users can build <code>OrgDb</code> via <code>AnnotationHub</code>.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/12/news-of-ggtree">News of ggtree</a></h2>
        <div class="article-style"><p><p>A new version of <a href="https://github.com/GuangchuangYu/ggtree">ggtree</a> that works with ggplot2 (version &gt;= 2.0.0) is now availabel.</p>

<h2 id="new-layers:04f7c01ab00975376c8371fb68bcfbb4">new layers</h2>

<p>Some functions, <em>add_legend</em>, <em>hilight</em>, <em>annotation_clade</em> and <em>annotation_clade2</em> were removed. Instead we provide layer functions, <code>geom_treescale</code>, <code>geom_hilight</code> and <code>geom_cladelabel</code>. You can use <code>+</code> operator to add layers using these layer functions.</p>

<p>In addtion, we provide <code>geom_point2</code>, <code>geom_text2</code> and <code>geom_segment2</code> which works exactly as <code>geom_point</code>, <code>geom_text</code> and <code>geom_segment</code> except they allow ggtree users to do <a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree/">subsetting</a>.</p>
</p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/12/use-emoji-font-in-r">use emoji font in R</a></h2>
        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot 2015-12-16 10.55.49.png" alt="" /></p>

<p>I have played with emoji in <code>R</code> for a while. My solution of using it is different from what implemented in <a href="https://github.com/dill/emoGG/">emoGG</a>.</p>

<p><code>emoGG</code> is a good attemp to add <code>emoji</code> in <code>ggplot2</code>. It render <code>emoji</code> picture (png) and creat a layer, <code>geom_emoji</code>, to add emoji.</p>

<p>In my opinion, <code>emoji</code> should be treated as ordinary font in user interface, albeit it maynot be true internally.</p>

<p>It would be more flexible if we can use emoji as ordinary font and in this way user don&rsquo;t need to learn extra stuff.</p>
</p></div>

    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="https://guangchuangyu.github.io//categories/r/page/2/">« Previous</a></li>
        
        
        <li><a href="https://guangchuangyu.github.io//categories/r/page/4/">Next »</a></li>
        
    </ul>
</nav>



</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

