<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/post/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/post/">

  

  <title>Posts | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Posts</h1>

  

  
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/09/ggtree-for-microbiome-data/">ggtree for microbiome data</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-09-01 03:17:48 &#43;0800 HKT" itemprop="datePublished">
      Thu, Sep 1, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/visualization">Visualization</a
    >, 
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/phyloseq">phyloseq</a
    >, 
    
    <a href="/tags/microbiome">microbiome</a
    >, 
    
    <a href="/tags/tree">tree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree</a> can parse many software outputs and the evolution evidences inferred by these software can be used directly for tree annotation. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree</a> not only works as an infrastructure that enables evolutionary data that inferred by commonly used software packages to be used in R, but also serves as a general tree visualization and annotation tool for the R community as it supports many S3/S4 objects defined by other R packages.</p>

<h2 id="phyloseq-for-microbiome-data">phyloseq for microbiome data</h2>

<p><code>phyloseq</code> class defined in the <a href="https://www.bioconductor.org/package/phyloseq" target="_blank">phyloseq</a> package was designed for microbiome data. <code>phyloseq</code> package implemented <code>plot_tree</code> function using <code>ggplot2</code>. Although the function was implemented by <code>ggplot2</code> and we can use <code>theme</code>, <code>scale_color_manual</code> etc for customization, the most valuable part of <code>ggplot2</code>, adding layer, is missing. <code>plot_tree</code> only provides limited parameters to control the output graph and it is hard to add layer unless user has expertise in both <code>phyloseq</code> and <code>ggplot2</code>.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/08/ggtree-paper-published/">ggtree paper published</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-08-16 22:10:00 &#43;0800 HKT" itemprop="datePublished">
      Tue, Aug 16, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>Today <a href="https://guangchuangyu.github.io/ggtree" target="_blank"><code>ggtree</code></a> received <a href="https://github.com/GuangchuangYu/ggtree/stargazers" target="_blank">100 stars</a> on <i class="fa fa-github"></i> and I found the paper was <a href="http://dx.doi.org/10.1111/2041-210X.12628" target="_blank">online</a> at the same day by the tweet:</p>

<p><blockquote class="twitter-tweet"><p lang="ja" dir="ltr">ggtree, Methods in Ecology and Evolutionに掲載されたのか。オープンアクセスっぽいな。<a href="https://t.co/JlET18BSZv">https://t.co/JlET18BSZv</a></p>&mdash; Hoxo_Mass_Spectrum (@siero5335) <a href="https://twitter.com/siero5335/status/765521239146233856?ref_src=twsrc%5Etfw">August 16, 2016</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script></p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/08/using-meshes-for-mesh-enrichment-analysis/">using meshes for MeSH Enrichment Analysis</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-08-03 00:19:15 &#43;0800 HKT" itemprop="datePublished">
      Wed, Aug 3, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/meshes">meshes</a
    >, 
    
    <a href="/tags/mesh">MeSH</a
    >, 
    
    <a href="/tags/gsea">GSEA</a
    >, 
    
    <a href="/tags/ora">ORA</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>MeSH (Medical Subject Headings) is the NLM (U.S. National Library of
Medicine) controlled vocabulary used to manually index articles for
MEDLINE/PubMed. MeSH is comprehensive life science vocabulary. MeSH has
19 categories and <code>MeSH.db</code> contains 16 of them. That is:</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/07/leading-edge-analysis/">leading edge analysis</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-07-20 18:58:10 &#43;0800 HKT" itemprop="datePublished">
      Wed, Jul 20, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/gsea">GSEA</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <h2 id="leading-edge-and-core-enrichment">leading edge and core enrichment</h2>

<p>Leading edge analysis reports <code>Tags</code> to indicate the percentage of genes contributing to the enrichment score, <code>List</code> to indicate where in the list the enrichment score is attained and <code>Signal</code> for enrichment signal strength.</p>

<p>It would also be very interesting to get the core enriched genes that contribute to the enrichment.</p>

<p>Now <a href="https://guangchuangyu.github.io/DOSE" target="_blank">DOSE</a>, <a href="https://guangchuangyu.github.io/clusterProfiler" target="_blank">clusterProfiler</a> and <a href="https://guangchuangyu.github.io/ReactomePA" target="_blank">ReactomePA</a> all support leading edge analysis and report core enriched genes.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/07/how-to-bug-author/">How to bug author</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-07-05 20:13:35 &#43;0800 HKT" itemprop="datePublished">
      Tue, Jul 5, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bug">bug</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>As an author of several Bioconductor packages. I found many questions from users are quite annoying. Some of them never use google and they are reluctant to read vignettes.</p>

<h3 id="step-1-make-sure-you-are-using-the-latest-release">Step 1: make sure you are using the latest release</h3>

<p>I found many peoples are using out-of-date packages. When they got an issue of an out-dated package, they never check whether the issue still exists in latest release.</p>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/06/identify-method-for-ggtree/">identify method for ggtree</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-06-28 14:30:00 &#43;0800 HKT" itemprop="datePublished">
      Tue, Jun 28, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/interactive">interactive</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p>We are happy to announce that <code>ggtree</code> supports interactive tree annotation/manipulation by implementing an <code>identify</code> method. Users can click on a node to highlight a clade, to label or rotate it <em>etc</em>.</p>

<p>Here is an example of highlighting clades using <code>geom_hilight</code> with <code>identify</code>:</p>

<div style="position: relative; padding-bottom: 56.25%; padding-top: 30px; height: 0; overflow: hidden;">
  <iframe src="//www.youtube.com/embed/KcF8Ec38mzI" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;" allowfullscreen frameborder="0" title="YouTube Video"></iframe>
 </div>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/05/bioc-33-news-of-my-bioc-packages/">[Bioc 33] NEWS of my BioC packages</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-05-05 20:13:57 &#43;0800 HKT" itemprop="datePublished">
      Thu, May 5, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><img src="http://guangchuangyu.github.io/blog_images/MISC/ke.jpg" width=600></p>

<p>Today is my birthday and it happened to be the release day of Bioconductor 3.3. It&rsquo;s again the time to reflect what I&rsquo;ve done in the past year.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker" target="_blank">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler" target="_blank">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE" target="_blank">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree" target="_blank">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim" target="_blank">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA" target="_blank">ReactomePA</a></li>
</ul>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/05/convert-biological-id-with-kegg-api-using-clusterprofiler/">convert biological ID with KEGG API using clusterProfiler</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-05-03 17:00:00 &#43;0800 HKT" itemprop="datePublished">
      Tue, May 3, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/kegg">KEGG</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <h2 id="bitr-kegg">bitr_kegg</h2>

<p><code>clusterProfiler</code> can convert biological IDs using <code>OrgDb</code> object via the <code>bitr</code> function. Now I implemented another function, <code>bitr_kegg</code> for converting IDs through KEGG API.</p>

<pre><code>library(clusterProfiler)
data(gcSample)
hg &lt;- gcSample[[1]]
head(hg)

## [1] &quot;4597&quot;  &quot;7111&quot;  &quot;5266&quot;  &quot;2175&quot;  &quot;755&quot;   &quot;23046&quot;

eg2np &lt;- bitr_kegg(hg, fromType='kegg', toType='ncbi-proteinid', organism='hsa')

## Warning in bitr_kegg(hg, fromType = &quot;kegg&quot;, toType = &quot;ncbi-proteinid&quot;,
## organism = &quot;hsa&quot;): 3.7% of input gene IDs are fail to map...

head(eg2np)

##     kegg ncbi-proteinid
## 1   8326      NP_003499
## 2  58487   NP_001034707
## 3 139081      NP_619647
## 4  59272      NP_068576
## 5    993      NP_001780
## 6   2676      NP_001487

np2up &lt;- bitr_kegg(eg2np[,2], fromType='ncbi-proteinid', toType='uniprot', organism='hsa')

head(np2up)

##   ncbi-proteinid uniprot
## 1      NP_005457  O75586
## 2      NP_005792  P41567
## 3      NP_005792  Q6IAV3
## 4      NP_037536  Q13421
## 5      NP_006054  O60662
## 6   NP_001092002  O95398
</code></pre>

<p>The ID type (both fromType &amp; toType) should be one of &lsquo;kegg&rsquo;, &lsquo;ncbi-geneid&rsquo;, &lsquo;ncbi-proteinid&rsquo; or &lsquo;uniprot&rsquo;. The &lsquo;kegg&rsquo; is the primary ID used in KEGG database. The data source of KEGG was from NCBI. A rule of thumb for the &lsquo;kegg&rsquo; ID is <code>entrezgene</code> ID for eukaryote species and <code>Locus</code> ID for prokaryotes.
</p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/04/kegg-module-enrichment-analysis/">KEGG Module Enrichment Analysis</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-04-13 20:48:15 &#43;0800 HKT" itemprop="datePublished">
      Wed, Apr 13, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/kegg">KEGG</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <blockquote>
<p>KEGG MODULE is a collection of manually defined functional units, called KEGG modules and identified by the M numbers, used for annotation and biological interpretation of sequenced genomes. There are four types of KEGG modules:</p>

<ul>
<li>pathway modules – representing tight functional units in KEGG metabolic pathway maps, such as M00002 (Glycolysis, core module involving three-carbon compounds)</li>
<li>structural complexes – often forming molecular machineries, such as M00072 (Oligosaccharyltransferase)</li>
<li>functional sets – for other types of essential sets, such as M00360 (Aminoacyl-tRNA synthases, prokaryotes)</li>
<li>signature modules – as markers of phenotypes, such as M00363 (EHEC pathogenicity signature, Shiga toxin)</li>
</ul>
</blockquote>

<p></p>

      
    </p></div>
  </div>
  
  <div>
    
    <h2><a href="https://guangchuangyu.github.io/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a></h2>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-03-30 13:40:22 &#43;0800 HKT" itemprop="datePublished">
      Wed, Mar 30, 2016
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/biology">Biology</a
    >, 
    
    <a href="/categories/software">Software</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioedit">BioEdit</a
    >, 
    
    <a href="/tags/alignment">alignment</a
    >, 
    
    <a href="/tags/osx">OSX</a
    >
    
  </span>
  
  

  

</div>

    <div class="article-style"><p>
      
      <p><a href="http://www.mbio.ncsu.edu/bioedit/bioedit.html" target="_blank"><img src="http://www.mbio.ncsu.edu/bioedit/Banner1.GIF" alt="" /></a></p>

<p>To my knowledge, <a href="http://www.mbio.ncsu.edu/bioedit/bioedit.html" target="_blank">BioEdit</a> is the most comprehensive biological sequence alignment editor. Most of my labmates run this software using <code>Parallels Desktop</code>. For some of them, BioEdit is the only reason to install <code>Parallels Desktop</code>.</p>

<p>I need to edit my alignment recently, and install it in my iMac using <a href="https://www.winehq.org/" target="_blank">Wine</a>, which is a compatibility layer for running Windows applications on POSIX-compliant OS. Although it is famous in Linux community for many years, many OSX users never heard of it.</p>

<p></p>

      
    </p></div>
  </div>
  

  
<nav>
  <ul class="pager">
    
    <li><a href="/post/page/2/">&laquo; Previous</a></li>
    
    
    <li><a href="/post/page/4/">Next &raquo;</a></li>
    
  </ul>
</nav>



</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    

    
    

  </body>
</html>

