<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="http://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="http://guangchuangyu.github.io/post/">

    <title>Posts | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/cn">‰∏≠ÊñáÂÖ•Âè£</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/09/subsetting-data-in-ggtree">subsetting data in ggtree</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-09-24 01:20:21 &#43;0800 HKT" itemprop="datePublished">Thu, Sep 24, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <p><p>Subsetting is commonly used in ggtree as we would like to for example
separating internal nodes from tips. We may also want to display
annotation to specific node(s)/tip(s).</p>

<p>Some software may stored clade information (e.g. bootstrap value) as
internal node labels. Indeed we want to manipulate such information and
taxa labels separately.</p>

<p>In current ggplot2 (version=1.0.1, access date:2015-09-23), it support
<em>subset</em>. For instance:</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/09/comic-phylogenetic-tree-with-ggtree-and-comicr">comic phylogenetic tree with ggtree and comicR</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-09-08 01:15:26 &#43;0800 HKT" itemprop="datePublished">Tue, Sep 8, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <p><p><a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
applies the concepts of grammar of graphic in phylogenetic tree
presentation and make it easy to add multiple layers of text and even
figures above a üå≤.</p>

<p>Here, I cartoonize a phylogenetic tree generated by
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
with <a href="https://github.com/timelyportfolio/comicR">comicR</a>, which is a
funny package to generate comic (xkcd-like) graph in R. Have fun with
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
and <a href="https://github.com/timelyportfolio/comicR">comicR</a>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/08/subview">subview</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-08-31 03:17:48 &#43;0800 HKT" itemprop="datePublished">Mon, Aug 31, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <p><p>I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
    geom_bar(stat=&quot;identity&quot;, width=1) + 
    coord_polar(theta=&quot;y&quot;) + theme_tree() + 
    xlab(NULL) + ylab(NULL) + 
    theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
    p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
    print(p)
}
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data">functional enrichment analysis with NGS data</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-08-21 03:22:21 &#43;0800 HKT" itemprop="datePublished">Fri, Aug 21, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>
        
    </span>
    
    



</div>

        <p><p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" /></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper">functional enrichment for GTEx paper</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-08-13 05:44:27 &#43;0800 HKT" itemprop="datePublished">Thu, Aug 13, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/dose">DOSE</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>
        
    </span>
    
    



</div>

        <p><p>The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" /></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/08/parsing-bed-coordinates">parsing BED coordinates</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-08-07 03:03:08 &#43;0800 HKT" itemprop="datePublished">Fri, Aug 7, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <p><p>In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/">PeakAnalyzer</a>.</p>

<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/08/ggtree-annotate-phylogenetic-tree-with-local-images">ggtree annotate phylogenetic tree with local images</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-08-01 12:52:34 &#43;0800 HKT" itemprop="datePublished">Sat, Aug 1, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <p><p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/ggtree/annotation_image/Screenshot%202015-08-01%2020.53.50.png" alt="" /></p>

<p>In <a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>,
we provide a function <em>annotation_image</em> for annotating phylogenetic
tree with images.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/07/upsetplot-in-chipseeker">upsetplot in ChIPseeker</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-07-28 06:45:01 &#43;0800 HKT" itemprop="datePublished">Tue, Jul 28, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>
        
    </span>
    
    



</div>

        <p><p><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-07-13 08:14:50 &#43;0800 HKT" itemprop="datePublished">Mon, Jul 13, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>, 
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/chipseq">ChIPseq</a>
        
    </span>
    
    



</div>

        <p><p>My R/Bioconductor package,
<a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>, published
in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145">Bioinformatics</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> had been
cited by <a href="http://www.biomedcentral.com/1471-2164/16/292">http://www.biomedcentral.com/1471-2164/16/292</a> and
<a href="http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short">http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short</a>, and
was used (not cited) in
<a href="http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract">http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract</a>
and
<a href="http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract">http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract</a>.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/2015/07/flip-and-rotate-branches-in-ggtree">flip and rotate branches in ggtree</a></h2>
        

<div class="article-metadata">

    <span class="article-date">
        <time datetime="2015-07-01 02:51:34 &#43;0800 HKT" itemprop="datePublished">Wed, Jul 1, 2015</time>
    </span>

    
    
    
    <span class="article-tags">
        <i class="fa fa-tags"></i>
        
        <a class="article-tag-link" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>
        
    </span>
    
    



</div>

        <p><p><strong><em>flip</em></strong> function is a feature request from ggtree user. It accepts a
tree view and two node numbers and exchange the positions of the
selected clades.</p>

<pre><code>require(ggtree)
set.seed(2015-07-01)
tr &lt;- rtree(30)
p &lt;- ggtree(tr) + geom_text(aes(label=node))
gridExtra::grid.arrange(p, flip(p, 38, 33), ncol=2)
</code></pre>
</p>
    </div>
    

    
<nav>
    <ul class="pager">
        
        <li><a href="http://guangchuangyu.github.io//post/page/2/">¬´ Previous</a></li>
        
        
        <li><a href="http://guangchuangyu.github.io//post/page/4/">Next ¬ª</a></li>
        
    </ul>
</nav>



    <footer class="site-footer">
    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">
        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

</div>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

