<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.15" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="http://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="http://guangchuangyu.github.io/categories/visualization/">

    <title>Visualization | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="http://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="http://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggtree/2016_inset_img/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a></h2>
        <p><p>I extended the <a href="http://guangchuangyu.github.io/2015/08/subview/">subview</a> function to support embed image file in a <code>ggplot</code> object.</p>

<pre><code>set.seed(123)
d = data.frame(x=rnorm(10), y=rnorm(10))

imgfile &lt;- tempfile(, fileext=&quot;.png&quot;)
download.file(&quot;https://avatars1.githubusercontent.com/u/626539?v=3&amp;u=e731426406dd3f45a73d96dd604bc45ae2e7c36f&amp;s=140&quot;,
              destfile=imgfile, mode='wb')

p = ggplot(d, aes(x, y))
subview(p, imgfile, x=d$x[1], y=d$y[1]) + geom_point(size=5)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/emojifont/emojifont-fontawesome/">Font Awesome supported in emojifont</a></h2>
        <p><p><a href="http://cran.r-project.org/package=emojifont"><img src="http://www.r-pkg.org/badges/version/emojifont" alt="CRAN_Status_Badge" /></a></p>

<p><a href="https://cran.r-project.org/web/packages/emojifont/index.html"><code>emojifont</code></a> is available in CRAN, you can use the following command to install it.</p>

<pre><code>install.packages(&quot;emojifont&quot;)
</code></pre>

<p>An example of using emoji font in R plot is showed below:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2016/douban_emoji.jpg" alt="" /></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2016_ChIPseeker_covplot/">covplot supports GRangesList</a></h2>
        <p><p>To answer the <a href="https://github.com/GuangchuangYu/ChIPseeker/issues/27">issue</a>, I extend the <code>covplot</code> function to support viewing coverage of a list of <code>GRanges</code> objects or <code>bed</code> files.</p>

<pre><code>library(ChIPseeker)
files &lt;- getSampleFiles()
peak=GenomicRanges::GRangesList(CBX6=readPeakFile(files[[4]]),
                                CBX7=readPeakFile(files[[5]]))

p &lt;- covplot(peak)
print(p)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/emojifont/emojifont/">use emoji font in R</a></h2>
        <p><p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot 2015-12-16 10.55.49.png" alt="" /></p>

<p>I have played with emoji in <code>R</code> for a while. My solution of using it is different from what implemented in <a href="https://github.com/dill/emoGG/">emoGG</a>.</p>

<p><code>emoGG</code> is a good attemp to add <code>emoji</code> in <code>ggplot2</code>. It render <code>emoji</code> picture (png) and creat a layer, <code>geom_emoji</code>, to add emoji.</p>

<p>In my opinion, <code>emoji</code> should be treated as ordinary font in user interface, albeit it maynot be true internally.</p>

<p>It would be more flexible if we can use emoji as ordinary font and in this way user don&rsquo;t need to learn extra stuff.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/clusterprofiler/2015_use-simplify-to-remove-redundancy-of-enriched-go-terms/">use simplify to remove redundancy of enriched GO terms</a></h2>
        <p><p>To simplify enriched GO result, we can use slim version of GO and use
<em><a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">enricher</a></em>
function to analyze.</p>

<p>Another strategy is to use
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> to calculate
similarity of GO terms and remove those highly similar terms by keeping
one representative term. To make this feature available to
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>
users, I develop a <em>simplify</em> method to reduce redundant GO terms from
output of <em>enrichGO</em> function.</p>

<pre><code>require(clusterProfiler)
data(geneList, package=&quot;DOSE&quot;)
de &lt;- names(geneList)[abs(geneList) &gt; 2]
bp &lt;- enrichGO(de, ont=&quot;BP&quot;)
enrichMap(bp)
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggtree/2015_subview/">subview</a></h2>
        <p><p>I implemented a function, <em>subview</em>, in
<a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
that make it easy to embed a subplot in ggplot.</p>

<p>An example is shown below:</p>

<pre><code>library(ggplot2)
library(ggtree)

dd &lt;- data.frame(x=LETTERS[1:3], y=1:3)
pie &lt;- ggplot(dd, aes(x=1, y, fill=x)) + 
    geom_bar(stat=&quot;identity&quot;, width=1) + 
    coord_polar(theta=&quot;y&quot;) + theme_tree() + 
    xlab(NULL) + ylab(NULL) + 
    theme_transparent()

x &lt;- sample(2:9)
y &lt;- sample(2:9)
width &lt;- sample(seq(0.05, 0.15, length.out=length(x)))
height &lt;- width

p &lt;- ggplot(data=data.frame(x=c(0, 10), y=c(0, 10)), aes(x, y))+geom_blank()
print(p)
for (i in seq_along(x)) {
    p %&lt;&gt;% subview(pie, x[i], y[i], width[i], height[i])
    print(p)
}
</code></pre>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/chipseeker/2015_upsetplot-in-chipseeker/">upsetplot in ChIPseeker</a></h2>
        <p><p><a href="http://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">ChIPseeker</a>
is an R package for ChIP peak annotation, comparison and visualization.</p>

<p>We have implemented <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">several visualization methods</a>,
including
<em><a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">vennpie</a></em>
that was designed for viewing annotation overlap as shown below:</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/clusterprofiler/2015_dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
        <p><p>This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.</p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggtree/2015_an-example-of-drawing-beast-tree-using-ggtree/">an example of drawing beast tree using ggtree</a></h2>
        <p><p><a href="http://beast.bio.ed.ac.uk/figtree">FigTree</a> is designed for viewing
beast output as demonstrated by their example data:
<img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-10%2018.15.57.png" alt="" /></p>
</p>
    </div>
    
    <div>
        <h2><a href="http://guangchuangyu.github.io/post/r/ggtree/2014_ggtree/">viewing and annotating phylogenetic tree with ggtree</a></h2>
        <p><p>When I need to annotate nucleotide substitutions in the phylogenetic
tree, I found that all the software are designed to display the tree but
not annotating it. Some of them may support annotating the tree with
specific data such as bootstrap values, but they are restricted to a few
supported data types. It is hard/impossible to inject user specific
data.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2014/12/2014-12-21-145753_579x436_scrot.png" alt="" /></p>
</p>
    </div>
    

    
<nav>
    <ul class="pager">
        
        
        <li><a href="http://guangchuangyu.github.io//categories/visualization/page/2/">Next »</a></li>
        
    </ul>
</nav>



    <footer class="site-footer">
    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">
        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

</div>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="http://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

