<!DOCTYPE html>
<html lang="en-us">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="theme" content="hugo-academic">
    <meta name="generator" content="Hugo 0.21" />
    <meta name="author" content="Guangchuang YU">
    <meta name="description" content="PhD student">

    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/highlight.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/font-awesome.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/academicons.min.css">
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-academic.css">
    


    <link rel="shortcut icon" href="https://guangchuangyu.github.io/img/favicon.ico" type="image/x-icon">
    <link rel="canonical" href="https://guangchuangyu.github.io/tags/dose/">

    <title>Dose | Guangchuang YU</title>

</head>
<body id="top">


<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
    <div class="container">

        
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-collapse" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="https://guangchuangyu.github.io/">Guangchuang YU</a>
        </div>

        
        <div class="collapse navbar-collapse" id="#navbar-collapse-1">

            
            <ul class="nav navbar-nav navbar-right">
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#top">Home</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#publications">Publications</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#posts">Posts</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#projects">Projects</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#teaching">Teaching</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/resume">Resume</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/#contact">Contact</a></li>
                
                <li class="nav-item"><a href="https://guangchuangyu.github.io/cn">中文入口</a></li>
                
            </ul>

        </div>
    </div>
</nav>

<div class="container">

    
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/12/dotplot-for-gsea-result/">dotplot for GSEA result</a></h2>
        <div class="article-style"><p><p><img src="https://guangchuangyu.github.io/blog_images/Bioconductor/DOSE/gseaplot.png" alt="" /></p>

<p>For GSEA analysis, we are familar with the above figure which shows the running enrichment score. But for most of the software, it lack of visualization method to summarize the whole enrichment result.</p>

<p>In <a href="https://guangchuangyu.github.io/DOSE">DOSE</a> (and related tools including <a href="https://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a>, <a href="https://guangchuangyu.github.io/ReactomePA">ReactomePA</a> and <a href="https://guangchuangyu.github.io/meshes">meshes</a>), we provide <code>enrichMap</code> and <code>cnetplot</code> to summarize GSEA result.</p>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/10/bioc-34-news-of-my-bioc-packages/">[Bioc 34] NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p>I have 7 packages published within the <code>Bioconductor</code> project.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/meshes">meshes</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA">ReactomePA</a></li>
</ul>

<p>A new package <a href="http://guangchuangyu.github.io/meshes">meshes</a> was included in BioC 3.4 release.</p>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/07/leading-edge-analysis/">leading edge analysis</a></h2>
        <div class="article-style"><p><h2 id="leading-edge-and-core-enrichment">leading edge and core enrichment</h2>

<p>Leading edge analysis reports <code>Tags</code> to indicate the percentage of genes contributing to the enrichment score, <code>List</code> to indicate where in the list the enrichment score is attained and <code>Signal</code> for enrichment signal strength.</p>

<p>It would also be very interesting to get the core enriched genes that contribute to the enrichment.</p>

<p>Now <a href="https://guangchuangyu.github.io/DOSE">DOSE</a>, <a href="https://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a> and <a href="https://guangchuangyu.github.io/ReactomePA">ReactomePA</a> all support leading edge analysis and report core enriched genes.</p>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2016/05/bioc-33-news-of-my-bioc-packages/">[Bioc 33] NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/MISC/ke.jpg" width=600></p>

<p>Today is my birthday and it happened to be the release day of Bioconductor 3.3. It&rsquo;s again the time to reflect what I&rsquo;ve done in the past year.</p>

<ul>
<li><a href="http://guangchuangyu.github.io/ChIPseeker">ChIPseeker</a></li>
<li><a href="http://guangchuangyu.github.io/clusterProfiler">clusterProfiler</a></li>
<li><a href="http://guangchuangyu.github.io/DOSE">DOSE</a></li>
<li><a href="http://guangchuangyu.github.io/ggtree">ggtree</a></li>
<li><a href="http://guangchuangyu.github.io/GOSemSim">GOSemSim</a></li>
<li><a href="http://guangchuangyu.github.io/ReactomePA">ReactomePA</a></li>
</ul>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/10/bioc-32-news-of-my-bioc-packages/">[Bioc 32] NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p>In BioC 3.2 release, all my packages including
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>,
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>,
and <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a> switch
from Sweave to R Markdown for package vignettes.</p>

<h3 id="gosemsim">GOSemSim</h3>

<p>To make it consistent between
<a href="http://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> and
<a href="http://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>,
&lsquo;worm&rsquo; was deprecated and instead we should use &lsquo;celegans&rsquo;. As usual,
information content data was updated.
</p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-for-gtex-paper/">functional enrichment for GTEx paper</a></h2>
        <div class="article-style"><p><p>The ENCODE consortium has recently published <a href="http://www.ncbi.nlm.nih.gov/pubmed/25954002">a great paper</a> on Gene Expression
from the GTEx dataset. A criticism raised on
<a href="https://pubpeer.com/publications/EB99BD086B27075CFF5278F1CEAA57">pubpeer</a>
is that the gene ontology enrichment analysis was done with DAVID which
has not been updated in the last five years.</p>

<p>The result is shown below:
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/Screenshot%202015-07-15%2019.49.01.png" alt="" />
</p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/06/dotplot-for-enrichment-result/">dotplot for enrichment result</a></h2>
        <div class="article-style"><p><p>This is a feature request from <a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
user. It&rsquo;s similar to what I implemented in
<a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
for comparing biological themes. For comparing different enrichment
results, the x-axis represent different gene clusters while for a single
enrichment result, the x-axis can be gene count or gene ratio. This is
actually similar to traditional barplot, with dot position as bar height
and dot color as bar color. But dotplot can represent one more feature
nicely by dot size and it can be a good alternative to barplot.
</p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages/">[Bioc 31] NEWS of my BioC packages</a></h2>
        <div class="article-style"><p><p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2015/03/dose-an-r/bioconductor-package-for-disease-ontology-semantic-and-enrichment-analysis/">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a></h2>
        <div class="article-style"><p><p>My R/Bioconductor package,
<a href="https://github.com/GuangchuangYu/DOSE">DOSE</a>, published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684">Bioinformatics</a>.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-01%2012.44.07.png" alt="" /></p>

<p></p></p></div>

    </div>
    
    <div>
        <h2><a href="https://guangchuangyu.github.io/2011/06/r-package-dose-released/">R package DOSE released</a></h2>
        <div class="article-style"><p><p>Disease Ontology (DO) provides an open source ontology for the
integration of biomedical data that is associated with human disease. DO
analysis can lead to interesting discoveries that deserve further
clinical investigation.</p>

<p>DOSE was designed for semantic similarity measure and enrichment
analysis.</p>

<p>Four information content (IC)-based methods, proposed by Resnik, Jiang,
Lin and Schlicker, and one graph structure-based method, proposed by
Wang, were implemented. The calculation details can be referred to the
vignette of R package
<a href="http://www.bioconductor.org/packages/2.9/bioc/html/GOSemSim.html" title="GOSemSim">GOSemSim</a>.</p>

<p></p></p></div>

    </div>
    

    


</div>
<footer class="site-footer">

    <div style="width:350px;margin:auto;">
  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
    <div class="inner">

    <div class="container">

        <p class="powered-by">

            &copy; 2016 Guangchuang YU &middot; 

            Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

            <span class="pull-right"><a href="#" id="back_to_top"><span class="button_icon"><i class="fa fa-chevron-up fa-2x" aria-hidden="true"></i></span></a></span>

        </p>
    </div>
</footer>

        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
        <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/jquery-1.12.3.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/bootstrap.min.js"></script>
        <script src="https://guangchuangyu.github.io/js/hugo-academic.js"></script>
        

        
        <script>
            (function(i,s,o,g,r,a,m){
                i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},
                    i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;
                m.parentNode.insertBefore(a,m)
            })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
            ga('create', 'UA-77901140-1', 'auto');
            ga('set', 'anonymizeIp', true);
            ga('send', 'pageview');

             
            var links = document.querySelectorAll('a');
            Array.prototype.map.call(links, function(item) {
                if (item.host != document.location.host) {
                    item.addEventListener('click', function() {
                        var action = item.getAttribute('data-action') || 'follow';
                        ga('send', 'event', 'outbound', action, item.href);
                    });
                }
            });
        </script>
        

        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.3.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>

        
        
        <script type="text/x-mathjax-config">
            MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
        </script>
        <script async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML"></script>
        

	
	<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
	<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>


    </body>
</html>

