<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/categories/bioinformatics/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/categories/bioinformatics/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/categories/bioinformatics/">

  

  <title>Bioinformatics | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>




<div class="universal-wrapper">

  <h1>Bioinformatics</h1>

  

  
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output/">showCategory parameter for visualizing compareCluster output</a></h2>
    <div class="article-style">
      
      <blockquote>
<p>I am using dotplot() to visualize results from enrichGO(), enrichDO(), enricher() and compareCluster() in clusterProfiler R package. When specifying showCategory, I get the right number of categories except with the results of compareCluser().</p>

<p>In my case, I use compareCluster() on a list of 3 elements:</p>

<pre><code>str(ClusterList)
List of 3
 $ All : chr [1:1450] &quot;89886&quot; &quot;29923&quot; &quot;100132891&quot; &quot;101410536&quot; ...
 $ g1  : chr [1:858] &quot;89886&quot; &quot;29923&quot; &quot;100132891&quot; &quot;101410536&quot; ...
 $ g2: chr [1:592] &quot;5325&quot; &quot;170691&quot; &quot;29953&quot; &quot;283392&quot; ...
CompareGO_BP=compareCluster(ClusterList, fun=&quot;enrichGO&quot;, pvalueCutoff=0.01, pAdjustMethod=&quot;BH&quot;, OrgDb=org.Hs.eg.db,ont=&quot;BP&quot;,readable=T)

dotplot(CompareGO_BP, showCategory=10, title=&quot;GO - Biological Process&quot;)
</code></pre>

<p>I ask for 10 categories, but I get 15 categories in All, 8 categories in g1 and 12 categories in g2. None of the categories, neither the sum of the categories are 10&hellip;</p>

<p>Is the option showCategory working in the case of comparison? Am I missing something here?</p>

<p>And which categories precisely will it plot? the most significant whatever my 3 cases or the most significant of each case?</p>
</blockquote>

<p>The question was <a href="https://support.bioconductor.org/p/88965/" target="_blank">posted</a> in Bioconductor support site. It seems quite confusing and I think I need to write a post to clarify it.</p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2016/02/reactomepa-an-r/bioconductor-package-for-reactome-pathway-analysis-and-visualization/">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a></h2>
    <div class="article-style">
      
      <p>My R/Bioconductor package,
<a href="https://github.com/GuangchuangYu/ReactomePA" target="_blank">ReactomePA</a>, published in
<a href="http://pubs.rsc.org/en/content/articlelanding/2016/mb/c5mb00663e" target="_blank">Molecular BioSystems</a>.</p>

<p><strong>ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</strong></p>

<p><strong>G Yu</strong>, QY He. <strong><em>Molecular BioSystems</em></strong> 2016, 12:477-479</p>

<p>Received: 05 Oct 2015 Accepted: 20 Nov 2015 Online: 23 Nov 2015</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">functional enrichment analysis with NGS data</a></h2>
    <div class="article-style">
      
      <p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/" target="_blank">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>

<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt="" />
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">parsing BED coordinates</a></h2>
    <div class="article-style">
      
      <p>In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf" target="_blank">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145" target="_blank">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno" target="_blank">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/" target="_blank">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/" target="_blank">PeakAnalyzer</a>.</p>

<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/07/chipseeker-an-r/bioconductor-package-for-chip-peak-annotation-comparison-and-visualization/">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a></h2>
    <div class="article-style">
      
      <p>My R/Bioconductor package,
<a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a>, published
in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145" target="_blank">Bioinformatics</a>.</p>

<p><a href="http://www.bioconductor.org/packages/ChIPseeker" target="_blank">ChIPseeker</a> had been
cited by <a href="http://www.biomedcentral.com/1471-2164/16/292" target="_blank">http://www.biomedcentral.com/1471-2164/16/292</a> and
<a href="http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short" target="_blank">http://www.jbc.org/content/early/2015/06/18/jbc.M115.668558.short</a>, and
was used (not cited) in
<a href="http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract" target="_blank">http://nar.oxfordjournals.org/content/early/2015/06/27/nar.gkv642.abstract</a>
and
<a href="http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract" target="_blank">http://emboj.embopress.org/content/early/2014/12/18/embj.201490061.abstract</a>.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/">BMC favors source code plagiarism</a></h2>
    <div class="article-style">
      
      <p>I found source code plagiarism a year ago and reported this case to <a href="http://www.biomedcentral.com/bmcsystbiol" target="_blank">BMC Systems Biology</a>:</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot 2015-05-27 19.56.58.png" alt="" /></p>

<p>I listed source code of many functions that are exactly copied from
<a href="http://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" target="_blank">GOSemSim</a>
with only function name changed in my email. The detail of source code
plagiarism can also be found at <strong><a href="http://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/" target="_blank">Proper use of
GOSemSim</a></strong>.</p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a></h2>
    <div class="article-style">
      
      <p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/" target="_blank">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/" target="_blank">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp" target="_blank">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/" target="_blank">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/" target="_blank">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
for human) and <a href="http://www.reactome.org/" target="_blank">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide" target="_blank">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/" target="_blank">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/" target="_blank">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.
</p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2015/03/dose-an-r/bioconductor-package-for-disease-ontology-semantic-and-enrichment-analysis/">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a></h2>
    <div class="article-style">
      
      <p>My R/Bioconductor package,
<a href="https://github.com/GuangchuangYu/DOSE" target="_blank">DOSE</a>, published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684" target="_blank">Bioinformatics</a>.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-01%2012.44.07.png" alt="" /></p>

<p></p>

      
    </div>
  </div>
  
  <div>
    <h2><a href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">why clusterProfiler fails</a></h2>
    <div class="article-style">
      
      <p><a href="https://github.com/kaji331" target="_blank">@kaji331</a>
compared <code>cluserProfiler</code> with <code>GeneAnswers</code> and found that clusterProfiler
gives larger p values.</p>

<p>It eventually came out that he passed the input
gene as numeric vector, which was supposed to be character and he used an <strong>old</strong> version of <code>clusterProfiler</code> which didn&rsquo;t use <code>as.character</code> to coerce the input</p>

<p>But his comment forces me to test it.</p>

<p></p>

      
    </div>
  </div>
  

  


</div>
<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

