<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="generator" content="Hugo 0.89.4 with theme Tranquilpeak 0.4.7-BETA">
<meta name="author" content="Guangchuang Yu">
<meta name="keywords" content="">
<meta name="description" content="Guangchuang&#39;s Blog">


<meta property="og:description" content="Guangchuang&#39;s Blog">
<meta property="og:type" content="website">
<meta property="og:title" content="Bioinformatics">
<meta name="twitter:title" content="Bioinformatics">
<meta property="og:url" content="https://guangchuangyu.github.io/categories/bioinformatics/">
<meta property="twitter:url" content="https://guangchuangyu.github.io/categories/bioinformatics/">
<meta property="og:site_name" content="Guangchuang Yu">
<meta property="og:description" content="Guangchuang&#39;s Blog">
<meta name="twitter:description" content="Guangchuang&#39;s Blog">
<meta property="og:locale" content="en-us">


<meta name="twitter:card" content="summary">

  <meta name="twitter:site" content="@guangchuangyu">


  <meta name="twitter:creator" content="@guangchuangyu">










  <meta property="og:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
  <meta property="twitter:image" content="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">


    <title>Bioinformatics</title>

    <link rel="icon" href="https://guangchuangyu.github.io/favicon.png">
    
      <link rel="alternate" type="application/rss+xml" title="RSS" href="https://guangchuangyu.github.io/categories/bioinformatics/index.xml">
    

    

    <link rel="canonical" href="https://guangchuangyu.github.io/categories/bioinformatics/">

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/css/style-twzjdbqhmnnacqs0pwwdzcdbt8yhv8giawvjqjmyfoqnvazl0dalmnhdkvp7.min.css" />
    
    

    
      
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-77901140-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="1">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/#about">
    
    
    
      
        <img class="header-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="1">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/#about">
          <img class="sidebar-profile-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">Bioinformatics Professor @ SMU</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://twitter.com/guangchuangyu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-twitter"></i>
      
      <span class="sidebar-button-desc">Twitter</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/blog_images/biobabble.jpg">
    
      <i class="sidebar-button-icon fa fa-lg fa-weixin"></i>
      
      <span class="sidebar-button-desc">WeChat</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>


    </ul>
  </div>
</nav>

      
        

      
      <div id="main" data-behavior="1"
        class="
               hasCoverMetaIn
               ">
        
          <section class="postShorten-group main-content-wrap">
            
            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output/">
          showCategory parameter for visualizing compareCluster output
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2016-11-03T10:30:00&#43;08:00">
        
  November 3, 2016

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/visualization">Visualization</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <blockquote>
<p>I am using dotplot() to visualize results from enrichGO(), enrichDO(), enricher() and compareCluster() in clusterProfiler R package. When specifying showCategory, I get the right number of categories except with the results of compareCluser().</p>
</blockquote>
<blockquote>
<p>In my case, I use compareCluster() on a list of 3 elements:</p>
</blockquote>
<blockquote>
<pre tabindex="0"><code></code></pre></blockquote>
<p>str(ClusterList)
List of 3
$ All : chr [1:1450] &ldquo;89886&rdquo; &ldquo;29923&rdquo; &ldquo;100132891&rdquo; &ldquo;101410536&rdquo; &hellip;
$ g1  : chr [1:858] &ldquo;89886&rdquo; &ldquo;29923&rdquo; &ldquo;100132891&rdquo; &ldquo;101410536&rdquo; &hellip;
$ g2: chr [1:592] &ldquo;5325&rdquo; &ldquo;170691&rdquo; &ldquo;29953&rdquo; &ldquo;283392&rdquo; &hellip;</p>
<blockquote>
<p>CompareGO_BP=compareCluster(ClusterList, fun=&ldquo;enrichGO&rdquo;, pvalueCutoff=0.01, pAdjustMethod=&ldquo;BH&rdquo;, OrgDb=org.Hs.eg.db,ont=&ldquo;BP&rdquo;,readable=T)</p>
</blockquote>
<blockquote>
<p>dotplot(CompareGO_BP, showCategory=10, title=&ldquo;GO - Biological Process&rdquo;)</p>
<pre tabindex="0"><code></code></pre></blockquote>
<blockquote>
<p>I ask for 10 categories, but I get 15 categories in All, 8 categories in g1 and 12 categories in g2. None of the categories, neither the sum of the categories are 10&hellip;</p>
</blockquote>
<blockquote>
<p>Is the option showCategory working in the case of comparison? Am I missing something here?</p>
</blockquote>
<blockquote>
<p>And which categories precisely will it plot? the most significant whatever my 3 cases or the most significant of each case?</p>
</blockquote>
<p>The question was <a href="https://support.bioconductor.org/p/88965/">posted</a> in Bioconductor support site. It seems quite confusing and I think I need to write a post to clarify it.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2016/11/showcategory-parameter-for-visualizing-comparecluster-output/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/">
          functional enrichment analysis with NGS data
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-08-21T03:22:21&#43;08:00">
        
  August 21, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/genomics">Genomics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>I found a Bioconductor package,
<a href="http://www.bioconductor.org/packages/seq2pathway/">seq2pathway</a>, that
can apply functional analysis to NGS data. It consists of two
components, <em>seq2gene</em> and <em>gene2pathway</em>. <em>seq2gene</em> converts genomic
coordination to genes while <em>gene2pathway</em> performs functional analysis
at gene level.</p>
<p>I think it would be interesting to incorporate <em>seq2gene</em> with
clusterProfiler. But it fail to run due to it call absolute path of
python installed in the author&rsquo;s computer.
<img src="http://guangchuangyu.github.io/blog_images/Bioconductor/others/Screenshot%202015-08-20%2015.25.52.png" alt=""></p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/08/functional-enrichment-analysis-with-ngs-data/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/">
          parsing BED coordinates
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-08-07T03:03:08&#43;08:00">
        
  August 7, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>In <a href="http://bioinformatics.oxfordjournals.org/content/suppl/2015/03/11/btv145.DC1/TSS_annotation_comparision.pdf">supplemental file</a>
of ChIPseeker <a href="http://doi.org/10.1093/bioinformatics/btv145">paper</a>, I
compare distances to TSS reported by several ChIP annotation software,
including <a href="http://www.bioconductor.org/packages/ChIPseeker">ChIPseeker</a>,
<a href="http://www.bioconductor.org/packages/ChIPpeakAnno">ChIPpeakAnno</a>,
<a href="http://homer.salk.edu/homer/chipseq/">HOMER</a> and
<a href="http://www.bioinformatics.org/peakanalyzer/wiki/">PeakAnalyzer</a>.</p>
<p>Although I found that the chromStart positions in HOMER output have a +1
shift compare to other software, I did not realize this issue since all
other software are consistent.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/08/parsing-bed-coordinates/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/">
          BMC favors source code plagiarism
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-05-27T13:38:44&#43;08:00">
        
  May 27, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/research">Research</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p>I found source code plagiarism a year ago and reported this case to <a href="http://www.biomedcentral.com/bmcsystbiol">BMC Systems Biology</a>:</p>
<p>![](<a href="http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot">http://guangchuangyu.github.io/blog_images/2015/plagiarism/Screenshot</a> 2015-05-27 19.56.58.png)</p>
<p>I listed source code of many functions that are exactly copied from
<a href="http://www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
with only function name changed in my email. The detail of source code
plagiarism can also be found at <strong><a href="http://guangchuangyu.github.io/2014/11/proper-use-of-gosemsim/">Proper use of
GOSemSim</a></strong>.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/05/bmc-favors-source-code-plagiarism/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">
          use clusterProfiler as an universal enrichment analysis tool
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2015-05-11T08:32:04&#43;08:00">
        
  May 11, 2015

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/systems-biology">Systems Biology</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
supports enrichment analysis of both hypergeometric test and gene set
enrichment analysis. It internally supports <a href="http://geneontology.org/">Gene Ontology</a> analysis of about 20 species, <a href="http://www.genome.jp/kegg/">Kyoto Encyclopedia of Genes and Genomes (KEGG)</a>
with all species that have annotation available in KEGG database, <a href="http://david.abcc.ncifcrf.gov/summary.jsp">DAVID annotation</a> (only
hypergeometric test supported), <a href="http://disease-ontology.org/">Disease Ontology</a> and <a href="http://ncg.kcl.ac.uk/">Network of Cancer Genes</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>
for human) and <a href="http://www.reactome.org/">Reactome Pathway</a> (via
<a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>
for several species). This is still not enough for users may want to
analyze their data with unsupported organisms, <a href="http://geneontology.org/page/go-slim-and-subset-guide">slim version of
GO</a>, novel
functional annotation (eg GO via <a href="https://www.blast2go.com/">blastgo</a>
and KEGG via <a href="http://www.genome.jp/tools/kaas/">KAAS</a>), unsupported
ontology/pathway or customized annotation.</p>
<p><a href="http://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
provides <em>enricher</em> function for hypergeometric test and <em>GSEA</em> function
for gene set enrichment analysis that are designed to accept user
defined annotation. They accept two additional parameters TERM2GENE and
TERM2NAME. As indicated in the parameter names, TERM2GENE is a
data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/">
          why clusterProfiler fails
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2014-08-07T05:46:35&#43;08:00">
        
  August 7, 2014

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/r">R</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        <p><a href="https://github.com/kaji331">@kaji331</a>
compared <code>cluserProfiler</code> with <code>GeneAnswers</code> and found that clusterProfiler
gives larger p values.</p>
<p>It eventually came out that he passed the input
gene as numeric vector, which was supposed to be character and he used an <strong>old</strong> version of <code>clusterProfiler</code> which didn&rsquo;t use <code>as.character</code> to coerce the input</p>
<p>But his comment forces me to test it.</p>
      
      <p>
        <a href="https://guangchuangyu.github.io/2014/08/why-clusterprofiler-fails/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
              
  
    
      
        
      
    
  


  

<article class="postShorten postShorten--thumbnailimg-bottom" itemscope itemType="http://schema.org/BlogPosting">
  <div class="postShorten-wrap">
    
    <div class="postShorten-header">
      <h1 class="postShorten-title" itemprop="headline">
        <a class="link-unstyled" href="https://guangchuangyu.github.io/2010/11/phosphoproteome-profile-of-human-lung-cancer-cell-line-a549/">
          Phosphoproteome profile of human lung cancer cell line A549
        </a>
      </h1>
      
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2010-11-23T02:22:22&#43;08:00">
        
  November 23, 2010

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/bioinformatics">Bioinformatics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/proteomics">Proteomics</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/categories/publication">Publication</a>
    
  

  </div>

    </div>
    <div class="postShorten-excerpt" itemprop="articleBody">
      
        As an in vitro model for type II human lung cancer, A549 cells resist cytotoxicity via phosphorylation of proteins as demonstrated by many studies. However, to date, no large-scale phosphoproteome investigation has been conducted on A549. Here, we performed a systematical analysis of the phosphoproteome of A549 by using mass spectrometry (MS)-based strategies. This investigation led to the identification of 337 phosphorylation sites on 181 phosphoproteins. Among them, 67 phosphoproteins and 230 phosphorylation sites identified appeared to be novel with no previous characterization in lung cancer.
      
      <p>
        <a href="https://guangchuangyu.github.io/2010/11/phosphoproteome-profile-of-human-lung-cancer-cell-line-a549/" class="postShorten-excerpt_link link">Continue reading</a>
        
      </p>
    </div>
  </div>
  
</article>

            
            
  <div class="pagination-bar">
    <ul class="pagination">
      
        
        
          <li class="pagination-next">
            <a class="btn btn--default btn--small" href="https://guangchuangyu.github.io/categories/bioinformatics/page/2/">
              <span>OLDER POSTS</span>
              <i class="fa fa-angle-right text-base icon-ml"></i>
            </a>
          </li>
        
      
      <li class="pagination-number">page 1 of 2</li>
    </ul>
  </div>


          </section>
        
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2021 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="https://www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">Bioinformatics Professor @ SMU</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Bioinformatics Professor
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Guangzhou
      </div>
    
  </div>
</div>

    

    
  
    
      <div id="cover" style="background-image:url('https://guangchuangyu.github.io/images/cover.jpg');"></div>
    
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.7/js/jquery.fancybox.min.js" integrity="sha256-GEAnjcTqVP+vBp3SSc8bEDQqvWAZMiHyUSIorrWwH50=" crossorigin="anonymous"></script>


<script src="https://guangchuangyu.github.io/js/script-pcw6v3xilnxydl1vddzazdverrnn9ctynvnxgwho987mfyqkuylcb1nlt.min.js"></script>


<script lang="javascript">
window.onload = updateMinWidth;
window.onresize = updateMinWidth;
document.getElementById("sidebar").addEventListener("transitionend", updateMinWidth);
function updateMinWidth() {
  var sidebar = document.getElementById("sidebar");
  var main = document.getElementById("main");
  main.style.minWidth = "";
  var w1 = getComputedStyle(main).getPropertyValue("min-width");
  var w2 = getComputedStyle(sidebar).getPropertyValue("width");
  var w3 = getComputedStyle(sidebar).getPropertyValue("left");
  main.style.minWidth = `calc(${w1} - ${w2} - ${w3})`;
}
</script>

<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight > code, pre > code').each(function(i, block) {
    if (!$(this).hasClass('codeblock')) {
      $(this).addClass('codeblock');
    }
    hljs.highlightBlock(block);
  });
});
</script>





    
  </body>
</html>

