<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Bioc 3.1: NEWS of my BioC packages</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            Bioc 3.1: NEWS of my BioC packages 
          </h1>
          <p class="meta">Apr 20, 2015 - 3 minute read -  <a href="http://guangchuangyu.github.io/2015/04/bioc-3.1-news-of-my-bioc-packages/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
          

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim:d7b2a91b2c7afdf12777a4b5cd75c8f0">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/a9ed4c29e8f66c07de917937694d7f43933d4fe5/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>

<h3 id="clusterprofiler:d7b2a91b2c7afdf12777a4b5cd75c8f0">clusterProfiler</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
was developed in
<a href="http://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8/">2011</a> and
published in
<a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118">OMICS</a> in
2012. I am very glad to find that
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
was presented in the book &lsquo;<a href="https://www.safaribooksonline.com/library/view/bioinformatics-with-r/9781783283132/ch02s08.html">Bioinformatics with R Cookbook</a>&lsquo;:
In this release, KEGG enrichment analysis (both hypergeometric and GSEA)
now supports <a href="http://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/">using latest online data</a>
and <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/">DAVID functional analysis</a>
is also supported. In addition, compareCluster function now supports
<a href="http://bioinfoblog.it/2015/02/a-formula-interface-for-geneontology-analysis/">formula
interface</a>
like x ~ y + z; this feature was contributed by
<a href="https://twitter.com/dalloliogm">@dalloliogm</a>. Considering that many new
users have problem of converting biological IDs,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
contains a new function <em>bitr</em>, stands for Biological Id TranslatoR.
NOW,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a>
is not restricted to GO and KEGG analyses. All kinds of annotation can
be supported via <em>enricher</em> function for hypergeometric test and <em>GSEA</em>
function for gene set enrichment analysis. These two functions are not
yet documented in vignette and will soon.</p>

<h3 id="dose:d7b2a91b2c7afdf12777a4b5cd75c8f0">DOSE</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>
was first released in
<a href="http://guangchuangyu.github.io/2011/06/r-package-dose-released/">2011</a>, and has been
published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684">Bioinformatics</a>
on February 2015. An example of applying DOSE for disease ontology
enrichment analysis can be found in <a href="http://www.sciencedirect.com/science/article/pii/S0002929714002626">Genome-wide Identification and Characterization of Fixed Human-Specific Regulatory Regions</a>.
Besides,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html">DOSE</a>
also supports <a href="http://bioinfoblog.it/2015/04/ncg-enrichment-implemented-in-dose/">Network of Cancer Genes database</a>.</p>

<h3 id="reactomepa:d7b2a91b2c7afdf12777a4b5cd75c8f0">ReactomePA</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html">ReactomePA</a>
was developed in 2012 when KEGG is no longer available in public domain.
In BioC 3.0, reactome.db has <a href="https://support.bioconductor.org/p/63024/">a lot of issues</a>. Until the early of
this April, Bioconductor restored reactome.db in Bioc 3.0 to Bioc 2.14.
This issue affects ReactomePA several months and was removed in BioC
3.1.</p>

<h3 id="chipseeker:d7b2a91b2c7afdf12777a4b5cd75c8f0">ChIPseeker</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
was developed in
<a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/">2014</a>,
after I found a <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/">bug of ChIPpeakAnno</a>
package. Many new features were added thereafter, including <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/">several visualization methods</a>,
<a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/">multiple annotations</a>,
ChIP-peak comparison and data mining with GEO database. ChIPseeker has
been published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145">Bioinformatics</a>.
In this release, several minor bugs were fixed, a new parameter
<em>genomicAnnotationPriority</em> was added in <em>annotatePeak</em> and
<em>plotAvgProf</em> supports estimating confidence interval as shown below:
<img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/dd399ab1d2287d0c7fee1009abf3b1b95553c73a/2015/Screenshot%202015-03-30%2020.09.44.png" alt="" />
This CI estimation feature was contributed by <a href="https://github.com/Puriney">Yan Yun</a>.
<a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html">ChIPseeker</a>
works fine with other ChIP-seq related tools as demonstrated in the
<a href="http://compbio.ucdenver.edu/Hunter_lab/Phang/resources/Tzu-Software/ChIPseq.Analysis.html">ChIPseq Analysis workflow</a>.
It has also been presented in <a href="http://physiology.med.cornell.edu/faculty/elemento/lab/data/courses/2014/CSHLseq/ChIP-seq.pdf">2014 CSHLseq course</a>
and <a href="https://github.com/Bioconductor/UseBioconductor/blob/master/vignettes/F_ChIPSeq.Rmd">2015 Use Bioconductor course</a>.</p>

<h3 id="ggtree:d7b2a91b2c7afdf12777a4b5cd75c8f0">ggtree</h3>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot-2015-02-10-10.27.27.png" alt="" /></p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>
was developed on <a href="http://guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/">December 2014</a>,
when I got tired of annotating tree by modifying node labels. This
release (version=1.0.0) is the first release, output of many common
software in evolution were supported and many new features were
introduced in the past few months. Please view the
<a href="http://www.bioconductor.org/packages/3.1/bioc/vignettes/ggtree/inst/doc/ggtree.html">vignette</a>
to find out more and I can guarantee that you won&rsquo;t be disappointed.</p>

<span class='st_twitter_hcount' displayText='Tweet'></span>
<span class='st_facebook_hcount' displayText='Facebook'></span>
<span class='st_sina_hcount' displayText='Sina'></span>
<span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Apr 20, 2015</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/bioc">BioC</a>  <a class="category" href="http://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>  <a class="category" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>  <a class="category" href="http://guangchuangyu.github.io/tags/dose">DOSE</a>  <a class="category" href="http://guangchuangyu.github.io/tags/reactomepa">ReactomePA</a>  <a class="category" href="http://guangchuangyu.github.io/tags/ggtree">ggtree</a>  <a class="category" href="http://guangchuangyu.github.io/tags/chipseeker">ChIPseeker</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/" title="an example of drawing beast tree using ggtree">an example of drawing beast tree using ggtree</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/" title="using GOSemSim to rank proteins obtained by co-IP">using GOSemSim to rank proteins obtained by co-IP</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>

  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/%E4%BD%BF%E7%94%A8ggtree%E5%AE%9E%E7%8E%B0%E8%BF%9B%E5%8C%96%E6%A0%91%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E5%92%8C%E6%B3%A8%E9%87%8A/">使用ggtree实现进化树的可视化和注释</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/covplot-supports-grangeslist/">covplot supports GRangesList</a>
      </li>
      
      <li class="post">
        <a href="/2016/02/tweets-of-ggtree/">tweets of ggtree</a>
      </li>
      
    </ul>
  </section>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>

<ins class="adsbygoogle"
     style="display:inline-block;width:728px;height:90px"
     data-ad-client="ca-pub-9874493018367687"
     data-ad-slot="5219143855"></ins>
<script>
(adsbygoogle = window.adsbygoogle || []).push({});
</script>


</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


