<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>GO analysis using clusterProfiler</title>

  
  <link rel="stylesheet" href="http://guangchuangyu.github.io/css/hugo-octopress.css">

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  

  
  <link href="http://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  

</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="http://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">


<ul class="main-navigation">
  
  
    
      <li><a href="http://guangchuangyu.github.io/" title="Blog">Blog</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="http://guangchuangyu.github.io/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="http://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>



<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            GO analysis using clusterProfiler 
          </h1>
          <p class="meta">Jan 4, 2016 - 9 minute read -  <a href="http://guangchuangyu.github.io/2016/01/go-analysis-using-clusterprofiler/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="http://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
	  	  
	  

<p><code>clusterProfiler</code> supports over-representation test and gene set
enrichment analysis of Gene Ontology. It supports GO annotation from
OrgDb object, GMT file and user&rsquo;s own data.</p>

<h2 id="support-many-species:bdc631894cab1e81ad84e5406d79b479">support many species</h2>

<p>In github version of clusterProfiler, <code>enrichGO</code> and <code>gseGO</code> functions
removed the parameter <em>organism</em> and add another parameter <em>OrgDb</em>, so
that any species that have <code>OrgDb</code> object available can be analyzed in
<strong>clusterProfiler</strong>. Bioconductor have already provide OrgDb for about
20 species, see
<a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">http://bioconductor.org/packages/release/BiocViews.html#___OrgDb</a>, and
users can build <code>OrgDb</code> via <code>AnnotationHub</code>.</p>

<pre><code>library(AnnotationHub)
hub &lt;- AnnotationHub()

## snapshotDate(): 2015-12-29

query(hub, &quot;Cricetulus&quot;)

## AnnotationHub with 4 records
## # snapshotDate(): 2015-12-29 
## # $dataprovider: UCSC, Inparanoid8, ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/
## # $species: Cricetulus griseus
## # $rdataclass: ChainFile, Inparanoid8Db, OrgDb, TwoBitFile
## # additional mcols(): taxonomyid, genome, description, tags,
## #   sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[[&quot;AH10393&quot;]]' 
## 
##             title                             
##   AH10393 | hom.Cricetulus_griseus.inp8.sqlite
##   AH13980 | criGri1.2bit                      
##   AH14346 | criGri1ToHg19.over.chain.gz       
##   AH48061 | org.Cricetulus_griseus.eg.sqlite

Cgriseus &lt;- hub[[&quot;AH48061&quot;]]

## loading from cache '/Users/guangchuangyu/.AnnotationHub/54367'

Cgriseus

## OrgDb object:
## | DBSCHEMAVERSION: 2.1
## | DBSCHEMA: NOSCHEMA_DB
## | ORGANISM: Cricetulus griseus
## | SPECIES: Cricetulus griseus
## | CENTRALID: GID
## | Taxonomy ID: 10029
## | Db type: OrgDb
## | Supporting package: AnnotationDbi

## 
## Please see: help('select') for usage information

sample_gene &lt;- sample(keys(Cgriseus), 100)
str(sample_gene)

##  chr [1:100] &quot;100762355&quot; &quot;100757285&quot; &quot;100773870&quot; &quot;100766902&quot; ...

library(clusterProfiler)

sample_test &lt;- enrichGO(sample_gene, OrgDb=Cgriseus, pvalueCutoff=1, qvalueCutoff=1)
head(summary(sample_test))

##                    ID                                 Description
## GO:0004983 GO:0004983            neuropeptide Y receptor activity
## GO:0005254 GO:0005254                   chloride channel activity
## GO:0005496 GO:0005496                             steroid binding
## GO:0005253 GO:0005253                      anion channel activity
## GO:0015108 GO:0015108 chloride transmembrane transporter activity
## GO:0019887 GO:0019887           protein kinase regulator activity
##            GeneRatio BgRatio     pvalue  p.adjust    qvalue    geneID
## GO:0004983      1/20  6/3946 0.03004660 0.6187407 0.6138746 100773047
## GO:0005254      1/20  6/3946 0.03004660 0.6187407 0.6138746 100773701
## GO:0005496      1/20  6/3946 0.03004660 0.6187407 0.6138746 100689048
## GO:0005253      1/20  8/3946 0.03987010 0.6187407 0.6138746 100773701
## GO:0015108      1/20  8/3946 0.03987010 0.6187407 0.6138746 100773701
## GO:0019887      1/20 12/3946 0.05923425 0.6187407 0.6138746 100763034
##            Count
## GO:0004983     1
## GO:0005254     1
## GO:0005496     1
## GO:0005253     1
## GO:0015108     1
## GO:0019887     1
</code></pre>

<h2 id="support-many-id-types:bdc631894cab1e81ad84e5406d79b479">support many ID types</h2>

<p>The input ID type can be any type that was supported in <code>OrgDb</code> object.</p>

<pre><code>library(org.Hs.eg.db)
data(geneList)
gene &lt;- names(geneList)[abs(geneList) &gt; 2]
gene.df &lt;- bitr(gene, fromType = &quot;ENTREZID&quot;, 
        toType = c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;),
        OrgDb = org.Hs.eg.db)

## 'select()' returned 1:many mapping between keys and columns

## Warning in bitr(gene, fromType = &quot;ENTREZID&quot;, toType = c(&quot;ENSEMBL&quot;,
## &quot;SYMBOL&quot;), : 0.48% of input gene IDs are fail to map...

head(gene.df)

##   ENTREZID         ENSEMBL SYMBOL
## 1     4312 ENSG00000196611   MMP1
## 2     8318 ENSG00000093009  CDC45
## 3    10874 ENSG00000109255    NMU
## 4    55143 ENSG00000134690  CDCA8
## 5    55388 ENSG00000065328  MCM10
## 6      991 ENSG00000117399  CDC20

ego &lt;- enrichGO(gene          = gene,
                universe      = names(geneList),
                OrgDb         = org.Hs.eg.db,
                ont           = &quot;CC&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05)
head(summary(ego))

##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    24/197
## GO:0005876 GO:0005876                      spindle microtubule    11/197
## GO:0000793 GO:0000793                     condensed chromosome    17/197
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    13/197
## GO:0005875 GO:0005875           microtubule associated complex    14/197
## GO:0015630 GO:0015630                 microtubule cytoskeleton    36/197
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0005819 222/11632 3.810608e-13 1.276554e-10 1.139171e-10
## GO:0005876  45/11632 1.527089e-10 2.557874e-08 2.282596e-08
## GO:0000793 150/11632 5.838332e-10 6.519471e-08 5.817847e-08
## GO:0000779  81/11632 8.684319e-10 7.273117e-08 6.490386e-08
## GO:0005875 109/11632 3.936298e-09 2.637319e-07 2.353492e-07
## GO:0015630 765/11632 1.719925e-08 9.602916e-07 8.569452e-07
##                                                                                                                                                                                                       geneID
## GO:0005819                                                                   55143/991/9493/1062/259266/9787/220134/51203/22974/4751/983/4085/81930/332/3832/7272/9212/9055/3833/146909/10112/6790/891/24137
## GO:0005876                                                                                                                                       220134/51203/983/81930/332/3832/9212/9055/146909/6790/24137
## GO:0000793                                                                                                       1062/10403/7153/23397/55355/220134/4751/79019/55839/54821/4085/332/64151/9212/1111/6790/891
## GO:0000779                                                                                                                             1062/10403/55355/220134/4751/79019/55839/54821/4085/332/9212/6790/891
## GO:0005875                                                                                                                        55143/9493/1062/81930/332/3832/9212/3833/146909/10112/6790/24137/4137/7802
## GO:0015630 8318/55143/991/9493/1062/9133/7153/259266/55165/9787/220134/51203/22974/10460/4751/983/54821/4085/81930/332/3832/7272/64151/9212/1111/9055/3833/146909/10112/51514/6790/891/24137/26289/4137/7802
##            Count
## GO:0005819    24
## GO:0005876    11
## GO:0000793    17
## GO:0000779    13
## GO:0005875    14
## GO:0015630    36

ego2 &lt;- enrichGO(gene         = gene.df$ENSEMBL,
                OrgDb         = org.Hs.eg.db,
                keytype       = 'ENSEMBL',
                ont           = &quot;CC&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05)
head(summary(ego2))

##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    28/220
## GO:0005875 GO:0005875           microtubule associated complex    19/220
## GO:0005876 GO:0005876                      spindle microtubule    12/220
## GO:0015630 GO:0015630                 microtubule cytoskeleton    43/220
## GO:0005874 GO:0005874                              microtubule    26/220
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    14/220
##               BgRatio       pvalue     p.adjust       qvalue
## GO:0005819  298/19428 6.831911e-18 2.336514e-15 1.812254e-15
## GO:0005875  157/19428 1.710039e-14 2.924167e-12 2.268052e-12
## GO:0005876   54/19428 7.427958e-13 8.467872e-11 6.567879e-11
## GO:0015630 1118/19428 1.228816e-12 1.050638e-10 8.148992e-11
## GO:0005874  421/19428 2.266638e-12 1.550381e-10 1.202511e-10
## GO:0000779  110/19428 2.652610e-11 1.444006e-09 1.120005e-09
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     geneID
## GO:0005819                                                                                                                                                                                                                                                 ENSG00000134690/ENSG00000117399/ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000126787/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000117650/ENSG00000170312/ENSG00000164109/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000112742/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000056678/ENSG00000233450/ENSG00000204197/ENSG00000186185/ENSG00000112984/ENSG00000087586/ENSG00000134057/ENSG00000090889
## GO:0005875                                                                                                                                                                                                                                                                                                                                                                                                 ENSG00000134690/ENSG00000137807/ENSG00000138778/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000178999/ENSG00000237649/ENSG00000056678/ENSG00000233450/ENSG00000204197/ENSG00000186185/ENSG00000112984/ENSG00000087586/ENSG00000090889/ENSG00000186868/ENSG00000276155/ENSG00000277956/ENSG00000163879
## GO:0005876                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000170312/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000178999/ENSG00000198901/ENSG00000186185/ENSG00000087586/ENSG00000090889
## GO:0015630 ENSG00000093009/ENSG00000134690/ENSG00000117399/ENSG00000137807/ENSG00000138778/ENSG00000157456/ENSG00000131747/ENSG00000066279/ENSG00000138180/ENSG00000126787/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000013810/ENSG00000117650/ENSG00000170312/ENSG00000186871/ENSG00000164109/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000112742/ENSG00000109805/ENSG00000178999/ENSG00000149554/ENSG00000198901/ENSG00000237649/ENSG00000056678/ENSG00000233450/ENSG00000204197/ENSG00000186185/ENSG00000112984/ENSG00000143476/ENSG00000087586/ENSG00000134057/ENSG00000090889/ENSG00000127603/ENSG00000154027/ENSG00000186868/ENSG00000276155/ENSG00000277956/ENSG00000163879
## GO:0005874                                                                                                                                                                                                                                                                                 ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000117650/ENSG00000170312/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000056678/ENSG00000233450/ENSG00000204197/ENSG00000186185/ENSG00000112984/ENSG00000087586/ENSG00000090889/ENSG00000127603/ENSG00000186868/ENSG00000276155/ENSG00000277956
## GO:0000779                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ENSG00000138778/ENSG00000080986/ENSG00000123485/ENSG00000154839/ENSG00000262634/ENSG00000117650/ENSG00000100162/ENSG00000166451/ENSG00000186871/ENSG00000164109/ENSG00000089685/ENSG00000178999/ENSG00000087586/ENSG00000134057
##            Count
## GO:0005819    28
## GO:0005875    19
## GO:0005876    12
## GO:0015630    43
## GO:0005874    26
## GO:0000779    14

ego3 &lt;- enrichGO(gene         = gene.df$SYMBOL,
                OrgDb         = org.Hs.eg.db,
                keytype       = 'SYMBOL',
                ont           = &quot;CC&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05)
head(summary(ego3))

##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    24/196
## GO:0005876 GO:0005876                      spindle microtubule    11/196
## GO:0000793 GO:0000793                     condensed chromosome    17/196
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    13/196
## GO:0015630 GO:0015630                 microtubule cytoskeleton    36/196
## GO:0005875 GO:0005875           microtubule associated complex    14/196
##               BgRatio       pvalue     p.adjust       qvalue
## GO:0005819  278/17761 6.023611e-15 2.042004e-12 1.769039e-12
## GO:0005876   52/17761 9.080301e-12 1.539111e-09 1.333370e-09
## GO:0000793  192/17761 4.363319e-11 4.930551e-09 4.271460e-09
## GO:0000779  103/17761 1.083989e-10 9.186804e-09 7.958759e-09
## GO:0015630 1034/17761 6.842818e-10 3.952252e-08 3.423935e-08
## GO:0005875  144/17761 6.995136e-10 3.952252e-08 3.423935e-08
##                                                                                                                                                                                                                        geneID
## GO:0005819                                                                      CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0005876                                                                                                                                                  SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
## GO:0000793                                                                                                              CENPE/NDC80/TOP2A/NCAPH/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/BIRC5/NCAPG/AURKB/CHEK1/AURKA/CCNB1
## GO:0000779                                                                                                                                      CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/BIRC5/AURKB/AURKA/CCNB1
## GO:0015630 CDC45/CDCA8/CDC20/KIF23/CENPE/CCNB2/TOP2A/ASPM/CEP55/DLGAP5/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/KIF11/TTK/NCAPG/AURKB/CHEK1/PRC1/KIFC1/KIF18B/KIF20A/DTL/AURKA/CCNB1/KIF4A/AK5/MAPT/DNALI1
## GO:0005875                                                                                                                             CDCA8/KIF23/CENPE/KIF18A/BIRC5/KIF11/AURKB/KIFC1/KIF18B/KIF20A/AURKA/KIF4A/MAPT/DNALI1
##            Count
## GO:0005819    24
## GO:0005876    11
## GO:0000793    17
## GO:0000779    13
## GO:0015630    36
## GO:0005875    14
</code></pre>

<p>Using <code>SYMBOL</code> directly is not recommended. User can use <code>setReadable</code>
function to translate <code>geneID</code> to gene symbol.</p>

<pre><code>ego &lt;- setReadable(ego, OrgDb = org.Hs.eg.db)
ego2 &lt;- setReadable(ego2, OrgDb = org.Hs.eg.db)
head(summary(ego), n=3)

##                    ID          Description GeneRatio   BgRatio
## GO:0005819 GO:0005819              spindle    24/197 222/11632
## GO:0005876 GO:0005876  spindle microtubule    11/197  45/11632
## GO:0000793 GO:0000793 condensed chromosome    17/197 150/11632
##                  pvalue     p.adjust       qvalue
## GO:0005819 3.810608e-13 1.276554e-10 1.139171e-10
## GO:0005876 1.527089e-10 2.557874e-08 2.282596e-08
## GO:0000793 5.838332e-10 6.519471e-08 5.817847e-08
##                                                                                                                                                   geneID
## GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0005876                                                                             SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
## GO:0000793                                         CENPE/NDC80/TOP2A/NCAPH/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/BIRC5/NCAPG/AURKB/CHEK1/AURKA/CCNB1
##            Count
## GO:0005819    24
## GO:0005876    11
## GO:0000793    17

head(summary(ego2), n=3)

##                    ID          Description GeneRatio   BgRatio
## GO:0005819 GO:0005819              spindle    24/197 222/11632
## GO:0005876 GO:0005876  spindle microtubule    11/197  45/11632
## GO:0000793 GO:0000793 condensed chromosome    17/197 150/11632
##                  pvalue     p.adjust       qvalue
## GO:0005819 3.810608e-13 1.276554e-10 1.139171e-10
## GO:0005876 1.527089e-10 2.557874e-08 2.282596e-08
## GO:0000793 5.838332e-10 6.519471e-08 5.817847e-08
##                                                                                                                                                   geneID
## GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0005876                                                                             SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
## GO:0000793                                         CENPE/NDC80/TOP2A/NCAPH/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/BIRC5/NCAPG/AURKB/CHEK1/AURKA/CCNB1
##            Count
## GO:0005819    24
## GO:0005876    11
## GO:0000793    17
</code></pre>

<p><code>enrichGO</code> test the whole GO corpus and enriched result may contains
very general terms. User can use <code>dropGO</code> function to remove specific GO
terms or GO level. If user want to restrict the result at sepcific GO
level, they can use <code>gofilter</code> function. We also provide a <code>simplify</code>
method to reduce redundancy of enriched GO terms, see <a href="http://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">the
post</a>.</p>

<h2 id="visualization-functions:bdc631894cab1e81ad84e5406d79b479">Visualization functions</h2>

<pre><code>dotplot(ego, showCategory=30)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/2016_GO_analysis_using_clusterProfiler_files/figure-markdown_strict/unnamed-chunk-7-1.png" alt="" /></p>

<pre><code>enrichMap(ego, vertex.label.cex=1.2, layout=igraph::layout.kamada.kawai)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/2016_GO_analysis_using_clusterProfiler_files/figure-markdown_strict/unnamed-chunk-7-2.png" alt="" /></p>

<pre><code>cnetplot(ego, foldChange=geneList)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/2016_GO_analysis_using_clusterProfiler_files/figure-markdown_strict/unnamed-chunk-7-3.png" alt="" /></p>

<pre><code>plotGOgraph(ego)

## 
## groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.
## 
## Building most specific GOs ..... ( 335 GO terms found. )
## 
## Build GO DAG topology .......... ( 335 GO terms and 667 relations. )
## 
## Annotating nodes ............... ( 11632 genes annotated to the GO terms. )
## $dag
## A graphNEL graph with directed edges
## Number of Nodes = 29 
## Number of Edges = 50 
## 
## $complete.dag
## [1] &quot;A graph with 29 nodes.&quot;
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/2016_GO_analysis_using_clusterProfiler_files/figure-markdown_strict/unnamed-chunk-7-4.png" alt="" /></p>

<h2 id="gene-set-enrichment-analysis:bdc631894cab1e81ad84e5406d79b479">Gene Set Enrichment Analysis</h2>

<pre><code>gsecc &lt;- gseGO(geneList=geneList, ont=&quot;CC&quot;, OrgDb=org.Hs.eg.db, verbose=F)
head(summary(gsecc))

##                    ID                              Description setSize
## GO:0031982 GO:0031982                                  vesicle    2880
## GO:0031988 GO:0031988                 membrane-bounded vesicle    2791
## GO:0005576 GO:0005576                     extracellular region    3296
## GO:0065010 GO:0065010 extracellular membrane-bounded organelle    2220
## GO:0070062 GO:0070062                    extracellular exosome    2220
## GO:0044421 GO:0044421                extracellular region part    2941
##            enrichmentScore       NES      pvalue   p.adjust    qvalues
## GO:0031982      -0.2561837 -1.222689 0.001002004 0.03721229 0.02816364
## GO:0031988      -0.2572169 -1.226003 0.001007049 0.03721229 0.02816364
## GO:0005576      -0.2746489 -1.312485 0.001009082 0.03721229 0.02816364
## GO:0065010      -0.2570342 -1.222048 0.001013171 0.03721229 0.02816364
## GO:0070062      -0.2570342 -1.222048 0.001013171 0.03721229 0.02816364
## GO:0044421      -0.2744658 -1.310299 0.001014199 0.03721229 0.02816364

gseaplot(gsecc, geneSetID=&quot;GO:0000779&quot;)
</code></pre>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/clusterProfiler/2016_GO_analysis_using_clusterProfiler_files/figure-markdown_strict/unnamed-chunk-8-1.png" alt="" /></p>

<h2 id="go-analysis-using-user-s-own-data:bdc631894cab1e81ad84e5406d79b479">GO analysis using user&rsquo;s own data</h2>

<p><code>clusterProfiler</code> provides enricher function for hypergeometric test and
GSEA function for gene set enrichment analysis that are designed to
accept user defined annotation. They accept two additional parameters
TERM2GENE and TERM2NAME. As indicated in the parameter names, TERM2GENE
is a data.frame with first column of term ID and second column of
corresponding mapped gene and TERM2NAME is a data.frame with first
column of term ID and second column of corresponding term name.
TERM2NAME is optional.</p>

<p>An example of using enricher and GSEA to analyze DisGeNet annotation is
presented in the post, <a href="http://guangchuangyu.github.io/2015/05/use-clusterprofiler-as-an-universal-enrichment-analysis-tool/">use clusterProfiler as an universal enrichment analysis tool</a>.</p>

<h3 id="gmt-files:bdc631894cab1e81ad84e5406d79b479">GMT files</h3>

<p>We provides a function, <code>read.gmt</code>, that can parse GMT file into a
TERM2GENE data.frame that is ready for both <code>enricher</code> and <code>GSEA</code>
functions.</p>

<pre><code>gmtfile &lt;- system.file(&quot;extdata&quot;, &quot;c5.cc.v5.0.entrez.gmt&quot;, package=&quot;clusterProfiler&quot;)
c5 &lt;- read.gmt(gmtfile)
egmt &lt;- enricher(gene, TERM2GENE=c5)
head(summary(egmt))

##                                                ID              Description
## SPINDLE                                   SPINDLE                  SPINDLE
## MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON
## CYTOSKELETAL_PART               CYTOSKELETAL_PART        CYTOSKELETAL_PART
## SPINDLE_MICROTUBULE           SPINDLE_MICROTUBULE      SPINDLE_MICROTUBULE
## MICROTUBULE                           MICROTUBULE              MICROTUBULE
## CYTOSKELETON                         CYTOSKELETON             CYTOSKELETON
##                          GeneRatio  BgRatio       pvalue     p.adjust
## SPINDLE                      11/82  39/5270 7.667674e-12 6.594200e-10
## MICROTUBULE_CYTOSKELETON     16/82 152/5270 8.449298e-10 3.633198e-08
## CYTOSKELETAL_PART            15/82 235/5270 2.414879e-06 6.623386e-05
## SPINDLE_MICROTUBULE           5/82  16/5270 3.080645e-06 6.623386e-05
## MICROTUBULE                   6/82  32/5270 7.740446e-06 1.331357e-04
## CYTOSKELETON                 16/82 367/5270 1.308357e-04 1.826293e-03
##                                qvalue
## SPINDLE                  5.327016e-10
## MICROTUBULE_CYTOSKELETON 2.935019e-08
## CYTOSKELETAL_PART        5.350593e-05
## SPINDLE_MICROTUBULE      5.350593e-05
## MICROTUBULE              1.075515e-04
## CYTOSKELETON             1.475340e-03
##                                                                                                  geneID
## SPINDLE                                           991/9493/9787/22974/983/332/3832/7272/9055/6790/24137
## MICROTUBULE_CYTOSKELETON 991/9493/9133/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
## CYTOSKELETAL_PART             991/9493/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
## SPINDLE_MICROTUBULE                                                             983/332/3832/9055/24137
## MICROTUBULE                                                                983/332/3832/9055/24137/4137
## CYTOSKELETON             991/9493/9133/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
##                          Count
## SPINDLE                     11
## MICROTUBULE_CYTOSKELETON    16
## CYTOSKELETAL_PART           15
## SPINDLE_MICROTUBULE          5
## MICROTUBULE                  6
## CYTOSKELETON                16

egmt &lt;- setReadable(egmt, OrgDb=org.Hs.eg.db, keytype=&quot;ENTREZID&quot;)
head(summary(egmt))

##                                                ID              Description
## SPINDLE                                   SPINDLE                  SPINDLE
## MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON
## CYTOSKELETAL_PART               CYTOSKELETAL_PART        CYTOSKELETAL_PART
## SPINDLE_MICROTUBULE           SPINDLE_MICROTUBULE      SPINDLE_MICROTUBULE
## MICROTUBULE                           MICROTUBULE              MICROTUBULE
## CYTOSKELETON                         CYTOSKELETON             CYTOSKELETON
##                          GeneRatio  BgRatio       pvalue     p.adjust
## SPINDLE                      11/82  39/5270 7.667674e-12 6.594200e-10
## MICROTUBULE_CYTOSKELETON     16/82 152/5270 8.449298e-10 3.633198e-08
## CYTOSKELETAL_PART            15/82 235/5270 2.414879e-06 6.623386e-05
## SPINDLE_MICROTUBULE           5/82  16/5270 3.080645e-06 6.623386e-05
## MICROTUBULE                   6/82  32/5270 7.740446e-06 1.331357e-04
## CYTOSKELETON                 16/82 367/5270 1.308357e-04 1.826293e-03
##                                qvalue
## SPINDLE                  5.327016e-10
## MICROTUBULE_CYTOSKELETON 2.935019e-08
## CYTOSKELETAL_PART        5.350593e-05
## SPINDLE_MICROTUBULE      5.350593e-05
## MICROTUBULE              1.075515e-04
## CYTOSKELETON             1.475340e-03
##                                                                                                              geneID
## SPINDLE                                               CDC20/KIF23/DLGAP5/TPX2/CDK1/BIRC5/KIF11/TTK/PRC1/AURKA/KIF4A
## MICROTUBULE_CYTOSKELETON CDC20/KIF23/CCNB2/TOP2A/DLGAP5/TPX2/NEK2/CDK1/BIRC5/KIF11/TTK/PRC1/AURKA/KIF4A/MAPT/DNALI1
## CYTOSKELETAL_PART              CDC20/KIF23/TOP2A/DLGAP5/TPX2/NEK2/CDK1/BIRC5/KIF11/TTK/PRC1/AURKA/KIF4A/MAPT/DNALI1
## SPINDLE_MICROTUBULE                                                                     CDK1/BIRC5/KIF11/PRC1/KIF4A
## MICROTUBULE                                                                        CDK1/BIRC5/KIF11/PRC1/KIF4A/MAPT
## CYTOSKELETON             CDC20/KIF23/CCNB2/TOP2A/DLGAP5/TPX2/NEK2/CDK1/BIRC5/KIF11/TTK/PRC1/AURKA/KIF4A/MAPT/DNALI1
##                          Count
## SPINDLE                     11
## MICROTUBULE_CYTOSKELETON    16
## CYTOSKELETAL_PART           15
## SPINDLE_MICROTUBULE          5
## MICROTUBULE                  6
## CYTOSKELETON                16

gsegmt &lt;- GSEA(geneList, TERM2GENE=c5, verbose=F)
head(summary(gsegmt))

##                                                                    ID
## EXTRACELLULAR_REGION                             EXTRACELLULAR_REGION
## EXTRACELLULAR_REGION_PART                   EXTRACELLULAR_REGION_PART
## CELL_PROJECTION                                       CELL_PROJECTION
## PROTEINACEOUS_EXTRACELLULAR_MATRIX PROTEINACEOUS_EXTRACELLULAR_MATRIX
## EXTRACELLULAR_MATRIX                             EXTRACELLULAR_MATRIX
## EXTRACELLULAR_MATRIX_PART                   EXTRACELLULAR_MATRIX_PART
##                                                           Description
## EXTRACELLULAR_REGION                             EXTRACELLULAR_REGION
## EXTRACELLULAR_REGION_PART                   EXTRACELLULAR_REGION_PART
## CELL_PROJECTION                                       CELL_PROJECTION
## PROTEINACEOUS_EXTRACELLULAR_MATRIX PROTEINACEOUS_EXTRACELLULAR_MATRIX
## EXTRACELLULAR_MATRIX                             EXTRACELLULAR_MATRIX
## EXTRACELLULAR_MATRIX_PART                   EXTRACELLULAR_MATRIX_PART
##                                    setSize enrichmentScore       NES
## EXTRACELLULAR_REGION                   401      -0.3860230 -1.694496
## EXTRACELLULAR_REGION_PART              310      -0.4101043 -1.761338
## CELL_PROJECTION                         87      -0.4729701 -1.739867
## PROTEINACEOUS_EXTRACELLULAR_MATRIX      93      -0.6355317 -2.365007
## EXTRACELLULAR_MATRIX                    95      -0.6229461 -2.318356
## EXTRACELLULAR_MATRIX_PART               54      -0.5908035 -2.002728
##                                         pvalue   p.adjust    qvalues
## EXTRACELLULAR_REGION               0.001310616 0.03192152 0.02442263
## EXTRACELLULAR_REGION_PART          0.001375516 0.03192152 0.02442263
## CELL_PROJECTION                    0.001503759 0.03192152 0.02442263
## PROTEINACEOUS_EXTRACELLULAR_MATRIX 0.001555210 0.03192152 0.02442263
## EXTRACELLULAR_MATRIX               0.001557632 0.03192152 0.02442263
## EXTRACELLULAR_MATRIX_PART          0.001631321 0.03192152 0.02442263
</code></pre>


	  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
	  
	  <ins class="adsbygoogle"
	       style="display:inline-block;width:728px;height:90px"
	       data-ad-client="ca-pub-9874493018367687"
	       data-ad-slot="5154541855"></ins>
	  <script>
	    (adsbygoogle = window.adsbygoogle || []).push({});
	  </script> 
	  
	  <span class='st_twitter_hcount' displayText='Tweet'></span>
	  <span class='st_facebook_hcount' displayText='Facebook'></span>
	  <span class='st_sina_hcount' displayText='Sina'></span>
	  <span class='st_linkedin_hcount' displayText='LinkedIn'></span>
        </div>
	
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Jan 4, 2016</time>
    <span class="categories">
      Tags: 
        
        
          <a class="category" href="http://guangchuangyu.github.io/tags/clusterprofiler">clusterProfiler</a>  <a class="category" href="http://guangchuangyu.github.io/tags/go">GO</a>  
        
    </span>
  </p>
    
  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="http://guangchuangyu.github.io/2015/12/news-of-ggtree/" title="News of ggtree">News of ggtree</a>
    

     
      <a class="basic-alignment right" href="http://guangchuangyu.github.io/2016/01/annotate-a-phylogenetic-tree-with-insets/" title="Annotate a phylogenetic tree with insets">Annotate a phylogenetic tree with insets</a>
    
  </p>

  <div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    
    

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'gcyu';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</footer>

 

	</article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    
    <h1>Who am I</h1>
    
    
    <p>I am Guangchuang YU. A PhD candidate studying evolution of influenza A virus at The University of Hong Kong.
      <br><br>View my profile in <a href="https://scholar.google.com.hk/citations?user=DO5oG40AAAAJ&hl=en">Google Scholar</a> and <a href="https://www.researchgate.net/profile/Guangchuang_Yu">ResearchGate</a>.
    </p>

    
  </section>



  



  <ul class="sidebar-nav">
    <li class="sidebar-nav-item">
    <a target="_blank" href="https://github.com/GuangchuangYu" title="https://github.com/GuangchuangYu"><i class="fa fa-github fa-3x"></i></a>
    <a target="_blank" href="https://www.linkedin.com/in/guangchuang-yu-86820120" title="https://www.linkedin.com/in/guangchuang-yu-86820120"><i class="fa fa-linkedin fa-3x"></i></a>
    <a target="_blank" href="https://www.facebook.com/guangchuang.yu" title="https://www.facebook.com/guangchuang.yu"><i class="fa fa-facebook fa-3x"></i></a>
    <a target="_blank" href="https://twitter.com/guangchuangyu" title="https://twitter.com/guangchuangyu"><i class="fa fa-twitter fa-3x"></i></a>
    <a target="_blank" href="http://www.weibo.com/uncleyy" title="http://www.weibo.com/uncleyy"><i class="fa fa-weibo fa-3x"></i></a>

    
    
    
    
    </li>
  </ul>

 
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="/2016/03/yet-an-unofficial-bioedit-for-osx/">yet an unofficial BioEdit for OSX</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/google-drive--hku/">Google Drive @ HKU</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/embed-images-in-ggplot2-via-subview-and-annotate-a-phylogenetic-tree-with-images-using-inset-function/">embed images in ggplot2 via subview and annotate a phylogenetic tree with images using inset function</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/font-awesome-supported-in-emojifont/">Font Awesome supported in emojifont</a>
      </li>
      
      <li class="post">
        <a href="/2016/03/%E4%BD%BF%E7%94%A8ggtree%E5%AE%9E%E7%8E%B0%E8%BF%9B%E5%8C%96%E6%A0%91%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E5%92%8C%E6%B3%A8%E9%87%8A/">使用ggtree实现进化树的可视化和注释</a>
      </li>
      
    </ul>
  </section>

  <iframe src="http://rcm-na.amazon-adsystem.com/e/cm?t=guanyu08-20&o=1&p=12&l=ur1&category=topratedbks&banner=0C8DH3FCRQMY6EWXQGR2&f=ifr&linkID=2NQBTUQLCAGY637S" width="290" height="250" scrolling="no" border="0" marginwidth="0" style="border:none;max-width:900px;max-height:600px;" frameborder="0"></iframe>

  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>
  </section>

  <img src="http://guangchuangyu.github.io/blog_images/wechatpay.jpg" />

  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2016 Guangchuang Yu - <a href="http://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>


