<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.30.2" />
  <meta name="author" content="Guangchuang YU">
  <meta name="description" content="Postdoc researcher">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  <link rel="stylesheet" href="/css/bootstrap.min.css">
  <link rel="stylesheet" href="/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/academicons.min.css">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  <link rel="alternate" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">
  <link rel="feed" href="https://guangchuangyu.github.io/index.xml" type="application/rss+xml" title="Guangchuang YU">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://guangchuangyu.github.io/2015/04/bioc-31-news-of-my-bioc-packages/">

  

  <title>[Bioc 31] NEWS of my BioC packages | Guangchuang YU</title>


</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Guangchuang YU</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/resume">
            
            <span>Resume</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/cn">
            
            <span>中文入口</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  

  <div class="article-container">
    <h1 itemprop="name">[Bioc 31] NEWS of my BioC packages</h1>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2015-04-20 11:44:01 &#43;0800 HKT" itemprop="datePublished">
      Mon, Apr 20, 2015
    </time>
  </span>

  
  
  
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r">R</a
    >
    
  </span>
  
  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/bioc">BioC</a
    >, 
    
    <a href="/tags/gosemsim">GOSemSim</a
    >, 
    
    <a href="/tags/clusterprofiler">clusterProfiler</a
    >, 
    
    <a href="/tags/dose">DOSE</a
    >, 
    
    <a href="/tags/reactomepa">ReactomePA</a
    >, 
    
    <a href="/tags/ggtree">ggtree</a
    >, 
    
    <a href="/tags/chipseeker">ChIPseeker</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2f2015%2f04%2fbioc-31-news-of-my-bioc-packages%2f"
         target="_blank">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages&amp;url=https%3a%2f%2fguangchuangyu.github.io%2f2015%2f04%2fbioc-31-news-of-my-bioc-packages%2f"
         target="_blank">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fguangchuangyu.github.io%2f2015%2f04%2fbioc-31-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fguangchuangyu.github.io%2f2015%2f04%2fbioc-31-news-of-my-bioc-packages%2f&amp;title=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages"
         target="_blank">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%5bBioc%2031%5d%20NEWS%20of%20my%20BioC%20packages&amp;body=https%3a%2f%2fguangchuangyu.github.io%2f2015%2f04%2fbioc-31-news-of-my-bioc-packages%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    <div class="article-style" itemprop="articleBody">
        <p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot%202015-03-12%2019.05.54.png" alt="" /></p>

<h3 id="gosemsim">GOSemSim</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/GOSemSim.html" target="_blank">GOSemSim</a>
was first implemented in <a href="http://guangchuangyu.github.io/2008/08/gosemsim/" target="_blank">2008</a>
and published in
<a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">Bioinformatics</a>
in 2010. It&rsquo;s now a mature package with no bugs found in the past half
year. Only vignette and Information content data were updated.</p>

<p><img src="http://guangchuangyu.github.io/blog_images/Bioconductor/GOSemSim/gosemsim_stats.png" alt="" /></p>

<p></p>

<h3 id="clusterprofiler">clusterProfiler</h3>

<p><a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
was developed in
<a href="http://guangchuangyu.github.io/2011/03/clusterprofiler-in-bioconductor-2.8/" target="_blank">2011</a> and
published in
<a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118" target="_blank">OMICS</a> in
2012. I am very glad to find that
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
was presented in the book &lsquo;<a href="https://www.safaribooksonline.com/library/view/bioinformatics-with-r/9781783283132/ch02s08.html" target="_blank">Bioinformatics with R Cookbook</a>&lsquo;:
In this release, KEGG enrichment analysis (both hypergeometric and GSEA)
now supports <a href="http://guangchuangyu.github.io/2015/02/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/" target="_blank">using latest online data</a>
and <a href="http://guangchuangyu.github.io/2015/03/david-functional-analysis-with-clusterprofiler/" target="_blank">DAVID functional analysis</a>
is also supported. In addition, compareCluster function now supports
<a href="http://bioinfoblog.it/2015/02/a-formula-interface-for-geneontology-analysis/" target="_blank">formula interface</a>
like x ~ y + z; this feature was contributed by
<a href="https://twitter.com/dalloliogm" target="_blank">@dalloliogm</a>. Considering that many new
users have problem of converting biological IDs,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
contains a new function <em>bitr</em>, stands for Biological Id TranslatoR.
NOW,
<a href="www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html" target="_blank">clusterProfiler</a>
is not restricted to GO and KEGG analyses. All kinds of annotation can
be supported via <em>enricher</em> function for hypergeometric test and <em>GSEA</em>
function for gene set enrichment analysis. These two functions are not
yet documented in vignette and will soon.</p>

<h3 id="dose">DOSE</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
was first released in
<a href="http://guangchuangyu.github.io/2011/06/r-package-dose-released/" target="_blank">2011</a>, and has been
published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684" target="_blank">Bioinformatics</a>
on February 2015. An example of applying DOSE for disease ontology
enrichment analysis can be found in <a href="http://www.sciencedirect.com/science/article/pii/S0002929714002626" target="_blank">Genome-wide Identification and Characterization of Fixed Human-Specific Regulatory Regions</a>.
Besides,
<a href="http://www.bioconductor.org/packages/release/bioc/html/DOSE.html" target="_blank">DOSE</a>
also supports <a href="http://bioinfoblog.it/2015/04/ncg-enrichment-implemented-in-dose/" target="_blank">Network of Cancer Genes database</a>.</p>

<h3 id="reactomepa">ReactomePA</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ReactomePA.html" target="_blank">ReactomePA</a>
was developed in 2012 when KEGG is no longer available in public domain.
In BioC 3.0, reactome.db has <a href="https://support.bioconductor.org/p/63024/" target="_blank">a lot of issues</a>. Until the early of
this April, Bioconductor restored reactome.db in Bioc 3.0 to Bioc 2.14.
This issue affects ReactomePA several months and was removed in BioC
3.1.</p>

<h3 id="chipseeker">ChIPseeker</h3>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
was developed in
<a href="http://guangchuangyu.github.io/2014/04/chipseeker-for-chip-peak-annotation/" target="_blank">2014</a>,
after I found a <a href="http://guangchuangyu.github.io/2014/01/bug-of-r-package-chippeakanno/" target="_blank">bug of ChIPpeakAnno</a>
package. Many new features were added thereafter, including <a href="http://guangchuangyu.github.io/2014/04/visualization-methods-in-chipseeker/" target="_blank">several visualization methods</a>,
<a href="http://guangchuangyu.github.io/2014/10/multiple-annotation-in-chipseeker/" target="_blank">multiple annotations</a>,
ChIP-peak comparison and data mining with GEO database. ChIPseeker has
been published in
<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145" target="_blank">Bioinformatics</a>.
In this release, several minor bugs were fixed, a new parameter
<em>genomicAnnotationPriority</em> was added in <em>annotatePeak</em> and
<em>plotAvgProf</em> supports estimating confidence interval as shown below:
<img src="https://raw.githubusercontent.com/GuangchuangYu/ygc_name/dd399ab1d2287d0c7fee1009abf3b1b95553c73a/2015/Screenshot%202015-03-30%2020.09.44.png" alt="" />
This CI estimation feature was contributed by <a href="https://github.com/Puriney" target="_blank">Yan Yun</a>.
<a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">ChIPseeker</a>
works fine with other ChIP-seq related tools as demonstrated in the
<a href="http://compbio.ucdenver.edu/Hunter_lab/Phang/resources/Tzu-Software/ChIPseq.Analysis.html" target="_blank">ChIPseq Analysis workflow</a>.
It has also been presented in <a href="http://physiology.med.cornell.edu/faculty/elemento/lab/data/courses/2014/CSHLseq/ChIP-seq.pdf" target="_blank">2014 CSHLseq course</a>
and <a href="https://github.com/Bioconductor/UseBioconductor/blob/master/vignettes/F_ChIPSeq.Rmd" target="_blank">2015 Use Bioconductor course</a>.</p>

<h3 id="ggtree">ggtree</h3>

<p><img src="http://guangchuangyu.github.io/blog_images/2015/Screenshot-2015-02-10-10.27.27.png" alt="" /></p>

<p><a href="http://www.bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank">ggtree</a>
was developed on <a href="http://guangchuangyu.github.io/2014/12/viewing-and-annotating-phylogenetic-tree-with-ggtree/" target="_blank">December 2014</a>,
when I got tired of annotating tree by modifying node labels. This
release (version=1.0.0) is the first release, output of many common
software in evolution were supported and many new features were
introduced in the past few months. Please view the
<a href="http://www.bioconductor.org/packages/3.1/bioc/vignettes/ggtree/inst/doc/ggtree.html" target="_blank">vignette</a>
to find out more and I can guarantee that you won&rsquo;t be disappointed.</p>

<h2 id="citation">Citation</h2>

<p><strong><em>Yu G</em></strong>, Wang LG and He QY<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/31/14/2382" target="_blank">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(14):2382-2383.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="3781087"></div></p>

<p><strong><em>Yu G</em></strong>, Wang L, Han Y and He Q<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118" target="_blank">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-type="donut" data-badge-popover="right" data-altmetric-id="681089"></div></p>

<p><strong><em>G Yu</em></strong>, LG Wang, GR Yan, QY He. <a href="http://bioinformatics.oxfordjournals.org/content/31/4/608" target="_blank">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong> 2015, 31(4):608-609.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="2788597"></div></p>

<p><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract" target="_blank">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. <a href="http://dx.doi.org/10.1111/2041-210X.12628" target="_blank"><code>doi:10.1111/2041-210X.12628</code></a>.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div data-badge-popover="right" data-badge-type="donut" data-doi="10.1111/2041-210X.12628" class="altmetric-embed"></div></p>

<p><strong>Yu G</strong><sup>†</sup>, Li F<sup>†</sup>, Qin Y, Bo X<sup>*</sup>, Wu Y and Wang S<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/content/26/7/976" target="_blank">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
<div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="100979"></div></p>

<p><strong><em>G Yu</em></strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E#!divAbstract" target="_blank">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong> 2016, 12(2):477-479.</p>

<p><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script><div class="altmetric-embed" data-badge-popover="right" data-badge-type="donut" data-altmetric-id="4796667"></div></p>
    </div>
  </div>

  <center><span class="zaln"><a href="https://guangchuangyu.github.io/2017/11/bmc-journal-protects-paper-of-plagiarism/" class="zb">Plea to against BMC</a></span></center>


</article>

<div class="container">
  <nav>
  <ul class="pager">
    
    <li class="previous"><a href="https://guangchuangyu.github.io/2015/04/an-example-of-drawing-beast-tree-using-ggtree/"><span
      aria-hidden="true">&larr;</span> an example of drawing beast tree using ggtree</a></li>
    

    
    <li class="next"><a href="https://guangchuangyu.github.io/2015/05/using-gosemsim-to-rank-proteins-obtained-by-co-ip/">using GOSemSim to rank proteins obtained by co-IP <span
      aria-hidden="true">&rarr;</span></a></li>
    
  </ul>
</nav>

</div>

<div class="article-container">
  
<section id="comments">
  <div id="disqus_thread">
    <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "gcyu" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
  </div>
</section>


</div>

<footer class="site-footer">
  <div style="width:320px;margin:auto;">
<script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>
</div>
  <div class="container">
    <p class="powered-by">

      &copy; 2016 Guangchuang YU &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/1.18.4/TweenMax.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/gsap/latest/plugins/ScrollToPlugin.min.js"></script>
    <script src="/js/jquery-1.12.3.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
    <script src="/js/isotope.pkgd.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.1/imagesloaded.pkgd.min.js"></script>
    <script src="/js/hugo-academic.js"></script>
    

    
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
        ga('create', 'UA-77901140-1', 'auto');
        ga('send', 'pageview');

         
        var links = document.querySelectorAll('a');
        Array.prototype.map.call(links, function(item) {
            if (item.host != document.location.host) {
                item.addEventListener('click', function() {
                    var action = item.getAttribute('data-action') || 'follow';
                    ga('send', 'event', 'outbound', action, item.href);
                });
            }
        });
    </script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

